<!-- build time:Thu Mar 30 2023 11:30:44 GMT+0800 (中国标准时间) --><!DOCTYPE html><html lang="zh-CN"><head><meta charset="UTF-8"><meta name="viewport" content="width=device-width,initial-scale=1,maximum-scale=2"><meta name="theme-color" content="#FFF"><link rel="apple-touch-icon" sizes="180x180" href="/images/apple-touch-icon.png"><link rel="icon" type="image/ico" sizes="32x32" href="/images/favicon.ico"><meta http-equiv="Cache-Control" content="no-transform"><meta http-equiv="Cache-Control" content="no-siteapp"><link rel="alternate" type="application/rss+xml" title="小钊の笔记" href="https://www.zhouxiaozhao.cn/rss.xml"><link rel="alternate" type="application/atom+xml" title="小钊の笔记" href="https://www.zhouxiaozhao.cn/atom.xml"><link rel="alternate" type="application/json" title="小钊の笔记" href="https://www.zhouxiaozhao.cn/feed.json"><link rel="stylesheet" href="//fonts.googleapis.com/css?family=Mulish:300,300italic,400,400italic,700,700italic%7CFredericka%20the%20Great:300,300italic,400,400italic,700,700italic%7CNoto%20Serif%20JP:300,300italic,400,400italic,700,700italic%7CNoto%20Serif%20SC:300,300italic,400,400italic,700,700italic%7CInconsolata:300,300italic,400,400italic,700,700italic&display=swap&subset=latin,latin-ext"><link rel="stylesheet" href="/css/app.css?v=0.2.5"><meta name="keywords" content="bioinformatics,RNAseq"><link rel="canonical" href="https://www.zhouxiaozhao.cn/2020/12/24/2020-12-24-RNAseq(13)/"><title>RNAseq 教程 (3.2) - 生信学习 | Zhou Xiaozhao = 小钊の笔记 = 前天是小兔子，昨天是小鹿，今天是你</title><meta name="generator" content="Hexo 5.4.1"></head><body itemscope itemtype="http://schema.org/WebPage"><div id="loading"><div class="cat"><div class="body"></div><div class="head"><div class="face"></div></div><div class="foot"><div class="tummy-end"></div><div class="bottom"></div><div class="legs left"></div><div class="legs right"></div></div><div class="paw"><div class="hands left"></div><div class="hands right"></div></div></div></div><div id="container"><header id="header" itemscope itemtype="http://schema.org/WPHeader"><div class="inner"><div id="brand"><div class="pjax"><h1 itemprop="name headline">RNAseq 教程 (3.2)</h1><div class="meta"><span class="item" title="创建时间：2020-12-24 18:38:45"><span class="icon"><i class="ic i-calendar"></i> </span><span class="text">发表于</span> <time itemprop="dateCreated datePublished" datetime="2020-12-24T18:38:45+08:00">2020-12-24</time> </span><span class="item" title="本文字数"><span class="icon"><i class="ic i-pen"></i> </span><span class="text">本文字数</span> <span>8.3k</span> <span class="text">字</span> </span><span class="item" title="阅读时长"><span class="icon"><i class="ic i-clock"></i> </span><span class="text">阅读时长</span> <span>8 分钟</span></span></div></div></div><nav id="nav"><div class="inner"><div class="toggle"><div class="lines" aria-label="切换导航栏"><span class="line"></span> <span class="line"></span> <span class="line"></span></div></div><ul class="menu"><li class="item title"><a href="/" rel="start">Zhou Xiaozhao</a></li></ul><ul class="right"><li class="item theme"><i class="ic i-sun"></i></li><li class="item search"><i class="ic i-search"></i></li></ul></div></nav></div><div id="imgs" class="pjax"><ul><li class="item" data-background-image="https://img1.imgtp.com/2023/02/01/ky7B2xxW.jpg"></li><li class="item" data-background-image="https://img1.imgtp.com/2023/02/01/4OefNOxP.jpg"></li><li class="item" data-background-image="https://img1.imgtp.com/2023/02/01/LoNchvnE.jpg"></li><li class="item" data-background-image="https://img1.imgtp.com/2023/02/01/U6TYwlFY.jpg"></li><li class="item" data-background-image="https://img1.imgtp.com/2023/02/01/xioKAX7K.jpg"></li><li class="item" data-background-image="https://img1.imgtp.com/2023/02/01/MRMqrWXd.jpg"></li></ul></div></header><div id="waves"><svg class="waves" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" viewBox="0 24 150 28" preserveAspectRatio="none" shape-rendering="auto"><defs><path id="gentle-wave" d="M-160 44c30 0 58-18 88-18s 58 18 88 18 58-18 88-18 58 18 88 18 v44h-352z"/></defs><g class="parallax"><use xlink:href="#gentle-wave" x="48" y="0"/><use xlink:href="#gentle-wave" x="48" y="3"/><use xlink:href="#gentle-wave" x="48" y="5"/><use xlink:href="#gentle-wave" x="48" y="7"/></g></svg></div><main><div class="inner"><div id="main" class="pjax"><div class="article wrap"><div class="breadcrumb" itemscope itemtype="https://schema.org/BreadcrumbList"><i class="ic i-home"></i> <span><a href="/">首页</a></span><i class="ic i-angle-right"></i> <span class="current" itemprop="itemListElement" itemscope itemtype="https://schema.org/ListItem"><a href="/categories/Bioinformation/" itemprop="item" rel="index" title="分类于 生信学习"><span itemprop="name">生信学习</span></a><meta itemprop="position" content="1"></span></div><article itemscope itemtype="http://schema.org/Article" class="post block" lang="zh-CN"><link itemprop="mainEntityOfPage" href="https://www.zhouxiaozhao.cn/2020/12/24/2020-12-24-RNAseq(13)/"><span hidden itemprop="author" itemscope itemtype="http://schema.org/Person"><meta itemprop="image" content="/images/avatar.jpg"><meta itemprop="name" content="amane"><meta itemprop="description" content="前天是小兔子，昨天是小鹿，今天是你, 生物学在读博士"></span><span hidden itemprop="publisher" itemscope itemtype="http://schema.org/Organization"><meta itemprop="name" content="小钊の笔记"></span><div class="body md" itemprop="articleBody"><h2 id="目录"><a class="anchor" href="#目录">#</a> 目录</h2><p><strong>1.Module 1 - Introduction to RNA sequencing</strong></p><ol><li><a href="https://www.zhouxiaozhao.cn/2020/11/17/2020-11-17-RNAseq(1)/">Installation</a></li><li><a href="https://www.zhouxiaozhao.cn/2020/11/28/2020-11-28-RNAseq(2)/">Reference Genomes</a></li><li><a href="https://www.zhouxiaozhao.cn/2020/12/01/2020-12-01-RNAseq(3)/">Annotations</a></li><li><a href="https://www.zhouxiaozhao.cn/2020/12/03/2020-12-03-RNAseq(4)/">Indexing</a></li><li><a href="https://www.zhouxiaozhao.cn/2020/12/05/2020-12-05-RNAseq(5)/">RNA-seq Data</a></li><li><a href="https://www.zhouxiaozhao.cn/2020/12/08/2020-12-08-RNAseq(6)/">Pre-Alignment QC</a></li></ol><p><strong>2.Module 2 - RNA-seq Alignment and Visualization</strong></p><ol><li><a href="https://www.zhouxiaozhao.cn/2020/12/10/2020-12-10-RNAseq(7)/">Adapter Trim</a></li><li><a href="https://www.zhouxiaozhao.cn/2020/12/12/2020-12-12-RNAseq(8)/">Alignment</a></li><li><a href="https://www.zhouxiaozhao.cn/2020/12/15/2020-12-15-RNAseq(9)/">IGV</a></li><li><a href="https://www.zhouxiaozhao.cn/2020/12/17/2020-12-17-RNAseq(10)/">Alignment Visualization</a></li><li><a href="https://www.zhouxiaozhao.cn/2020/12/19/2020-12-19-RNAseq(11)/">Alignment QC</a></li></ol><p><strong>3.Module 3 - Expression and Differential Expression</strong></p><ol><li><a href="https://www.zhouxiaozhao.cn/2020/12/22/2020-12-22-RNAseq(12)/">Expression</a></li><li><a href="https://www.zhouxiaozhao.cn/2020/12/24/2020-12-24-RNAseq(13)/">Differential Expression</a></li><li><a href="https://www.zhouxiaozhao.cn/2020/12/26/2020-12-26-RNAseq(14)/">DE Visualization</a></li><li><a href="https://www.zhouxiaozhao.cn/2020/12/29/2020-12-29-RNAseq(15)/">Kallisto for Reference-Free Abundance Estimation</a></li></ol><p><strong>4.Module 4 - Isoform Discovery and Alternative Expression</strong></p><ol><li><a href="https://www.zhouxiaozhao.cn/2020/12/31/2020-12-31-RNAseq(16)/">Reference Guided Transcript Assembly</a></li><li><a href="https://www.zhouxiaozhao.cn/2021/01/02/2021-01-02-RNAseq(17)/">de novo Transcript Assembly</a></li><li><a href="https://www.zhouxiaozhao.cn/2021/01/05/2020-01-05-RNAseq(18)/">Transcript Assembly Merge</a></li><li><a href="https://www.zhouxiaozhao.cn/2021/01/07/2021-01-07-RNAseq(19)/">Differential Splicing</a></li><li><a href="https://www.zhouxiaozhao.cn/2021/01/09/2021-01-09-RNAseq(20)/">Splicing Visualization</a></li></ol><p><strong>5.Module 5 - De novo transcript reconstruction</strong></p><ol><li><a href="https://www.zhouxiaozhao.cn/2021/01/12/2021-01-12-RNAseq(21)/">De novo RNA-Seq Assembly and Analysis Using Trinity</a></li></ol><p><strong>6.Module 6 - Functional Annotation of Transcripts</strong></p><ol><li><a href="https://www.zhouxiaozhao.cn/2021/01/14/2021-01-14RNAseq(22)/">Functional Annotation of Assembled Transcripts Using Trinotate</a></li></ol><h2 id="32-differential-expression"><a class="anchor" href="#32-differential-expression">#</a> 3.2 Differential Expression</h2><h3 id="ballgown"><a class="anchor" href="#ballgown">#</a> Ballgown</h3><p>用 Ballgown 比较肿瘤和正常情况。详情请参考手册:</p><p><span class="exturl" data-url="aHR0cHM6Ly93d3cuYmlvY29uZHVjdG9yLm9yZy9wYWNrYWdlcy9yZWxlYXNlL2Jpb2MvaHRtbC9iYWxsZ293bi5odG1s">https://www.bioconductor.org/packages/release/bioc/html/ballgown.html</span> `</p><p>使用所有复制，对已知 (仅参考模式) 转录本进行 UHR 与 HBR 比较:</p><p>首先创建一个文件，列出我们的 6 个表达式文件，然后查看该文件，检查这些结果:</p><figure class="highlight raw"><figcaption data-lang></figcaption><table><tr><td data-num="1"></td><td><pre>printf &quot;\&quot;ids\&quot;,\&quot;type\&quot;,\&quot;path\&quot;\n\&quot;UHR_Rep1\&quot;,\&quot;UHR\&quot;,\&quot;UHR_Rep1\&quot;\n\&quot;UHR_Rep2\&quot;,\&quot;UHR\&quot;,\&quot;UHR_Rep2\&quot;\n\&quot;UHR_Rep3\&quot;,\&quot;UHR\&quot;,\&quot;UHR_Rep3\&quot;\n\&quot;HBR_Rep1\&quot;,\&quot;HBR\&quot;,\&quot;HBR_Rep1\&quot;\n\&quot;HBR_Rep2\&quot;,\&quot;HBR\&quot;,\&quot;HBR_Rep2\&quot;\n\&quot;HBR_Rep3\&quot;,\&quot;HBR\&quot;,\&quot;HBR_Rep3\&quot;\n&quot; &gt; UHR_vs_HBR.csv</pre></td></tr><tr><td data-num="2"></td><td><pre>cat UHR_vs_HBR.csv</pre></td></tr><tr><td data-num="3"></td><td><pre></pre></td></tr><tr><td data-num="4"></td><td><pre>R</pre></td></tr></table></figure><figure class="highlight raw"><figcaption data-lang></figcaption><table><tr><td data-num="1"></td><td><pre>#Jason Walker, jason.walker[AT]wustl.edu</pre></td></tr><tr><td data-num="2"></td><td><pre>#Malachi Griffith, mgriffit[AT]wustl.edu</pre></td></tr><tr><td data-num="3"></td><td><pre>#Obi Griffith, obigriffith[AT]wustl.edu</pre></td></tr><tr><td data-num="4"></td><td><pre>#The Genome McDonnell Institute, Washington University School of Medicine</pre></td></tr><tr><td data-num="5"></td><td><pre></pre></td></tr><tr><td data-num="6"></td><td><pre>#R tutorial for Informatics for RNA-sequence Analysis workshops</pre></td></tr><tr><td data-num="7"></td><td><pre></pre></td></tr><tr><td data-num="8"></td><td><pre>library(ballgown)</pre></td></tr><tr><td data-num="9"></td><td><pre>library(genefilter)</pre></td></tr><tr><td data-num="10"></td><td><pre>library(dplyr)</pre></td></tr><tr><td data-num="11"></td><td><pre>library(devtools)</pre></td></tr><tr><td data-num="12"></td><td><pre></pre></td></tr><tr><td data-num="13"></td><td><pre># Load phenotype data from a file we saved in the current working directory</pre></td></tr><tr><td data-num="14"></td><td><pre>pheno_data &#x3D; read.csv(&quot;UHR_vs_HBR.csv&quot;)</pre></td></tr><tr><td data-num="15"></td><td><pre></pre></td></tr><tr><td data-num="16"></td><td><pre># Load ballgown data structure and save it to a variable &quot;bg&quot;</pre></td></tr><tr><td data-num="17"></td><td><pre>bg &#x3D; ballgown(samples&#x3D;as.vector(pheno_data$path), pData&#x3D;pheno_data)</pre></td></tr><tr><td data-num="18"></td><td><pre></pre></td></tr><tr><td data-num="19"></td><td><pre># Display a description of this object</pre></td></tr><tr><td data-num="20"></td><td><pre>bg</pre></td></tr><tr><td data-num="21"></td><td><pre></pre></td></tr><tr><td data-num="22"></td><td><pre># Load all attributes including gene name</pre></td></tr><tr><td data-num="23"></td><td><pre>bg_table &#x3D; texpr(bg, &#39;all&#39;)</pre></td></tr><tr><td data-num="24"></td><td><pre>bg_gene_names &#x3D; unique(bg_table[, 9:10])</pre></td></tr><tr><td data-num="25"></td><td><pre></pre></td></tr><tr><td data-num="26"></td><td><pre># Save the ballgown object to a file for later use</pre></td></tr><tr><td data-num="27"></td><td><pre>save(bg, file&#x3D;&#39;bg.rda&#39;)</pre></td></tr><tr><td data-num="28"></td><td><pre></pre></td></tr><tr><td data-num="29"></td><td><pre># Perform differential expression (DE) analysis with no filtering</pre></td></tr><tr><td data-num="30"></td><td><pre>results_transcripts &#x3D; stattest(bg, feature&#x3D;&quot;transcript&quot;, covariate&#x3D;&quot;type&quot;, getFC&#x3D;TRUE, meas&#x3D;&quot;FPKM&quot;)</pre></td></tr><tr><td data-num="31"></td><td><pre>results_genes &#x3D; stattest(bg, feature&#x3D;&quot;gene&quot;, covariate&#x3D;&quot;type&quot;, getFC&#x3D;TRUE, meas&#x3D;&quot;FPKM&quot;)</pre></td></tr><tr><td data-num="32"></td><td><pre>results_genes &#x3D; merge(results_genes, bg_gene_names, by.x&#x3D;c(&quot;id&quot;), by.y&#x3D;c(&quot;gene_id&quot;))</pre></td></tr><tr><td data-num="33"></td><td><pre></pre></td></tr><tr><td data-num="34"></td><td><pre># Save a tab delimited file for both the transcript and gene results</pre></td></tr><tr><td data-num="35"></td><td><pre>write.table(results_transcripts, &quot;UHR_vs_HBR_transcript_results.tsv&quot;, sep&#x3D;&quot;\t&quot;, quote&#x3D;FALSE, row.names &#x3D; FALSE)</pre></td></tr><tr><td data-num="36"></td><td><pre>write.table(results_genes, &quot;UHR_vs_HBR_gene_results.tsv&quot;, sep&#x3D;&quot;\t&quot;, quote&#x3D;FALSE, row.names &#x3D; FALSE)</pre></td></tr><tr><td data-num="37"></td><td><pre></pre></td></tr><tr><td data-num="38"></td><td><pre># Filter low-abundance genes. Here we remove all transcripts with a variance across the samples of less than one</pre></td></tr><tr><td data-num="39"></td><td><pre>bg_filt &#x3D; subset (bg,&quot;rowVars(texpr(bg)) &gt; 1&quot;, genomesubset&#x3D;TRUE)</pre></td></tr><tr><td data-num="40"></td><td><pre></pre></td></tr><tr><td data-num="41"></td><td><pre># Load all attributes including gene name</pre></td></tr><tr><td data-num="42"></td><td><pre>bg_filt_table &#x3D; texpr(bg_filt , &#39;all&#39;)</pre></td></tr><tr><td data-num="43"></td><td><pre>bg_filt_gene_names &#x3D; unique(bg_filt_table[, 9:10])</pre></td></tr><tr><td data-num="44"></td><td><pre></pre></td></tr><tr><td data-num="45"></td><td><pre># Perform DE analysis now using the filtered data</pre></td></tr><tr><td data-num="46"></td><td><pre>results_transcripts &#x3D; stattest(bg_filt, feature&#x3D;&quot;transcript&quot;, covariate&#x3D;&quot;type&quot;, getFC&#x3D;TRUE, meas&#x3D;&quot;FPKM&quot;)</pre></td></tr><tr><td data-num="47"></td><td><pre>results_genes &#x3D; stattest(bg_filt, feature&#x3D;&quot;gene&quot;, covariate&#x3D;&quot;type&quot;, getFC&#x3D;TRUE, meas&#x3D;&quot;FPKM&quot;)</pre></td></tr><tr><td data-num="48"></td><td><pre>results_genes &#x3D; merge(results_genes, bg_filt_gene_names, by.x&#x3D;c(&quot;id&quot;), by.y&#x3D;c(&quot;gene_id&quot;))</pre></td></tr><tr><td data-num="49"></td><td><pre></pre></td></tr><tr><td data-num="50"></td><td><pre># Output the filtered list of genes and transcripts and save to tab delimited files</pre></td></tr><tr><td data-num="51"></td><td><pre>write.table(results_transcripts, &quot;UHR_vs_HBR_transcript_results_filtered.tsv&quot;, sep&#x3D;&quot;\t&quot;, quote&#x3D;FALSE, row.names &#x3D; FALSE)</pre></td></tr><tr><td data-num="52"></td><td><pre>write.table(results_genes, &quot;UHR_vs_HBR_gene_results_filtered.tsv&quot;, sep&#x3D;&quot;\t&quot;, quote&#x3D;FALSE, row.names &#x3D; FALSE)</pre></td></tr><tr><td data-num="53"></td><td><pre></pre></td></tr><tr><td data-num="54"></td><td><pre># Identify the significant genes with p-value &lt; 0.05</pre></td></tr><tr><td data-num="55"></td><td><pre>sig_transcripts &#x3D; subset(results_transcripts, results_transcripts$pval&lt;0.05)</pre></td></tr><tr><td data-num="56"></td><td><pre>sig_genes &#x3D; subset(results_genes, results_genes$pval&lt;0.05)</pre></td></tr><tr><td data-num="57"></td><td><pre></pre></td></tr><tr><td data-num="58"></td><td><pre># Output the signifant gene results to a pair of tab delimited files</pre></td></tr><tr><td data-num="59"></td><td><pre>write.table(sig_transcripts, &quot;UHR_vs_HBR_transcript_results_sig.tsv&quot;, sep&#x3D;&quot;\t&quot;, quote&#x3D;FALSE, row.names &#x3D; FALSE)</pre></td></tr><tr><td data-num="60"></td><td><pre>write.table(sig_genes, &quot;UHR_vs_HBR_gene_results_sig.tsv&quot;, sep&#x3D;&quot;\t&quot;, quote&#x3D;FALSE, row.names &#x3D; FALSE)</pre></td></tr><tr><td data-num="61"></td><td><pre></pre></td></tr><tr><td data-num="62"></td><td><pre># Exit the R session</pre></td></tr><tr><td data-num="63"></td><td><pre>quit(save&#x3D;&quot;no&quot;)</pre></td></tr></table></figure><p>运行结束后查看结果</p><figure class="highlight raw"><figcaption data-lang></figcaption><table><tr><td data-num="1"></td><td><pre>head UHR_vs_HBR_gene_results.tsv</pre></td></tr></table></figure><p>这条染色体上有多少个基因？</p><figure class="highlight raw"><figcaption data-lang></figcaption><table><tr><td data-num="1"></td><td><pre>grep -v feature UHR_vs_HBR_gene_results.tsv | wc -l</pre></td></tr></table></figure><p>在 UHR 或 HBR 中有多少通过过滤？</p><figure class="highlight raw"><figcaption data-lang></figcaption><table><tr><td data-num="1"></td><td><pre>grep -v feature UHR_vs_HBR_gene_results_filtered.tsv | wc -l</pre></td></tr></table></figure><p>在这条染色体上发现了多少差异表达基因 (p 值 &lt; 0.05)?</p><figure class="highlight raw"><figcaption data-lang></figcaption><table><tr><td data-num="1"></td><td><pre>grep -v feature UHR_vs_HBR_gene_results_sig.tsv | wc -l</pre></td></tr></table></figure><p>展示前 20 个 DE 基因。看看 IGV 中的一些基因 —— 它们有意义吗？</p><figure class="highlight raw"><figcaption data-lang></figcaption><table><tr><td data-num="1"></td><td><pre>grep -v feature UHR_vs_HBR_gene_results_sig.tsv | sort -rnk 3 | head -n 20 #Higher abundance in UHR</pre></td></tr><tr><td data-num="2"></td><td><pre>grep -v feature UHR_vs_HBR_gene_results_sig.tsv | sort -nk 3 | head -n 20 #Higher abundance in HBR</pre></td></tr></table></figure><p>将 P&lt;0.05 的所有基因保存到一个新文件中。</p><figure class="highlight raw"><figcaption data-lang></figcaption><table><tr><td data-num="1"></td><td><pre>grep -v feature UHR_vs_HBR_gene_results_sig.tsv | cut -f 6 | sed &#39;s&#x2F;\&quot;&#x2F;&#x2F;g&#39; &gt; DE_genes.txt</pre></td></tr><tr><td data-num="2"></td><td><pre>head DE_genes.txt</pre></td></tr></table></figure><h3 id="edger"><a class="anchor" href="#edger">#</a> edgeR</h3><figure class="highlight raw"><figcaption data-lang></figcaption><table><tr><td data-num="1"></td><td><pre>mkdir -p de&#x2F;htseq_counts</pre></td></tr><tr><td data-num="2"></td><td><pre>cd de&#x2F;htseq_counts</pre></td></tr><tr><td data-num="3"></td><td><pre>perl -ne &#39;if ($_ &#x3D;~ &#x2F;gene_id\s\&quot;(ENSG\S+)\&quot;\;&#x2F;) &#123; $id &#x3D; $1; $name &#x3D; undef; if ($_ &#x3D;~ &#x2F;gene_name\s\&quot;(\S+)&quot;\;&#x2F;) &#123; $name &#x3D; $1; &#125;; &#125;; if ($id &amp;&amp; $name) &#123;print &quot;$id\t$name\n&quot;;&#125; if ($_&#x3D;~&#x2F;gene_id\s\&quot;(ERCC\S+)\&quot;&#x2F;)&#123;print &quot;$1\t$1\n&quot;;&#125;&#39; $RNA_REF_GTF | sort | uniq &gt; ENSG_ID2Name.txt</pre></td></tr><tr><td data-num="4"></td><td><pre>head ENSG_ID2Name.txt</pre></td></tr></table></figure><figure class="highlight raw"><figcaption data-lang></figcaption><table><tr><td data-num="1"></td><td><pre>cut -f 1 ENSG_ID2Name.txt | sort | uniq | wc</pre></td></tr><tr><td data-num="2"></td><td><pre>cut -f 2 ENSG_ID2Name.txt | sort | uniq | wc</pre></td></tr><tr><td data-num="3"></td><td><pre>cut -f 2 ENSG_ID2Name.txt | sort | uniq -c | sort -r | head</pre></td></tr></table></figure><figure class="highlight raw"><figcaption data-lang></figcaption><table><tr><td data-num="1"></td><td><pre>#Malachi Griffith, mgriffit[AT]wustl.edu</pre></td></tr><tr><td data-num="2"></td><td><pre>#Obi Griffith, obigriffith[AT]wustl.edu</pre></td></tr><tr><td data-num="3"></td><td><pre>#The McDonnell Genome Institute, Washington University School of Medicine</pre></td></tr><tr><td data-num="4"></td><td><pre></pre></td></tr><tr><td data-num="5"></td><td><pre>#R tutorial for Informatics for RNA-sequence Analysis workshops</pre></td></tr><tr><td data-num="6"></td><td><pre></pre></td></tr><tr><td data-num="7"></td><td><pre>#######################</pre></td></tr><tr><td data-num="8"></td><td><pre># Loading Data into R #</pre></td></tr><tr><td data-num="9"></td><td><pre>#######################</pre></td></tr><tr><td data-num="10"></td><td><pre></pre></td></tr><tr><td data-num="11"></td><td><pre>#Set working directory where output will go</pre></td></tr><tr><td data-num="12"></td><td><pre>working_dir &#x3D; &quot;path_to&#x2F;de&#x2F;htseq_counts&quot;</pre></td></tr><tr><td data-num="13"></td><td><pre>setwd(working_dir)</pre></td></tr><tr><td data-num="14"></td><td><pre></pre></td></tr><tr><td data-num="15"></td><td><pre>#Read in gene mapping</pre></td></tr><tr><td data-num="16"></td><td><pre>mapping&#x3D;read.table(&quot;~&#x2F;workspace&#x2F;rnaseq&#x2F;de&#x2F;htseq_counts&#x2F;ENSG_ID2Name.txt&quot;, header&#x3D;FALSE, stringsAsFactors&#x3D;FALSE, row.names&#x3D;1)</pre></td></tr><tr><td data-num="17"></td><td><pre></pre></td></tr><tr><td data-num="18"></td><td><pre># Read in count matrix</pre></td></tr><tr><td data-num="19"></td><td><pre>rawdata&#x3D;read.table(&quot;~&#x2F;workspace&#x2F;rnaseq&#x2F;expression&#x2F;htseq_counts&#x2F;gene_read_counts_table_all_final.tsv&quot;, header&#x3D;TRUE, stringsAsFactors&#x3D;FALSE, row.names&#x3D;1)</pre></td></tr><tr><td data-num="20"></td><td><pre></pre></td></tr><tr><td data-num="21"></td><td><pre># Check dimensions</pre></td></tr><tr><td data-num="22"></td><td><pre>dim(rawdata)</pre></td></tr><tr><td data-num="23"></td><td><pre></pre></td></tr><tr><td data-num="24"></td><td><pre># Require at least 25% of samples to have count &gt; 25</pre></td></tr><tr><td data-num="25"></td><td><pre>quant &lt;- apply(rawdata,1,quantile,0.75)</pre></td></tr><tr><td data-num="26"></td><td><pre>keep &lt;- which((quant &gt;&#x3D; 25) &#x3D;&#x3D; 1)</pre></td></tr><tr><td data-num="27"></td><td><pre>rawdata &lt;- rawdata[keep,]</pre></td></tr><tr><td data-num="28"></td><td><pre>dim(rawdata)</pre></td></tr><tr><td data-num="29"></td><td><pre></pre></td></tr><tr><td data-num="30"></td><td><pre>#################</pre></td></tr><tr><td data-num="31"></td><td><pre># Running edgeR #</pre></td></tr><tr><td data-num="32"></td><td><pre>#################</pre></td></tr><tr><td data-num="33"></td><td><pre></pre></td></tr><tr><td data-num="34"></td><td><pre># load edgeR</pre></td></tr><tr><td data-num="35"></td><td><pre>library(&#39;edgeR&#39;)</pre></td></tr><tr><td data-num="36"></td><td><pre></pre></td></tr><tr><td data-num="37"></td><td><pre># make class labels</pre></td></tr><tr><td data-num="38"></td><td><pre>class &lt;- factor( c( rep(&quot;UHR&quot;,3), rep(&quot;HBR&quot;,3) ))</pre></td></tr><tr><td data-num="39"></td><td><pre></pre></td></tr><tr><td data-num="40"></td><td><pre># Get common gene names</pre></td></tr><tr><td data-num="41"></td><td><pre>genes&#x3D;rownames(rawdata)</pre></td></tr><tr><td data-num="42"></td><td><pre>gene_names&#x3D;mapping[genes,1]</pre></td></tr><tr><td data-num="43"></td><td><pre></pre></td></tr><tr><td data-num="44"></td><td><pre></pre></td></tr><tr><td data-num="45"></td><td><pre># Make DGEList object</pre></td></tr><tr><td data-num="46"></td><td><pre>y &lt;- DGEList(counts&#x3D;rawdata, genes&#x3D;genes, group&#x3D;class)</pre></td></tr><tr><td data-num="47"></td><td><pre>nrow(y)</pre></td></tr><tr><td data-num="48"></td><td><pre></pre></td></tr><tr><td data-num="49"></td><td><pre># TMM Normalization</pre></td></tr><tr><td data-num="50"></td><td><pre>y &lt;- calcNormFactors(y)</pre></td></tr><tr><td data-num="51"></td><td><pre></pre></td></tr><tr><td data-num="52"></td><td><pre># Estimate dispersion</pre></td></tr><tr><td data-num="53"></td><td><pre>y &lt;- estimateCommonDisp(y, verbose&#x3D;TRUE)</pre></td></tr><tr><td data-num="54"></td><td><pre>y &lt;- estimateTagwiseDisp(y)</pre></td></tr><tr><td data-num="55"></td><td><pre></pre></td></tr><tr><td data-num="56"></td><td><pre># Differential expression test</pre></td></tr><tr><td data-num="57"></td><td><pre>et &lt;- exactTest(y)</pre></td></tr><tr><td data-num="58"></td><td><pre></pre></td></tr><tr><td data-num="59"></td><td><pre># Print top genes</pre></td></tr><tr><td data-num="60"></td><td><pre>topTags(et)</pre></td></tr><tr><td data-num="61"></td><td><pre></pre></td></tr><tr><td data-num="62"></td><td><pre># Print number of up&#x2F;down significant genes at FDR &#x3D; 0.05  significance level</pre></td></tr><tr><td data-num="63"></td><td><pre>summary(de &lt;- decideTestsDGE(et, p&#x3D;.05))</pre></td></tr><tr><td data-num="64"></td><td><pre>detags &lt;- rownames(y)[as.logical(de)]</pre></td></tr><tr><td data-num="65"></td><td><pre></pre></td></tr><tr><td data-num="66"></td><td><pre></pre></td></tr><tr><td data-num="67"></td><td><pre># Output DE genes</pre></td></tr><tr><td data-num="68"></td><td><pre># Matrix of significantly DE genes</pre></td></tr><tr><td data-num="69"></td><td><pre>mat &lt;- cbind(</pre></td></tr><tr><td data-num="70"></td><td><pre> genes,gene_names,</pre></td></tr><tr><td data-num="71"></td><td><pre> sprintf(&#39;%0.3f&#39;,log10(et$table$PValue)),</pre></td></tr><tr><td data-num="72"></td><td><pre> sprintf(&#39;%0.3f&#39;,et$table$logFC)</pre></td></tr><tr><td data-num="73"></td><td><pre>)[as.logical(de),]</pre></td></tr><tr><td data-num="74"></td><td><pre>colnames(mat) &lt;- c(&quot;Gene&quot;, &quot;Gene_Name&quot;, &quot;Log10_Pvalue&quot;, &quot;Log_fold_change&quot;)</pre></td></tr><tr><td data-num="75"></td><td><pre></pre></td></tr><tr><td data-num="76"></td><td><pre># Order by log fold change</pre></td></tr><tr><td data-num="77"></td><td><pre>o &lt;- order(et$table$logFC[as.logical(de)],decreasing&#x3D;TRUE)</pre></td></tr><tr><td data-num="78"></td><td><pre>mat &lt;- mat[o,]</pre></td></tr><tr><td data-num="79"></td><td><pre></pre></td></tr><tr><td data-num="80"></td><td><pre># Save table</pre></td></tr><tr><td data-num="81"></td><td><pre>write.table(mat, file&#x3D;&quot;DE_genes.txt&quot;, quote&#x3D;FALSE, row.names&#x3D;FALSE, sep&#x3D;&quot;\t&quot;)</pre></td></tr><tr><td data-num="82"></td><td><pre></pre></td></tr><tr><td data-num="83"></td><td><pre>#To exit R type the following</pre></td></tr><tr><td data-num="84"></td><td><pre>quit(save&#x3D;&quot;no&quot;)</pre></td></tr></table></figure><p>与其他结果比较</p><figure class="highlight raw"><figcaption data-lang></figcaption><table><tr><td data-num="1"></td><td><pre>cat DE_genes.txt</pre></td></tr><tr><td data-num="2"></td><td><pre>cat &#x2F;de&#x2F;htseq_counts&#x2F;DE_genes.txt</pre></td></tr></table></figure><figure class="highlight raw"><figcaption data-lang></figcaption><table><tr><td data-num="1"></td><td><pre>cut -f 1 DE_genes.txt | sort  &gt; ballgown_DE_gene_symbols.txt</pre></td></tr><tr><td data-num="2"></td><td><pre>cut -f 2 de&#x2F;htseq_counts&#x2F;DE_genes.txt | sort &gt; htseq_counts_edgeR_DE_gene_symbols.txt</pre></td></tr></table></figure><div class="tags"><a href="/tags/bioinformatics/" rel="tag"><i class="ic i-tag"></i> bioinformatics</a> <a href="/tags/RNAseq/" rel="tag"><i class="ic i-tag"></i> RNAseq</a></div></div><footer><div class="meta"><span class="item"><span class="icon"><i class="ic i-calendar-check"></i> </span><span class="text">更新于</span> <time title="修改时间：2023-03-08 15:08:23" itemprop="dateModified" datetime="2023-03-08T15:08:23+08:00">2023-03-08</time> </span><span id="2020/12/24/2020-12-24-RNAseq(13)/" class="item leancloud_visitors" data-flag-title="RNAseq 教程 (3.2)" title="阅读次数"><span class="icon"><i class="ic i-eye"></i> </span><span class="text">阅读次数</span> <span class="leancloud-visitors-count"></span> <span class="text">次</span></span></div><div class="reward"><button><i class="ic i-heartbeat"></i> 赞赏</button><p>请我喝[茶]~(￣▽￣)~*</p><div id="qr"><div><img data-src="/images/wechatpay.png" alt="amane 微信支付"><p>微信支付</p></div><div><img data-src="/images/alipay.png" alt="amane 支付宝"><p>支付宝</p></div></div></div><div id="copyright"><ul><li class="author"><strong>本文作者： </strong>amane <i class="ic i-at"><em>@</em></i>小钊の笔记</li><li class="link"><strong>本文链接：</strong> <a href="https://www.zhouxiaozhao.cn/2020/12/24/2020-12-24-RNAseq(13)/" title="RNAseq 教程 (3.2)">https://www.zhouxiaozhao.cn/2020/12/24/2020-12-24-RNAseq(13)/</a></li><li class="license"><strong>版权声明： </strong>本站所有文章除特别声明外，均采用 <span class="exturl" data-url="aHR0cHM6Ly9jcmVhdGl2ZWNvbW1vbnMub3JnL2xpY2Vuc2VzL2J5LW5jLXNhLzQuMC9kZWVkLnpo"><i class="ic i-creative-commons"><em>(CC)</em></i>BY-NC-SA</span> 许可协议。转载请注明出处！</li></ul></div></footer></article></div><div class="post-nav"><div class="item left"><a href="/2020/12/22/2020-12-22-RNAseq(12)/" itemprop="url" rel="prev" data-background-image="https:&#x2F;&#x2F;img1.imgtp.com&#x2F;2023&#x2F;02&#x2F;01&#x2F;UJhRWazi.jpg" title="RNAseq教程(3.1)"><span class="type">上一篇</span> <span class="category"><i class="ic i-flag"></i> 生信学习</span><h3>RNAseq教程(3.1)</h3></a></div><div class="item right"><a href="/2020/12/26/2020-12-26-RNAseq(14)/" itemprop="url" rel="next" data-background-image="https:&#x2F;&#x2F;img1.imgtp.com&#x2F;2023&#x2F;02&#x2F;01&#x2F;bIGA6nyQ.jpg" title="RNAseq教程(3.3)"><span class="type">下一篇</span> <span class="category"><i class="ic i-flag"></i> 生信学习</span><h3>RNAseq教程(3.3)</h3></a></div></div><div class="wrap" id="comments"></div></div><div id="sidebar"><div class="inner"><div class="panels"><div class="inner"><div class="contents panel pjax" data-title="文章目录"><ol class="toc"><li class="toc-item toc-level-2"><a class="toc-link" href="#%E7%9B%AE%E5%BD%95"><span class="toc-number">1.</span> <span class="toc-text">目录</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#32-differential-expression"><span class="toc-number">2.</span> <span class="toc-text">3.2 Differential Expression</span></a><ol class="toc-child"><li class="toc-item toc-level-3"><a class="toc-link" href="#ballgown"><span class="toc-number">2.1.</span> <span class="toc-text">Ballgown</span></a></li><li class="toc-item toc-level-3"><a class="toc-link" href="#edger"><span class="toc-number">2.2.</span> <span class="toc-text">edgeR</span></a></li></ol></li></ol></div><div class="related panel pjax" data-title="系列文章"><ul><li><a href="/2020/07/08/2020-07-08-repeat/" rel="bookmark" title="RepeatModeler+RepeatMasker的安装与使用">RepeatModeler+RepeatMasker的安装与使用</a></li><li><a href="/2020/07/09/2020-07-09-github+jekyll/" rel="bookmark" title="利用Github+Jekyll构建自己的博客">利用Github+Jekyll构建自己的博客</a></li><li><a href="/2020/07/18/2020-07-18-orthofinder/" rel="bookmark" title="利用orthofinder寻找单拷贝基因构建系统发育树">利用orthofinder寻找单拷贝基因构建系统发育树</a></li><li><a href="/2020/07/20/2020-07-20-fungi/" rel="bookmark" title="文献导读：真菌进化">文献导读：真菌进化</a></li><li><a href="/2020/08/03/2020-08-03-WGCNA1/" rel="bookmark" title="WGCNA学习：WGCNA分析原理">WGCNA学习：WGCNA分析原理</a></li><li><a href="/2020/08/05/2020-08-05-WGCNA2/" rel="bookmark" title="WGCNA学习：WGCNA应用场景与实例解读">WGCNA学习：WGCNA应用场景与实例解读</a></li><li><a href="/2020/08/08/2020-08-08-WGCNA3/" rel="bookmark" title="WGCNA学习：WGCNA分析实战">WGCNA学习：WGCNA分析实战</a></li><li><a href="/2020/08/10/2020-08-10-cytoscape/" rel="bookmark" title="基因共表达网络图的绘制">基因共表达网络图的绘制</a></li><li><a href="/2020/08/18/2020-08-18-sys/" rel="bookmark" title="根据转录组数据鉴定共生物种">根据转录组数据鉴定共生物种</a></li><li><a href="/2020/08/20/2020-08-20-nt/" rel="bookmark" title="使用本地nt数据库对reads和Trinity结果进行blast">使用本地nt数据库对reads和Trinity结果进行blast</a></li><li><a href="/2020/08/23/2020-08-23-IGV/" rel="bookmark" title="使用IGV简单绘制甲基化分布图">使用IGV简单绘制甲基化分布图</a></li><li><a href="/2020/08/25/2020-08-25-evolview1/" rel="bookmark" title="进化树美化之Evolview--热图（1）">进化树美化之Evolview--热图（1）</a></li><li><a href="/2020/08/28/2020-08-28-evolview/" rel="bookmark" title="进化树美化之Evolview--热图（2）">进化树美化之Evolview--热图（2）</a></li><li><a href="/2020/08/30/2020-08-30-evolview3/" rel="bookmark" title="进化树+蛋白结构域">进化树+蛋白结构域</a></li><li><a href="/2020/09/01/2020-09-01-iTOL1/" rel="bookmark" title="进化树在线制作工具iTOL教程（1）：基本操作">进化树在线制作工具iTOL教程（1）：基本操作</a></li><li><a href="/2020/09/03/2020-09-03-iTOL2/" rel="bookmark" title="进化树在线制作工具iTOL教程（2）：添加文本">进化树在线制作工具iTOL教程（2）：添加文本</a></li><li><a href="/2020/09/06/2020-09-06-iTOL3/" rel="bookmark" title="进化树在线制作工具iTOL教程（3）：添加图片">进化树在线制作工具iTOL教程（3）：添加图片</a></li><li><a href="/2020/09/08/2020-09-08-iTOL4/" rel="bookmark" title="进化树在线制作工具iTOL教程（4）：添加点图">进化树在线制作工具iTOL教程（4）：添加点图</a></li><li><a href="/2020/09/29/2020-09-29-ScRNAseq(1)/" rel="bookmark" title="单细胞实战(1)：原始测序数据下载">单细胞实战(1)：原始测序数据下载</a></li><li><a href="/2020/10/01/2020-10-01-ScRNAseq(2)/" rel="bookmark" title="单细胞实战(2)：cellranger使用">单细胞实战(2)：cellranger使用</a></li><li><a href="/2020/10/03/2020-10-03-ScRNAseq(3)/" rel="bookmark" title="单细胞实战(3)：STAR分析单细胞数据">单细胞实战(3)：STAR分析单细胞数据</a></li><li><a href="/2020/10/06/2020-10-06-ScRNAseq(4)/" rel="bookmark" title="单细胞实战(4)：STAR与cellranger结果比较">单细胞实战(4)：STAR与cellranger结果比较</a></li><li><a href="/2020/10/10/2020-10-10-ScRNAseq(5)/" rel="bookmark" title="单细胞实战(5)：复现文章中的聚类图（1）">单细胞实战(5)：复现文章中的聚类图（1）</a></li><li><a href="/2020/10/13/2020-10-13-ScRNAseq(6)/" rel="bookmark" title="单细胞实战(5)：复现文章中的聚类图（2）">单细胞实战(5)：复现文章中的聚类图（2）</a></li><li><a href="/2020/10/24/2020-10-24-ScRNAseq(7)/" rel="bookmark" title="生信技能树单细胞数据挖掘笔记(1)">生信技能树单细胞数据挖掘笔记(1)</a></li><li><a href="/2020/10/27/2020-10-27-ScRNAseq(8)/" rel="bookmark" title="生信技能树单细胞数据挖掘笔记(2)">生信技能树单细胞数据挖掘笔记(2)</a></li><li><a href="/2020/10/29/2020-10-29-ScRNAseq(9)/" rel="bookmark" title="生信技能树单细胞数据挖掘笔记(3)">生信技能树单细胞数据挖掘笔记(3)</a></li><li><a href="/2020/10/31/2020-10-31-ScRNAseq(10)/" rel="bookmark" title="生信技能树单细胞数据挖掘笔记(4)">生信技能树单细胞数据挖掘笔记(4)</a></li><li><a href="/2020/11/03/2020-11-03-ScRNAseq(11)/" rel="bookmark" title="生信技能树单细胞数据挖掘笔记(5)">生信技能树单细胞数据挖掘笔记(5)</a></li><li><a href="/2020/11/05/2020-11-05-ScRNAseq(12)/" rel="bookmark" title="生信技能树单细胞数据挖掘笔记(6)">生信技能树单细胞数据挖掘笔记(6)</a></li><li><a href="/2020/11/07/2020-11-07-ScRNAseq(13)/" rel="bookmark" title="生信技能树单细胞数据挖掘笔记(7)">生信技能树单细胞数据挖掘笔记(7)</a></li><li><a href="/2020/11/10/2020-11-10-RNAvelocity(1)/" rel="bookmark" title="RNA速率：软件下载与loom文件准备">RNA速率：软件下载与loom文件准备</a></li><li><a href="/2020/11/12/2020-11-12-RNAvelocity(2)/" rel="bookmark" title="RNA速率：数据读入">RNA速率：数据读入</a></li><li><a href="/2020/11/14/2020-11-14-RNAvelocity(3)/" rel="bookmark" title="RNA速率：使用Seurat的结果做RNA velocity">RNA速率：使用Seurat的结果做RNA velocity</a></li><li><a href="/2020/11/17/2020-11-17-RNAseq(1)/" rel="bookmark" title="RNAseq教程(1.1)">RNAseq教程(1.1)</a></li><li><a href="/2020/11/19/2020-11-19-annotion(1)/" rel="bookmark" title="基因结构注释（1）：从头注释">基因结构注释（1）：从头注释</a></li><li><a href="/2020/11/21/2020-11-21-annotion(2)/" rel="bookmark" title="基因结构注释（2）：同源注释">基因结构注释（2）：同源注释</a></li><li><a href="/2020/11/24/2020-11-24-annotion(3)/" rel="bookmark" title="基因结构注释（3）：转录组注释">基因结构注释（3）：转录组注释</a></li><li><a href="/2020/11/26/2020-11-26-annotion(4)/" rel="bookmark" title="基因结构注释（4）：整合预测结果">基因结构注释（4）：整合预测结果</a></li><li><a href="/2020/11/28/2020-11-28-RNAseq(2)/" rel="bookmark" title="RNAseq教程(1.2)">RNAseq教程(1.2)</a></li><li><a href="/2020/12/01/2020-12-01-RNAseq(3)/" rel="bookmark" title="RNAseq教程(1.3)">RNAseq教程(1.3)</a></li><li><a href="/2020/12/03/2020-12-03-RNAseq(4)/" rel="bookmark" title="RNAseq教程(1.4)">RNAseq教程(1.4)</a></li><li><a href="/2020/12/05/2020-12-05-RNAseq(5)/" rel="bookmark" title="RNAseq教程(1.5)">RNAseq教程(1.5)</a></li><li><a href="/2020/12/08/2020-12-08-RNAseq(6)/" rel="bookmark" title="RNAseq教程(1.6)">RNAseq教程(1.6)</a></li><li><a href="/2020/12/10/2020-12-10-RNAseq(7)/" rel="bookmark" title="RNAseq教程(2.1)">RNAseq教程(2.1)</a></li><li><a href="/2020/12/12/2020-12-12-RNAseq(8)/" rel="bookmark" title="RNAseq教程(2.2)">RNAseq教程(2.2)</a></li><li><a href="/2020/12/15/2020-12-15-RNAseq(9)/" rel="bookmark" title="RNAseq教程(2.3)">RNAseq教程(2.3)</a></li><li><a href="/2020/12/17/2020-12-17-RNAseq(10)/" rel="bookmark" title="RNAseq教程(2.4)">RNAseq教程(2.4)</a></li><li><a href="/2020/12/19/2020-12-19-RNAseq(11)/" rel="bookmark" title="RNAseq教程(2.5)">RNAseq教程(2.5)</a></li><li><a href="/2020/12/22/2020-12-22-RNAseq(12)/" rel="bookmark" title="RNAseq教程(3.1)">RNAseq教程(3.1)</a></li><li class="active"><a href="/2020/12/24/2020-12-24-RNAseq(13)/" rel="bookmark" title="RNAseq教程(3.2)">RNAseq教程(3.2)</a></li><li><a href="/2020/12/26/2020-12-26-RNAseq(14)/" rel="bookmark" title="RNAseq教程(3.3)">RNAseq教程(3.3)</a></li><li><a href="/2020/12/29/2020-12-29-RNAseq(15)/" rel="bookmark" title="RNAseq教程(3.4)">RNAseq教程(3.4)</a></li><li><a href="/2020/12/31/2020-12-31-RNAseq(16)/" rel="bookmark" title="RNAseq教程(4.1)">RNAseq教程(4.1)</a></li><li><a href="/2021/01/02/2021-01-02-RNAseq(17)/" rel="bookmark" title="RNAseq教程(4.2)">RNAseq教程(4.2)</a></li><li><a href="/2021/01/05/2021-01-05-RNAseq(18)/" rel="bookmark" title="RNAseq教程(4.3)">RNAseq教程(4.3)</a></li><li><a href="/2021/01/07/2021-01-07-RNAseq(19)/" rel="bookmark" title="RNAseq教程(4.4)">RNAseq教程(4.4)</a></li><li><a href="/2021/01/09/2021-01-09-RNAseq(20)/" rel="bookmark" title="RNAseq教程(4.5)">RNAseq教程(4.5)</a></li><li><a href="/2021/01/12/2021-01-12-RNAseq(21)/" rel="bookmark" title="RNAseq教程(5)">RNAseq教程(5)</a></li><li><a href="/2022/12/08/2022-12-08-SingleCopy/" rel="bookmark" title="比较基因组学分析1：构建单拷贝基因树">比较基因组学分析1：构建单拷贝基因树</a></li><li><a href="/2022/12/14/2022-12-14-CAFE/" rel="bookmark" title="比较基因组学分析2：基因家族收缩与扩张分析">比较基因组学分析2：基因家族收缩与扩张分析</a></li><li><a href="/2022/12/15/2022-12-15-GO/" rel="bookmark" title="比较基因组学分析3：特异节点基因家族富集分析（非模式物种GO/KEEG库构建）">比较基因组学分析3：特异节点基因家族富集分析（非模式物种GO/KEEG库构建）</a></li><li><a href="/2023/02/22/2023-02-22-ST(1)/" rel="bookmark" title="植物空间转录组分析1：Seurat基本流程">植物空间转录组分析1：Seurat基本流程</a></li><li><a href="/2023/02/24/2023-02-24-ST(2)/" rel="bookmark" title="植物空间转录组分析2：STEEL+Seurat">植物空间转录组分析2：STEEL+Seurat</a></li></ul></div><div class="overview panel" data-title="站点概览"><div class="author" itemprop="author" itemscope itemtype="http://schema.org/Person"><img class="image" itemprop="image" alt="amane" data-src="/images/avatar.jpg"><p class="name" itemprop="name">amane</p><div class="description" itemprop="description">生物学在读博士</div></div><nav class="state"><div class="item posts"><a href="/archives/"><span class="count">84</span> <span class="name">文章</span></a></div><div class="item categories"><a href="/categories/"><span class="count">8</span> <span class="name">分类</span></a></div><div class="item tags"><a href="/tags/"><span class="count">21</span> <span class="name">标签</span></a></div></nav><div class="social"><span class="exturl item github" data-url="aHR0cHM6Ly9naXRodWIuY29tL0Jpb2luZm9ybWF0aWNzLXJvb2tpZQ==" title="https:&#x2F;&#x2F;github.com&#x2F;Bioinformatics-rookie"><i class="ic i-github"></i></span> <span class="exturl item bilibili" data-url="aHR0cHM6Ly9zcGFjZS5iaWxpYmlsaS5jb20vMTU3Nzg1NTM/c3BtX2lkX2Zyb209MzMzLjEwMDcuMC4w" title="https:&#x2F;&#x2F;space.bilibili.com&#x2F;15778553?spm_id_from&#x3D;333.1007.0.0"><i class="ic i-bilibili"></i></span> <a href="https://www.zhouxiaozhao.cn/about/" title="https:&#x2F;&#x2F;www.zhouxiaozhao.cn&#x2F;about&#x2F;" class="item about"><i class="ic i-address-card"></i></a> <span class="exturl item email" data-url="bWFpbHRvOnpob3VzaGl6aGFvQG1haWwua2liLmFjLmNu" title="mailto:zhoushizhao@mail.kib.ac.cn"><i class="ic i-envelope"></i></span></div><ul class="menu"><li class="item"><a href="/" rel="section"><i class="ic i-home"></i>首页</a></li><li class="item"><a href="/about/" rel="section"><i class="ic i-user"></i>关于</a></li><li class="item dropdown"><a href="javascript:void(0);"><i class="ic i-feather"></i>文章</a><ul class="submenu"><li class="item"><a href="/archives/" rel="section"><i class="ic i-list-alt"></i>归档</a></li><li class="item"><a href="/categories/" rel="section"><i class="ic i-th"></i>分类</a></li><li class="item"><a href="/tags/" rel="section"><i class="ic i-tags"></i>标签</a></li></ul></li><li class="item dropdown"><a href="/webstack/" rel="section"><i class="ic i-magic"></i>AI学习</a><ul class="submenu"><li class="item"><a href="/AIpainting/" rel="section"><i class="ic i-pen"></i>AI绘画</a></li><li class="item"><a href="/AIfile/" rel="section"><i class="ic i-file"></i>AI图包</a></li><li class="item"><a href="/AIbookmark/" rel="section"><i class="ic i-comments"></i>AI书签</a></li></ul></li><li class="item"><a href="/links/" rel="section"><i class="ic i-heart"></i>友達</a></li></ul></div></div></div><ul id="quick"><li class="prev pjax"><a href="/2020/12/22/2020-12-22-RNAseq(12)/" rel="prev" title="上一篇"><i class="ic i-chevron-left"></i></a></li><li class="up"><i class="ic i-arrow-up"></i></li><li class="down"><i class="ic i-arrow-down"></i></li><li class="next pjax"><a href="/2020/12/26/2020-12-26-RNAseq(14)/" rel="next" title="下一篇"><i class="ic i-chevron-right"></i></a></li><li class="percent"></li></ul></div></div><div class="dimmer"></div></div></main><footer id="footer"><div class="inner"><div class="widgets"><div class="rpost pjax"><h2>随机文章</h2><ul><li class="item"><div class="breadcrumb"><a href="/categories/Bioinformation/" title="分类于 生信学习">生信学习</a></div><span><a href="/2020/12/15/2020-12-15-RNAseq(9)/" title="RNAseq教程(2.3)">RNAseq教程(2.3)</a></span></li><li class="item"><div class="breadcrumb"><a href="/categories/Bioinformation/" title="分类于 生信学习">生信学习</a></div><span><a href="/2020/12/12/2020-12-12-RNAseq(8)/" title="RNAseq教程(2.2)">RNAseq教程(2.2)</a></span></li><li class="item"><div class="breadcrumb"><a href="/categories/Bioinformation/" title="分类于 生信学习">生信学习</a> <i class="ic i-angle-right"></i> <a href="/categories/Bioinformation/SingleCell/" title="分类于 单细胞">单细胞</a></div><span><a href="/2020/11/10/2020-11-10-RNAvelocity(1)/" title="RNA速率：软件下载与loom文件准备">RNA速率：软件下载与loom文件准备</a></span></li><li class="item"><div class="breadcrumb"><a href="/categories/Bioinformation/" title="分类于 生信学习">生信学习</a> <i class="ic i-angle-right"></i> <a href="/categories/Bioinformation/SingleCell/" title="分类于 单细胞">单细胞</a></div><span><a href="/2020/11/14/2020-11-14-RNAvelocity(3)/" title="RNA速率：使用Seurat的结果做RNA velocity">RNA速率：使用Seurat的结果做RNA velocity</a></span></li><li class="item"><div class="breadcrumb"><a href="/categories/Bioinformation/" title="分类于 生信学习">生信学习</a> <i class="ic i-angle-right"></i> <a href="/categories/Bioinformation/SingleCell/" title="分类于 单细胞">单细胞</a></div><span><a href="/2020/11/07/2020-11-07-ScRNAseq(13)/" title="生信技能树单细胞数据挖掘笔记(7)">生信技能树单细胞数据挖掘笔记(7)</a></span></li><li class="item"><div class="breadcrumb"><a href="/categories/Bioinformation/" title="分类于 生信学习">生信学习</a> <i class="ic i-angle-right"></i> <a href="/categories/Bioinformation/SingleCell/" title="分类于 单细胞">单细胞</a></div><span><a href="/2020/10/10/2020-10-10-ScRNAseq(5)/" title="单细胞实战(5)：复现文章中的聚类图（1）">单细胞实战(5)：复现文章中的聚类图（1）</a></span></li><li class="item"><div class="breadcrumb"><a href="/categories/Bioinformation/" title="分类于 生信学习">生信学习</a> <i class="ic i-angle-right"></i> <a href="/categories/Bioinformation/SingleCell/" title="分类于 单细胞">单细胞</a></div><span><a href="/2020/10/27/2020-10-27-ScRNAseq(8)/" title="生信技能树单细胞数据挖掘笔记(2)">生信技能树单细胞数据挖掘笔记(2)</a></span></li><li class="item"><div class="breadcrumb"><a href="/categories/Bioinformation/" title="分类于 生信学习">生信学习</a> <i class="ic i-angle-right"></i> <a href="/categories/Bioinformation/SingleCell/" title="分类于 单细胞">单细胞</a></div><span><a href="/2020/10/24/2020-10-24-ScRNAseq(7)/" title="生信技能树单细胞数据挖掘笔记(1)">生信技能树单细胞数据挖掘笔记(1)</a></span></li><li class="item"><div class="breadcrumb"><a href="/categories/Bioinformation/" title="分类于 生信学习">生信学习</a></div><span><a href="/2020/08/05/2020-08-05-WGCNA2/" title="WGCNA学习：WGCNA应用场景与实例解读">WGCNA学习：WGCNA应用场景与实例解读</a></span></li><li class="item"><div class="breadcrumb"><a href="/categories/Bioinformation/" title="分类于 生信学习">生信学习</a> <i class="ic i-angle-right"></i> <a href="/categories/Bioinformation/SingleCell/" title="分类于 单细胞">单细胞</a></div><span><a href="/2020/10/29/2020-10-29-ScRNAseq(9)/" title="生信技能树单细胞数据挖掘笔记(3)">生信技能树单细胞数据挖掘笔记(3)</a></span></li></ul></div><div><h2>最新评论</h2><ul class="leancloud-recent-comment"></ul></div></div><div class="status"><div class="copyright">&copy; 2010 – <span itemprop="copyrightYear">2023</span> <span class="with-love"><i class="ic i-sakura rotate"></i> </span><span class="author" itemprop="copyrightHolder">amane @ Zhou Xiaozhao</span></div><div class="count"><span class="post-meta-item-icon"><i class="ic i-chart-area"></i> </span><span title="站点总字数">381k 字</span> <span class="post-meta-divider">|</span> <span class="post-meta-item-icon"><i class="ic i-coffee"></i> </span><span title="站点阅读时长">5:46</span></div><div class="powered-by">基于 <span class="exturl" data-url="aHR0cHM6Ly9oZXhvLmlv">Hexo</span> & Theme.<span class="exturl" data-url="aHR0cHM6Ly9naXRodWIuY29tL2FtZWhpbWUvaGV4by10aGVtZS1zaG9rYQ==">Shoka</span></div></div></div></footer></div><script data-config type="text/javascript">var LOCAL={path:"2020/12/24/2020-12-24-RNAseq(13)/",favicon:{show:"（●´3｀●）やれやれだぜ",hide:"(´Д｀)大変だ！"},search:{placeholder:"文章搜索",empty:"关于 「 ${query} 」，什么也没搜到",stats:"${time} ms 内找到 ${hits} 条结果"},valine:!0,fancybox:!0,copyright:'复制成功，转载请遵守 <i class="ic i-creative-commons"></i>BY-NC-SA 协议。',ignores:[function(e){return e.includes("#")},function(e){return new RegExp(LOCAL.path+"$").test(e)}]}</script><script src="https://cdn.polyfill.io/v2/polyfill.js"></script><script src="//cdn.jsdelivr.net/combine/npm/pace-js@1.0.2/pace.min.js,npm/pjax@0.2.8/pjax.min.js,npm/whatwg-fetch@3.4.0/dist/fetch.umd.min.js,npm/animejs@3.2.0/lib/anime.min.js,npm/algoliasearch@4/dist/algoliasearch-lite.umd.js,npm/instantsearch.js@4/dist/instantsearch.production.min.js,npm/lozad@1/dist/lozad.min.js,npm/quicklink@2/dist/quicklink.umd.js"></script><script src="/js/app.js?v=0.2.5"></script></body></html><!-- rebuild by hrmmi -->