<!-- build time:Fri Jul 07 2023 15:48:10 GMT+0800 (中国标准时间) --><!DOCTYPE html><html lang="zh-CN"><head><meta charset="UTF-8"><meta name="viewport" content="width=device-width,initial-scale=1,maximum-scale=2"><meta name="theme-color" content="#FFF"><link rel="apple-touch-icon" sizes="180x180" href="/images/apple-touch-icon.png"><link rel="icon" type="image/ico" sizes="32x32" href="/images/favicon.ico"><meta http-equiv="Cache-Control" content="no-transform"><meta http-equiv="Cache-Control" content="no-siteapp"><link rel="alternate" type="application/rss+xml" title="小钊の笔记" href="https://www.zhouxiaozhao.cn/rss.xml"><link rel="alternate" type="application/atom+xml" title="小钊の笔记" href="https://www.zhouxiaozhao.cn/atom.xml"><link rel="alternate" type="application/json" title="小钊の笔记" href="https://www.zhouxiaozhao.cn/feed.json"><link rel="stylesheet" href="//fonts.googleapis.com/css?family=Mulish:300,300italic,400,400italic,700,700italic%7CFredericka%20the%20Great:300,300italic,400,400italic,700,700italic%7CNoto%20Serif%20JP:300,300italic,400,400italic,700,700italic%7CNoto%20Serif%20SC:300,300italic,400,400italic,700,700italic%7CInconsolata:300,300italic,400,400italic,700,700italic&display=swap&subset=latin,latin-ext"><link rel="stylesheet" href="/css/app.css?v=0.2.5"><meta name="keywords" content="bioinformatics,Comparative_Genomics"><link rel="canonical" href="https://www.zhouxiaozhao.cn/2022/12/08/2022-12-08-SingleCopy/"><title>比较基因组学分析 1：构建单拷贝基因树 - 基因组 - 生信学习 | Zhou Xiaozhao = 小钊の笔记 = 前天是小兔子，昨天是小鹿，今天是你</title><meta name="generator" content="Hexo 5.4.1"></head><body itemscope itemtype="http://schema.org/WebPage"><div id="loading"><div class="cat"><div class="body"></div><div class="head"><div class="face"></div></div><div class="foot"><div class="tummy-end"></div><div class="bottom"></div><div class="legs left"></div><div class="legs right"></div></div><div class="paw"><div class="hands left"></div><div class="hands right"></div></div></div></div><div id="container"><header id="header" itemscope itemtype="http://schema.org/WPHeader"><div class="inner"><div id="brand"><div class="pjax"><h1 itemprop="name headline">比较基因组学分析 1：构建单拷贝基因树</h1><div class="meta"><span class="item" title="创建时间：2022-12-08 00:00:00"><span class="icon"><i class="ic i-calendar"></i> </span><span class="text">发表于</span> <time itemprop="dateCreated datePublished" datetime="2022-12-08T00:00:00+08:00">2022-12-08</time> </span><span class="item" title="本文字数"><span class="icon"><i class="ic i-pen"></i> </span><span class="text">本文字数</span> <span>7.1k</span> <span class="text">字</span> </span><span class="item" title="阅读时长"><span class="icon"><i class="ic i-clock"></i> </span><span class="text">阅读时长</span> <span>6 分钟</span></span></div></div></div><nav id="nav"><div class="inner"><div class="toggle"><div class="lines" aria-label="切换导航栏"><span class="line"></span> <span class="line"></span> <span class="line"></span></div></div><ul class="menu"><li class="item title"><a href="/" rel="start">Zhou Xiaozhao</a></li></ul><ul class="right"><li class="item theme"><i class="ic i-sun"></i></li><li class="item search"><i class="ic i-search"></i></li></ul></div></nav></div><div id="imgs" class="pjax"><ul><li class="item" data-background-image="https://img1.imgtp.com/2023/02/01/DnAfhAaN.jpg"></li><li class="item" data-background-image="https://img1.imgtp.com/2023/02/01/ItRCVT7v.jpg"></li><li class="item" data-background-image="https://img1.imgtp.com/2023/02/01/Vghr1YDl.jpg"></li><li class="item" data-background-image="https://img1.imgtp.com/2023/02/01/82Elo9NQ.jpg"></li><li class="item" data-background-image="https://img1.imgtp.com/2023/02/01/y8HR7QqH.jpg"></li><li class="item" data-background-image="https://img1.imgtp.com/2023/02/01/szJGZgtF.jpg"></li></ul></div></header><div id="waves"><svg class="waves" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" viewBox="0 24 150 28" preserveAspectRatio="none" shape-rendering="auto"><defs><path id="gentle-wave" d="M-160 44c30 0 58-18 88-18s 58 18 88 18 58-18 88-18 58 18 88 18 v44h-352z"/></defs><g class="parallax"><use xlink:href="#gentle-wave" x="48" y="0"/><use xlink:href="#gentle-wave" x="48" y="3"/><use xlink:href="#gentle-wave" x="48" y="5"/><use xlink:href="#gentle-wave" x="48" y="7"/></g></svg></div><main><div class="inner"><div id="main" class="pjax"><div class="article wrap"><div class="breadcrumb" itemscope itemtype="https://schema.org/BreadcrumbList"><i class="ic i-home"></i> <span><a href="/">首页</a></span><i class="ic i-angle-right"></i> <span itemprop="itemListElement" itemscope itemtype="https://schema.org/ListItem"><a href="/categories/Bioinformation/" itemprop="item" rel="index" title="分类于 生信学习"><span itemprop="name">生信学习</span></a><meta itemprop="position" content="1"></span><i class="ic i-angle-right"></i> <span class="current" itemprop="itemListElement" itemscope itemtype="https://schema.org/ListItem"><a href="/categories/Bioinformation/genome/" itemprop="item" rel="index" title="分类于 基因组"><span itemprop="name">基因组</span></a><meta itemprop="position" content="2"></span></div><article itemscope itemtype="http://schema.org/Article" class="post block" lang="zh-CN"><link itemprop="mainEntityOfPage" href="https://www.zhouxiaozhao.cn/2022/12/08/2022-12-08-SingleCopy/"><span hidden itemprop="author" itemscope itemtype="http://schema.org/Person"><meta itemprop="image" content="/images/avatar.jpg"><meta itemprop="name" content="amane"><meta itemprop="description" content="前天是小兔子，昨天是小鹿，今天是你, 生物学在读博士"></span><span hidden itemprop="publisher" itemscope itemtype="http://schema.org/Organization"><meta itemprop="name" content="小钊の笔记"></span><div class="body md" itemprop="articleBody"><h1 id="前言"><a class="anchor" href="#前言">#</a> 前言</h1><p>之前有写过<a href="https://www.zhouxiaozhao.cn/2020/07/18/2020-07-18-orthofinder/">单拷贝基因构建物种树</a>的流程，现在我对流程进行优化，而且将会增加后续的基因家族收缩与扩张分析，希望对大家的分析有所帮助</p><h1 id="物种选择"><a class="anchor" href="#物种选择">#</a> 物种选择</h1><p>在此我不在介绍软件的原理和安装了，大家可以去看我之前的推文<br>物种选择此次包含了整个绿色植物，从绿藻门到被子植物<br>大部分基因组是在 jgi phytozome 上下载的<br>绿藻门 Chlorophyte：小球藻：Coccomyxa subellipsoidea (Csu), 团藻：Volvox carteri (Vca)<br>轮藻门 Charophyta：布氏轮藻：Chara braunii (Cba), Klebsormidium nitens (Kle)<br>苔藓植物门 Bryophyta：地钱：Marchantia polymorpha (Mpo), 小立碗藓：Physcomitrium patens (Ppa)<br>石松类 Lycopsida： 江南卷柏：Selaginella moellendorffii (Smo)<br>蕨类 Ferns： 水蕨：Ceratopteris richardii (Cri)<br>裸子植物门 Gymnospermae：北美乔柏 Thuja plicata (Tpl)<br>基部被子植物 Basial Angiosperm：无油樟：Amborella trichopoda (Atr), 睡莲：Nymphaea colorata (Nco)<br>双子叶植物 Eudicot：葡萄：Vitis vinifera (Vvi), 拟南芥：Arabidopsis thaliana (Ath), 黄瓜：Cucumis sativus (Csa)<br>单子叶植物 Monocot: 芦笋：Asparagus officinalis (Aof), 水稻：Oryza sativa (Osa), 玉米：Zea mays (Zma)<br>下载完成后我对每个蛋白序列的 ID 前面加上了物种简写，方便后续使用</p><pre><code class="language-bash">sed -i &quot;s/&gt;/&gt;Ath/g&quot; Ath.pro.fasta
</code></pre><h1 id="orthofinder获得单拷贝基因"><a class="anchor" href="#orthofinder获得单拷贝基因">#</a> Orthofinder 获得单拷贝基因</h1><p>将所有物种的蛋白文件放到 CGA 文件夹下</p><figure class="highlight raw"><figcaption data-lang><span>命令行提示符</span></figcaption><table><tr><td data-num="1"></td><td><pre>orthofinder -t 16 -f CGA&#x2F;</pre></td></tr></table></figure><p>我们主要使用 Orthoogroups 查看正交群的基因和使用 Single_Copy_Orthologue_Sequences 里的单拷贝基因构建系统发育树</p><p>WorkingDirectory 其中包含运算过程的中间文件，例如 diamond 结果，如果我们想去掉某一物种，在 SpeciesIDs.txt 中将该物种注释掉</p><figure class="highlight raw"><figcaption data-lang><span>命令行提示符</span></figcaption><table><tr><td data-num="1"></td><td><pre>0: Aof.pro.fasta</pre></td></tr><tr><td data-num="2"></td><td><pre>1: Ath.pro.fasta</pre></td></tr><tr><td data-num="3"></td><td><pre>2: Atr.pro.fasta</pre></td></tr><tr><td data-num="4"></td><td><pre>3: Cba.pro.fasta</pre></td></tr><tr><td data-num="5"></td><td><pre>4: Cri.pro.fasta</pre></td></tr><tr><td data-num="6"></td><td><pre>5: Csa.pro.fasta</pre></td></tr><tr><td data-num="7"></td><td><pre>6: Csu.pro.fasta</pre></td></tr><tr><td data-num="8"></td><td><pre>7: Kle.pro.fasta</pre></td></tr><tr><td data-num="9"></td><td><pre>8: Mpo.pro.fasta</pre></td></tr><tr><td data-num="10"></td><td><pre>9: Nco.pro.fasta</pre></td></tr><tr><td data-num="11"></td><td><pre>10: Osa.pro.fasta</pre></td></tr><tr><td data-num="12"></td><td><pre>11: Ppa.pro.fasta</pre></td></tr><tr><td data-num="13"></td><td><pre>#12: Pum.pro.fasta</pre></td></tr><tr><td data-num="14"></td><td><pre>13: Smo.pro.fasta</pre></td></tr><tr><td data-num="15"></td><td><pre>14: Tpl.pro.fasta</pre></td></tr><tr><td data-num="16"></td><td><pre>15: Vca.pro.fasta</pre></td></tr><tr><td data-num="17"></td><td><pre>16: Vvi.pro.fasta</pre></td></tr><tr><td data-num="18"></td><td><pre>17: Zma.pro.fasta</pre></td></tr></table></figure><figure class="highlight raw"><figcaption data-lang><span>命令行提示符</span></figcaption><table><tr><td data-num="1"></td><td><pre>orthofinder -b WorkingDirectory</pre></td></tr></table></figure><p>如果想增加额外的物种进行分析，可以按照如下方式运行</p><figure class="highlight raw"><figcaption data-lang><span>命令行提示符</span></figcaption><table><tr><td data-num="1"></td><td><pre>orthofinder -b WorkingDirectory -f new_fasta_directory</pre></td></tr></table></figure><h1 id="利用单拷贝基因构建系统发育树"><a class="anchor" href="#利用单拷贝基因构建系统发育树">#</a> 利用单拷贝基因构建系统发育树</h1><p>Orthofinder 的 Single_Copy_Orthologue_Sequences 下存放着单拷贝同源基因的序列，我们可以利用这些序列构建系统发育树<br>建树方法可以分为串联法与并联法<br>不同的是，<span class="pink">串联法是将得到的单拷贝同源基因比对后进行了串联，每个物种都得到一个很大的序列，然后进行建树；并联法是将每个单拷贝同源基因集比对后建树，然后再利用 Astral 构建了物种树</span>，这里目前只讲串联法，并联法正在研究</p><h2 id="串联法建树"><a class="anchor" href="#串联法建树">#</a> 串联法建树</h2><figure class="highlight raw"><figcaption data-lang><span>命令行提示符</span></figcaption><table><tr><td data-num="1"></td><td><pre>vi test.sh</pre></td></tr><tr><td data-num="2"></td><td><pre></pre></td></tr><tr><td data-num="3"></td><td><pre>for i in *.fa</pre></td></tr><tr><td data-num="4"></td><td><pre>do</pre></td></tr><tr><td data-num="5"></td><td><pre>    file&#x3D;$&#123;i%.fa*&#125;</pre></td></tr><tr><td data-num="6"></td><td><pre>    mafft --maxiterate 1000 --localpair &quot;$&#123;file&#125;.fa&quot; &gt; &quot;$&#123;file&#125;.aln&quot;        ## 多序列比对</pre></td></tr><tr><td data-num="7"></td><td><pre>   (也可以添加一步，将蛋白文件转为 cds 文件再进行后续的分析，pal2nal.pl &quot;$&#123;file&#125;.aln&quot; &quot;$&#123;file&#125;.cds&quot; -output fasta &gt; &quot;$&#123;file&#125;.cds.aln)</pre></td></tr><tr><td data-num="8"></td><td><pre>    Gblocks &quot;$&#123;file&#125;.aln&quot; -b4&#x3D;5 -b5&#x3D;h -t&#x3D;p -e&#x3D;.gb          ## 提取保守序列</pre></td></tr><tr><td data-num="9"></td><td><pre>    seqkit seq &quot;$&#123;file&#125;.aln.gb&quot; -w 0 &gt; &quot;$&#123;file&#125;.new.aln&quot;           ## 多行序列并为一行</pre></td></tr><tr><td data-num="10"></td><td><pre>    awk &#39;&#123;if (NR%2&#x3D;&#x3D;1) print substr ($1, 1, 4); else print $0&#125;&#39; &quot;$&#123;file&#125;.new.aln&quot; &gt; &quot;$&#123;file&#125;.final.aln&quot; ## ID 修饰</pre></td></tr><tr><td data-num="11"></td><td><pre>done</pre></td></tr></table></figure><p>这一步需要的软件 (mafft,Gblocks,seqkit) 请自行安装</p><p>接下来将所有的 aln 文件串联</p><figure class="highlight raw"><figcaption data-lang><span>命令行提示符</span></figcaption><table><tr><td data-num="1"></td><td><pre>seqkit concat *.final.aln &gt; all.fa</pre></td></tr></table></figure><p>将所有单拷贝基因串联在一起后，进行模型预测，这次使用 modeltestng 进行预测</p><figure class="highlight raw"><figcaption data-lang><span>命令行提示符</span></figcaption><table><tr><td data-num="1"></td><td><pre>modeltest-ng -i all.fa -d aa -o modeltest -p 16</pre></td></tr></table></figure><p>在 modeltest.out 文件中，我们可以看到我们需要的模型并且可以直接 copy 命令<br><img data-src="/img/posts/2022.12.8/1.png" alt="modeltest输出结果"></p><p>接下来进行建树</p><figure class="highlight raw"><figcaption data-lang><span>命令行提示符</span></figcaption><table><tr><td data-num="1"></td><td><pre>raxmlHPC-PTHREADS-SSE3 -T 32 -f a -x 123 -p 123 -N 10000 -m PROTGAMMAILGF -k -O -o Csu,Vca -n all.tre -s all.fa</pre></td></tr></table></figure><p>得到<span class="label primary"> RAxML_bipartitions.all.tre</span> 文件查看树结构<br><img data-src="/img/posts/2022.12.8/2.png" alt="树结构"><br>结果和进化关系是一致的</p><h2 id="并联法建树"><a class="anchor" href="#并联法建树">#</a> 并联法建树</h2><p>并联法建树使用 Astral，将所有的单拷贝基因树合并为一颗物种树</p><figure class="highlight raw"><figcaption data-lang><span>命令行提示符</span></figcaption><table><tr><td data-num="1"></td><td><pre>orthDir&#x3D;~&#x2F;CGApaper&#x2F;protein&#x2F;OrthoFinder&#x2F;Results_May02   #基于 Orthofinder 结果</pre></td></tr><tr><td data-num="2"></td><td><pre></pre></td></tr><tr><td data-num="3"></td><td><pre>cat  $orthDir&#x2F;Orthogroups&#x2F;Orthogroups_SingleCopyOrthologues.txt |while read aa; do cat  $orthDir&#x2F;Gene_Trees&#x2F;$aa\_tree.txt |awk &#39;&#123;print $0&#125;&#39;   ;done &gt; SingleCopy.trees</pre></td></tr><tr><td data-num="4"></td><td><pre></pre></td></tr><tr><td data-num="5"></td><td><pre>sed -r  &#39;s&#x2F;([(,]&#123;1&#125;[A-Za-z]+)_[^:]+&#x2F;\1&#x2F;g&#39; SingleCopy.trees &gt; Astral_input.trees</pre></td></tr><tr><td data-num="6"></td><td><pre></pre></td></tr><tr><td data-num="7"></td><td><pre>java -jar   ~&#x2F;soft&#x2F;ASTRAL-5.7.1&#x2F;astral.5.7.1.jar  -i  Astral_input.trees  -o Astral_output.tree -t 8</pre></td></tr></table></figure><p>最后得到 Astral_output.tree，里面包含了不同拓扑结构的可能性，目前先到这一步</p><figure class="highlight raw"><figcaption data-lang><span>命令行提示符</span></figcaption><table><tr><td data-num="1"></td><td><pre>(Csu,(Vca,(Kle,(Cba,((Smo,(Cri,(Tpl,((Atr,Nco)&#39;[q1&#x3D;0.56;q2&#x3D;0.24;q3&#x3D;0.21]&#39;:0.3957605767189867,((Ath,(Vvi,Csa)&#39;[q1&#x3D;0.59;q2&#x3D;0.24;q3&#x3D;0.17]&#39;:0.47671301702525964)&#39;[q1&#x3D;0.9;q2&#x3D;0.06;q3&#x3D;0.04]&#39;:1.82319624265162,(Aof,(Zma,Osa)&#39;[q1&#x3D;0.98;q2&#x3D;0.01;q3&#x3D;0.01]&#39;:3.3756896221149395)&#39;[q1&#x3D;0.73;q2&#x3D;0.11;q3&#x3D;0.16]&#39;:0.8957898742947543)&#39;[q1&#x3D;0.59;q2&#x3D;0.13;q3&#x3D;0.28]&#39;:0.4793457920367108)&#39;[q1&#x3D;0.95;q2&#x3D;0.01;q3&#x3D;0.04]&#39;:2.471696418051194)&#39;[q1&#x3D;0.92;q2&#x3D;0.02;q3&#x3D;0.06]&#39;:2.040688555382602)&#39;[q1&#x3D;0.62;q2&#x3D;0.18;q3&#x3D;0.2]&#39;:0.5471348976356571)&#39;[q1&#x3D;0.37;q2&#x3D;0.32;q3&#x3D;0.32]&#39;:0.04487496622879318,(Ppa,Mpo)&#39;[q1&#x3D;0.74;q2&#x3D;0.15;q3&#x3D;0.1]&#39;:0.9286981407683164)&#39;[q1&#x3D;0.86;q2&#x3D;0.07;q3&#x3D;0.07]&#39;:1.5475625087160123)&#39;[q1&#x3D;0.56;q2&#x3D;0.21;q3&#x3D;0.23]&#39;:0.4056697116894409)&#39;[q1&#x3D;0.98;q2&#x3D;0.02;q3&#x3D;0.01]&#39;:3.121909101338843):0.0);</pre></td></tr></table></figure><h1 id="物种分化时间计算"><a class="anchor" href="#物种分化时间计算">#</a> 物种分化时间计算</h1><p>此次使用 mcmctree 计算物种的分化时间，首先在 timetree 上查询物种的分化时间进行标定，修改树文件，需要注意在标定时间是尽量较早的节点和较晚的节点都进行标定</p><figure class="highlight raw"><figcaption data-lang><span>命令行提示符</span></figcaption><table><tr><td data-num="1"></td><td><pre>17      1</pre></td></tr><tr><td data-num="2"></td><td><pre>((Kle,(((Mpo,Ppa),((Cri,(Tpl,(Atr,(Nco,(((Osa,Zma)&#39;&gt;42&lt;52&#39;,Aof)&#39;&gt;104&lt;125&#39;,(Vvi,(Csa,Ath))&#39;&gt;98&lt;117&#39;)&#39;&gt;85&lt;128&#39;)&#39;&gt;173&lt;199&#39;)&#39;&gt;136&gt;247&#39;)&#39;&gt;289&lt;330&#39;)&#39;&gt;392&lt;422&#39;,Smo)&#39;410&lt;468&#39;)&#39;&gt;472&lt;515&#39;,Cba)),(Csu,Vca)&#39;&gt;800&lt;1000&#39;)&#39;&gt;725&lt;1200&#39;;</pre></td></tr></table></figure><p>配置文件书写</p><figure class="highlight raw"><figcaption data-lang><span>命令行提示符</span></figcaption><table><tr><td data-num="1"></td><td><pre>cp ~&#x2F;soft&#x2F;paml4.9i&#x2F;mcmctree.ctl mcmctree1.ctl</pre></td></tr><tr><td data-num="2"></td><td><pre>vi mcmctree1.ctl</pre></td></tr><tr><td data-num="3"></td><td><pre></pre></td></tr><tr><td data-num="4"></td><td><pre>          seed &#x3D; -1</pre></td></tr><tr><td data-num="5"></td><td><pre>       seqfile &#x3D; all.fa</pre></td></tr><tr><td data-num="6"></td><td><pre>      treefile &#x3D; all.tre</pre></td></tr><tr><td data-num="7"></td><td><pre>       outfile &#x3D; out</pre></td></tr><tr><td data-num="8"></td><td><pre></pre></td></tr><tr><td data-num="9"></td><td><pre>         ndata &#x3D; 1 ## 必须注意为 1 </pre></td></tr><tr><td data-num="10"></td><td><pre>       seqtype &#x3D; 2  * 0: nucleotides; 1:codons; 2:AAs ## 序列格式，蛋白就选择 AA</pre></td></tr><tr><td data-num="11"></td><td><pre>       usedata &#x3D; 3    * 0: no data; 1:seq like; 2:use in.BV; 3: out.BV ## 此步骤先设为 3</pre></td></tr><tr><td data-num="12"></td><td><pre>         clock &#x3D; 3    * 1: global clock; 2: independent rates; 3: correlated rates</pre></td></tr><tr><td data-num="13"></td><td><pre>       RootAge &#x3D; &lt;1.0  * safe constraint on root age, used if no fossil for root.</pre></td></tr><tr><td data-num="14"></td><td><pre></pre></td></tr><tr><td data-num="15"></td><td><pre>         model &#x3D; 0    * 0:JC69, 1:K80, 2:F81, 3:F84, 4:HKY85</pre></td></tr><tr><td data-num="16"></td><td><pre>         alpha &#x3D; 0    * alpha for gamma rates at sites</pre></td></tr><tr><td data-num="17"></td><td><pre>         ncatG &#x3D; 5    * No. categories in discrete gamma</pre></td></tr><tr><td data-num="18"></td><td><pre></pre></td></tr><tr><td data-num="19"></td><td><pre>     cleandata &#x3D; 0    * remove sites with ambiguity data (1:yes, 0:no)?</pre></td></tr><tr><td data-num="20"></td><td><pre></pre></td></tr><tr><td data-num="21"></td><td><pre>       BDparas &#x3D; 1 1 0    * birth, death, sampling</pre></td></tr><tr><td data-num="22"></td><td><pre>   kappa_gamma &#x3D; 6 2      * gamma prior for kappa</pre></td></tr><tr><td data-num="23"></td><td><pre>   alpha_gamma &#x3D; 1 1      * gamma prior for alpha</pre></td></tr><tr><td data-num="24"></td><td><pre></pre></td></tr><tr><td data-num="25"></td><td><pre>   rgene_gamma &#x3D; 2 2   * gamma prior for overall rates for genes</pre></td></tr><tr><td data-num="26"></td><td><pre>  sigma2_gamma &#x3D; 1 10    * gamma prior for sigma^2     (for clock&#x3D;2 or 3)</pre></td></tr><tr><td data-num="27"></td><td><pre></pre></td></tr><tr><td data-num="28"></td><td><pre>      finetune &#x3D; 1: 0.1  0.1  0.1  0.01 .5  * auto (0 or 1) : times, musigma2, rates, mixing, paras, FossilErr</pre></td></tr><tr><td data-num="29"></td><td><pre></pre></td></tr><tr><td data-num="30"></td><td><pre>         print &#x3D; 1</pre></td></tr><tr><td data-num="31"></td><td><pre>        burnin &#x3D; 20000</pre></td></tr><tr><td data-num="32"></td><td><pre>      sampfreq &#x3D; 2</pre></td></tr><tr><td data-num="33"></td><td><pre>       nsample &#x3D; 100000</pre></td></tr><tr><td data-num="34"></td><td><pre></pre></td></tr><tr><td data-num="35"></td><td><pre>*** Note: Make your window wider (100 columns) before running the program.</pre></td></tr></table></figure><p>之后运行</p><figure class="highlight raw"><figcaption data-lang><span>命令行提示符</span></figcaption><table><tr><td data-num="1"></td><td><pre>mcmctree mcmctree.ctl</pre></td></tr></table></figure><p><span class="exturl" data-url="aHR0cDovL3huLS1vdXQtcDE4ZG9ocGR6OWFodHAwOGFpNTdibzFlODkwY3RtMGE4eTFhdHR4LkJW">运行结果产生一个文件名为 out.BV</span>，我们拷贝到两次分化时间估计的目录里，<span class="exturl" data-url="aHR0cDovL3huLS1pbi0xNTJkMjM0Zi5CVg==">并用 in.BV</span> 作为新的命名</p><figure class="highlight raw"><figcaption data-lang><span>命令行提示符</span></figcaption><table><tr><td data-num="1"></td><td><pre>cp out.BV ..&#x2F;02.Time&#x2F;rep1&#x2F;in.BV</pre></td></tr><tr><td data-num="2"></td><td><pre>cp out.BV ..&#x2F;02.Time&#x2F;rep2&#x2F;in.BV</pre></td></tr></table></figure><p>之后在运行两次重复<br>首先修改配置文件</p><figure class="highlight raw"><figcaption data-lang><span>命令行提示符</span></figcaption><table><tr><td data-num="1"></td><td><pre>seed &#x3D; -1</pre></td></tr><tr><td data-num="2"></td><td><pre>       seqfile &#x3D; all.fa</pre></td></tr><tr><td data-num="3"></td><td><pre>      treefile &#x3D; all.tre</pre></td></tr><tr><td data-num="4"></td><td><pre>       outfile &#x3D; out</pre></td></tr><tr><td data-num="5"></td><td><pre></pre></td></tr><tr><td data-num="6"></td><td><pre>         ndata &#x3D; 1</pre></td></tr><tr><td data-num="7"></td><td><pre>       seqtype &#x3D; 2  * 0: nucleotides; 1:codons; 2:AAs</pre></td></tr><tr><td data-num="8"></td><td><pre>       usedata &#x3D; 2    * 0: no data; 1:seq like; 2:use in.BV; 3: out.BV</pre></td></tr><tr><td data-num="9"></td><td><pre>         clock &#x3D; 3    * 1: global clock; 2: independent rates; 3: correlated rates</pre></td></tr><tr><td data-num="10"></td><td><pre>       RootAge &#x3D; &lt;1.0  * safe constraint on root age, used if no fossil for root.</pre></td></tr><tr><td data-num="11"></td><td><pre></pre></td></tr><tr><td data-num="12"></td><td><pre>         model &#x3D; 0    * 0:JC69, 1:K80, 2:F81, 3:F84, 4:HKY85</pre></td></tr><tr><td data-num="13"></td><td><pre>         alpha &#x3D; 0    * alpha for gamma rates at sites</pre></td></tr><tr><td data-num="14"></td><td><pre>         ncatG &#x3D; 5    * No. categories in discrete gamma</pre></td></tr><tr><td data-num="15"></td><td><pre></pre></td></tr><tr><td data-num="16"></td><td><pre>     cleandata &#x3D; 0    * remove sites with ambiguity data (1:yes, 0:no)?</pre></td></tr><tr><td data-num="17"></td><td><pre></pre></td></tr><tr><td data-num="18"></td><td><pre>       BDparas &#x3D; 1 1 0    * birth, death, sampling</pre></td></tr><tr><td data-num="19"></td><td><pre>   kappa_gamma &#x3D; 6 2      * gamma prior for kappa</pre></td></tr><tr><td data-num="20"></td><td><pre>   alpha_gamma &#x3D; 1 1      * gamma prior for alpha</pre></td></tr><tr><td data-num="21"></td><td><pre></pre></td></tr><tr><td data-num="22"></td><td><pre>   rgene_gamma &#x3D; 2 2   * gamma prior for overall rates for genes</pre></td></tr><tr><td data-num="23"></td><td><pre>  sigma2_gamma &#x3D; 1 10    * gamma prior for sigma^2     (for clock&#x3D;2 or 3)</pre></td></tr><tr><td data-num="24"></td><td><pre></pre></td></tr><tr><td data-num="25"></td><td><pre>      finetune &#x3D; 1: 0.1  0.1  0.1  0.01 .5  * auto (0 or 1) : times, musigma2, rates, mixing, paras, FossilErr</pre></td></tr><tr><td data-num="26"></td><td><pre></pre></td></tr><tr><td data-num="27"></td><td><pre>         print &#x3D; 1</pre></td></tr><tr><td data-num="28"></td><td><pre>        burnin &#x3D; 800000</pre></td></tr><tr><td data-num="29"></td><td><pre>      sampfreq &#x3D; 40</pre></td></tr><tr><td data-num="30"></td><td><pre>       nsample &#x3D; 6500000</pre></td></tr><tr><td data-num="31"></td><td><pre></pre></td></tr><tr><td data-num="32"></td><td><pre>*** Note: Make your window wider (100 columns) before running the program.</pre></td></tr></table></figure><p>最后我们会得到如下文件</p><figure class="highlight raw"><figcaption data-lang><span>命令行提示符</span></figcaption><table><tr><td data-num="1"></td><td><pre>all.fa  all.phy  all.tre  FigTree.tre  in.BV  mcmctree.ctl  mcmc.txt  out  SeedUsed</pre></td></tr></table></figure><p>检查两次结果的 FigTree.tre 文件，如果相差不大就可以使用<br><img data-src="/img/posts/2022.12.8/3.png" alt="分化时间"></p><p>之后可以进行基因家族的收缩与扩张分析了，这部分之后在写</p><div class="tags"><a href="/tags/bioinformatics/" rel="tag"><i class="ic i-tag"></i> bioinformatics</a> <a href="/tags/Comparative-Genomics/" rel="tag"><i class="ic i-tag"></i> Comparative_Genomics</a></div></div><footer><div class="meta"><span class="item"><span class="icon"><i class="ic i-calendar-check"></i> </span><span class="text">更新于</span> <time title="修改时间：2023-03-08 15:10:47" itemprop="dateModified" datetime="2023-03-08T15:10:47+08:00">2023-03-08</time> </span><span id="2022/12/08/2022-12-08-SingleCopy/" class="item leancloud_visitors" data-flag-title="比较基因组学分析 1：构建单拷贝基因树" title="阅读次数"><span class="icon"><i class="ic i-eye"></i> </span><span class="text">阅读次数</span> <span class="leancloud-visitors-count"></span> <span class="text">次</span></span></div><div class="reward"><button><i class="ic i-heartbeat"></i> 赞赏</button><p>请我喝[茶]~(￣▽￣)~*</p><div id="qr"><div><img data-src="/images/wechatpay.png" alt="amane 微信支付"><p>微信支付</p></div><div><img data-src="/images/alipay.png" alt="amane 支付宝"><p>支付宝</p></div></div></div><div id="copyright"><ul><li class="author"><strong>本文作者： </strong>amane <i class="ic i-at"><em>@</em></i>小钊の笔记</li><li class="link"><strong>本文链接：</strong> <a href="https://www.zhouxiaozhao.cn/2022/12/08/2022-12-08-SingleCopy/" title="比较基因组学分析 1：构建单拷贝基因树">https://www.zhouxiaozhao.cn/2022/12/08/2022-12-08-SingleCopy/</a></li><li class="license"><strong>版权声明： </strong>本站所有文章除特别声明外，均采用 <span class="exturl" data-url="aHR0cHM6Ly9jcmVhdGl2ZWNvbW1vbnMub3JnL2xpY2Vuc2VzL2J5LW5jLXNhLzQuMC9kZWVkLnpo"><i class="ic i-creative-commons"><em>(CC)</em></i>BY-NC-SA</span> 许可协议。转载请注明出处！</li></ul></div></footer></article></div><div class="post-nav"><div class="item left"><a href="/2021/01/12/2021-01-12-RNAseq(21)/" itemprop="url" rel="prev" data-background-image="https:&#x2F;&#x2F;img1.imgtp.com&#x2F;2023&#x2F;02&#x2F;01&#x2F;piDPxYxl.jpg" title="RNAseq教程(5)"><span class="type">上一篇</span> <span class="category"><i class="ic i-flag"></i> 生信学习</span><h3>RNAseq教程(5)</h3></a></div><div class="item right"><a href="/2022/12/14/2022-12-14-CAFE/" itemprop="url" rel="next" data-background-image="https:&#x2F;&#x2F;img1.imgtp.com&#x2F;2023&#x2F;02&#x2F;01&#x2F;RJVmoMVH.jpg" title="比较基因组学分析2：基因家族收缩与扩张分析"><span class="type">下一篇</span> <span class="category"><i class="ic i-flag"></i> 基因组</span><h3>比较基因组学分析2：基因家族收缩与扩张分析</h3></a></div></div><div class="wrap" id="comments"></div></div><div id="sidebar"><div class="inner"><div class="panels"><div class="inner"><div class="contents panel pjax" data-title="文章目录"><ol class="toc"><li class="toc-item toc-level-1"><a class="toc-link" href="#%E5%89%8D%E8%A8%80"><span class="toc-number">1.</span> <span class="toc-text">前言</span></a></li><li class="toc-item toc-level-1"><a class="toc-link" href="#%E7%89%A9%E7%A7%8D%E9%80%89%E6%8B%A9"><span class="toc-number">2.</span> <span class="toc-text">物种选择</span></a></li><li class="toc-item toc-level-1"><a class="toc-link" href="#orthofinder%E8%8E%B7%E5%BE%97%E5%8D%95%E6%8B%B7%E8%B4%9D%E5%9F%BA%E5%9B%A0"><span class="toc-number">3.</span> <span class="toc-text">Orthofinder 获得单拷贝基因</span></a></li><li class="toc-item toc-level-1"><a class="toc-link" href="#%E5%88%A9%E7%94%A8%E5%8D%95%E6%8B%B7%E8%B4%9D%E5%9F%BA%E5%9B%A0%E6%9E%84%E5%BB%BA%E7%B3%BB%E7%BB%9F%E5%8F%91%E8%82%B2%E6%A0%91"><span class="toc-number">4.</span> <span class="toc-text">利用单拷贝基因构建系统发育树</span></a><ol class="toc-child"><li class="toc-item toc-level-2"><a class="toc-link" href="#%E4%B8%B2%E8%81%94%E6%B3%95%E5%BB%BA%E6%A0%91"><span class="toc-number">4.1.</span> <span class="toc-text">串联法建树</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#%E5%B9%B6%E8%81%94%E6%B3%95%E5%BB%BA%E6%A0%91"><span class="toc-number">4.2.</span> <span class="toc-text">并联法建树</span></a></li></ol></li><li class="toc-item toc-level-1"><a class="toc-link" href="#%E7%89%A9%E7%A7%8D%E5%88%86%E5%8C%96%E6%97%B6%E9%97%B4%E8%AE%A1%E7%AE%97"><span class="toc-number">5.</span> <span class="toc-text">物种分化时间计算</span></a></li></ol></div><div class="related panel pjax" data-title="系列文章"><ul><li><a href="/2020/07/08/2020-07-08-repeat/" rel="bookmark" title="RepeatModeler+RepeatMasker的安装与使用">RepeatModeler+RepeatMasker的安装与使用</a></li><li><a href="/2020/07/18/2020-07-18-orthofinder/" rel="bookmark" title="利用orthofinder寻找单拷贝基因构建系统发育树">利用orthofinder寻找单拷贝基因构建系统发育树</a></li><li class="active"><a href="/2022/12/08/2022-12-08-SingleCopy/" rel="bookmark" title="比较基因组学分析1：构建单拷贝基因树">比较基因组学分析1：构建单拷贝基因树</a></li><li><a href="/2022/12/14/2022-12-14-CAFE/" rel="bookmark" title="比较基因组学分析2：基因家族收缩与扩张分析">比较基因组学分析2：基因家族收缩与扩张分析</a></li><li><a href="/2022/12/15/2022-12-15-GO/" rel="bookmark" title="比较基因组学分析3：特异节点基因家族富集分析（非模式物种GO/KEEG库构建）">比较基因组学分析3：特异节点基因家族富集分析（非模式物种GO/KEEG库构建）</a></li></ul></div><div class="overview panel" data-title="站点概览"><div class="author" itemprop="author" itemscope itemtype="http://schema.org/Person"><img class="image" itemprop="image" alt="amane" data-src="/images/avatar.jpg"><p class="name" itemprop="name">amane</p><div class="description" itemprop="description">生物学在读博士</div></div><nav class="state"><div class="item posts"><a href="/archives/"><span class="count">85</span> <span class="name">文章</span></a></div><div class="item categories"><a href="/categories/"><span class="count">8</span> <span class="name">分类</span></a></div><div class="item tags"><a href="/tags/"><span class="count">22</span> <span class="name">标签</span></a></div></nav><div class="social"><span class="exturl item github" data-url="aHR0cHM6Ly9naXRodWIuY29tL0Jpb2luZm9ybWF0aWNzLXJvb2tpZQ==" title="https:&#x2F;&#x2F;github.com&#x2F;Bioinformatics-rookie"><i class="ic i-github"></i></span> <span class="exturl item bilibili" data-url="aHR0cHM6Ly9zcGFjZS5iaWxpYmlsaS5jb20vMTU3Nzg1NTM/c3BtX2lkX2Zyb209MzMzLjEwMDcuMC4w" title="https:&#x2F;&#x2F;space.bilibili.com&#x2F;15778553?spm_id_from&#x3D;333.1007.0.0"><i class="ic i-bilibili"></i></span> <a href="https://www.zhouxiaozhao.cn/about/" title="https:&#x2F;&#x2F;www.zhouxiaozhao.cn&#x2F;about&#x2F;" class="item about"><i class="ic i-address-card"></i></a> <span class="exturl item email" data-url="bWFpbHRvOnpob3VzaGl6aGFvQG1haWwua2liLmFjLmNu" title="mailto:zhoushizhao@mail.kib.ac.cn"><i class="ic i-envelope"></i></span></div><ul class="menu"><li class="item"><a href="/" rel="section"><i class="ic i-home"></i>首页</a></li><li class="item"><a href="/about/" rel="section"><i class="ic i-user"></i>关于</a></li><li class="item dropdown"><a href="javascript:void(0);"><i class="ic i-feather"></i>文章</a><ul class="submenu"><li class="item"><a href="/archives/" rel="section"><i class="ic i-list-alt"></i>归档</a></li><li class="item"><a href="/categories/" rel="section"><i class="ic i-th"></i>分类</a></li><li class="item"><a href="/tags/" rel="section"><i class="ic i-tags"></i>标签</a></li></ul></li><li class="item dropdown"><a href="/webstack/" rel="section"><i class="ic i-magic"></i>AI学习</a><ul class="submenu"><li class="item"><a href="/AIpainting/" rel="section"><i class="ic i-pen"></i>AI绘画</a></li><li class="item"><a href="/AIfile/" rel="section"><i class="ic i-file"></i>AI图包</a></li><li class="item"><a href="/AIbookmark/" rel="section"><i class="ic i-comments"></i>AI书签</a></li></ul></li><li class="item"><a href="/links/" rel="section"><i class="ic i-heart"></i>友達</a></li></ul></div></div></div><ul id="quick"><li class="prev pjax"><a href="/2021/01/12/2021-01-12-RNAseq(21)/" rel="prev" title="上一篇"><i class="ic i-chevron-left"></i></a></li><li class="up"><i class="ic i-arrow-up"></i></li><li class="down"><i class="ic i-arrow-down"></i></li><li class="next pjax"><a href="/2022/12/14/2022-12-14-CAFE/" rel="next" title="下一篇"><i class="ic i-chevron-right"></i></a></li><li class="percent"></li></ul></div></div><div class="dimmer"></div></div></main><footer id="footer"><div class="inner"><div class="widgets"><div class="rpost pjax"><h2>随机文章</h2><ul><li class="item"><div class="breadcrumb"><a href="/categories/Bioinformation/" title="分类于 生信学习">生信学习</a> <i class="ic i-angle-right"></i> <a href="/categories/Bioinformation/SingleCell/" title="分类于 单细胞">单细胞</a></div><span><a href="/2020/11/07/2020-11-07-ScRNAseq(13)/" title="生信技能树单细胞数据挖掘笔记(7)">生信技能树单细胞数据挖掘笔记(7)</a></span></li><li class="item"><div class="breadcrumb"><a href="/categories/Python/" title="分类于 Python">Python</a></div><span><a href="/2020/10/20/2020-10-20-python8/" title="马士兵教育Python入门基础:4.流程控制（2）">马士兵教育Python入门基础:4.流程控制（2）</a></span></li><li class="item"><div class="breadcrumb"><a href="/categories/Bioinformation/" title="分类于 生信学习">生信学习</a></div><span><a href="/2020/11/17/2020-11-17-RNAseq(1)/" title="RNAseq教程(1.1)">RNAseq教程(1.1)</a></span></li><li class="item"><div class="breadcrumb"><a href="/categories/Bioinformation/" title="分类于 生信学习">生信学习</a></div><span><a href="/2021/01/09/2021-01-09-RNAseq(20)/" title="RNAseq教程(4.5)">RNAseq教程(4.5)</a></span></li><li class="item"><div class="breadcrumb"><a href="/categories/Python/" title="分类于 Python">Python</a></div><span><a href="/2020/09/15/2020-09-15-python3/" title="马士兵教育Python入门基础:2.七十二变（1）">马士兵教育Python入门基础:2.七十二变（1）</a></span></li><li class="item"><div class="breadcrumb"><a href="/categories/R/" title="分类于 R">R</a></div><span><a href="/2020/07/25/2020-07-25-R4ds2/" title="R数据科学笔记:2">R数据科学笔记:2</a></span></li><li class="item"><div class="breadcrumb"><a href="/categories/Bioinformation/" title="分类于 生信学习">生信学习</a> <i class="ic i-angle-right"></i> <a href="/categories/Bioinformation/SingleCell/" title="分类于 单细胞">单细胞</a></div><span><a href="/2020/10/13/2020-10-13-ScRNAseq(6)/" title="单细胞实战(5)：复现文章中的聚类图（2）">单细胞实战(5)：复现文章中的聚类图（2）</a></span></li><li class="item"><div class="breadcrumb"><a href="/categories/Bioinformation/" title="分类于 生信学习">生信学习</a></div><span><a href="/2020/07/20/2020-07-20-fungi/" title="文献导读：真菌进化">文献导读：真菌进化</a></span></li><li class="item"><div class="breadcrumb"><a href="/categories/R/" title="分类于 R">R</a></div><span><a href="/2020/07/22/2020-07-22-R4ds1/" title="R数据科学笔记：1">R数据科学笔记：1</a></span></li><li class="item"><div class="breadcrumb"><a href="/categories/Bioinformation/" title="分类于 生信学习">生信学习</a> <i class="ic i-angle-right"></i> <a href="/categories/Bioinformation/SingleCell/" title="分类于 单细胞">单细胞</a></div><span><a href="/2020/11/12/2020-11-12-RNAvelocity(2)/" title="RNA速率：数据读入">RNA速率：数据读入</a></span></li></ul></div><div><h2>最新评论</h2><ul class="leancloud-recent-comment"></ul></div></div><div class="status"><div class="copyright">&copy; 2010 – <span itemprop="copyrightYear">2023</span> <span class="with-love"><i class="ic i-sakura rotate"></i> </span><span class="author" itemprop="copyrightHolder">amane @ Zhou Xiaozhao</span></div><div class="count"><span class="post-meta-item-icon"><i class="ic i-chart-area"></i> </span><span title="站点总字数">385k 字</span> <span class="post-meta-divider">|</span> <span class="post-meta-item-icon"><i class="ic i-coffee"></i> </span><span title="站点阅读时长">5:50</span></div><div class="powered-by">基于 <span class="exturl" data-url="aHR0cHM6Ly9oZXhvLmlv">Hexo</span> & Theme.<span class="exturl" data-url="aHR0cHM6Ly9naXRodWIuY29tL2FtZWhpbWUvaGV4by10aGVtZS1zaG9rYQ==">Shoka</span></div></div></div></footer></div><script data-config type="text/javascript">var LOCAL={path:"2022/12/08/2022-12-08-SingleCopy/",favicon:{show:"（●´3｀●）やれやれだぜ",hide:"(´Д｀)大変だ！"},search:{placeholder:"文章搜索",empty:"关于 「 ${query} 」，什么也没搜到",stats:"${time} ms 内找到 ${hits} 条结果"},valine:!0,fancybox:!0,copyright:'复制成功，转载请遵守 <i class="ic i-creative-commons"></i>BY-NC-SA 协议。',ignores:[function(e){return e.includes("#")},function(e){return new RegExp(LOCAL.path+"$").test(e)}]}</script><script src="https://cdn.polyfill.io/v2/polyfill.js"></script><script src="//cdn.jsdelivr.net/combine/npm/pace-js@1.0.2/pace.min.js,npm/pjax@0.2.8/pjax.min.js,npm/whatwg-fetch@3.4.0/dist/fetch.umd.min.js,npm/animejs@3.2.0/lib/anime.min.js,npm/algoliasearch@4/dist/algoliasearch-lite.umd.js,npm/instantsearch.js@4/dist/instantsearch.production.min.js,npm/lozad@1/dist/lozad.min.js,npm/quicklink@2/dist/quicklink.umd.js"></script><script src="/js/app.js?v=0.2.5"></script></body></html><!-- rebuild by hrmmi -->