<!-- build time:Fri Feb 24 2023 14:40:38 GMT+0800 (中国标准时间) --><!DOCTYPE html><html lang="zh-CN"><head><meta charset="UTF-8"><meta name="viewport" content="width=device-width,initial-scale=1,maximum-scale=2"><meta name="theme-color" content="#FFF"><link rel="apple-touch-icon" sizes="180x180" href="/images/apple-touch-icon.png"><link rel="icon" type="image/ico" sizes="32x32" href="/images/favicon.ico"><meta http-equiv="Cache-Control" content="no-transform"><meta http-equiv="Cache-Control" content="no-siteapp"><link rel="alternate" type="application/rss+xml" title="小钊の笔记" href="https://www.zhouxiaozhao.cn/rss.xml"><link rel="alternate" type="application/atom+xml" title="小钊の笔记" href="https://www.zhouxiaozhao.cn/atom.xml"><link rel="alternate" type="application/json" title="小钊の笔记" href="https://www.zhouxiaozhao.cn/feed.json"><link rel="stylesheet" href="//fonts.googleapis.com/css?family=Mulish:300,300italic,400,400italic,700,700italic%7CFredericka%20the%20Great:300,300italic,400,400italic,700,700italic%7CNoto%20Serif%20JP:300,300italic,400,400italic,700,700italic%7CNoto%20Serif%20SC:300,300italic,400,400italic,700,700italic%7CInconsolata:300,300italic,400,400italic,700,700italic&display=swap&subset=latin,latin-ext"><link rel="stylesheet" href="/css/app.css?v=0.2.5"><meta name="keywords" content="bioinformatics,spatial_transcriptome"><link rel="canonical" href="https://www.zhouxiaozhao.cn/2023/02/22/2023-02-22-ST(1)/"><title>植物空间转录组分析 1：Seurat 基本流程 - 单细胞 - 生信学习 | Zhou Xiaozhao = 小钊の笔记 = 前天是小兔子，昨天是小鹿，今天是你</title><meta name="generator" content="Hexo 5.4.1"></head><body itemscope itemtype="http://schema.org/WebPage"><div id="loading"><div class="cat"><div class="body"></div><div class="head"><div class="face"></div></div><div class="foot"><div class="tummy-end"></div><div class="bottom"></div><div class="legs left"></div><div class="legs right"></div></div><div class="paw"><div class="hands left"></div><div class="hands right"></div></div></div></div><div id="container"><header id="header" itemscope itemtype="http://schema.org/WPHeader"><div class="inner"><div id="brand"><div class="pjax"><h1 itemprop="name headline">植物空间转录组分析 1：Seurat 基本流程</h1><div class="meta"><span class="item" title="创建时间：2023-02-22 00:00:00"><span class="icon"><i class="ic i-calendar"></i> </span><span class="text">发表于</span> <time itemprop="dateCreated datePublished" datetime="2023-02-22T00:00:00+08:00">2023-02-22</time> </span><span class="item" title="本文字数"><span class="icon"><i class="ic i-pen"></i> </span><span class="text">本文字数</span> <span>6.4k</span> <span class="text">字</span> </span><span class="item" title="阅读时长"><span class="icon"><i class="ic i-clock"></i> </span><span class="text">阅读时长</span> <span>6 分钟</span></span></div></div></div><nav id="nav"><div class="inner"><div class="toggle"><div class="lines" aria-label="切换导航栏"><span class="line"></span> <span class="line"></span> <span class="line"></span></div></div><ul class="menu"><li class="item title"><a href="/" rel="start">Zhou Xiaozhao</a></li></ul><ul class="right"><li class="item theme"><i class="ic i-sun"></i></li><li class="item search"><i class="ic i-search"></i></li></ul></div></nav></div><div id="imgs" class="pjax"><ul><li class="item" data-background-image="https://img1.imgtp.com/2023/02/01/udJuerr1.jpg"></li><li class="item" data-background-image="https://img1.imgtp.com/2023/02/01/szJGZgtF.jpg"></li><li class="item" data-background-image="https://img1.imgtp.com/2023/02/01/73RQFvJi.jpg"></li><li class="item" data-background-image="https://img1.imgtp.com/2023/02/01/u2N1wYKS.jpg"></li><li class="item" data-background-image="https://img1.imgtp.com/2023/02/01/6NZoUuLt.jpg"></li><li class="item" data-background-image="https://img1.imgtp.com/2023/02/01/ZsnbIfQs.jpg"></li></ul></div></header><div id="waves"><svg class="waves" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" viewBox="0 24 150 28" preserveAspectRatio="none" shape-rendering="auto"><defs><path id="gentle-wave" d="M-160 44c30 0 58-18 88-18s 58 18 88 18 58-18 88-18 58 18 88 18 v44h-352z"/></defs><g class="parallax"><use xlink:href="#gentle-wave" x="48" y="0"/><use xlink:href="#gentle-wave" x="48" y="3"/><use xlink:href="#gentle-wave" x="48" y="5"/><use xlink:href="#gentle-wave" x="48" y="7"/></g></svg></div><main><div class="inner"><div id="main" class="pjax"><div class="article wrap"><div class="breadcrumb" itemscope itemtype="https://schema.org/BreadcrumbList"><i class="ic i-home"></i> <span><a href="/">首页</a></span><i class="ic i-angle-right"></i> <span itemprop="itemListElement" itemscope itemtype="https://schema.org/ListItem"><a href="/categories/Bioinformation/" itemprop="item" rel="index" title="分类于 生信学习"><span itemprop="name">生信学习</span></a><meta itemprop="position" content="1"></span><i class="ic i-angle-right"></i> <span class="current" itemprop="itemListElement" itemscope itemtype="https://schema.org/ListItem"><a href="/categories/Bioinformation/SingleCell/" itemprop="item" rel="index" title="分类于 单细胞"><span itemprop="name">单细胞</span></a><meta itemprop="position" content="2"></span></div><article itemscope itemtype="http://schema.org/Article" class="post block" lang="zh-CN"><link itemprop="mainEntityOfPage" href="https://www.zhouxiaozhao.cn/2023/02/22/2023-02-22-ST(1)/"><span hidden itemprop="author" itemscope itemtype="http://schema.org/Person"><meta itemprop="image" content="/images/avatar.png"><meta itemprop="name" content="amane"><meta itemprop="description" content="前天是小兔子，昨天是小鹿，今天是你, 生物学在读博士"></span><span hidden itemprop="publisher" itemscope itemtype="http://schema.org/Organization"><meta itemprop="name" content="小钊の笔记"></span><div class="body md" itemprop="articleBody"><p>空间转录组是 2022 生命科学十大创新产品名单，因此将来会在生物学领域有非常大的应用空间，目前植物类的相关文章较少，在本次的系列教程中，将使用复旦大学戚继团队兰花空间转录组的数据，希望大家一起学习，掌握植物空间转录组基本的分析方法<br>数据连接<span class="exturl" data-url="aHR0cHM6Ly9vc2YuaW8vbXFzYjQvP3ZpZXdfb25seT03ODdiM2Q1YTgxZTU0YzI3YjNmMzNkNTg0MWI1OWE1MQ=="> OSF | Spatiotemporal atlas of organogenesis in development of orchid flowers</span>, 与单细胞的数据结构基本一致，多了 spatial 这个文件夹，主要包含的就是切片信息和 spot 定位信息</p><p><img data-src="/img/posts/2023.2.23/1.png" alt="文章"><br><img data-src="/img/posts/2023.2.23/2.png" alt="数据"></p><h1 id="前言"><a class="anchor" href="#前言">#</a> 前言</h1><p>在本篇文章中，作者一共制作了兰花空间转录组切片，并使用了 STEEL 方法进行聚类，出于学习考虑，本次分析先使用 Seurat 的常见流程分析其中一个切片，后续推文中将使用文章中的 STELL 方法进行聚类并使用 Seurat 的其他函数进行后续分析<br>本次只分析 Slide1，感兴趣的可以试试其他切片<br><img data-src="/img/posts/2023.2.23/3.png" alt="Slide1"></p><h1 id="数据载入"><a class="anchor" href="#数据载入">#</a> 数据载入</h1><p>和单细胞一样，只是多了 spatial 这个文件夹需要输入</p><pre><code class="language-r">##=============================1.安装依赖包=====================================
##BiocManager::install(c('Seurat','ggplot2','cowplot','dplyr'))
##BiocManager::install(&quot;monocle&quot;,force = TRUE) 
library(Seurat)
library(ggplot2)
library(cowplot)
library(dplyr)
library(magrittr)
library(gtools)
library(stringr)
library(Matrix)
library(tidyverse)
library(patchwork)
##=============================2.载入数据=======================================
orc1&lt;- Load10X_Spatial(&quot;Slide_1&quot;,filename = &quot;filtered_feature_bc_matrix.h5&quot;,assay = &quot;spatial&quot;)
</code></pre><h1 id="质量控制"><a class="anchor" href="#质量控制">#</a> 质量控制</h1><p>不只是能体现小提琴图，还能将每个 spot 的测序质量体现在切片上，帮我们确定哪些组织可能会出现问题</p><pre><code>##=============================3.质量控制=======================================
dir.create(&quot;QC&quot;)
##UMI统计画图
plot1 &lt;- VlnPlot(orc1, features = &quot;nCount_Spatial&quot;, pt.size = 0.1) + NoLegend()
plot2 &lt;- SpatialFeaturePlot(orc1, features = &quot;nCount_Spatial&quot;) + theme(legend.position = &quot;right&quot;)
pearplot &lt;- plot_grid(plot1,plot2) 
ggsave(&quot;QC/Slide1_UMI.pdf&quot;, plot = pearplot, width = 12, height = 5) 
##Feature/Gene统计画图
plot1 &lt;- VlnPlot(orc1, features = &quot;nFeature_Spatial&quot;, pt.size = 0.1) + NoLegend()
plot2 &lt;- SpatialFeaturePlot(orc1, features = &quot;nFeature_Spatial&quot;) + theme(legend.position = &quot;right&quot;)
pearplot &lt;- plot_grid(plot1,plot2) 
ggsave(&quot;QC/Slide1_Feature.pdf&quot;, plot = pearplot, width = 12, height = 5)
</code></pre><p><img data-src="/img/posts/2023.2.23/4.png" alt="UMI"><br><img data-src="/img/posts/2023.2.23/5.png" alt="feature"><br>可以明显的看到，不管是 UMI 还是 Feature，在 spot 中差异主要是有与切片组织的细胞数目以及细胞类型导致的。比如，在花瓣中可以看到非常少的 UMI 以及 Feature，在中间的分生组织中，UMI 和 Feature 的数目非常多。因此，单细胞的标准方法 (如 LogNormalize 函数) 可能会有问题，在空间转录组分析中，一般使用其他方法进行标准化。</p><h1 id="数据标准化"><a class="anchor" href="#数据标准化">#</a> 数据标准化</h1><p>目前空间转录组数据推荐使用 sctransform，具体的原理我还没有看懂，所以先根据流程走一遍，看看效果<br>为了探究标准化方法的不同，sctransform 和 log 规范化结果如何与 UMIs 的数量相关。为了进行比较，首先运行 sctransform 来存储所有基因的值，并通过 NormalizeData 运行一个 log 规范化过程。</p><pre><code class="language-r">##============================4.数据标准化======================================
## 目前空间转录组推荐使用SCTransform，集合了单细胞标准化的NormalizeData()，ScaleData()，FindVariableFeatures()，
dir.create(&quot;Normalize&quot;)
orc1 &lt;- SCTransform(orc1, assay = &quot;Spatial&quot;, return.only.var.genes = FALSE, verbose = FALSE)
## 比较SCTransform与log Normalization的区别
orc1 &lt;- NormalizeData(orc1, verbose = FALSE, assay = &quot;Spatial&quot;)
## 计算UMI与Feature的相关性，分为5组
orc1 &lt;- GroupCorrelation(orc1, group.assay = &quot;Spatial&quot;, assay = &quot;Spatial&quot;, slot = &quot;data&quot;, do.plot = FALSE)
orc1 &lt;- GroupCorrelation(orc1, group.assay = &quot;Spatial&quot;, assay = &quot;SCT&quot;, slot = &quot;scale.data&quot;, do.plot = FALSE)
p1 &lt;- GroupCorrelationPlot(orc1, assay = &quot;Spatial&quot;, cor = &quot;nCount_Spatial_cor&quot;) + ggtitle(&quot;Log Normalization&quot;) + 
  theme(plot.title = element_text(hjust = 0.5))
p2 &lt;- GroupCorrelationPlot(orc1, assay = &quot;SCT&quot;, cor = &quot;nCount_Spatial_cor&quot;) + ggtitle(&quot;SCTransform Normalization&quot;) + 
  theme(plot.title = element_text(hjust = 0.5))
p3 &lt;- plot_grid(p1, p2)
ggsave(&quot;Normalize/Normalization_cor_Slide1.pdf&quot;, plot = p3, width = 12, height = 5)  
</code></pre><p><img data-src="/img/posts/2023.2.23/6.png" alt="标准化"></p><p>对于上面的箱形图，计算每个特征 (基因) 与 UMIs 数量 (这里的 nCount_Spatial 变量) 的相关性。然后，根据基因的平均表达将它们分组，并生成这些相关性的箱形图。<br>同时这里需要注意，SCT 标准化后有个单独的 assay (SCT)，里面包含三个矩阵，分别为 count,data 以及 scale.data，log 标准化也有一个 assay (Spatial)，里面也包含三个矩阵，但是 scale.data 的数据为 0<br>标准化之后进行高变基因的筛选，这部分没什么可说的，与单细胞是一致的</p><pre><code class="language-r">top10 &lt;- head(VariableFeatures(orc1), 10) 
plot1 &lt;- VariableFeaturePlot(orc1) 
plot2 &lt;- LabelPoints(plot = plot1, points = top10, repel = TRUE, size=2.5) 
ggsave(&quot;Normalize/VariableFeature_Slide1.pdf&quot;, plot = plot2, width = 9, height = 6)
</code></pre><p><img data-src="/img/posts/2023.2.23/7.png" alt="高变基因筛选"></p><h1 id="数据降维与聚类"><a class="anchor" href="#数据降维与聚类">#</a> 数据降维与聚类</h1><p>由于文章中使用了 STEEL 方法，所以这次的分析只是走走流程，之后会按照文章中的 STEEL 方法进行测试</p><pre><code class="language-r">##============================5.数据降维与聚类==================================
dir.create(&quot;cluster&quot;)
## PCA降维并提取主成分
orc1 &lt;- RunPCA(orc1, features = VariableFeatures(orc1_new)) 
plot1 &lt;- DimPlot(orc1, reduction = &quot;pca&quot;, group.by=&quot;orig.ident&quot;) 
plot2 &lt;- ElbowPlot(orc1, ndims=40, reduction=&quot;pca&quot;) 
pearplot &lt;- plot_grid(plot1,plot2)
ggsave(&quot;cluster/PCA_Slide1.pdf&quot;, plot = pearplot, width = 13, height = 6)

pdf(&quot;cluster/PC_heatmap_Slide1.pdf&quot;,height = 12,width = 12)
DimHeatmap(orc1, dims = 1:9, nfeatures=10, cells = 200, balanced = TRUE)
dev.off()

## 细胞聚类
## 此步利用 细胞-PC值 矩阵计算细胞之间的距离，
## 然后利用距离矩阵来聚类。其中有两个参数需要人工选择，
## 第一个是FindNeighbors()函数中的dims参数，需要指定哪些pc轴用于分析，选择依据是之前介绍的cluster/pca.png文件中的右图。
## 第二个是FindClusters()函数中的resolution参数，需要指定一个数值，用于决定clusters的相对数量，数值越大cluters越多。
orc1 &lt;- FindNeighbors(orc1, dims = 1:20) 
orc1 &lt;- FindClusters(orc1, resolution = 0.9)

metadata &lt;- orc1@meta.data
table(orc1@meta.data$seurat_clusters)
cell_cluster &lt;- data.frame(cell_ID=rownames(metadata), cluster_ID=metadata$seurat_clusters)
write.csv(cell_cluster,'cluster/cell_cluster_Slide1.csv',row.names = F)
</code></pre><p><img data-src="/img/posts/2023.2.23/8.png" alt="PCA"></p><p>使用 resolution = 0.9 只生成了 12 个聚类，原文中有 40 个，大家在设置的时候可以把 resolution 调到 2 甚至 3 左右看看效果</p><pre><code class="language-r">## 非线性降维
## tsne
orc1 &lt;- RunTSNE(orc1, dims = 1:20)
embed_tsne &lt;- Embeddings(orc1, 'tsne')
write.csv(embed_tsne,'cluster/embed_tsne_Slide1.csv')

p1 &lt;- DimPlot(orc1, reduction = &quot;tsne&quot;, label = TRUE)
p2 &lt;- SpatialDimPlot(orc1, label.size = 3, pt.size.factor = 1.3)
pearplot &lt;- plot_grid(p1,p2)
ggsave(&quot;cluster/tsne_Slide1.pdf&quot;, plot = p2, width = 6, height = 6)

## UMAP
orc1 &lt;- RunUMAP(orc1,dims = 1:20)
embed_umap &lt;- Embeddings(orc1, 'umap')
write.csv(embed_umap,'cluster/embed_umap_Slide1.csv')

p1 &lt;- DimPlot(orc1, reduction = &quot;umap&quot;, label = TRUE)
p2 &lt;- SpatialDimPlot(orc1, label = TRUE, label.size = 3, pt.size.factor = 1.3)
pearplot &lt;- plot_grid(p1,p2)
ggsave(&quot;cluster/umap_Slide1.pdf&quot;, plot = pearplot, width = 13, height = 6)
</code></pre><p><img data-src="/img/posts/2023.2.23/9.png" alt="TSNE"><br><img data-src="/img/posts/2023.2.23/10.png" alt="UMAP"><br>#6. 部分基因展示<br>此处是我比较纠结的地方，因为一开始得到的数据和原文中有点不符，后来发现可能的原因是文章中使用的展示数据是 counts</p><pre><code class="language-r">## featureplot
dir.create(&quot;featureplot&quot;)
p1 &lt;- SpatialFeaturePlot(orc1, features = &quot;PAXXG070630&quot;,slot=&quot;counts&quot;,pt.size.factor = 1.6,min.cutoff = 0.1)
p2 &lt;- SpatialFeaturePlot(orc1, features = &quot;PAXXG070630&quot;,slot=&quot;data&quot;,pt.size.factor = 1.6,min.cutoff = 0.1)
p3 &lt;- SpatialFeaturePlot(orc1, features = &quot;PAXXG070630&quot;,slot=&quot;scale.data&quot;,pt.size.factor = 1.6,min.cutoff = 0.1)
pearplot &lt;- plot_grid(p1,p2,p3,align = &quot;v&quot;,axis = &quot;b&quot;,ncol = 3)
ggsave(&quot;featureplot/SCTSlide1.pdf&quot;, plot = pearplot, width = 18, height = 7)
</code></pre><p><img data-src="/img/posts/2023.2.23/11.png" alt="三种展示方式"><br><img data-src="/img/posts/2023.2.23/12.png" alt="原文"><br>#总结<br>目前先把前期的工作的进行了一下，原文中进行了两个拟时分析，但因为聚类不一样所以先不进行分析，研究完 STEEL 之后再进行拟时分析。<br>总体来讲与单细胞变化不大，有优势也有劣势吧，对于细胞分型来说，空间转录组有先天优势，可以直接根据切片信息判断细胞类型，但是空间转录组最小的单位 spot，不能代表单个细胞，分辨率可能还存在问题</p><div class="tags"><a href="/tags/bioinformatics/" rel="tag"><i class="ic i-tag"></i> bioinformatics</a> <a href="/tags/spatial-transcriptome/" rel="tag"><i class="ic i-tag"></i> spatial_transcriptome</a></div></div><footer><div class="meta"><span class="item"><span class="icon"><i class="ic i-calendar-check"></i> </span><span class="text">更新于</span> <time title="修改时间：2023-02-23 10:03:49" itemprop="dateModified" datetime="2023-02-23T10:03:49+08:00">2023-02-23</time> </span><span id="2023/02/22/2023-02-22-ST(1)/" class="item leancloud_visitors" data-flag-title="植物空间转录组分析 1：Seurat 基本流程" title="阅读次数"><span class="icon"><i class="ic i-eye"></i> </span><span class="text">阅读次数</span> <span class="leancloud-visitors-count"></span> <span class="text">次</span></span></div><div class="reward"><button><i class="ic i-heartbeat"></i> 赞赏</button><p>请我喝[茶]~(￣▽￣)~*</p><div id="qr"><div><img data-src="/images/wechatpay.png" alt="amane 微信支付"><p>微信支付</p></div><div><img data-src="/images/alipay.png" alt="amane 支付宝"><p>支付宝</p></div></div></div><div id="copyright"><ul><li class="author"><strong>本文作者： </strong>amane <i class="ic i-at"><em>@</em></i>小钊の笔记</li><li class="link"><strong>本文链接：</strong> <a href="https://www.zhouxiaozhao.cn/2023/02/22/2023-02-22-ST(1)/" title="植物空间转录组分析 1：Seurat 基本流程">https://www.zhouxiaozhao.cn/2023/02/22/2023-02-22-ST(1)/</a></li><li class="license"><strong>版权声明： </strong>本站所有文章除特别声明外，均采用 <span class="exturl" data-url="aHR0cHM6Ly9jcmVhdGl2ZWNvbW1vbnMub3JnL2xpY2Vuc2VzL2J5LW5jLXNhLzQuMC9kZWVkLnpo"><i class="ic i-creative-commons"><em>(CC)</em></i>BY-NC-SA</span> 许可协议。转载请注明出处！</li></ul></div></footer></article></div><div class="post-nav"><div class="item left"><a href="/2022/12/15/2022-12-15-GO/" itemprop="url" rel="prev" data-background-image="https:&#x2F;&#x2F;img1.imgtp.com&#x2F;2023&#x2F;02&#x2F;01&#x2F;Vghr1YDl.jpg" title="比较基因组学分析3：特异节点基因家族富集分析（非模式物种GO&#x2F;KEEG库构建）"><span class="type">上一篇</span> <span class="category"><i class="ic i-flag"></i> 基因组</span><h3>比较基因组学分析3：特异节点基因家族富集分析（非模式物种GO/KEEG库构建）</h3></a></div><div class="item right"></div></div><div class="wrap" id="comments"></div></div><div id="sidebar"><div class="inner"><div class="panels"><div class="inner"><div class="contents panel pjax" data-title="文章目录"><ol class="toc"><li class="toc-item toc-level-1"><a class="toc-link" href="#%E5%89%8D%E8%A8%80"><span class="toc-number">1.</span> <span class="toc-text">前言</span></a></li><li class="toc-item toc-level-1"><a class="toc-link" href="#%E6%95%B0%E6%8D%AE%E8%BD%BD%E5%85%A5"><span class="toc-number">2.</span> <span class="toc-text">数据载入</span></a></li><li class="toc-item toc-level-1"><a class="toc-link" href="#%E8%B4%A8%E9%87%8F%E6%8E%A7%E5%88%B6"><span class="toc-number">3.</span> <span class="toc-text">质量控制</span></a></li><li class="toc-item toc-level-1"><a class="toc-link" href="#%E6%95%B0%E6%8D%AE%E6%A0%87%E5%87%86%E5%8C%96"><span class="toc-number">4.</span> <span class="toc-text">数据标准化</span></a></li><li class="toc-item toc-level-1"><a class="toc-link" href="#%E6%95%B0%E6%8D%AE%E9%99%8D%E7%BB%B4%E4%B8%8E%E8%81%9A%E7%B1%BB"><span class="toc-number">5.</span> <span class="toc-text">数据降维与聚类</span></a></li></ol></div><div class="related panel pjax" data-title="系列文章"><ul><li><a href="/2020/09/29/2020-09-29-ScRNAseq(1)/" rel="bookmark" title="单细胞实战(1)：原始测序数据下载">单细胞实战(1)：原始测序数据下载</a></li><li><a href="/2020/10/01/2020-10-01-ScRNAseq(2)/" rel="bookmark" title="单细胞实战(2)：cellranger使用">单细胞实战(2)：cellranger使用</a></li><li><a href="/2020/10/03/2020-10-03-ScRNAseq(3)/" rel="bookmark" title="单细胞实战(3)：STAR分析单细胞数据">单细胞实战(3)：STAR分析单细胞数据</a></li><li><a href="/2020/10/06/2020-10-06-ScRNAseq(4)/" rel="bookmark" title="单细胞实战(4)：STAR与cellranger结果比较">单细胞实战(4)：STAR与cellranger结果比较</a></li><li><a href="/2020/10/10/2020-10-10-ScRNAseq(5)/" rel="bookmark" title="单细胞实战(5)：复现文章中的聚类图（1）">单细胞实战(5)：复现文章中的聚类图（1）</a></li><li><a href="/2020/10/13/2020-10-13-ScRNAseq(6)/" rel="bookmark" title="单细胞实战(5)：复现文章中的聚类图（2）">单细胞实战(5)：复现文章中的聚类图（2）</a></li><li><a href="/2020/10/24/2020-10-24-ScRNAseq(7)/" rel="bookmark" title="生信技能树单细胞数据挖掘笔记(1)">生信技能树单细胞数据挖掘笔记(1)</a></li><li><a href="/2020/10/27/2020-10-27-ScRNAseq(8)/" rel="bookmark" title="生信技能树单细胞数据挖掘笔记(2)">生信技能树单细胞数据挖掘笔记(2)</a></li><li><a href="/2020/10/29/2020-10-29-ScRNAseq(9)/" rel="bookmark" title="生信技能树单细胞数据挖掘笔记(3)">生信技能树单细胞数据挖掘笔记(3)</a></li><li><a href="/2020/10/31/2020-10-31-ScRNAseq(10)/" rel="bookmark" title="生信技能树单细胞数据挖掘笔记(4)">生信技能树单细胞数据挖掘笔记(4)</a></li><li><a href="/2020/11/03/2020-11-03-ScRNAseq(11)/" rel="bookmark" title="生信技能树单细胞数据挖掘笔记(5)">生信技能树单细胞数据挖掘笔记(5)</a></li><li><a href="/2020/11/05/2020-11-05-ScRNAseq(12)/" rel="bookmark" title="生信技能树单细胞数据挖掘笔记(6)">生信技能树单细胞数据挖掘笔记(6)</a></li><li><a href="/2020/11/07/2020-11-07-ScRNAseq(13)/" rel="bookmark" title="生信技能树单细胞数据挖掘笔记(7)">生信技能树单细胞数据挖掘笔记(7)</a></li><li><a href="/2020/11/10/2020-11-10-RNAvelocity(1)/" rel="bookmark" title="RNA速率：软件下载与loom文件准备">RNA速率：软件下载与loom文件准备</a></li><li><a href="/2020/11/12/2020-11-12-RNAvelocity(2)/" rel="bookmark" title="RNA速率：数据读入">RNA速率：数据读入</a></li><li><a href="/2020/11/14/2020-11-14-RNAvelocity(3)/" rel="bookmark" title="RNA速率：使用Seurat的结果做RNA velocity">RNA速率：使用Seurat的结果做RNA velocity</a></li><li class="active"><a href="/2023/02/22/2023-02-22-ST(1)/" rel="bookmark" title="植物空间转录组分析1：Seurat基本流程">植物空间转录组分析1：Seurat基本流程</a></li></ul></div><div class="overview panel" data-title="站点概览"><div class="author" itemprop="author" itemscope itemtype="http://schema.org/Person"><img class="image" itemprop="image" alt="amane" data-src="/images/avatar.png"><p class="name" itemprop="name">amane</p><div class="description" itemprop="description">生物学在读博士</div></div><nav class="state"><div class="item posts"><a href="/archives/"><span class="count">84</span> <span class="name">文章</span></a></div><div class="item categories"><a href="/categories/"><span class="count">7</span> <span class="name">分类</span></a></div><div class="item tags"><a href="/tags/"><span class="count">21</span> <span class="name">标签</span></a></div></nav><div class="social"><span class="exturl item github" data-url="aHR0cHM6Ly9naXRodWIuY29tL0Jpb2luZm9ybWF0aWNzLXJvb2tpZQ==" title="https:&#x2F;&#x2F;github.com&#x2F;Bioinformatics-rookie"><i class="ic i-github"></i></span> <a href="https://www.zhouxiaozhao.cn/about/" title="https:&#x2F;&#x2F;www.zhouxiaozhao.cn&#x2F;about&#x2F;" class="item about"><i class="ic i-address-card"></i></a> <span class="exturl item email" data-url="bWFpbHRvOnpob3VzaGl6aGFvQG1haWwua2liLmFjLmNu" title="mailto:zhoushizhao@mail.kib.ac.cn"><i class="ic i-envelope"></i></span></div><ul class="menu"><li class="item"><a href="/" rel="section"><i class="ic i-home"></i>首页</a></li><li class="item"><a href="/about/" rel="section"><i class="ic i-user"></i>关于</a></li><li class="item dropdown"><a href="javascript:void(0);"><i class="ic i-feather"></i>文章</a><ul class="submenu"><li class="item"><a href="/archives/" rel="section"><i class="ic i-list-alt"></i>归档</a></li><li class="item"><a href="/categories/" rel="section"><i class="ic i-th"></i>分类</a></li><li class="item"><a href="/tags/" rel="section"><i class="ic i-tags"></i>标签</a></li></ul></li><li class="item"><a href="/links/" rel="section"><i class="ic i-heart"></i>友達</a></li></ul></div></div></div><ul id="quick"><li class="prev pjax"></li><li class="up"><i class="ic i-arrow-up"></i></li><li class="down"><i class="ic i-arrow-down"></i></li><li class="next pjax"></li><li class="percent"></li></ul></div></div><div class="dimmer"></div></div></main><footer id="footer"><div class="inner"><div class="widgets"><div class="rpost pjax"><h2>随机文章</h2><ul><li class="item"><div class="breadcrumb"><a href="/categories/Bioinformation/" title="分类于 生信学习">生信学习</a></div><span><a href="/2020/08/20/2020-08-20-nt/" title="使用本地nt数据库对reads和Trinity结果进行blast">使用本地nt数据库对reads和Trinity结果进行blast</a></span></li><li class="item"><div class="breadcrumb"><a href="/categories/Bioinformation/" title="分类于 生信学习">生信学习</a> <i class="ic i-angle-right"></i> <a href="/categories/Bioinformation/SingleCell/" title="分类于 单细胞">单细胞</a></div><span><a href="/2020/10/01/2020-10-01-ScRNAseq(2)/" title="单细胞实战(2)：cellranger使用">单细胞实战(2)：cellranger使用</a></span></li><li class="item"><div class="breadcrumb"><a href="/categories/Bioinformation/" title="分类于 生信学习">生信学习</a></div><span><a href="/2020/11/19/2020-11-19-annotion(1)/" title="基因结构注释（1）：从头注释">基因结构注释（1）：从头注释</a></span></li><li class="item"><div class="breadcrumb"><a href="/categories/Bioinformation/" title="分类于 生信学习">生信学习</a></div><span><a href="/2020/07/20/2020-07-20-fungi/" title="文献导读：真菌进化">文献导读：真菌进化</a></span></li><li class="item"><div class="breadcrumb"><a href="/categories/R/" title="分类于 R">R</a></div><span><a href="/2020/08/16/2020-08-16-R4ds4/" title="R数据科学笔记:4">R数据科学笔记:4</a></span></li><li class="item"><div class="breadcrumb"><a href="/categories/Bioinformation/" title="分类于 生信学习">生信学习</a></div><span><a href="/2020/11/21/2020-11-21-annotion(2)/" title="基因结构注释（2）：同源注释">基因结构注释（2）：同源注释</a></span></li><li class="item"><div class="breadcrumb"><a href="/categories/Bioinformation/" title="分类于 生信学习">生信学习</a></div><span><a href="/2020/12/01/2020-12-01-RNAseq(3)/" title="RNAseq教程(1.3)">RNAseq教程(1.3)</a></span></li><li class="item"><div class="breadcrumb"><a href="/categories/Bioinformation/" title="分类于 生信学习">生信学习</a> <i class="ic i-angle-right"></i> <a href="/categories/Bioinformation/genome/" title="分类于 基因组">基因组</a></div><span><a href="/2022/12/08/2022-12-08-SingleCopy/" title="比较基因组学分析1：构建单拷贝基因树">比较基因组学分析1：构建单拷贝基因树</a></span></li><li class="item"><div class="breadcrumb"><a href="/categories/Bioinformation/" title="分类于 生信学习">生信学习</a> <i class="ic i-angle-right"></i> <a href="/categories/Bioinformation/SingleCell/" title="分类于 单细胞">单细胞</a></div><span><a href="/2020/10/27/2020-10-27-ScRNAseq(8)/" title="生信技能树单细胞数据挖掘笔记(2)">生信技能树单细胞数据挖掘笔记(2)</a></span></li><li class="item"><div class="breadcrumb"><a href="/categories/Python/" title="分类于 Python">Python</a></div><span><a href="/2020/09/26/2020-09-26-python5/" title="马士兵教育Python入门基础:3.运算（1）">马士兵教育Python入门基础:3.运算（1）</a></span></li></ul></div><div><h2>最新评论</h2><ul class="leancloud-recent-comment"></ul></div></div><div class="status"><div class="copyright">&copy; 2010 – <span itemprop="copyrightYear">2023</span> <span class="with-love"><i class="ic i-sakura rotate"></i> </span><span class="author" itemprop="copyrightHolder">amane @ Zhou Xiaozhao</span></div><div class="count"><span class="post-meta-item-icon"><i class="ic i-chart-area"></i> </span><span title="站点总字数">351k 字</span> <span class="post-meta-divider">|</span> <span class="post-meta-item-icon"><i class="ic i-coffee"></i> </span><span title="站点阅读时长">5:19</span></div><div class="powered-by">基于 <span class="exturl" data-url="aHR0cHM6Ly9oZXhvLmlv">Hexo</span> & Theme.<span class="exturl" data-url="aHR0cHM6Ly9naXRodWIuY29tL2FtZWhpbWUvaGV4by10aGVtZS1zaG9rYQ==">Shoka</span></div></div></div></footer></div><script data-config type="text/javascript">var LOCAL={path:"2023/02/22/2023-02-22-ST(1)/",favicon:{show:"（●´3｀●）やれやれだぜ",hide:"(´Д｀)大変だ！"},search:{placeholder:"文章搜索",empty:"关于 「 ${query} 」，什么也没搜到",stats:"${time} ms 内找到 ${hits} 条结果"},valine:!0,fancybox:!0,copyright:'复制成功，转载请遵守 <i class="ic i-creative-commons"></i>BY-NC-SA 协议。',ignores:[function(e){return e.includes("#")},function(e){return new RegExp(LOCAL.path+"$").test(e)}]}</script><script src="https://cdn.polyfill.io/v2/polyfill.js"></script><script src="//cdn.jsdelivr.net/combine/npm/pace-js@1.0.2/pace.min.js,npm/pjax@0.2.8/pjax.min.js,npm/whatwg-fetch@3.4.0/dist/fetch.umd.min.js,npm/animejs@3.2.0/lib/anime.min.js,npm/algoliasearch@4/dist/algoliasearch-lite.umd.js,npm/instantsearch.js@4/dist/instantsearch.production.min.js,npm/lozad@1/dist/lozad.min.js,npm/quicklink@2/dist/quicklink.umd.js"></script><script src="/js/app.js?v=0.2.5"></script></body></html><!-- rebuild by hrmmi -->