<!-- build time:Thu Mar 30 2023 11:00:38 GMT+0800 (中国标准时间) --><!DOCTYPE html><html lang="zh-CN"><head><meta charset="UTF-8"><meta name="viewport" content="width=device-width,initial-scale=1,maximum-scale=2"><meta name="theme-color" content="#FFF"><link rel="apple-touch-icon" sizes="180x180" href="/images/apple-touch-icon.png"><link rel="icon" type="image/ico" sizes="32x32" href="/images/favicon.ico"><meta http-equiv="Cache-Control" content="no-transform"><meta http-equiv="Cache-Control" content="no-siteapp"><link rel="alternate" type="application/rss+xml" title="小钊の笔记" href="https://www.zhouxiaozhao.cn/rss.xml"><link rel="alternate" type="application/atom+xml" title="小钊の笔记" href="https://www.zhouxiaozhao.cn/atom.xml"><link rel="alternate" type="application/json" title="小钊の笔记" href="https://www.zhouxiaozhao.cn/feed.json"><link rel="stylesheet" href="//fonts.googleapis.com/css?family=Mulish:300,300italic,400,400italic,700,700italic%7CFredericka%20the%20Great:300,300italic,400,400italic,700,700italic%7CNoto%20Serif%20JP:300,300italic,400,400italic,700,700italic%7CNoto%20Serif%20SC:300,300italic,400,400italic,700,700italic%7CInconsolata:300,300italic,400,400italic,700,700italic&display=swap&subset=latin,latin-ext"><link rel="stylesheet" href="/css/app.css?v=0.2.5"><meta name="keywords" content="bioinformatics,RNAseq"><link rel="canonical" href="https://www.zhouxiaozhao.cn/2020/12/26/2020-12-26-RNAseq(14)/"><title>RNAseq 教程 (3.3) - 生信学习 | Zhou Xiaozhao = 小钊の笔记 = 前天是小兔子，昨天是小鹿，今天是你</title><meta name="generator" content="Hexo 5.4.1"></head><body itemscope itemtype="http://schema.org/WebPage"><div id="loading"><div class="cat"><div class="body"></div><div class="head"><div class="face"></div></div><div class="foot"><div class="tummy-end"></div><div class="bottom"></div><div class="legs left"></div><div class="legs right"></div></div><div class="paw"><div class="hands left"></div><div class="hands right"></div></div></div></div><div id="container"><header id="header" itemscope itemtype="http://schema.org/WPHeader"><div class="inner"><div id="brand"><div class="pjax"><h1 itemprop="name headline">RNAseq 教程 (3.3)</h1><div class="meta"><span class="item" title="创建时间：2020-12-26 18:38:45"><span class="icon"><i class="ic i-calendar"></i> </span><span class="text">发表于</span> <time itemprop="dateCreated datePublished" datetime="2020-12-26T18:38:45+08:00">2020-12-26</time> </span><span class="item" title="本文字数"><span class="icon"><i class="ic i-pen"></i> </span><span class="text">本文字数</span> <span>19k</span> <span class="text">字</span> </span><span class="item" title="阅读时长"><span class="icon"><i class="ic i-clock"></i> </span><span class="text">阅读时长</span> <span>17 分钟</span></span></div></div></div><nav id="nav"><div class="inner"><div class="toggle"><div class="lines" aria-label="切换导航栏"><span class="line"></span> <span class="line"></span> <span class="line"></span></div></div><ul class="menu"><li class="item title"><a href="/" rel="start">Zhou Xiaozhao</a></li></ul><ul class="right"><li class="item theme"><i class="ic i-sun"></i></li><li class="item search"><i class="ic i-search"></i></li></ul></div></nav></div><div id="imgs" class="pjax"><ul><li class="item" data-background-image="https://img1.imgtp.com/2023/02/01/wDa0q6Dr.jpg"></li><li class="item" data-background-image="https://img1.imgtp.com/2023/02/01/tzn2O8cD.jpg"></li><li class="item" data-background-image="https://img1.imgtp.com/2023/02/01/4OefNOxP.jpg"></li><li class="item" data-background-image="https://img1.imgtp.com/2023/02/01/Vghr1YDl.jpg"></li><li class="item" data-background-image="https://img1.imgtp.com/2023/02/01/708Aox8K.jpg"></li><li class="item" data-background-image="https://img1.imgtp.com/2023/02/01/RJVmoMVH.jpg"></li></ul></div></header><div id="waves"><svg class="waves" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" viewBox="0 24 150 28" preserveAspectRatio="none" shape-rendering="auto"><defs><path id="gentle-wave" d="M-160 44c30 0 58-18 88-18s 58 18 88 18 58-18 88-18 58 18 88 18 v44h-352z"/></defs><g class="parallax"><use xlink:href="#gentle-wave" x="48" y="0"/><use xlink:href="#gentle-wave" x="48" y="3"/><use xlink:href="#gentle-wave" x="48" y="5"/><use xlink:href="#gentle-wave" x="48" y="7"/></g></svg></div><main><div class="inner"><div id="main" class="pjax"><div class="article wrap"><div class="breadcrumb" itemscope itemtype="https://schema.org/BreadcrumbList"><i class="ic i-home"></i> <span><a href="/">首页</a></span><i class="ic i-angle-right"></i> <span class="current" itemprop="itemListElement" itemscope itemtype="https://schema.org/ListItem"><a href="/categories/Bioinformation/" itemprop="item" rel="index" title="分类于 生信学习"><span itemprop="name">生信学习</span></a><meta itemprop="position" content="1"></span></div><article itemscope itemtype="http://schema.org/Article" class="post block" lang="zh-CN"><link itemprop="mainEntityOfPage" href="https://www.zhouxiaozhao.cn/2020/12/26/2020-12-26-RNAseq(14)/"><span hidden itemprop="author" itemscope itemtype="http://schema.org/Person"><meta itemprop="image" content="/images/avatar.jpg"><meta itemprop="name" content="amane"><meta itemprop="description" content="前天是小兔子，昨天是小鹿，今天是你, 生物学在读博士"></span><span hidden itemprop="publisher" itemscope itemtype="http://schema.org/Organization"><meta itemprop="name" content="小钊の笔记"></span><div class="body md" itemprop="articleBody"><h2 id="目录"><a class="anchor" href="#目录">#</a> 目录</h2><p><strong>1.Module 1 - Introduction to RNA sequencing</strong></p><ol><li><a href="https://www.zhouxiaozhao.cn/2020/11/17/2020-11-17-RNAseq(1)/">Installation</a></li><li><a href="https://www.zhouxiaozhao.cn/2020/11/28/2020-11-28-RNAseq(2)/">Reference Genomes</a></li><li><a href="https://www.zhouxiaozhao.cn/2020/12/01/2020-12-01-RNAseq(3)/">Annotations</a></li><li><a href="https://www.zhouxiaozhao.cn/2020/12/03/2020-12-03-RNAseq(4)/">Indexing</a></li><li><a href="https://www.zhouxiaozhao.cn/2020/12/05/2020-12-05-RNAseq(5)/">RNA-seq Data</a></li><li><a href="https://www.zhouxiaozhao.cn/2020/12/08/2020-12-08-RNAseq(6)/">Pre-Alignment QC</a></li></ol><p><strong>2.Module 2 - RNA-seq Alignment and Visualization</strong></p><ol><li><a href="https://www.zhouxiaozhao.cn/2020/12/10/2020-12-10-RNAseq(7)/">Adapter Trim</a></li><li><a href="https://www.zhouxiaozhao.cn/2020/12/12/2020-12-12-RNAseq(8)/">Alignment</a></li><li><a href="https://www.zhouxiaozhao.cn/2020/12/15/2020-12-15-RNAseq(9)/">IGV</a></li><li><a href="https://www.zhouxiaozhao.cn/2020/12/17/2020-12-17-RNAseq(10)/">Alignment Visualization</a></li><li><a href="https://www.zhouxiaozhao.cn/2020/12/19/2020-12-19-RNAseq(11)/">Alignment QC</a></li></ol><p><strong>3.Module 3 - Expression and Differential Expression</strong></p><ol><li><a href="https://www.zhouxiaozhao.cn/2020/12/22/2020-12-22-RNAseq(12)/">Expression</a></li><li><a href="https://www.zhouxiaozhao.cn/2020/12/24/2020-12-24-RNAseq(13)/">Differential Expression</a></li><li><a href="https://www.zhouxiaozhao.cn/2020/12/26/2020-12-26-RNAseq(14)/">DE Visualization</a></li><li><a href="https://www.zhouxiaozhao.cn/2020/12/29/2020-12-29-RNAseq(15)/">Kallisto for Reference-Free Abundance Estimation</a></li></ol><p><strong>4.Module 4 - Isoform Discovery and Alternative Expression</strong></p><ol><li><a href="https://www.zhouxiaozhao.cn/2020/12/31/2020-12-31-RNAseq(16)/">Reference Guided Transcript Assembly</a></li><li><a href="https://www.zhouxiaozhao.cn/2021/01/02/2021-01-02-RNAseq(17)/">de novo Transcript Assembly</a></li><li><a href="https://www.zhouxiaozhao.cn/2021/01/05/2020-01-05-RNAseq(18)/">Transcript Assembly Merge</a></li><li><a href="https://www.zhouxiaozhao.cn/2021/01/07/2021-01-07-RNAseq(19)/">Differential Splicing</a></li><li><a href="https://www.zhouxiaozhao.cn/2021/01/09/2021-01-09-RNAseq(20)/">Splicing Visualization</a></li></ol><p><strong>5.Module 5 - De novo transcript reconstruction</strong></p><ol><li><a href="https://www.zhouxiaozhao.cn/2021/01/12/2021-01-12-RNAseq(21)/">De novo RNA-Seq Assembly and Analysis Using Trinity</a></li></ol><p><strong>6.Module 6 - Functional Annotation of Transcripts</strong></p><ol><li><a href="https://www.zhouxiaozhao.cn/2021/01/14/2021-01-14RNAseq(22)/">Functional Annotation of Assembled Transcripts Using Trinotate</a></li></ol><h2 id="33-de-visualization"><a class="anchor" href="#33-de-visualization">#</a> 3.3 DE Visualization</h2><p>在正确的目录（之前的 UHR_vs_HBR.csv），然后启动 R:</p><figure class="highlight raw"><figcaption data-lang></figcaption><table><tr><td data-num="1"></td><td><pre># Load libraries needed for this analysis</pre></td></tr><tr><td data-num="2"></td><td><pre>library(ballgown)</pre></td></tr><tr><td data-num="3"></td><td><pre>library(genefilter)</pre></td></tr><tr><td data-num="4"></td><td><pre>library(dplyr)</pre></td></tr><tr><td data-num="5"></td><td><pre>library(devtools)</pre></td></tr><tr><td data-num="6"></td><td><pre></pre></td></tr><tr><td data-num="7"></td><td><pre># Define a path for the output PDF to be written</pre></td></tr><tr><td data-num="8"></td><td><pre>outfile&#x3D;&quot;~&#x2F;workspace&#x2F;rnaseq&#x2F;de&#x2F;ballgown&#x2F;ref_only&#x2F;Tutorial_Part2_ballgown_output.pdf&quot;</pre></td></tr><tr><td data-num="9"></td><td><pre></pre></td></tr><tr><td data-num="10"></td><td><pre># Load phenotype data</pre></td></tr><tr><td data-num="11"></td><td><pre>pheno_data &#x3D; read.csv(&quot;UHR_vs_HBR.csv&quot;)</pre></td></tr><tr><td data-num="12"></td><td><pre></pre></td></tr><tr><td data-num="13"></td><td><pre># Display the phenotype data</pre></td></tr><tr><td data-num="14"></td><td><pre>pheno_data</pre></td></tr><tr><td data-num="15"></td><td><pre></pre></td></tr><tr><td data-num="16"></td><td><pre># Load the ballgown object from file</pre></td></tr><tr><td data-num="17"></td><td><pre>load(&#39;bg.rda&#39;)</pre></td></tr><tr><td data-num="18"></td><td><pre></pre></td></tr><tr><td data-num="19"></td><td><pre># The load command, loads an R object from a file into memory in our R session.</pre></td></tr><tr><td data-num="20"></td><td><pre># You can use ls() to view the names of variables that have been loaded</pre></td></tr><tr><td data-num="21"></td><td><pre>ls()</pre></td></tr><tr><td data-num="22"></td><td><pre></pre></td></tr><tr><td data-num="23"></td><td><pre># Print a summary of the ballgown object</pre></td></tr><tr><td data-num="24"></td><td><pre>bg</pre></td></tr><tr><td data-num="25"></td><td><pre></pre></td></tr><tr><td data-num="26"></td><td><pre># Open a PDF file where we will save some plots. We will save all figures and then view the PDF at the end</pre></td></tr><tr><td data-num="27"></td><td><pre>pdf(file&#x3D;outfile)</pre></td></tr><tr><td data-num="28"></td><td><pre></pre></td></tr><tr><td data-num="29"></td><td><pre># Extract FPKM values from the &#39;bg&#39; object</pre></td></tr><tr><td data-num="30"></td><td><pre>fpkm &#x3D; texpr(bg,meas&#x3D;&quot;FPKM&quot;)</pre></td></tr><tr><td data-num="31"></td><td><pre></pre></td></tr><tr><td data-num="32"></td><td><pre># View the last several rows of the FPKM table</pre></td></tr><tr><td data-num="33"></td><td><pre>tail(fpkm)</pre></td></tr><tr><td data-num="34"></td><td><pre></pre></td></tr><tr><td data-num="35"></td><td><pre># Transform the FPKM values by adding 1 and convert to a log2 scale</pre></td></tr><tr><td data-num="36"></td><td><pre>fpkm &#x3D; log2(fpkm+1)</pre></td></tr><tr><td data-num="37"></td><td><pre></pre></td></tr><tr><td data-num="38"></td><td><pre># View the last several rows of the transformed FPKM table</pre></td></tr><tr><td data-num="39"></td><td><pre>tail(fpkm)</pre></td></tr><tr><td data-num="40"></td><td><pre></pre></td></tr><tr><td data-num="41"></td><td><pre># Create boxplots to display summary statistics for the FPKM values for each sample</pre></td></tr><tr><td data-num="42"></td><td><pre>boxplot(fpkm,col&#x3D;as.numeric(pheno_data$type)+1,las&#x3D;2,ylab&#x3D;&#39;log2(FPKM+1)&#39;)</pre></td></tr><tr><td data-num="43"></td><td><pre></pre></td></tr><tr><td data-num="44"></td><td><pre># Display the transcript ID for a single row of data</pre></td></tr><tr><td data-num="45"></td><td><pre>ballgown::transcriptNames(bg)[2763]</pre></td></tr><tr><td data-num="46"></td><td><pre></pre></td></tr><tr><td data-num="47"></td><td><pre># Display the gene name for a single row of data</pre></td></tr><tr><td data-num="48"></td><td><pre>ballgown::geneNames(bg)[2763]</pre></td></tr><tr><td data-num="49"></td><td><pre></pre></td></tr><tr><td data-num="50"></td><td><pre># Create a BoxPlot comparing the expression of a single gene for all replicates of both conditions</pre></td></tr><tr><td data-num="51"></td><td><pre>plot(fpkm[2763,] ~ pheno_data$type, border&#x3D;c(2,3), main&#x3D;paste(ballgown::geneNames(bg)[2763],&#39; : &#39;, ballgown::transcriptNames(bg)[2763]),pch&#x3D;19, xlab&#x3D;&quot;Type&quot;, ylab&#x3D;&#39;log2(FPKM+1)&#39;)</pre></td></tr><tr><td data-num="52"></td><td><pre></pre></td></tr><tr><td data-num="53"></td><td><pre># Add the FPKM values for each sample onto the plot</pre></td></tr><tr><td data-num="54"></td><td><pre>points(fpkm[2763,] ~ jitter(as.numeric(pheno_data$type)), col&#x3D;as.numeric(pheno_data$type)+1, pch&#x3D;16)</pre></td></tr><tr><td data-num="55"></td><td><pre></pre></td></tr><tr><td data-num="56"></td><td><pre># Create a plot of transcript structures observed in each replicate and color transcripts by expression level</pre></td></tr><tr><td data-num="57"></td><td><pre>plotTranscripts(ballgown::geneIDs(bg)[2763], bg, main&#x3D;c(&#39;Gene in sample HBR_Rep1&#39;), sample&#x3D;c(&#39;HBR_Rep1&#39;))</pre></td></tr><tr><td data-num="58"></td><td><pre>plotTranscripts(ballgown::geneIDs(bg)[2763], bg, main&#x3D;c(&#39;Gene in sample HBR_Rep2&#39;), sample&#x3D;c(&#39;HBR_Rep2&#39;))</pre></td></tr><tr><td data-num="59"></td><td><pre>plotTranscripts(ballgown::geneIDs(bg)[2763], bg, main&#x3D;c(&#39;Gene in sample HBR_Rep3&#39;), sample&#x3D;c(&#39;HBR_Rep3&#39;))</pre></td></tr><tr><td data-num="60"></td><td><pre>plotTranscripts(ballgown::geneIDs(bg)[2763], bg, main&#x3D;c(&#39;Gene in sample UHR_Rep1&#39;), sample&#x3D;c(&#39;UHR_Rep1&#39;))</pre></td></tr><tr><td data-num="61"></td><td><pre>plotTranscripts(ballgown::geneIDs(bg)[2763], bg, main&#x3D;c(&#39;Gene in sample UHR_Rep2&#39;), sample&#x3D;c(&#39;UHR_Rep2&#39;))</pre></td></tr><tr><td data-num="62"></td><td><pre>plotTranscripts(ballgown::geneIDs(bg)[2763], bg, main&#x3D;c(&#39;Gene in sample UHR_Rep3&#39;), sample&#x3D;c(&#39;UHR_Rep3&#39;))</pre></td></tr><tr><td data-num="63"></td><td><pre></pre></td></tr><tr><td data-num="64"></td><td><pre>#plotMeans(&#39;TST&#39;,bg,groupvar&#x3D;&quot;type&quot;,legend&#x3D;FALSE)</pre></td></tr><tr><td data-num="65"></td><td><pre></pre></td></tr><tr><td data-num="66"></td><td><pre># Close the PDF device where we have been saving our plots</pre></td></tr><tr><td data-num="67"></td><td><pre>dev.off()</pre></td></tr><tr><td data-num="68"></td><td><pre></pre></td></tr><tr><td data-num="69"></td><td><pre># Exit the R session</pre></td></tr><tr><td data-num="70"></td><td><pre>quit(save&#x3D;&quot;no&quot;)</pre></td></tr></table></figure><p>查看一下输出的 pdf 文件</p><p><img data-src="/img/posts/2020.12.26/image-20210105164226503.png" alt="image-20210105164226503"></p><h3 id="额外分析"><a class="anchor" href="#额外分析">#</a> 额外分析</h3><p>运行 R 脚本</p><figure class="highlight raw"><figcaption data-lang></figcaption><table><tr><td data-num="1"></td><td><pre>#Tutorial_Part3_Supplementary_R.R</pre></td></tr><tr><td data-num="2"></td><td><pre></pre></td></tr><tr><td data-num="3"></td><td><pre>#Malachi Griffith, mgriffit[AT]wustl.edu</pre></td></tr><tr><td data-num="4"></td><td><pre>#Obi Griffith, obigriffith[AT]wustl.edu</pre></td></tr><tr><td data-num="5"></td><td><pre>#Jason Walker, jason.walker[AT]wustl.edu</pre></td></tr><tr><td data-num="6"></td><td><pre>#McDonnell Genome Institute, Washington University School of Medicine</pre></td></tr><tr><td data-num="7"></td><td><pre></pre></td></tr><tr><td data-num="8"></td><td><pre>#R tutorial for Informatics for RNA-sequence Analysis workshops</pre></td></tr><tr><td data-num="9"></td><td><pre></pre></td></tr><tr><td data-num="10"></td><td><pre>#Starting from the output of the RNA-seq Tutorial Part 1.</pre></td></tr><tr><td data-num="11"></td><td><pre></pre></td></tr><tr><td data-num="12"></td><td><pre>#Load libraries</pre></td></tr><tr><td data-num="13"></td><td><pre>library(ggplot2)</pre></td></tr><tr><td data-num="14"></td><td><pre>library(gplots)</pre></td></tr><tr><td data-num="15"></td><td><pre>library(GenomicRanges)</pre></td></tr><tr><td data-num="16"></td><td><pre>library(ballgown)</pre></td></tr><tr><td data-num="17"></td><td><pre></pre></td></tr><tr><td data-num="18"></td><td><pre>#If X11 not available, open a pdf device for output of all plots</pre></td></tr><tr><td data-num="19"></td><td><pre>pdf(file&#x3D;&quot;Tutorial_Part3_Supplementary_R_output.pdf&quot;)</pre></td></tr><tr><td data-num="20"></td><td><pre></pre></td></tr><tr><td data-num="21"></td><td><pre>#### Import the gene expression data from the HISAT2&#x2F;StringTie&#x2F;Ballgown tutorial</pre></td></tr><tr><td data-num="22"></td><td><pre></pre></td></tr><tr><td data-num="23"></td><td><pre>#Set working directory where results files exist</pre></td></tr><tr><td data-num="24"></td><td><pre>working_dir &#x3D; &quot;~&#x2F;workspace&#x2F;rnaseq&#x2F;de&#x2F;ballgown&#x2F;ref_only&quot;</pre></td></tr><tr><td data-num="25"></td><td><pre>setwd(working_dir)</pre></td></tr><tr><td data-num="26"></td><td><pre></pre></td></tr><tr><td data-num="27"></td><td><pre># List the current contents of this directory</pre></td></tr><tr><td data-num="28"></td><td><pre>dir()</pre></td></tr><tr><td data-num="29"></td><td><pre></pre></td></tr><tr><td data-num="30"></td><td><pre>#Import expression and differential expression results from the HISAT2&#x2F;StringTie&#x2F;Ballgown pipeline</pre></td></tr><tr><td data-num="31"></td><td><pre>load(&#39;bg.rda&#39;)</pre></td></tr><tr><td data-num="32"></td><td><pre></pre></td></tr><tr><td data-num="33"></td><td><pre># View a summary of the ballgown object</pre></td></tr><tr><td data-num="34"></td><td><pre>bg</pre></td></tr><tr><td data-num="35"></td><td><pre></pre></td></tr><tr><td data-num="36"></td><td><pre># Load gene names for lookup later in the tutorial</pre></td></tr><tr><td data-num="37"></td><td><pre>bg_table &#x3D; texpr(bg, &#39;all&#39;)</pre></td></tr><tr><td data-num="38"></td><td><pre>bg_gene_names &#x3D; unique(bg_table[, 9:10])</pre></td></tr><tr><td data-num="39"></td><td><pre></pre></td></tr><tr><td data-num="40"></td><td><pre># Pull the gene_expression data frame from the ballgown object</pre></td></tr><tr><td data-num="41"></td><td><pre>gene_expression &#x3D; as.data.frame(gexpr(bg))</pre></td></tr><tr><td data-num="42"></td><td><pre></pre></td></tr><tr><td data-num="43"></td><td><pre>#### Working with &#39;dataframes&#39;</pre></td></tr><tr><td data-num="44"></td><td><pre>#View the first five rows of data (all columns) in one of the dataframes created</pre></td></tr><tr><td data-num="45"></td><td><pre>head(gene_expression)</pre></td></tr><tr><td data-num="46"></td><td><pre></pre></td></tr><tr><td data-num="47"></td><td><pre>#View the column names</pre></td></tr><tr><td data-num="48"></td><td><pre>colnames(gene_expression)</pre></td></tr><tr><td data-num="49"></td><td><pre></pre></td></tr><tr><td data-num="50"></td><td><pre>#View the row names</pre></td></tr><tr><td data-num="51"></td><td><pre>row.names(gene_expression)</pre></td></tr><tr><td data-num="52"></td><td><pre></pre></td></tr><tr><td data-num="53"></td><td><pre>#Determine the dimensions of the dataframe.  &#39;dim()&#39; will return the number of rows and columns</pre></td></tr><tr><td data-num="54"></td><td><pre>dim(gene_expression)</pre></td></tr><tr><td data-num="55"></td><td><pre></pre></td></tr><tr><td data-num="56"></td><td><pre>#Get the first 3 rows of data and a selection of columns</pre></td></tr><tr><td data-num="57"></td><td><pre>gene_expression[1:3,c(1:3,6)]</pre></td></tr><tr><td data-num="58"></td><td><pre></pre></td></tr><tr><td data-num="59"></td><td><pre>#Do the same thing, but using the column names instead of numbers</pre></td></tr><tr><td data-num="60"></td><td><pre>gene_expression[1:3, c(&quot;FPKM.UHR_Rep1&quot;,&quot;FPKM.UHR_Rep2&quot;,&quot;FPKM.UHR_Rep3&quot;,&quot;FPKM.HBR_Rep3&quot;)]</pre></td></tr><tr><td data-num="61"></td><td><pre></pre></td></tr><tr><td data-num="62"></td><td><pre>#Assign colors to each.  You can specify color by RGB, Hex code, or name</pre></td></tr><tr><td data-num="63"></td><td><pre>#To get a list of color names:</pre></td></tr><tr><td data-num="64"></td><td><pre>colours()</pre></td></tr><tr><td data-num="65"></td><td><pre>data_colors&#x3D;c(&quot;tomato1&quot;,&quot;tomato2&quot;,&quot;tomato3&quot;,&quot;royalblue1&quot;,&quot;royalblue2&quot;,&quot;royalblue3&quot;)</pre></td></tr><tr><td data-num="66"></td><td><pre></pre></td></tr><tr><td data-num="67"></td><td><pre>#View expression values for the transcripts of a particular gene symbol of chromosome 22.  e.g. &#39;TST&#39;</pre></td></tr><tr><td data-num="68"></td><td><pre>#First determine the rows in the data.frame that match &#39;TST&#39;, aka. ENSG00000128311, then display only those rows of the data.frame</pre></td></tr><tr><td data-num="69"></td><td><pre>i &#x3D; row.names(gene_expression) &#x3D;&#x3D; &quot;ENSG00000128311&quot;</pre></td></tr><tr><td data-num="70"></td><td><pre>gene_expression[i,]</pre></td></tr><tr><td data-num="71"></td><td><pre></pre></td></tr><tr><td data-num="72"></td><td><pre>#What if we want to view values for a list of genes of interest all at once?</pre></td></tr><tr><td data-num="73"></td><td><pre>#genes_of_interest &#x3D; c(&quot;TST&quot;, &quot;MMP11&quot;, &quot;LGALS2&quot;, &quot;ISX&quot;)</pre></td></tr><tr><td data-num="74"></td><td><pre>genes_of_interest &#x3D; c(&quot;ENSG00000128311&quot;,&quot;ENSG00000099953&quot;,&quot;ENSG00000100079&quot;,&quot;ENSG00000175329&quot;)</pre></td></tr><tr><td data-num="75"></td><td><pre>i &#x3D; which(row.names(gene_expression) %in% genes_of_interest)</pre></td></tr><tr><td data-num="76"></td><td><pre>gene_expression[i,]</pre></td></tr><tr><td data-num="77"></td><td><pre></pre></td></tr><tr><td data-num="78"></td><td><pre># Load the transcript to gene index from the ballgown object</pre></td></tr><tr><td data-num="79"></td><td><pre>transcript_gene_table &#x3D; indexes(bg)$t2g</pre></td></tr><tr><td data-num="80"></td><td><pre>head(transcript_gene_table)</pre></td></tr><tr><td data-num="81"></td><td><pre></pre></td></tr><tr><td data-num="82"></td><td><pre>#Each row of data represents a transcript. Many of these transcripts represent the same gene. Determine the numbers of transcripts and unique genes  </pre></td></tr><tr><td data-num="83"></td><td><pre>length(row.names(transcript_gene_table)) #Transcript count</pre></td></tr><tr><td data-num="84"></td><td><pre>length(unique(transcript_gene_table[,&quot;g_id&quot;])) #Unique Gene count</pre></td></tr><tr><td data-num="85"></td><td><pre></pre></td></tr><tr><td data-num="86"></td><td><pre>#### Plot #1 - the number of transcripts per gene.  </pre></td></tr><tr><td data-num="87"></td><td><pre>#Many genes will have only 1 transcript, some genes will have several transcripts</pre></td></tr><tr><td data-num="88"></td><td><pre>#Use the &#39;table()&#39; command to count the number of times each gene symbol occurs (i.e. the # of transcripts that have each gene symbol)</pre></td></tr><tr><td data-num="89"></td><td><pre>#Then use the &#39;hist&#39; command to create a histogram of these counts</pre></td></tr><tr><td data-num="90"></td><td><pre>#How many genes have 1 transcript?  More than one transcript?  What is the maximum number of transcripts for a single gene?</pre></td></tr><tr><td data-num="91"></td><td><pre>counts&#x3D;table(transcript_gene_table[,&quot;g_id&quot;])</pre></td></tr><tr><td data-num="92"></td><td><pre>c_one &#x3D; length(which(counts &#x3D;&#x3D; 1))</pre></td></tr><tr><td data-num="93"></td><td><pre>c_more_than_one &#x3D; length(which(counts &gt; 1))</pre></td></tr><tr><td data-num="94"></td><td><pre>c_max &#x3D; max(counts)</pre></td></tr><tr><td data-num="95"></td><td><pre>hist(counts, breaks&#x3D;50, col&#x3D;&quot;bisque4&quot;, xlab&#x3D;&quot;Transcripts per gene&quot;, main&#x3D;&quot;Distribution of transcript count per gene&quot;)</pre></td></tr><tr><td data-num="96"></td><td><pre>legend_text &#x3D; c(paste(&quot;Genes with one transcript &#x3D;&quot;, c_one), paste(&quot;Genes with more than one transcript &#x3D;&quot;, c_more_than_one), paste(&quot;Max transcripts for single gene &#x3D; &quot;, c_max))</pre></td></tr><tr><td data-num="97"></td><td><pre>legend(&quot;topright&quot;, legend_text, lty&#x3D;NULL)</pre></td></tr><tr><td data-num="98"></td><td><pre></pre></td></tr><tr><td data-num="99"></td><td><pre></pre></td></tr><tr><td data-num="100"></td><td><pre>#### Plot #2 - the distribution of transcript sizes as a histogram</pre></td></tr><tr><td data-num="101"></td><td><pre>#In this analysis we supplied StringTie with transcript models so the lengths will be those of known transcripts</pre></td></tr><tr><td data-num="102"></td><td><pre>#However, if we had used a de novo transcript discovery mode, this step would give us some idea of how well transcripts were being assembled</pre></td></tr><tr><td data-num="103"></td><td><pre>#If we had a low coverage library, or other problems, we might get short &#39;transcripts&#39; that are actually only pieces of real transcripts</pre></td></tr><tr><td data-num="104"></td><td><pre></pre></td></tr><tr><td data-num="105"></td><td><pre>full_table &lt;- texpr(bg , &#39;all&#39;)</pre></td></tr><tr><td data-num="106"></td><td><pre>hist(full_table$length, breaks&#x3D;50, xlab&#x3D;&quot;Transcript length (bp)&quot;, main&#x3D;&quot;Distribution of transcript lengths&quot;, col&#x3D;&quot;steelblue&quot;)</pre></td></tr><tr><td data-num="107"></td><td><pre></pre></td></tr><tr><td data-num="108"></td><td><pre>#### Summarize FPKM values for all 6 replicates</pre></td></tr><tr><td data-num="109"></td><td><pre>#What are the minimum and maximum FPKM values for a particular library?</pre></td></tr><tr><td data-num="110"></td><td><pre>min(gene_expression[,&quot;FPKM.UHR_Rep1&quot;])</pre></td></tr><tr><td data-num="111"></td><td><pre>max(gene_expression[,&quot;FPKM.UHR_Rep1&quot;])</pre></td></tr><tr><td data-num="112"></td><td><pre></pre></td></tr><tr><td data-num="113"></td><td><pre>#Set the minimum non-zero FPKM values for use later.</pre></td></tr><tr><td data-num="114"></td><td><pre>#Do this by grabbing a copy of all data values, coverting 0&#39;s to NA, and calculating the minimum or all non NA values</pre></td></tr><tr><td data-num="115"></td><td><pre>#zz &#x3D; fpkm_matrix[,data_columns]</pre></td></tr><tr><td data-num="116"></td><td><pre>#zz[zz&#x3D;&#x3D;0] &#x3D; NA</pre></td></tr><tr><td data-num="117"></td><td><pre>#min_nonzero &#x3D; min(zz, na.rm&#x3D;TRUE)</pre></td></tr><tr><td data-num="118"></td><td><pre>#min_nonzero</pre></td></tr><tr><td data-num="119"></td><td><pre></pre></td></tr><tr><td data-num="120"></td><td><pre>#Alternatively just set min value to 1</pre></td></tr><tr><td data-num="121"></td><td><pre>min_nonzero&#x3D;1</pre></td></tr><tr><td data-num="122"></td><td><pre></pre></td></tr><tr><td data-num="123"></td><td><pre># Set the columns for finding FPKM and create shorter names for figures</pre></td></tr><tr><td data-num="124"></td><td><pre>data_columns&#x3D;c(1:6)</pre></td></tr><tr><td data-num="125"></td><td><pre>short_names&#x3D;c(&quot;UHR_1&quot;,&quot;UHR_2&quot;,&quot;UHR_3&quot;,&quot;HBR_1&quot;,&quot;HBR_2&quot;,&quot;HBR_3&quot;)</pre></td></tr><tr><td data-num="126"></td><td><pre></pre></td></tr><tr><td data-num="127"></td><td><pre>#### Plot #3 - View the range of values and general distribution of FPKM values for all 4 libraries</pre></td></tr><tr><td data-num="128"></td><td><pre>#Create boxplots for this purpose</pre></td></tr><tr><td data-num="129"></td><td><pre>#Display on a log2 scale and add the minimum non-zero value to avoid log2(0)</pre></td></tr><tr><td data-num="130"></td><td><pre>boxplot(log2(gene_expression[,data_columns]+min_nonzero), col&#x3D;data_colors, names&#x3D;short_names, las&#x3D;2, ylab&#x3D;&quot;log2(FPKM)&quot;, main&#x3D;&quot;Distribution of FPKMs for all 6 libraries&quot;)</pre></td></tr><tr><td data-num="131"></td><td><pre>#Note that the bold horizontal line on each boxplot is the median</pre></td></tr><tr><td data-num="132"></td><td><pre></pre></td></tr><tr><td data-num="133"></td><td><pre>#### Plot #4 - plot a pair of replicates to assess reproducibility of technical replicates</pre></td></tr><tr><td data-num="134"></td><td><pre>#Tranform the data by converting to log2 scale after adding an arbitrary small value to avoid log2(0)</pre></td></tr><tr><td data-num="135"></td><td><pre>x &#x3D; gene_expression[,&quot;FPKM.UHR_Rep1&quot;]</pre></td></tr><tr><td data-num="136"></td><td><pre>y &#x3D; gene_expression[,&quot;FPKM.UHR_Rep2&quot;]</pre></td></tr><tr><td data-num="137"></td><td><pre>plot(x&#x3D;log2(x+min_nonzero), y&#x3D;log2(y+min_nonzero), pch&#x3D;16, col&#x3D;&quot;blue&quot;, cex&#x3D;0.25, xlab&#x3D;&quot;FPKM (UHR, Replicate 1)&quot;, ylab&#x3D;&quot;FPKM (UHR, Replicate 2)&quot;, main&#x3D;&quot;Comparison of expression values for a pair of replicates&quot;)</pre></td></tr><tr><td data-num="138"></td><td><pre></pre></td></tr><tr><td data-num="139"></td><td><pre>#Add a straight line of slope 1, and intercept 0</pre></td></tr><tr><td data-num="140"></td><td><pre>abline(a&#x3D;0,b&#x3D;1)</pre></td></tr><tr><td data-num="141"></td><td><pre></pre></td></tr><tr><td data-num="142"></td><td><pre>#Calculate the correlation coefficient and display in a legend</pre></td></tr><tr><td data-num="143"></td><td><pre>rs&#x3D;cor(x,y)^2</pre></td></tr><tr><td data-num="144"></td><td><pre>legend(&quot;topleft&quot;, paste(&quot;R squared &#x3D; &quot;, round(rs, digits&#x3D;3), sep&#x3D;&quot;&quot;), lwd&#x3D;1, col&#x3D;&quot;black&quot;)</pre></td></tr><tr><td data-num="145"></td><td><pre></pre></td></tr><tr><td data-num="146"></td><td><pre>#### Plot #5 - Scatter plots with a large number of data points can be misleading ... regenerate this figure as a density scatter plot</pre></td></tr><tr><td data-num="147"></td><td><pre>colors &#x3D; colorRampPalette(c(&quot;white&quot;, &quot;blue&quot;, &quot;#007FFF&quot;, &quot;cyan&quot;,&quot;#7FFF7F&quot;, &quot;yellow&quot;, &quot;#FF7F00&quot;, &quot;red&quot;, &quot;#7F0000&quot;))</pre></td></tr><tr><td data-num="148"></td><td><pre>smoothScatter(x&#x3D;log2(x+min_nonzero), y&#x3D;log2(y+min_nonzero), xlab&#x3D;&quot;FPKM (UHR, Replicate 1)&quot;, ylab&#x3D;&quot;FPKM (UHR, Replicate 2)&quot;, main&#x3D;&quot;Comparison of expression values for a pair of replicates&quot;, colramp&#x3D;colors, nbin&#x3D;200)</pre></td></tr><tr><td data-num="149"></td><td><pre></pre></td></tr><tr><td data-num="150"></td><td><pre></pre></td></tr><tr><td data-num="151"></td><td><pre>#### Plot all sets of replicates on a single plot</pre></td></tr><tr><td data-num="152"></td><td><pre>#Create an function that generates an R plot.  This function will take as input the two libraries to be compared and a plot name and color</pre></td></tr><tr><td data-num="153"></td><td><pre>plotCor &#x3D; function(lib1, lib2, name, color)&#123;</pre></td></tr><tr><td data-num="154"></td><td><pre>	x&#x3D;gene_expression[,lib1]</pre></td></tr><tr><td data-num="155"></td><td><pre>	y&#x3D;gene_expression[,lib2]</pre></td></tr><tr><td data-num="156"></td><td><pre>	zero_count &#x3D; length(which(x&#x3D;&#x3D;0)) + length(which(y&#x3D;&#x3D;0))</pre></td></tr><tr><td data-num="157"></td><td><pre>	plot(x&#x3D;log2(x+min_nonzero), y&#x3D;log2(y+min_nonzero), pch&#x3D;16, col&#x3D;color, cex&#x3D;0.25, xlab&#x3D;lib1, ylab&#x3D;lib2, main&#x3D;name)</pre></td></tr><tr><td data-num="158"></td><td><pre>	abline(a&#x3D;0,b&#x3D;1)</pre></td></tr><tr><td data-num="159"></td><td><pre>	rs&#x3D;cor(x,y, method&#x3D;&quot;pearson&quot;)^2</pre></td></tr><tr><td data-num="160"></td><td><pre>	legend_text &#x3D; c(paste(&quot;R squared &#x3D; &quot;, round(rs, digits&#x3D;3), sep&#x3D;&quot;&quot;), paste(&quot;Zero count &#x3D; &quot;, zero_count, sep&#x3D;&quot;&quot;))</pre></td></tr><tr><td data-num="161"></td><td><pre>	legend(&quot;topleft&quot;, legend_text, lwd&#x3D;c(1,NA), col&#x3D;&quot;black&quot;, bg&#x3D;&quot;white&quot;, cex&#x3D;0.8)</pre></td></tr><tr><td data-num="162"></td><td><pre>&#125;</pre></td></tr><tr><td data-num="163"></td><td><pre>#Open a plotting page with room for two plots on one page</pre></td></tr><tr><td data-num="164"></td><td><pre>par(mfrow&#x3D;c(1,2))</pre></td></tr><tr><td data-num="165"></td><td><pre></pre></td></tr><tr><td data-num="166"></td><td><pre>#Plot #6 - Now make a call to our custom function created above, once for each library comparison</pre></td></tr><tr><td data-num="167"></td><td><pre>plotCor(&quot;FPKM.UHR_Rep1&quot;, &quot;FPKM.HBR_Rep1&quot;, &quot;UHR_1 vs HBR_1&quot;, &quot;tomato2&quot;)</pre></td></tr><tr><td data-num="168"></td><td><pre>plotCor(&quot;FPKM.UHR_Rep2&quot;, &quot;FPKM.HBR_Rep2&quot;, &quot;UHR_2 vs HBR_2&quot;, &quot;royalblue2&quot;)</pre></td></tr><tr><td data-num="169"></td><td><pre></pre></td></tr><tr><td data-num="170"></td><td><pre></pre></td></tr><tr><td data-num="171"></td><td><pre>##### One problem with these plots is that there are so many data points on top of each other, that information is being lost</pre></td></tr><tr><td data-num="172"></td><td><pre>#Regenerate these plots using a density scatter plot</pre></td></tr><tr><td data-num="173"></td><td><pre>plotCor2 &#x3D; function(lib1, lib2, name, color)&#123;</pre></td></tr><tr><td data-num="174"></td><td><pre>	x&#x3D;gene_expression[,lib1]</pre></td></tr><tr><td data-num="175"></td><td><pre>	y&#x3D;gene_expression[,lib2]</pre></td></tr><tr><td data-num="176"></td><td><pre>	zero_count &#x3D; length(which(x&#x3D;&#x3D;0)) + length(which(y&#x3D;&#x3D;0))</pre></td></tr><tr><td data-num="177"></td><td><pre>	colors &#x3D; colorRampPalette(c(&quot;white&quot;, &quot;blue&quot;, &quot;#007FFF&quot;, &quot;cyan&quot;,&quot;#7FFF7F&quot;, &quot;yellow&quot;, &quot;#FF7F00&quot;, &quot;red&quot;, &quot;#7F0000&quot;))</pre></td></tr><tr><td data-num="178"></td><td><pre>	smoothScatter(x&#x3D;log2(x+min_nonzero), y&#x3D;log2(y+min_nonzero), xlab&#x3D;lib1, ylab&#x3D;lib2, main&#x3D;name, colramp&#x3D;colors, nbin&#x3D;275)</pre></td></tr><tr><td data-num="179"></td><td><pre>	abline(a&#x3D;0,b&#x3D;1)</pre></td></tr><tr><td data-num="180"></td><td><pre>	rs&#x3D;cor(x,y, method&#x3D;&quot;pearson&quot;)^2</pre></td></tr><tr><td data-num="181"></td><td><pre>	legend_text &#x3D; c(paste(&quot;R squared &#x3D; &quot;, round(rs, digits&#x3D;3), sep&#x3D;&quot;&quot;), paste(&quot;Zero count &#x3D; &quot;, zero_count, sep&#x3D;&quot;&quot;))</pre></td></tr><tr><td data-num="182"></td><td><pre>	legend(&quot;topleft&quot;, legend_text, lwd&#x3D;c(1,NA), col&#x3D;&quot;black&quot;, bg&#x3D;&quot;white&quot;, cex&#x3D;0.8)</pre></td></tr><tr><td data-num="183"></td><td><pre>&#125;</pre></td></tr><tr><td data-num="184"></td><td><pre></pre></td></tr><tr><td data-num="185"></td><td><pre>#### Plot #7 - Now make a call to our custom function created above, once for each library comparison</pre></td></tr><tr><td data-num="186"></td><td><pre>par(mfrow&#x3D;c(1,2))</pre></td></tr><tr><td data-num="187"></td><td><pre>plotCor2(&quot;FPKM.UHR_Rep1&quot;, &quot;FPKM.HBR_Rep1&quot;, &quot;UHR_1 vs HBR_1&quot;, &quot;tomato2&quot;)</pre></td></tr><tr><td data-num="188"></td><td><pre>plotCor2(&quot;FPKM.UHR_Rep2&quot;, &quot;FPKM.HBR_Rep2&quot;, &quot;UHR_2 vs HBR_2&quot;, &quot;royalblue2&quot;)</pre></td></tr><tr><td data-num="189"></td><td><pre></pre></td></tr><tr><td data-num="190"></td><td><pre></pre></td></tr><tr><td data-num="191"></td><td><pre>#### Compare the correlation &#39;distance&#39; between all replicates</pre></td></tr><tr><td data-num="192"></td><td><pre>#Do we see the expected pattern for all eight libraries (i.e. replicates most similar, then tumor vs. normal)?</pre></td></tr><tr><td data-num="193"></td><td><pre></pre></td></tr><tr><td data-num="194"></td><td><pre>#Calculate the FPKM sum for all 6 libraries</pre></td></tr><tr><td data-num="195"></td><td><pre>gene_expression[,&quot;sum&quot;]&#x3D;apply(gene_expression[,data_columns], 1, sum)</pre></td></tr><tr><td data-num="196"></td><td><pre></pre></td></tr><tr><td data-num="197"></td><td><pre>#Identify the genes with a grand sum FPKM of at least 5 - we will filter out the genes with very low expression across the board</pre></td></tr><tr><td data-num="198"></td><td><pre>i &#x3D; which(gene_expression[,&quot;sum&quot;] &gt; 5)</pre></td></tr><tr><td data-num="199"></td><td><pre></pre></td></tr><tr><td data-num="200"></td><td><pre>#Calculate the correlation between all pairs of data</pre></td></tr><tr><td data-num="201"></td><td><pre>r&#x3D;cor(gene_expression[i,data_columns], use&#x3D;&quot;pairwise.complete.obs&quot;, method&#x3D;&quot;pearson&quot;)</pre></td></tr><tr><td data-num="202"></td><td><pre></pre></td></tr><tr><td data-num="203"></td><td><pre>#Print out these correlation values</pre></td></tr><tr><td data-num="204"></td><td><pre>r</pre></td></tr><tr><td data-num="205"></td><td><pre></pre></td></tr><tr><td data-num="206"></td><td><pre>#### Plot #8 - Convert correlation to &#39;distance&#39;, and use &#39;multi-dimensional scaling&#39; to display the relative differences between libraries</pre></td></tr><tr><td data-num="207"></td><td><pre>#This step calculates 2-dimensional coordinates to plot points for each library</pre></td></tr><tr><td data-num="208"></td><td><pre>#Libraries with similar expression patterns (highly correlated to each other) should group together</pre></td></tr><tr><td data-num="209"></td><td><pre>#What pattern do we expect to see, given the types of libraries we have (technical replicates, biologal replicates, tumor&#x2F;normal)?</pre></td></tr><tr><td data-num="210"></td><td><pre>d&#x3D;1-r</pre></td></tr><tr><td data-num="211"></td><td><pre>mds&#x3D;cmdscale(d, k&#x3D;2, eig&#x3D;TRUE)</pre></td></tr><tr><td data-num="212"></td><td><pre>par(mfrow&#x3D;c(1,1))</pre></td></tr><tr><td data-num="213"></td><td><pre>plot(mds$points, type&#x3D;&quot;n&quot;, xlab&#x3D;&quot;&quot;, ylab&#x3D;&quot;&quot;, main&#x3D;&quot;MDS distance plot (all non-zero genes)&quot;, xlim&#x3D;c(-0.12,0.12), ylim&#x3D;c(-0.12,0.12))</pre></td></tr><tr><td data-num="214"></td><td><pre>points(mds$points[,1], mds$points[,2], col&#x3D;&quot;grey&quot;, cex&#x3D;2, pch&#x3D;16)</pre></td></tr><tr><td data-num="215"></td><td><pre>text(mds$points[,1], mds$points[,2], short_names, col&#x3D;data_colors)</pre></td></tr><tr><td data-num="216"></td><td><pre></pre></td></tr><tr><td data-num="217"></td><td><pre># Calculate the differential expression results including significance</pre></td></tr><tr><td data-num="218"></td><td><pre>results_genes &#x3D; stattest(bg, feature&#x3D;&quot;gene&quot;, covariate&#x3D;&quot;type&quot;, getFC&#x3D;TRUE, meas&#x3D;&quot;FPKM&quot;)</pre></td></tr><tr><td data-num="219"></td><td><pre>results_genes &#x3D; merge(results_genes,bg_gene_names,by.x&#x3D;c(&quot;id&quot;),by.y&#x3D;c(&quot;gene_id&quot;))</pre></td></tr><tr><td data-num="220"></td><td><pre></pre></td></tr><tr><td data-num="221"></td><td><pre>#### Plot #9 - View the distribution of differential expression values as a histogram</pre></td></tr><tr><td data-num="222"></td><td><pre>#Display only those that are significant according to Ballgown</pre></td></tr><tr><td data-num="223"></td><td><pre></pre></td></tr><tr><td data-num="224"></td><td><pre>sig&#x3D;which(results_genes$pval&lt;0.05)</pre></td></tr><tr><td data-num="225"></td><td><pre>results_genes[,&quot;de&quot;] &#x3D; log2(results_genes[,&quot;fc&quot;])</pre></td></tr><tr><td data-num="226"></td><td><pre>hist(results_genes[sig,&quot;de&quot;], breaks&#x3D;50, col&#x3D;&quot;seagreen&quot;, xlab&#x3D;&quot;log2(Fold change) UHR vs HBR&quot;, main&#x3D;&quot;Distribution of differential expression values&quot;)</pre></td></tr><tr><td data-num="227"></td><td><pre>abline(v&#x3D;-2, col&#x3D;&quot;black&quot;, lwd&#x3D;2, lty&#x3D;2)</pre></td></tr><tr><td data-num="228"></td><td><pre>abline(v&#x3D;2, col&#x3D;&quot;black&quot;, lwd&#x3D;2, lty&#x3D;2)</pre></td></tr><tr><td data-num="229"></td><td><pre>legend(&quot;topleft&quot;, &quot;Fold-change &gt; 4&quot;, lwd&#x3D;2, lty&#x3D;2)</pre></td></tr><tr><td data-num="230"></td><td><pre></pre></td></tr><tr><td data-num="231"></td><td><pre>#### Plot #10 - Display the grand expression values from UHR and HBR and mark those that are significantly differentially expressed</pre></td></tr><tr><td data-num="232"></td><td><pre>gene_expression[,&quot;UHR&quot;]&#x3D;apply(gene_expression[,c(1:3)], 1, mean)</pre></td></tr><tr><td data-num="233"></td><td><pre>gene_expression[,&quot;HBR&quot;]&#x3D;apply(gene_expression[,c(4:6)], 1, mean)</pre></td></tr><tr><td data-num="234"></td><td><pre></pre></td></tr><tr><td data-num="235"></td><td><pre>x&#x3D;log2(gene_expression[,&quot;UHR&quot;]+min_nonzero)</pre></td></tr><tr><td data-num="236"></td><td><pre>y&#x3D;log2(gene_expression[,&quot;HBR&quot;]+min_nonzero)</pre></td></tr><tr><td data-num="237"></td><td><pre>plot(x&#x3D;x, y&#x3D;y, pch&#x3D;16, cex&#x3D;0.25, xlab&#x3D;&quot;UHR FPKM (log2)&quot;, ylab&#x3D;&quot;HBR FPKM (log2)&quot;, main&#x3D;&quot;UHR vs HBR FPKMs&quot;)</pre></td></tr><tr><td data-num="238"></td><td><pre>abline(a&#x3D;0, b&#x3D;1)</pre></td></tr><tr><td data-num="239"></td><td><pre>xsig&#x3D;x[sig]</pre></td></tr><tr><td data-num="240"></td><td><pre>ysig&#x3D;y[sig]</pre></td></tr><tr><td data-num="241"></td><td><pre>points(x&#x3D;xsig, y&#x3D;ysig, col&#x3D;&quot;magenta&quot;, pch&#x3D;16, cex&#x3D;0.5)</pre></td></tr><tr><td data-num="242"></td><td><pre>legend(&quot;topleft&quot;, &quot;Significant&quot;, col&#x3D;&quot;magenta&quot;, pch&#x3D;16)</pre></td></tr><tr><td data-num="243"></td><td><pre></pre></td></tr><tr><td data-num="244"></td><td><pre>#Get the gene symbols for the top N (according to corrected p-value) and display them on the plot</pre></td></tr><tr><td data-num="245"></td><td><pre>topn &#x3D; order(abs(results_genes[sig,&quot;fc&quot;]), decreasing&#x3D;TRUE)[1:25]</pre></td></tr><tr><td data-num="246"></td><td><pre>topn &#x3D; order(results_genes[sig,&quot;qval&quot;])[1:25]</pre></td></tr><tr><td data-num="247"></td><td><pre>text(x[topn], y[topn], results_genes[topn,&quot;gene_name&quot;], col&#x3D;&quot;black&quot;, cex&#x3D;0.75, srt&#x3D;45)</pre></td></tr><tr><td data-num="248"></td><td><pre></pre></td></tr><tr><td data-num="249"></td><td><pre></pre></td></tr><tr><td data-num="250"></td><td><pre>#### Write a simple table of differentially expressed transcripts to an output file</pre></td></tr><tr><td data-num="251"></td><td><pre>#Each should be significant with a log2 fold-change &gt;&#x3D; 2</pre></td></tr><tr><td data-num="252"></td><td><pre>sigpi &#x3D; which(results_genes[,&quot;pval&quot;]&lt;0.05)</pre></td></tr><tr><td data-num="253"></td><td><pre>sigp &#x3D; results_genes[sigpi,]</pre></td></tr><tr><td data-num="254"></td><td><pre>sigde &#x3D; which(abs(sigp[,&quot;de&quot;]) &gt;&#x3D; 2)</pre></td></tr><tr><td data-num="255"></td><td><pre>sig_tn_de &#x3D; sigp[sigde,]</pre></td></tr><tr><td data-num="256"></td><td><pre></pre></td></tr><tr><td data-num="257"></td><td><pre>#Order the output by or p-value and then break ties using fold-change</pre></td></tr><tr><td data-num="258"></td><td><pre>o &#x3D; order(sig_tn_de[,&quot;qval&quot;], -abs(sig_tn_de[,&quot;de&quot;]), decreasing&#x3D;FALSE)</pre></td></tr><tr><td data-num="259"></td><td><pre></pre></td></tr><tr><td data-num="260"></td><td><pre>output &#x3D; sig_tn_de[o,c(&quot;gene_name&quot;,&quot;id&quot;,&quot;fc&quot;,&quot;pval&quot;,&quot;qval&quot;,&quot;de&quot;)]</pre></td></tr><tr><td data-num="261"></td><td><pre>write.table(output, file&#x3D;&quot;SigDE_supplementary_R.txt&quot;, sep&#x3D;&quot;\t&quot;, row.names&#x3D;FALSE, quote&#x3D;FALSE)</pre></td></tr><tr><td data-num="262"></td><td><pre></pre></td></tr><tr><td data-num="263"></td><td><pre>#View selected columns of the first 25 lines of output</pre></td></tr><tr><td data-num="264"></td><td><pre>output[1:25,c(1,4,5)]</pre></td></tr><tr><td data-num="265"></td><td><pre></pre></td></tr><tr><td data-num="266"></td><td><pre>#You can open the file &quot;SigDE.txt&quot; in Excel, Calc, etc.</pre></td></tr><tr><td data-num="267"></td><td><pre>#It should have been written to the current working directory that you set at the beginning of the R tutorial</pre></td></tr><tr><td data-num="268"></td><td><pre>dir()</pre></td></tr><tr><td data-num="269"></td><td><pre></pre></td></tr><tr><td data-num="270"></td><td><pre></pre></td></tr><tr><td data-num="271"></td><td><pre>#### Plot #11 - Create a heatmap to vizualize expression differences between the eight samples</pre></td></tr><tr><td data-num="272"></td><td><pre>#Define custom dist and hclust functions for use with heatmaps</pre></td></tr><tr><td data-num="273"></td><td><pre>mydist&#x3D;function(c) &#123;dist(c,method&#x3D;&quot;euclidian&quot;)&#125;</pre></td></tr><tr><td data-num="274"></td><td><pre>myclust&#x3D;function(c) &#123;hclust(c,method&#x3D;&quot;average&quot;)&#125;</pre></td></tr><tr><td data-num="275"></td><td><pre></pre></td></tr><tr><td data-num="276"></td><td><pre>main_title&#x3D;&quot;sig DE Transcripts&quot;</pre></td></tr><tr><td data-num="277"></td><td><pre>par(cex.main&#x3D;0.8)</pre></td></tr><tr><td data-num="278"></td><td><pre>sig_genes&#x3D;results_genes[sig,&quot;id&quot;]</pre></td></tr><tr><td data-num="279"></td><td><pre>sig_gene_names&#x3D;results_genes[sig,&quot;gene_name&quot;]</pre></td></tr><tr><td data-num="280"></td><td><pre>data&#x3D;log2(as.matrix(gene_expression[sig_genes,data_columns])+1)</pre></td></tr><tr><td data-num="281"></td><td><pre>heatmap.2(data, hclustfun&#x3D;myclust, distfun&#x3D;mydist, na.rm &#x3D; TRUE, scale&#x3D;&quot;none&quot;, dendrogram&#x3D;&quot;both&quot;, margins&#x3D;c(6,7), Rowv&#x3D;TRUE, Colv&#x3D;TRUE, symbreaks&#x3D;FALSE, key&#x3D;TRUE, symkey&#x3D;FALSE, density.info&#x3D;&quot;none&quot;, trace&#x3D;&quot;none&quot;, main&#x3D;main_title, cexRow&#x3D;0.3, cexCol&#x3D;1, labRow&#x3D;sig_gene_names,col&#x3D;rev(heat.colors(75)))</pre></td></tr><tr><td data-num="282"></td><td><pre></pre></td></tr><tr><td data-num="283"></td><td><pre>dev.off()</pre></td></tr><tr><td data-num="284"></td><td><pre></pre></td></tr><tr><td data-num="285"></td><td><pre>#The output file can be viewed in your browser at the following url:</pre></td></tr><tr><td data-num="286"></td><td><pre>#Note, you must replace __YOUR_IP_ADDRESS__ with your own amazon instance IP</pre></td></tr><tr><td data-num="287"></td><td><pre>#http:&#x2F;&#x2F;__YOUR_IP_ADDRESS__&#x2F;workspace&#x2F;rnaseq&#x2F;de&#x2F;ballgown&#x2F;ref_only&#x2F;Tutorial_Part3_Supplementary_R_output.pdf</pre></td></tr><tr><td data-num="288"></td><td><pre>#To exit R type:</pre></td></tr><tr><td data-num="289"></td><td><pre>quit(save&#x3D;&quot;no&quot;)</pre></td></tr></table></figure><p><img data-src="/img/posts/2020.12.26/image-20210105164626539.png" alt="image-20210105164626539"></p><p><img data-src="/img/posts/2020.12.26/image-20210105164645781.png" alt="image-20210105164645781"></p><div class="tags"><a href="/tags/bioinformatics/" rel="tag"><i class="ic i-tag"></i> bioinformatics</a> <a href="/tags/RNAseq/" rel="tag"><i class="ic i-tag"></i> RNAseq</a></div></div><footer><div class="meta"><span class="item"><span class="icon"><i class="ic i-calendar-check"></i> </span><span class="text">更新于</span> <time title="修改时间：2023-03-08 15:08:37" itemprop="dateModified" datetime="2023-03-08T15:08:37+08:00">2023-03-08</time> </span><span id="2020/12/26/2020-12-26-RNAseq(14)/" class="item leancloud_visitors" data-flag-title="RNAseq 教程 (3.3)" title="阅读次数"><span class="icon"><i class="ic i-eye"></i> </span><span class="text">阅读次数</span> <span class="leancloud-visitors-count"></span> <span class="text">次</span></span></div><div class="reward"><button><i class="ic i-heartbeat"></i> 赞赏</button><p>请我喝[茶]~(￣▽￣)~*</p><div id="qr"><div><img data-src="/images/wechatpay.png" alt="amane 微信支付"><p>微信支付</p></div><div><img data-src="/images/alipay.png" alt="amane 支付宝"><p>支付宝</p></div></div></div><div id="copyright"><ul><li class="author"><strong>本文作者： </strong>amane <i class="ic i-at"><em>@</em></i>小钊の笔记</li><li class="link"><strong>本文链接：</strong> <a href="https://www.zhouxiaozhao.cn/2020/12/26/2020-12-26-RNAseq(14)/" title="RNAseq 教程 (3.3)">https://www.zhouxiaozhao.cn/2020/12/26/2020-12-26-RNAseq(14)/</a></li><li class="license"><strong>版权声明： </strong>本站所有文章除特别声明外，均采用 <span class="exturl" data-url="aHR0cHM6Ly9jcmVhdGl2ZWNvbW1vbnMub3JnL2xpY2Vuc2VzL2J5LW5jLXNhLzQuMC9kZWVkLnpo"><i class="ic i-creative-commons"><em>(CC)</em></i>BY-NC-SA</span> 许可协议。转载请注明出处！</li></ul></div></footer></article></div><div class="post-nav"><div class="item left"><a href="/2020/12/24/2020-12-24-RNAseq(13)/" itemprop="url" rel="prev" data-background-image="https:&#x2F;&#x2F;img1.imgtp.com&#x2F;2023&#x2F;02&#x2F;01&#x2F;GOGkua2f.jpg" title="RNAseq教程(3.2)"><span class="type">上一篇</span> <span class="category"><i class="ic i-flag"></i> 生信学习</span><h3>RNAseq教程(3.2)</h3></a></div><div class="item right"><a href="/2020/12/29/2020-12-29-RNAseq(15)/" itemprop="url" rel="next" data-background-image="https:&#x2F;&#x2F;img1.imgtp.com&#x2F;2023&#x2F;02&#x2F;01&#x2F;6iJnFub2.jpg" title="RNAseq教程(3.4)"><span class="type">下一篇</span> <span class="category"><i class="ic i-flag"></i> 生信学习</span><h3>RNAseq教程(3.4)</h3></a></div></div><div class="wrap" id="comments"></div></div><div id="sidebar"><div class="inner"><div class="panels"><div class="inner"><div class="contents panel pjax" data-title="文章目录"><ol class="toc"><li class="toc-item toc-level-2"><a class="toc-link" href="#%E7%9B%AE%E5%BD%95"><span class="toc-number">1.</span> <span class="toc-text">目录</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#33-de-visualization"><span class="toc-number">2.</span> <span class="toc-text">3.3 DE Visualization</span></a><ol class="toc-child"><li class="toc-item toc-level-3"><a class="toc-link" href="#%E9%A2%9D%E5%A4%96%E5%88%86%E6%9E%90"><span class="toc-number">2.1.</span> <span class="toc-text">额外分析</span></a></li></ol></li></ol></div><div class="related panel pjax" data-title="系列文章"><ul><li><a href="/2020/07/08/2020-07-08-repeat/" rel="bookmark" title="RepeatModeler+RepeatMasker的安装与使用">RepeatModeler+RepeatMasker的安装与使用</a></li><li><a href="/2020/07/09/2020-07-09-github+jekyll/" rel="bookmark" title="利用Github+Jekyll构建自己的博客">利用Github+Jekyll构建自己的博客</a></li><li><a href="/2020/07/18/2020-07-18-orthofinder/" rel="bookmark" title="利用orthofinder寻找单拷贝基因构建系统发育树">利用orthofinder寻找单拷贝基因构建系统发育树</a></li><li><a href="/2020/07/20/2020-07-20-fungi/" rel="bookmark" title="文献导读：真菌进化">文献导读：真菌进化</a></li><li><a href="/2020/08/03/2020-08-03-WGCNA1/" rel="bookmark" title="WGCNA学习：WGCNA分析原理">WGCNA学习：WGCNA分析原理</a></li><li><a href="/2020/08/05/2020-08-05-WGCNA2/" rel="bookmark" title="WGCNA学习：WGCNA应用场景与实例解读">WGCNA学习：WGCNA应用场景与实例解读</a></li><li><a href="/2020/08/08/2020-08-08-WGCNA3/" rel="bookmark" title="WGCNA学习：WGCNA分析实战">WGCNA学习：WGCNA分析实战</a></li><li><a href="/2020/08/10/2020-08-10-cytoscape/" rel="bookmark" title="基因共表达网络图的绘制">基因共表达网络图的绘制</a></li><li><a href="/2020/08/18/2020-08-18-sys/" rel="bookmark" title="根据转录组数据鉴定共生物种">根据转录组数据鉴定共生物种</a></li><li><a href="/2020/08/20/2020-08-20-nt/" rel="bookmark" title="使用本地nt数据库对reads和Trinity结果进行blast">使用本地nt数据库对reads和Trinity结果进行blast</a></li><li><a href="/2020/08/23/2020-08-23-IGV/" rel="bookmark" title="使用IGV简单绘制甲基化分布图">使用IGV简单绘制甲基化分布图</a></li><li><a href="/2020/08/25/2020-08-25-evolview1/" rel="bookmark" title="进化树美化之Evolview--热图（1）">进化树美化之Evolview--热图（1）</a></li><li><a href="/2020/08/28/2020-08-28-evolview/" rel="bookmark" title="进化树美化之Evolview--热图（2）">进化树美化之Evolview--热图（2）</a></li><li><a href="/2020/08/30/2020-08-30-evolview3/" rel="bookmark" title="进化树+蛋白结构域">进化树+蛋白结构域</a></li><li><a href="/2020/09/01/2020-09-01-iTOL1/" rel="bookmark" title="进化树在线制作工具iTOL教程（1）：基本操作">进化树在线制作工具iTOL教程（1）：基本操作</a></li><li><a href="/2020/09/03/2020-09-03-iTOL2/" rel="bookmark" title="进化树在线制作工具iTOL教程（2）：添加文本">进化树在线制作工具iTOL教程（2）：添加文本</a></li><li><a href="/2020/09/06/2020-09-06-iTOL3/" rel="bookmark" title="进化树在线制作工具iTOL教程（3）：添加图片">进化树在线制作工具iTOL教程（3）：添加图片</a></li><li><a href="/2020/09/08/2020-09-08-iTOL4/" rel="bookmark" title="进化树在线制作工具iTOL教程（4）：添加点图">进化树在线制作工具iTOL教程（4）：添加点图</a></li><li><a href="/2020/09/29/2020-09-29-ScRNAseq(1)/" rel="bookmark" title="单细胞实战(1)：原始测序数据下载">单细胞实战(1)：原始测序数据下载</a></li><li><a href="/2020/10/01/2020-10-01-ScRNAseq(2)/" rel="bookmark" title="单细胞实战(2)：cellranger使用">单细胞实战(2)：cellranger使用</a></li><li><a href="/2020/10/03/2020-10-03-ScRNAseq(3)/" rel="bookmark" title="单细胞实战(3)：STAR分析单细胞数据">单细胞实战(3)：STAR分析单细胞数据</a></li><li><a href="/2020/10/06/2020-10-06-ScRNAseq(4)/" rel="bookmark" title="单细胞实战(4)：STAR与cellranger结果比较">单细胞实战(4)：STAR与cellranger结果比较</a></li><li><a href="/2020/10/10/2020-10-10-ScRNAseq(5)/" rel="bookmark" title="单细胞实战(5)：复现文章中的聚类图（1）">单细胞实战(5)：复现文章中的聚类图（1）</a></li><li><a href="/2020/10/13/2020-10-13-ScRNAseq(6)/" rel="bookmark" title="单细胞实战(5)：复现文章中的聚类图（2）">单细胞实战(5)：复现文章中的聚类图（2）</a></li><li><a href="/2020/10/24/2020-10-24-ScRNAseq(7)/" rel="bookmark" title="生信技能树单细胞数据挖掘笔记(1)">生信技能树单细胞数据挖掘笔记(1)</a></li><li><a href="/2020/10/27/2020-10-27-ScRNAseq(8)/" rel="bookmark" title="生信技能树单细胞数据挖掘笔记(2)">生信技能树单细胞数据挖掘笔记(2)</a></li><li><a href="/2020/10/29/2020-10-29-ScRNAseq(9)/" rel="bookmark" title="生信技能树单细胞数据挖掘笔记(3)">生信技能树单细胞数据挖掘笔记(3)</a></li><li><a href="/2020/10/31/2020-10-31-ScRNAseq(10)/" rel="bookmark" title="生信技能树单细胞数据挖掘笔记(4)">生信技能树单细胞数据挖掘笔记(4)</a></li><li><a href="/2020/11/03/2020-11-03-ScRNAseq(11)/" rel="bookmark" title="生信技能树单细胞数据挖掘笔记(5)">生信技能树单细胞数据挖掘笔记(5)</a></li><li><a href="/2020/11/05/2020-11-05-ScRNAseq(12)/" rel="bookmark" title="生信技能树单细胞数据挖掘笔记(6)">生信技能树单细胞数据挖掘笔记(6)</a></li><li><a href="/2020/11/07/2020-11-07-ScRNAseq(13)/" rel="bookmark" title="生信技能树单细胞数据挖掘笔记(7)">生信技能树单细胞数据挖掘笔记(7)</a></li><li><a href="/2020/11/10/2020-11-10-RNAvelocity(1)/" rel="bookmark" title="RNA速率：软件下载与loom文件准备">RNA速率：软件下载与loom文件准备</a></li><li><a href="/2020/11/12/2020-11-12-RNAvelocity(2)/" rel="bookmark" title="RNA速率：数据读入">RNA速率：数据读入</a></li><li><a href="/2020/11/14/2020-11-14-RNAvelocity(3)/" rel="bookmark" title="RNA速率：使用Seurat的结果做RNA velocity">RNA速率：使用Seurat的结果做RNA velocity</a></li><li><a href="/2020/11/17/2020-11-17-RNAseq(1)/" rel="bookmark" title="RNAseq教程(1.1)">RNAseq教程(1.1)</a></li><li><a href="/2020/11/19/2020-11-19-annotion(1)/" rel="bookmark" title="基因结构注释（1）：从头注释">基因结构注释（1）：从头注释</a></li><li><a href="/2020/11/21/2020-11-21-annotion(2)/" rel="bookmark" title="基因结构注释（2）：同源注释">基因结构注释（2）：同源注释</a></li><li><a href="/2020/11/24/2020-11-24-annotion(3)/" rel="bookmark" title="基因结构注释（3）：转录组注释">基因结构注释（3）：转录组注释</a></li><li><a href="/2020/11/26/2020-11-26-annotion(4)/" rel="bookmark" title="基因结构注释（4）：整合预测结果">基因结构注释（4）：整合预测结果</a></li><li><a href="/2020/11/28/2020-11-28-RNAseq(2)/" rel="bookmark" title="RNAseq教程(1.2)">RNAseq教程(1.2)</a></li><li><a href="/2020/12/01/2020-12-01-RNAseq(3)/" rel="bookmark" title="RNAseq教程(1.3)">RNAseq教程(1.3)</a></li><li><a href="/2020/12/03/2020-12-03-RNAseq(4)/" rel="bookmark" title="RNAseq教程(1.4)">RNAseq教程(1.4)</a></li><li><a href="/2020/12/05/2020-12-05-RNAseq(5)/" rel="bookmark" title="RNAseq教程(1.5)">RNAseq教程(1.5)</a></li><li><a href="/2020/12/08/2020-12-08-RNAseq(6)/" rel="bookmark" title="RNAseq教程(1.6)">RNAseq教程(1.6)</a></li><li><a href="/2020/12/10/2020-12-10-RNAseq(7)/" rel="bookmark" title="RNAseq教程(2.1)">RNAseq教程(2.1)</a></li><li><a href="/2020/12/12/2020-12-12-RNAseq(8)/" rel="bookmark" title="RNAseq教程(2.2)">RNAseq教程(2.2)</a></li><li><a href="/2020/12/15/2020-12-15-RNAseq(9)/" rel="bookmark" title="RNAseq教程(2.3)">RNAseq教程(2.3)</a></li><li><a href="/2020/12/17/2020-12-17-RNAseq(10)/" rel="bookmark" title="RNAseq教程(2.4)">RNAseq教程(2.4)</a></li><li><a href="/2020/12/19/2020-12-19-RNAseq(11)/" rel="bookmark" title="RNAseq教程(2.5)">RNAseq教程(2.5)</a></li><li><a href="/2020/12/22/2020-12-22-RNAseq(12)/" rel="bookmark" title="RNAseq教程(3.1)">RNAseq教程(3.1)</a></li><li><a href="/2020/12/24/2020-12-24-RNAseq(13)/" rel="bookmark" title="RNAseq教程(3.2)">RNAseq教程(3.2)</a></li><li class="active"><a href="/2020/12/26/2020-12-26-RNAseq(14)/" rel="bookmark" title="RNAseq教程(3.3)">RNAseq教程(3.3)</a></li><li><a href="/2020/12/29/2020-12-29-RNAseq(15)/" rel="bookmark" title="RNAseq教程(3.4)">RNAseq教程(3.4)</a></li><li><a href="/2020/12/31/2020-12-31-RNAseq(16)/" rel="bookmark" title="RNAseq教程(4.1)">RNAseq教程(4.1)</a></li><li><a href="/2021/01/02/2021-01-02-RNAseq(17)/" rel="bookmark" title="RNAseq教程(4.2)">RNAseq教程(4.2)</a></li><li><a href="/2021/01/05/2021-01-05-RNAseq(18)/" rel="bookmark" title="RNAseq教程(4.3)">RNAseq教程(4.3)</a></li><li><a href="/2021/01/07/2021-01-07-RNAseq(19)/" rel="bookmark" title="RNAseq教程(4.4)">RNAseq教程(4.4)</a></li><li><a href="/2021/01/09/2021-01-09-RNAseq(20)/" rel="bookmark" title="RNAseq教程(4.5)">RNAseq教程(4.5)</a></li><li><a href="/2021/01/12/2021-01-12-RNAseq(21)/" rel="bookmark" title="RNAseq教程(5)">RNAseq教程(5)</a></li><li><a href="/2022/12/08/2022-12-08-SingleCopy/" rel="bookmark" title="比较基因组学分析1：构建单拷贝基因树">比较基因组学分析1：构建单拷贝基因树</a></li><li><a href="/2022/12/14/2022-12-14-CAFE/" rel="bookmark" title="比较基因组学分析2：基因家族收缩与扩张分析">比较基因组学分析2：基因家族收缩与扩张分析</a></li><li><a href="/2022/12/15/2022-12-15-GO/" rel="bookmark" title="比较基因组学分析3：特异节点基因家族富集分析（非模式物种GO/KEEG库构建）">比较基因组学分析3：特异节点基因家族富集分析（非模式物种GO/KEEG库构建）</a></li><li><a href="/2023/02/22/2023-02-22-ST(1)/" rel="bookmark" title="植物空间转录组分析1：Seurat基本流程">植物空间转录组分析1：Seurat基本流程</a></li><li><a href="/2023/02/24/2023-02-24-ST(2)/" rel="bookmark" title="植物空间转录组分析2：STEEL+Seurat">植物空间转录组分析2：STEEL+Seurat</a></li></ul></div><div class="overview panel" data-title="站点概览"><div class="author" itemprop="author" itemscope itemtype="http://schema.org/Person"><img class="image" itemprop="image" alt="amane" data-src="/images/avatar.jpg"><p class="name" itemprop="name">amane</p><div class="description" itemprop="description">生物学在读博士</div></div><nav class="state"><div class="item posts"><a href="/archives/"><span class="count">84</span> <span class="name">文章</span></a></div><div class="item categories"><a href="/categories/"><span class="count">8</span> <span class="name">分类</span></a></div><div class="item tags"><a href="/tags/"><span class="count">21</span> <span class="name">标签</span></a></div></nav><div class="social"><span class="exturl item github" data-url="aHR0cHM6Ly9naXRodWIuY29tL0Jpb2luZm9ybWF0aWNzLXJvb2tpZQ==" title="https:&#x2F;&#x2F;github.com&#x2F;Bioinformatics-rookie"><i class="ic i-github"></i></span> <span class="exturl item bilibili" data-url="aHR0cHM6Ly9zcGFjZS5iaWxpYmlsaS5jb20vMTU3Nzg1NTM/c3BtX2lkX2Zyb209MzMzLjEwMDcuMC4w" title="https:&#x2F;&#x2F;space.bilibili.com&#x2F;15778553?spm_id_from&#x3D;333.1007.0.0"><i class="ic i-bilibili"></i></span> <a href="https://www.zhouxiaozhao.cn/about/" title="https:&#x2F;&#x2F;www.zhouxiaozhao.cn&#x2F;about&#x2F;" class="item about"><i class="ic i-address-card"></i></a> <span class="exturl item email" data-url="bWFpbHRvOnpob3VzaGl6aGFvQG1haWwua2liLmFjLmNu" title="mailto:zhoushizhao@mail.kib.ac.cn"><i class="ic i-envelope"></i></span></div><ul class="menu"><li class="item"><a href="/" rel="section"><i class="ic i-home"></i>首页</a></li><li class="item"><a href="/about/" rel="section"><i class="ic i-user"></i>关于</a></li><li class="item dropdown"><a href="javascript:void(0);"><i class="ic i-feather"></i>文章</a><ul class="submenu"><li class="item"><a href="/archives/" rel="section"><i class="ic i-list-alt"></i>归档</a></li><li class="item"><a href="/categories/" rel="section"><i class="ic i-th"></i>分类</a></li><li class="item"><a href="/tags/" rel="section"><i class="ic i-tags"></i>标签</a></li></ul></li><li class="item dropdown"><a href="/webstack/" rel="section"><i class="ic i-magic"></i>AI学习</a><ul class="submenu"><li class="item"><a href="/AIpainting/" rel="section"><i class="ic i-pen"></i>AI绘画</a></li><li class="item"><a href="/AIfile/" rel="section"><i class="ic i-file"></i>AI图包</a></li><li class="item"><a href="/AIbookmark/" rel="section"><i class="ic i-comments"></i>AI书签</a></li></ul></li><li class="item"><a href="/links/" rel="section"><i class="ic i-heart"></i>友達</a></li></ul></div></div></div><ul id="quick"><li class="prev pjax"><a href="/2020/12/24/2020-12-24-RNAseq(13)/" rel="prev" title="上一篇"><i class="ic i-chevron-left"></i></a></li><li class="up"><i class="ic i-arrow-up"></i></li><li class="down"><i class="ic i-arrow-down"></i></li><li class="next pjax"><a href="/2020/12/29/2020-12-29-RNAseq(15)/" rel="next" title="下一篇"><i class="ic i-chevron-right"></i></a></li><li class="percent"></li></ul></div></div><div class="dimmer"></div></div></main><footer id="footer"><div class="inner"><div class="widgets"><div class="rpost pjax"><h2>随机文章</h2><ul><li class="item"><div class="breadcrumb"><a href="/categories/Bioinformation/" title="分类于 生信学习">生信学习</a> <i class="ic i-angle-right"></i> <a href="/categories/Bioinformation/Blog/" title="分类于 博客">博客</a></div><span><a href="/2020/07/09/2020-07-09-github+jekyll/" title="利用Github+Jekyll构建自己的博客">利用Github+Jekyll构建自己的博客</a></span></li><li class="item"><div class="breadcrumb"><a href="/categories/Python/" title="分类于 Python">Python</a></div><span><a href="/2020/10/17/2020-10-17-python7/" title="马士兵教育Python入门基础:4.流程控制（1）">马士兵教育Python入门基础:4.流程控制（1）</a></span></li><li class="item"><div class="breadcrumb"><a href="/categories/Bioinformation/" title="分类于 生信学习">生信学习</a> <i class="ic i-angle-right"></i> <a href="/categories/Bioinformation/genome/" title="分类于 基因组">基因组</a></div><span><a href="/2022/12/15/2022-12-15-GO/" title="比较基因组学分析3：特异节点基因家族富集分析（非模式物种GO&#x2F;KEEG库构建）">比较基因组学分析3：特异节点基因家族富集分析（非模式物种GO/KEEG库构建）</a></span></li><li class="item"><div class="breadcrumb"><a href="/categories/Python/" title="分类于 Python">Python</a></div><span><a href="/2020/09/12/2020-09-12-python2/" title="马士兵教育Python入门基础:1.python入门（二）">马士兵教育Python入门基础:1.python入门（二）</a></span></li><li class="item"><div class="breadcrumb"><a href="/categories/Bioinformation/" title="分类于 生信学习">生信学习</a></div><span><a href="/2020/12/19/2020-12-19-RNAseq(11)/" title="RNAseq教程(2.5)">RNAseq教程(2.5)</a></span></li><li class="item"><div class="breadcrumb"><a href="/categories/Bioinformation/" title="分类于 生信学习">生信学习</a></div><span><a href="/2021/01/02/2021-01-02-RNAseq(17)/" title="RNAseq教程(4.2)">RNAseq教程(4.2)</a></span></li><li class="item"><div class="breadcrumb"><a href="/categories/Python/" title="分类于 Python">Python</a></div><span><a href="/2020/09/15/2020-09-15-python3/" title="马士兵教育Python入门基础:2.七十二变（1）">马士兵教育Python入门基础:2.七十二变（1）</a></span></li><li class="item"><div class="breadcrumb"><a href="/categories/Bioinformation/" title="分类于 生信学习">生信学习</a></div><span><a href="/2021/01/05/2021-01-05-RNAseq(18)/" title="RNAseq教程(4.3)">RNAseq教程(4.3)</a></span></li><li class="item"><div class="breadcrumb"><a href="/categories/Bioinformation/" title="分类于 生信学习">生信学习</a> <i class="ic i-angle-right"></i> <a href="/categories/Bioinformation/SingleCell/" title="分类于 单细胞">单细胞</a></div><span><a href="/2020/11/10/2020-11-10-RNAvelocity(1)/" title="RNA速率：软件下载与loom文件准备">RNA速率：软件下载与loom文件准备</a></span></li><li class="item"><div class="breadcrumb"><a href="/categories/Bioinformation/" title="分类于 生信学习">生信学习</a></div><span><a href="/2020/12/03/2020-12-03-RNAseq(4)/" title="RNAseq教程(1.4)">RNAseq教程(1.4)</a></span></li></ul></div><div><h2>最新评论</h2><ul class="leancloud-recent-comment"></ul></div></div><div class="status"><div class="copyright">&copy; 2010 – <span itemprop="copyrightYear">2023</span> <span class="with-love"><i class="ic i-sakura rotate"></i> </span><span class="author" itemprop="copyrightHolder">amane @ Zhou Xiaozhao</span></div><div class="count"><span class="post-meta-item-icon"><i class="ic i-chart-area"></i> </span><span title="站点总字数">381k 字</span> <span class="post-meta-divider">|</span> <span class="post-meta-item-icon"><i class="ic i-coffee"></i> </span><span title="站点阅读时长">5:46</span></div><div class="powered-by">基于 <span class="exturl" data-url="aHR0cHM6Ly9oZXhvLmlv">Hexo</span> & Theme.<span class="exturl" data-url="aHR0cHM6Ly9naXRodWIuY29tL2FtZWhpbWUvaGV4by10aGVtZS1zaG9rYQ==">Shoka</span></div></div></div></footer></div><script data-config type="text/javascript">var LOCAL={path:"2020/12/26/2020-12-26-RNAseq(14)/",favicon:{show:"（●´3｀●）やれやれだぜ",hide:"(´Д｀)大変だ！"},search:{placeholder:"文章搜索",empty:"关于 「 ${query} 」，什么也没搜到",stats:"${time} ms 内找到 ${hits} 条结果"},valine:!0,fancybox:!0,copyright:'复制成功，转载请遵守 <i class="ic i-creative-commons"></i>BY-NC-SA 协议。',ignores:[function(e){return e.includes("#")},function(e){return new RegExp(LOCAL.path+"$").test(e)}]}</script><script src="https://cdn.polyfill.io/v2/polyfill.js"></script><script src="//cdn.jsdelivr.net/combine/npm/pace-js@1.0.2/pace.min.js,npm/pjax@0.2.8/pjax.min.js,npm/whatwg-fetch@3.4.0/dist/fetch.umd.min.js,npm/animejs@3.2.0/lib/anime.min.js,npm/algoliasearch@4/dist/algoliasearch-lite.umd.js,npm/instantsearch.js@4/dist/instantsearch.production.min.js,npm/lozad@1/dist/lozad.min.js,npm/quicklink@2/dist/quicklink.umd.js"></script><script src="/js/app.js?v=0.2.5"></script></body></html><!-- rebuild by hrmmi -->