<!-- build time:Wed Feb 01 2023 16:39:25 GMT+0800 (中国标准时间) --><!DOCTYPE html><html lang="zh-CN"><head><meta charset="UTF-8"><meta name="viewport" content="width=device-width,initial-scale=1,maximum-scale=2"><meta name="theme-color" content="#FFF"><link rel="apple-touch-icon" sizes="180x180" href="/images/apple-touch-icon.png"><link rel="icon" type="image/ico" sizes="32x32" href="/images/favicon.ico"><meta http-equiv="Cache-Control" content="no-transform"><meta http-equiv="Cache-Control" content="no-siteapp"><link rel="alternate" type="application/rss+xml" title="小钊の笔记" href="https://www.zhouxiaozhao.cn/rss.xml"><link rel="alternate" type="application/atom+xml" title="小钊の笔记" href="https://www.zhouxiaozhao.cn/atom.xml"><link rel="alternate" type="application/json" title="小钊の笔记" href="https://www.zhouxiaozhao.cn/feed.json"><link rel="stylesheet" href="//fonts.googleapis.com/css?family=Mulish:300,300italic,400,400italic,700,700italic%7CFredericka%20the%20Great:300,300italic,400,400italic,700,700italic%7CNoto%20Serif%20JP:300,300italic,400,400italic,700,700italic%7CNoto%20Serif%20SC:300,300italic,400,400italic,700,700italic%7CInconsolata:300,300italic,400,400italic,700,700italic&display=swap&subset=latin,latin-ext"><link rel="stylesheet" href="/css/app.css?v=0.2.5"><meta name="keywords" content="bioinformatics,WGCNA"><link rel="canonical" href="https://www.zhouxiaozhao.cn/2020/08/08/2020-08-08-WGCNA3/"><title>WGCNA 学习：WGCNA 分析实战 - 生信学习 | Zhou Xiaozhao = 小钊の笔记 = 前天是小兔子，昨天是小鹿，今天是你</title><meta name="generator" content="Hexo 5.4.1"></head><body itemscope itemtype="http://schema.org/WebPage"><div id="loading"><div class="cat"><div class="body"></div><div class="head"><div class="face"></div></div><div class="foot"><div class="tummy-end"></div><div class="bottom"></div><div class="legs left"></div><div class="legs right"></div></div><div class="paw"><div class="hands left"></div><div class="hands right"></div></div></div></div><div id="container"><header id="header" itemscope itemtype="http://schema.org/WPHeader"><div class="inner"><div id="brand"><div class="pjax"><h1 itemprop="name headline">WGCNA 学习：WGCNA 分析实战</h1><div class="meta"><span class="item" title="创建时间：2020-08-08 18:38:45"><span class="icon"><i class="ic i-calendar"></i> </span><span class="text">发表于</span> <time itemprop="dateCreated datePublished" datetime="2020-08-08T18:38:45+08:00">2020-08-08</time> </span><span class="item" title="本文字数"><span class="icon"><i class="ic i-pen"></i> </span><span class="text">本文字数</span> <span>12k</span> <span class="text">字</span> </span><span class="item" title="阅读时长"><span class="icon"><i class="ic i-clock"></i> </span><span class="text">阅读时长</span> <span>11 分钟</span></span></div></div></div><nav id="nav"><div class="inner"><div class="toggle"><div class="lines" aria-label="切换导航栏"><span class="line"></span> <span class="line"></span> <span class="line"></span></div></div><ul class="menu"><li class="item title"><a href="/" rel="start">Zhou Xiaozhao</a></li></ul><ul class="right"><li class="item theme"><i class="ic i-sun"></i></li><li class="item search"><i class="ic i-search"></i></li></ul></div></nav></div><div id="imgs" class="pjax"><ul><li class="item" data-background-image="https://img1.imgtp.com/2023/02/01/UJhRWazi.jpg"></li><li class="item" data-background-image="https://img1.imgtp.com/2023/02/01/ID2PmCx1.jpg"></li><li class="item" data-background-image="https://img1.imgtp.com/2023/02/01/FwUn4GrO.jpg"></li><li class="item" data-background-image="https://img1.imgtp.com/2023/02/01/J7Hm1Qfg.jpg"></li><li class="item" data-background-image="https://img1.imgtp.com/2023/02/01/ljlU8tAi.jpg"></li><li class="item" data-background-image="https://img1.imgtp.com/2023/02/01/tzn2O8cD.jpg"></li></ul></div></header><div id="waves"><svg class="waves" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" viewBox="0 24 150 28" preserveAspectRatio="none" shape-rendering="auto"><defs><path id="gentle-wave" d="M-160 44c30 0 58-18 88-18s 58 18 88 18 58-18 88-18 58 18 88 18 v44h-352z"/></defs><g class="parallax"><use xlink:href="#gentle-wave" x="48" y="0"/><use xlink:href="#gentle-wave" x="48" y="3"/><use xlink:href="#gentle-wave" x="48" y="5"/><use xlink:href="#gentle-wave" x="48" y="7"/></g></svg></div><main><div class="inner"><div id="main" class="pjax"><div class="article wrap"><div class="breadcrumb" itemscope itemtype="https://schema.org/BreadcrumbList"><i class="ic i-home"></i> <span><a href="/">首页</a></span><i class="ic i-angle-right"></i> <span class="current" itemprop="itemListElement" itemscope itemtype="https://schema.org/ListItem"><a href="/categories/Bioinformation/" itemprop="item" rel="index" title="分类于 生信学习"><span itemprop="name">生信学习</span></a><meta itemprop="position" content="1"></span></div><article itemscope itemtype="http://schema.org/Article" class="post block" lang="zh-CN"><link itemprop="mainEntityOfPage" href="https://www.zhouxiaozhao.cn/2020/08/08/2020-08-08-WGCNA3/"><span hidden itemprop="author" itemscope itemtype="http://schema.org/Person"><meta itemprop="image" content="/images/avatar.png"><meta itemprop="name" content="amane"><meta itemprop="description" content="前天是小兔子，昨天是小鹿，今天是你, 生物学在读博士"></span><span hidden itemprop="publisher" itemscope itemtype="http://schema.org/Organization"><meta itemprop="name" content="小钊の笔记"></span><div class="body md" itemprop="articleBody"><p>WGCNA 笔记第三弹</p><ul><li><span class="exturl" data-url="aHR0cHM6Ly93d3cuemhvdXhpYW96aGFvLmNuLzIwMjAvMDgvMDMvV0dDTkExLw==">WGCNA 分析原理</span></li><li><span class="exturl" data-url="aHR0cHM6Ly93d3cuemhvdXhpYW96aGFvLmNuLzIwMjAvMDgvMDUvV0dDTkEyLw==">WGCNA 应用场景</span></li><li><span class="exturl" data-url="aHR0cHM6Ly93d3cuemhvdXhpYW96aGFvLmNuLzIwMjAvMDgvMDgvV0dDTkEzLw==">WGCNA 分析实践</span></li></ul><p>本代码借鉴了<span class="exturl" data-url="aHR0cHM6Ly9naXRodWIuY29tL2ptemVuZzEzMTQvbXlfV0dDTkE=">生信技能树的 WGCNA 教程</span>以及<span class="exturl" data-url="aHR0cHM6Ly9rZS5xcS5jb20vY291cnNlLzQzMDIwMQ=="> PlanTech 的 WGCNA 课程</span>编写，转载请注明出处</p><h1 id="1wgcna安装"><a class="anchor" href="#1wgcna安装">#</a> 1.WGCNA 安装</h1><pre><code>&gt; install.packages(&quot;BiocManager&quot;)
&gt; BiocManager::install(&quot;WGCNA&quot;)
&gt; library(WGCNA)
载入需要的程辑包：dynamicTreeCut
载入需要的程辑包：fastcluster

载入程辑包：‘fastcluster’

The following object is masked from ‘package:stats’:

    hclust



载入程辑包：‘WGCNA’

The following object is masked from ‘package:stats’:

    cor
</code></pre><h1 id="2数据准备与读入"><a class="anchor" href="#2数据准备与读入">#</a> 2. 数据准备与读入</h1><h2 id="21数据准备"><a class="anchor" href="#21数据准备">#</a> 2.1 数据准备</h2><p>需要两个数据</p><ol><li><p>表达矩阵（All_fpkm.list）</p><p><img data-src="/img/posts/2020.8.8/image-20200802142219001.png" alt="image-20200802142219001"></p></li><li><p>表型文件（pheno.txt）, 需要注意表型文件分为两类，连续变量型与分类变量型.</p></li></ol><p><img data-src="/img/posts/2020.8.8/image-20200802142335624.png" alt="image-20200802142335624"></p><p>&lt;center&gt;<br>连续变量<br>&lt;/center&gt;</p><p><img data-src="/img/posts/2020.8.8/image-20200803165842485.png" alt="image-20200803165842485"></p><p>&lt;center&gt;<br>分类变量<br>&lt;/center&gt;</p><h2 id="22-数据读入"><a class="anchor" href="#22-数据读入">#</a> 2.2 数据读入</h2><pre><code>library(WGCNA)
library(reshape2)
library(stringr)
setwd('D:/data/wgcna/Categorical')
options(stringsAsFactors = FALSE)# 在读入数据时，遇到字符串后，不要将其转换成因子，仍然保留为字符串格式
enableWGCNAThreads()#打开多线程
##====================step 1 :数据读入

RNAseq_voom &lt;- fpkm

## 因为WGCNA针对的是基因进行聚类，而一般我们的聚类是针对样本用hclust即可，所以这个时候需要转置
WGCNA_matrix = t(RNAseq_voom[order(apply(RNAseq_voom,1,mad), decreasing = T)[1:5000],])
datExpr &lt;- WGCNA_matrix  ## top 5000 mad genes

#明确样本数和基因
nGenes = ncol(datExpr)
nSamples = nrow(datExpr)

#首先针对样本做个系统聚类
datExpr_tree&lt;-hclust(dist(datExpr), method = &quot;average&quot;)
par(mar = c(0,5,2,0))
png(&quot;img/step1-sample-cluster.png&quot;,width = 800,height = 600)
plot(datExpr_tree, main = &quot;Sample clustering&quot;, sub=&quot;&quot;, xlab=&quot;&quot;, cex.lab = 2,
       cex.axis = 1, cex.main = 1,cex.lab=1)
dev.off()
</code></pre><p><img data-src="/img/posts/2020.8.8/step1-sample-cluster.png" alt></p><h1 id="3-β值估计"><a class="anchor" href="#3-β值估计">#</a> 3. β 值估计</h1><pre><code>##====================step 2：β值确定====

datExpr[1:4,1:4]
powers = c(c(1:10), seq(from = 12, to=20, by=2))
#设置beta值的取值范围
sft = pickSoftThreshold(datExpr, RsquaredCut = 0.9,powerVector = powers, verbose = 5)
#设置网络构建参数选择范围，计算无尺度分布拓扑矩阵
png(&quot;img/step2-beta-value.png&quot;,width = 800,height = 600)
par(mfrow = c(1,2));
cex1 = 0.9;
# Scale-free topology fit index as a function of the soft-thresholding power
plot(sft$fitIndices[,1], -sign(sft$fitIndices[,3])*sft$fitIndices[,2],
       xlab=&quot;Soft Threshold (power)&quot;,ylab=&quot;Scale Free Topology Model Fit,signed R^2&quot;,type=&quot;n&quot;,
       main = paste(&quot;Scale independence&quot;));
text(sft$fitIndices[,1], -sign(sft$fitIndices[,3])*sft$fitIndices[,2],
       labels=powers,cex=cex1,col=&quot;red&quot;);
# this line corresponds to using an R^2 cut-off of h
abline(h=0.90,col=&quot;red&quot;)
# Mean connectivity as a function of the soft-thresholding power
plot(sft$fitIndices[,1], sft$fitIndices[,5],
       xlab=&quot;Soft Threshold (power)&quot;,ylab=&quot;Mean Connectivity&quot;, type=&quot;n&quot;,
       main = paste(&quot;Mean connectivity&quot;))
text(sft$fitIndices[,1], sft$fitIndices[,5], labels=powers, cex=cex1,col=&quot;red&quot;)
dev.off()
</code></pre><p><img data-src="/img/posts/2020.8.8/step2-beta-value.png" alt></p><p>可以确定最佳 β 值为 6</p><h1 id="4-一步法构建共表达矩阵"><a class="anchor" href="#4-一步法构建共表达矩阵">#</a> 4. 一步法构建共表达矩阵</h1><p>最核心的一步，同时也是最耗费计算资源的一步</p><pre><code>##====================step 3:自动构建WGCNA模型==================
# 首先是一步法完成网络构建
net = blockwiseModules(
  datExpr,
  power = sft$powerEstimate,             #软阈值，前面计算出来的
  maxBlockSize = 6000,                   #最大block大小，将所有基因放在一个block中
  TOMType = &quot;unsigned&quot;,                  #选择unsigned，使用标准TOM矩阵
  deepSplit = 2, minModuleSize = 30,     #剪切树参数，deepSplit取值0-4
  mergeCutHeight = 0.25,                 # 模块合并参数，越大模块越少
  numericLabels = TRUE,                  # T返回数字，F返回颜色
  pamRespectsDendro = FALSE,  
  saveTOMs = TRUE,
  saveTOMFileBase = &quot;FPKM-TOM&quot;,
  loadTOMs = TRUE,
  verbose = 3
)
</code></pre><h1 id="5-模块可视化"><a class="anchor" href="#5-模块可视化">#</a> 5. 模块可视化</h1><pre><code>##===============================step4:模块可视化=========================
# Convert labels to colors for plotting
mergedColors = labels2colors(net$colors)
table(mergedColors)
moduleColors=mergedColors
# Plot the dendrogram and the module colors underneath
png(&quot;img/step4-genes-modules.png&quot;,width = 800,height = 600)
plotDendroAndColors(net$dendrograms[[1]], mergedColors[net$blockGenes[[1]]],
                    &quot;Module colors&quot;,
                    dendroLabels = FALSE, hang = 0.03,
                    addGuide = TRUE, guideHang = 0.05)
dev.off()
## assign all of the gene to their corresponding module
## hclust for the genes.



#明确样本数和基因
nGenes = ncol(datExpr)
nSamples = nrow(datExpr)
#首先针对样本做个系统聚类
datExpr_tree&lt;-hclust(dist(datExpr), method = &quot;average&quot;)
par(mar = c(0,5,2,0))
plot(datExpr_tree, main = &quot;Sample clustering&quot;, sub=&quot;&quot;, xlab=&quot;&quot;, cex.lab = 2,
     cex.axis = 1, cex.main = 1,cex.lab=1)
## 如果这个时候样本是有性状，或者临床表型的，可以加进去看看是否聚类合理
#针对前面构造的样品矩阵添加对应颜色
sample_colors &lt;- numbers2colors(as.numeric(factor(datTraits$subtype)),
                                colors = c(&quot;white&quot;,&quot;blue&quot;,&quot;red&quot;,&quot;green&quot;),signed = FALSE)
## 这个给样品添加对应颜色的代码需要自行修改以适应自己的数据分析项目
#  sample_colors &lt;- numbers2colors( datTraits ,signed = FALSE)
## 如果样品有多种分类情况，而且 datTraits 里面都是分类信息，那么可以直接用上面代码，当然，这样给的颜色不明显，意义不大
#10个样品的系统聚类树及性状热图
par(mar = c(1,4,3,1),cex=0.8)

png(&quot;img/sample-subtype-cluster.png&quot;,width = 800,height = 600)
plotDendroAndColors(datExpr_tree, sample_colors,
                    groupLabels = colnames(sample),
                    cex.dendroLabels = 0.8,
                    marAll = c(1, 4, 3, 1),
                    cex.rowText = 0.01,
                    main = &quot;Sample dendrogram and trait heatmap&quot;)
dev.off()
</code></pre><p>可以看到模块用不同的颜色来标注，灰色模块是无法归类于任何模块的基因，在后续分析的时候不需要考虑</p><p><img data-src="/img/posts/2020.8.8/step4-genes-modules.png" alt></p><p><img data-src="/img/posts/2020.8.8/sample-subtype-cluster.png" alt></p><h1 id="6模块与性状的关系"><a class="anchor" href="#6模块与性状的关系">#</a> 6. 模块与性状的关系</h1><pre><code>##===============================step5:模块与性状的关系===================
table(datTraits$subtype)
nGenes = ncol(datExpr)
nSamples = nrow(datExpr)
design=model.matrix(~0+ datTraits$subtype)
colnames(design)=levels(datTraits$subtype)
moduleColors &lt;- labels2colors(net$colors)
# Recalculate MEs with color labels
MEs0 = moduleEigengenes(datExpr, moduleColors)$eigengenes
MEs = orderMEs(MEs0); ##不同颜色的模块的ME值矩 (样本vs模块)
# 输出每个基因所在的模块，以及与该模块的KME值
file.remove('All_Gene_KME.txt')
for(module in substring(colnames(MEs),3))&#123;
  if(module == &quot;grey&quot;) next
  ME=as.data.frame(MEs[,paste(&quot;ME&quot;,module,sep=&quot;&quot;)])
  colnames(ME)=module
  datModExpr=datExpr[,moduleColors==module]
  datKME = signedKME(datModExpr, ME)
  datKME=cbind(datKME,rep(module,length(datKME)))
  write.table(datKME,quote = F,row.names = T,append = T,file = &quot;All_Gene_KME.txt&quot;,col.names = F)
&#125;
moduleTraitCor = cor(MEs, design , use = &quot;p&quot;);
moduleTraitPvalue = corPvalueStudent(moduleTraitCor, nSamples)

sizeGrWindow(10,6)
# Will display correlations and their p-values
textMatrix = paste(signif(moduleTraitCor, 2), &quot;\n(&quot;,
                   signif(moduleTraitPvalue, 1), &quot;)&quot;, sep = &quot;&quot;);
dim(textMatrix) = dim(moduleTraitCor)
png(&quot;img/step5-Module-trait-relationships.png&quot;,width = 800,height = 600)
par(mar = c(6, 8.5, 3, 3));
# Display the correlation values within a heatmap plot
labeledHeatmap(Matrix = moduleTraitCor,
               xLabels = colnames(design),
               yLabels = names(MEs),
               ySymbols = names(MEs),
               colorLabels = FALSE,
               colors = blueWhiteRed(50),
               textMatrix = textMatrix,
               setStdMargins = FALSE,
               cex.text = 0.5,
               zlim = c(-1,1),
               main = paste(&quot;Module-trait relationships&quot;))
dev.off()
#绘制两两模块间的邻接矩阵
png(&quot;img/wgcna.adjacency.heatmap.pdf&quot;,height = 800,width = 600)
plotEigengeneNetworks(MEs, &quot;Eigengene adjacency heatmap&quot;,plotDendrograms = F,
                      marDendro = c(4,4,2,4))
dev.off()
</code></pre><p>可以看到不同的模块与不同的性状是有不同的相关性的，在后续分析的时候我们可以选择感兴趣的模块进行分析。</p><p><img data-src="/img/posts/2020.8.8/step5-Module-trait-relationships.png" alt></p><p>两两模块之间的邻接矩阵，主要看不同模块之间的相关性</p><p><img data-src="/img/posts/2020.8.8/wgcna.adjacency.heatmap.png" alt></p><h1 id="7-选择感兴趣的模块进行分析"><a class="anchor" href="#7-选择感兴趣的模块进行分析">#</a> 7. 选择感兴趣的模块进行分析</h1><pre><code>##===============================step 6：感兴趣性状的模块的具体基因分析=====
# 查看第五步出图：step5-Module-trait-relationships.png
# 发现跟 Luminal 亚型 最相关的是  pink 模块
# 所以接下来就分析这两个
Luminal = as.data.frame(design[,3])
names(Luminal) = &quot;Luminal&quot;
module = &quot;pink&quot;
# names (colors) of the modules
modNames = substring(names(MEs), 3)
geneModuleMembership = as.data.frame(cor(datExpr, MEs, use = &quot;p&quot;))
## 算出每个模块跟基因的皮尔森相关系数矩阵
## MEs是每个模块在每个样本里面的
## datExpr是每个基因在每个样本的表达量
MMPvalue = as.data.frame(corPvalueStudent(as.matrix(geneModuleMembership), nSamples))
names(geneModuleMembership) = paste(&quot;MM&quot;, modNames, sep=&quot;&quot;)
names(MMPvalue) = paste(&quot;p.MM&quot;, modNames, sep=&quot;&quot;)
geneModuleMembership[1:4,1:4]
## 只有连续型性状才能只有计算
## 这里把是否属 Luminal 表型这个变量0,1进行数值化
Luminal = as.data.frame(design[,3])
names(Luminal) = &quot;Luminal&quot;
geneTraitSignificance = as.data.frame(cor(datExpr, Luminal, use = &quot;p&quot;))
GSPvalue = as.data.frame(corPvalueStudent(as.matrix(geneTraitSignificance), nSamples))
names(geneTraitSignificance) = paste(&quot;GS.&quot;, names(Luminal), sep=&quot;&quot;)
names(GSPvalue) = paste(&quot;p.GS.&quot;, names(Luminal), sep=&quot;&quot;)

module = &quot;pink&quot;
column = match(module, modNames)
moduleGenes = moduleColors==module
png(&quot;step6-Module_membership-gene_significance.png&quot;,width = 800,height = 600)
#sizeGrWindow(7, 7)
par(mfrow = c(1,1))
verboseScatterplot(abs(geneModuleMembership[moduleGenes, column]),
                     abs(geneTraitSignificance[moduleGenes, 1]),
                     xlab = paste(&quot;Module Membership in&quot;, module, &quot;module&quot;),
                     ylab = &quot;Gene significance for Luminal&quot;,
                     main = paste(&quot;Module membership vs. gene significance\n&quot;),
                     cex.main = 1.2, cex.lab = 1.2, cex.axis = 1.2, col =module)
dev.off()
</code></pre><p>由于是分类变量，只能按照 0 至 1 量化，可以看出模块内的基因与表型有很好的线性关系</p><p><img data-src="/img/posts/2020.8.8/step6-Module_membership-gene_significance.png" alt></p><p>然后再绘制性状与模块的关系</p><pre><code># Recalculate module eigengenes
MEs = moduleEigengenes(datExpr, moduleColors)$eigengenes
## 只有连续型性状才能只有计算
## 这里把是否属 Luminal 表型这个变量0,1进行数值化
Luminal = as.data.frame(design[,3]);
names(Luminal) = &quot;Luminal&quot;
# Add the weight to existing module eigengenes
MET = orderMEs(cbind(MEs, Luminal))
# Plot the relationships among the eigengenes and the trait
sizeGrWindow(5,7.5);

par(cex = 0.9)
png(&quot;img/step6-Eigengene-dendrogram.png&quot;,width = 800,height = 600)
plotEigengeneNetworks(MET, &quot;&quot;, marDendro = c(0,4,1,2), marHeatmap = c(3,4,1,2), cex.lab = 0.8, xLabelsAngle
                        = 90)
dev.off()

# Plot the dendrogram
sizeGrWindow(6,6);
par(cex = 1.0)
## 模块的进化树
png(&quot;img/step6-Eigengene-dendrogram-hclust.png&quot;,width = 800,height = 600)
plotEigengeneNetworks(MET, &quot;Eigengene dendrogram&quot;, marDendro = c(0,4,2,0),
                        plotHeatmaps = FALSE)
dev.off()
# Plot the heatmap matrix (note: this plot will overwrite the dendrogram plot)
par(cex = 1.0)
## 性状与模块热

png(&quot;img/step6-Eigengene-adjacency-heatmap.png&quot;,width = 800,height = 600)
plotEigengeneNetworks(MET, &quot;Eigengene adjacency heatmap&quot;, marHeatmap = c(3,4,2,2),
                        plotDendrograms = FALSE, xLabelsAngle = 90)
dev.off()# Recalculate module eigengenes
MEs = moduleEigengenes(datExpr, moduleColors)$eigengenes
## 只有连续型性状才能只有计算
## 这里把是否属 Luminal 表型这个变量0,1进行数值化
Luminal = as.data.frame(design[,3]);
names(Luminal) = &quot;Luminal&quot;
# Add the weight to existing module eigengenes
MET = orderMEs(cbind(MEs, Luminal))
# Plot the relationships among the eigengenes and the trait
sizeGrWindow(5,7.5);

par(cex = 0.9)
png(&quot;img/step6-Eigengene-dendrogram.png&quot;,width = 800,height = 600)
plotEigengeneNetworks(MET, &quot;&quot;, marDendro = c(0,4,1,2), marHeatmap = c(3,4,1,2), cex.lab = 0.8, xLabelsAngle
                        = 90)
dev.off()

# Plot the dendrogram
sizeGrWindow(6,6);
par(cex = 1.0)
## 模块的进化树
png(&quot;img/step6-Eigengene-dendrogram-hclust.png&quot;,width = 800,height = 600)
plotEigengeneNetworks(MET, &quot;Eigengene dendrogram&quot;, marDendro = c(0,4,2,0),
                        plotHeatmaps = FALSE)
dev.off()
# Plot the heatmap matrix (note: this plot will overwrite the dendrogram plot)
par(cex = 1.0)
## 性状与模块热

png(&quot;img/step6-Eigengene-adjacency-heatmap.png&quot;,width = 800,height = 600)
plotEigengeneNetworks(MET, &quot;Eigengene adjacency heatmap&quot;, marHeatmap = c(3,4,2,2),
                        plotDendrograms = FALSE, xLabelsAngle = 90)
dev.off()
</code></pre><p><img data-src="/img/posts/2020.8.8/step6-Eigengene-dendrogram.png" alt></p><h1 id="8网络的可视化"><a class="anchor" href="#8网络的可视化">#</a> 8. 网络的可视化</h1><pre><code># 主要是可视化 TOM矩阵，WGCNA的标准配图
# 然后可视化不同 模块 的相关性 热图
# 不同模块的层次聚类图
# 还有模块诊断，主要是 intramodular connectivity

nGenes = ncol(datExpr)
nSamples = nrow(datExpr)
geneTree = net$dendrograms[[1]];
dissTOM = 1-TOMsimilarityFromExpr(datExpr, power = 6);
plotTOM = dissTOM^7;
diag(plotTOM) = NA;
#TOMplot(plotTOM, geneTree, moduleColors, main = &quot;Network heatmap plot, all genes&quot;)
nSelect = 400
# For reproducibility, we set the random seed
set.seed(10);
select = sample(nGenes, size = nSelect);
selectTOM = dissTOM[select, select];
# There’s no simple way of restricting a clustering tree to a subset of genes, so we must re-cluster.
selectTree = hclust(as.dist(selectTOM), method = &quot;average&quot;)
selectColors = moduleColors[select];
# Open a graphical window
sizeGrWindow(9,9)
# Taking the dissimilarity to a power, say 10, makes the plot more informative by effectively changing
# the color palette; setting the diagonal to NA also improves the clarity of the plot
plotDiss = selectTOM^7;
diag(plotDiss) = NA;

png(&quot;img/step7-Network-heatmap.png&quot;,width = 800,height = 600)
  TOMplot(plotDiss, selectTree, selectColors, main = &quot;Network heatmap plot, selected genes&quot;)
dev.off()
</code></pre><p>不知道为啥，这张图有点奇怪</p><p><img data-src="/img/posts/2020.8.8/step7-Network-heatmap.png" alt></p><h1 id="9-模块的导出"><a class="anchor" href="#9-模块的导出">#</a> 9. 模块的导出</h1><pre><code>##===============================step 8：模块导出 =========
# 导出模块内部基因的连接关系，进入其它可视化软件
# 比如 cytoscape软件等等。
# Recalculate topological overlap
TOM = TOMsimilarityFromExpr(datExpr, power = 6);
# Select module
module = &quot;pink&quot;;
# Select module probes
probes = colnames(datExpr) ## 我们例子里面的probe就是基因
inModule = (moduleColors==module)
modProbes = probes[inModule]
## 也是提取指定模块的基因名
# Select the corresponding Topological Overlap
modTOM = TOM[inModule, inModule]
dimnames(modTOM) = list(modProbes, modProbes)
## 模块对应的基因关系矩
cyt = exportNetworkToCytoscape(
    modTOM,
    edgeFile = paste(&quot;CytoscapeInput-edges-&quot;, paste(module, collapse=&quot;-&quot;), &quot;.txt&quot;, sep=&quot;&quot;),
    nodeFile = paste(&quot;CytoscapeInput-nodes-&quot;, paste(module, collapse=&quot;-&quot;), &quot;.txt&quot;, sep=&quot;&quot;),
    weighted = TRUE,
    threshold = 0.02,
    nodeNames = modProbes,
    nodeAttr = moduleColors[inModule]
)
</code></pre><p>模块导出后可以用 cytoscape 构建网络，后面的教程会教大家利用这个文件构建网络图</p><h1 id="后记"><a class="anchor" href="#后记">#</a> 后记</h1><p>本次 WGCNA 的代码结合了<span class="exturl" data-url="aHR0cHM6Ly9naXRodWIuY29tL2ptemVuZzEzMTQvbXlfV0dDTkE=">生信技能树</span>和<span class="exturl" data-url="aHR0cHM6Ly9wbGFudGVjaC5rZS5xcS5jb20="> PlantTech</span> 的 WGCNA 教程，原始数据也来自这两个教程，我将代码和原始数据上传到自己的<span class="exturl" data-url="aHR0cHM6Ly9naXRodWIuY29tL0Jpb2luZm9ybWF0aWNzLXJvb2tpZS9XR0NOQQ=="> github</span> 中，其中 PlantTech 课程是收费课程，我已将其下载，大家有需要可以去<span class="exturl" data-url="aHR0cHM6Ly9wYW4uYmFpZHUuY29tL3MvMTE0ZGNmWmhvMnQ2cHQzWE9oa1lYdGc=">百度云</span>下载（提取码：es41）</p><p>视频压缩包解压密码是我博客 about 界面下的一行小字（出自《蒲公英女孩》，标点是英文），如果链接炸了去博客找我联系方式。</p><div class="tags"><a href="/tags/bioinformatics/" rel="tag"><i class="ic i-tag"></i> bioinformatics</a> <a href="/tags/WGCNA/" rel="tag"><i class="ic i-tag"></i> WGCNA</a></div></div><footer><div class="meta"><span class="item"><span class="icon"><i class="ic i-calendar-check"></i> </span><span class="text">更新于</span> <time title="修改时间：2022-12-14 10:55:44" itemprop="dateModified" datetime="2022-12-14T10:55:44+08:00">2022-12-14</time> </span><span id="2020/08/08/2020-08-08-WGCNA3/" class="item leancloud_visitors" data-flag-title="WGCNA 学习：WGCNA 分析实战" title="阅读次数"><span class="icon"><i class="ic i-eye"></i> </span><span class="text">阅读次数</span> <span class="leancloud-visitors-count"></span> <span class="text">次</span></span></div><div class="reward"><button><i class="ic i-heartbeat"></i> 赞赏</button><p>请我喝[茶]~(￣▽￣)~*</p><div id="qr"><div><img data-src="/images/wechatpay.png" alt="amane 微信支付"><p>微信支付</p></div><div><img data-src="/images/alipay.png" alt="amane 支付宝"><p>支付宝</p></div></div></div><div id="copyright"><ul><li class="author"><strong>本文作者： </strong>amane <i class="ic i-at"><em>@</em></i>小钊の笔记</li><li class="link"><strong>本文链接：</strong> <a href="https://www.zhouxiaozhao.cn/2020/08/08/2020-08-08-WGCNA3/" title="WGCNA 学习：WGCNA 分析实战">https://www.zhouxiaozhao.cn/2020/08/08/2020-08-08-WGCNA3/</a></li><li class="license"><strong>版权声明： </strong>本站所有文章除特别声明外，均采用 <span class="exturl" data-url="aHR0cHM6Ly9jcmVhdGl2ZWNvbW1vbnMub3JnL2xpY2Vuc2VzL2J5LW5jLXNhLzQuMC9kZWVkLnpo"><i class="ic i-creative-commons"><em>(CC)</em></i>BY-NC-SA</span> 许可协议。转载请注明出处！</li></ul></div></footer></article></div><div class="post-nav"><div class="item left"><a href="/2020/08/05/2020-08-05-WGCNA2/" itemprop="url" rel="prev" data-background-image="https:&#x2F;&#x2F;img1.imgtp.com&#x2F;2023&#x2F;02&#x2F;01&#x2F;3G4nGWGt.jpg" title="WGCNA学习：WGCNA应用场景与实例解读"><span class="type">上一篇</span> <span class="category"><i class="ic i-flag"></i> 生信学习</span><h3>WGCNA学习：WGCNA应用场景与实例解读</h3></a></div><div class="item right"><a href="/2020/08/10/2020-08-13-R4ds3/" itemprop="url" rel="next" data-background-image="https:&#x2F;&#x2F;img1.imgtp.com&#x2F;2023&#x2F;02&#x2F;01&#x2F;3G4nGWGt.jpg" title="R数据科学笔记:3"><span class="type">下一篇</span> <span class="category"><i class="ic i-flag"></i> R</span><h3>R数据科学笔记:3</h3></a></div></div><div class="wrap" id="comments"></div></div><div id="sidebar"><div class="inner"><div class="panels"><div class="inner"><div class="contents panel pjax" data-title="文章目录"><ol class="toc"><li class="toc-item toc-level-1"><a class="toc-link" href="#1wgcna%E5%AE%89%E8%A3%85"><span class="toc-number">1.</span> <span class="toc-text">1.WGCNA 安装</span></a></li><li class="toc-item toc-level-1"><a class="toc-link" href="#2%E6%95%B0%E6%8D%AE%E5%87%86%E5%A4%87%E4%B8%8E%E8%AF%BB%E5%85%A5"><span class="toc-number">2.</span> <span class="toc-text">2. 数据准备与读入</span></a><ol class="toc-child"><li class="toc-item toc-level-2"><a class="toc-link" href="#21%E6%95%B0%E6%8D%AE%E5%87%86%E5%A4%87"><span class="toc-number">2.1.</span> <span class="toc-text">2.1 数据准备</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#22-%E6%95%B0%E6%8D%AE%E8%AF%BB%E5%85%A5"><span class="toc-number">2.2.</span> <span class="toc-text">2.2 数据读入</span></a></li></ol></li><li class="toc-item toc-level-1"><a class="toc-link" href="#3-%CE%B2%E5%80%BC%E4%BC%B0%E8%AE%A1"><span class="toc-number">3.</span> <span class="toc-text">3. β 值估计</span></a></li><li class="toc-item toc-level-1"><a class="toc-link" href="#4-%E4%B8%80%E6%AD%A5%E6%B3%95%E6%9E%84%E5%BB%BA%E5%85%B1%E8%A1%A8%E8%BE%BE%E7%9F%A9%E9%98%B5"><span class="toc-number">4.</span> <span class="toc-text">4. 一步法构建共表达矩阵</span></a></li><li class="toc-item toc-level-1"><a class="toc-link" href="#5-%E6%A8%A1%E5%9D%97%E5%8F%AF%E8%A7%86%E5%8C%96"><span class="toc-number">5.</span> <span class="toc-text">5. 模块可视化</span></a></li><li class="toc-item toc-level-1"><a class="toc-link" href="#6%E6%A8%A1%E5%9D%97%E4%B8%8E%E6%80%A7%E7%8A%B6%E7%9A%84%E5%85%B3%E7%B3%BB"><span class="toc-number">6.</span> <span class="toc-text">6. 模块与性状的关系</span></a></li><li class="toc-item toc-level-1"><a class="toc-link" href="#7-%E9%80%89%E6%8B%A9%E6%84%9F%E5%85%B4%E8%B6%A3%E7%9A%84%E6%A8%A1%E5%9D%97%E8%BF%9B%E8%A1%8C%E5%88%86%E6%9E%90"><span class="toc-number">7.</span> <span class="toc-text">7. 选择感兴趣的模块进行分析</span></a></li><li class="toc-item toc-level-1"><a class="toc-link" href="#8%E7%BD%91%E7%BB%9C%E7%9A%84%E5%8F%AF%E8%A7%86%E5%8C%96"><span class="toc-number">8.</span> <span class="toc-text">8. 网络的可视化</span></a></li><li class="toc-item toc-level-1"><a class="toc-link" href="#9-%E6%A8%A1%E5%9D%97%E7%9A%84%E5%AF%BC%E5%87%BA"><span class="toc-number">9.</span> <span class="toc-text">9. 模块的导出</span></a></li><li class="toc-item toc-level-1"><a class="toc-link" href="#%E5%90%8E%E8%AE%B0"><span class="toc-number">10.</span> <span class="toc-text">后记</span></a></li></ol></div><div class="related panel pjax" data-title="系列文章"><ul><li><a href="/2020/07/08/2020-07-08-repeat/" rel="bookmark" title="RepeatModeler+RepeatMasker的安装与使用">RepeatModeler+RepeatMasker的安装与使用</a></li><li><a href="/2020/07/09/2020-07-09-github+jekyll/" rel="bookmark" title="利用Github+Jekyll构建自己的博客">利用Github+Jekyll构建自己的博客</a></li><li><a href="/2020/07/18/2020-07-18-orthofinder/" rel="bookmark" title="利用orthofinder寻找单拷贝基因构建系统发育树">利用orthofinder寻找单拷贝基因构建系统发育树</a></li><li><a href="/2020/07/20/2020-07-20-fungi/" rel="bookmark" title="文献导读：真菌进化">文献导读：真菌进化</a></li><li><a href="/2020/08/03/2020-08-03-WGCNA1/" rel="bookmark" title="WGCNA学习：WGCNA分析原理">WGCNA学习：WGCNA分析原理</a></li><li><a href="/2020/08/05/2020-08-05-WGCNA2/" rel="bookmark" title="WGCNA学习：WGCNA应用场景与实例解读">WGCNA学习：WGCNA应用场景与实例解读</a></li><li class="active"><a href="/2020/08/08/2020-08-08-WGCNA3/" rel="bookmark" title="WGCNA学习：WGCNA分析实战">WGCNA学习：WGCNA分析实战</a></li><li><a href="/2020/08/10/2020-08-10-cytoscape/" rel="bookmark" title="基因共表达网络图的绘制">基因共表达网络图的绘制</a></li><li><a href="/2020/08/18/2020-08-18-sys/" rel="bookmark" title="根据转录组数据鉴定共生物种">根据转录组数据鉴定共生物种</a></li><li><a href="/2020/08/20/2020-08-20-nt/" rel="bookmark" title="使用本地nt数据库对reads和Trinity结果进行blast">使用本地nt数据库对reads和Trinity结果进行blast</a></li><li><a href="/2020/08/23/2020-08-23-IGV/" rel="bookmark" title="使用IGV简单绘制甲基化分布图">使用IGV简单绘制甲基化分布图</a></li><li><a href="/2020/08/25/2020-08-25-evolview1/" rel="bookmark" title="进化树美化之Evolview--热图（1）">进化树美化之Evolview--热图（1）</a></li><li><a href="/2020/08/28/2020-08-28-evolview/" rel="bookmark" title="进化树美化之Evolview--热图（2）">进化树美化之Evolview--热图（2）</a></li><li><a href="/2020/08/30/2020-08-30-evolview3/" rel="bookmark" title="进化树+蛋白结构域">进化树+蛋白结构域</a></li><li><a href="/2020/09/01/2020-09-01-iTOL1/" rel="bookmark" title="进化树在线制作工具iTOL教程（1）：基本操作">进化树在线制作工具iTOL教程（1）：基本操作</a></li><li><a href="/2020/09/03/2020-09-03-iTOL2/" rel="bookmark" title="进化树在线制作工具iTOL教程（2）：添加文本">进化树在线制作工具iTOL教程（2）：添加文本</a></li><li><a href="/2020/09/06/2020-09-06-iTOL3/" rel="bookmark" title="进化树在线制作工具iTOL教程（3）：添加图片">进化树在线制作工具iTOL教程（3）：添加图片</a></li><li><a href="/2020/09/08/2020-09-08-iTOL4/" rel="bookmark" title="进化树在线制作工具iTOL教程（4）：添加点图">进化树在线制作工具iTOL教程（4）：添加点图</a></li><li><a href="/2020/09/29/2020-09-29-ScRNAseq(1)/" rel="bookmark" title="单细胞实战(1)：原始测序数据下载">单细胞实战(1)：原始测序数据下载</a></li><li><a href="/2020/10/01/2020-10-01-ScRNAseq(2)/" rel="bookmark" title="单细胞实战(2)：cellranger使用">单细胞实战(2)：cellranger使用</a></li><li><a href="/2020/10/03/2020-10-03-ScRNAseq(3)/" rel="bookmark" title="单细胞实战(3)：STAR分析单细胞数据">单细胞实战(3)：STAR分析单细胞数据</a></li><li><a href="/2020/10/06/2020-10-06-ScRNAseq(4)/" rel="bookmark" title="单细胞实战(4)：STAR与cellranger结果比较">单细胞实战(4)：STAR与cellranger结果比较</a></li><li><a href="/2020/10/10/2020-10-10-ScRNAseq(5)/" rel="bookmark" title="单细胞实战(5)：复现文章中的聚类图（1）">单细胞实战(5)：复现文章中的聚类图（1）</a></li><li><a href="/2020/10/13/2020-10-13-ScRNAseq(6)/" rel="bookmark" title="单细胞实战(5)：复现文章中的聚类图（2）">单细胞实战(5)：复现文章中的聚类图（2）</a></li><li><a href="/2020/10/24/2020-10-24-ScRNAseq(7)/" rel="bookmark" title="生信技能树单细胞数据挖掘笔记(1)">生信技能树单细胞数据挖掘笔记(1)</a></li><li><a href="/2020/10/27/2020-10-27-ScRNAseq(8)/" rel="bookmark" title="生信技能树单细胞数据挖掘笔记(2)">生信技能树单细胞数据挖掘笔记(2)</a></li><li><a href="/2020/10/29/2020-10-29-ScRNAseq(9)/" rel="bookmark" title="生信技能树单细胞数据挖掘笔记(3)">生信技能树单细胞数据挖掘笔记(3)</a></li><li><a href="/2020/10/31/2020-10-31-ScRNAseq(10)/" rel="bookmark" title="生信技能树单细胞数据挖掘笔记(4)">生信技能树单细胞数据挖掘笔记(4)</a></li><li><a href="/2020/11/03/2020-11-03-ScRNAseq(11)/" rel="bookmark" title="生信技能树单细胞数据挖掘笔记(5)">生信技能树单细胞数据挖掘笔记(5)</a></li><li><a href="/2020/11/05/2020-11-05-ScRNAseq(12)/" rel="bookmark" title="生信技能树单细胞数据挖掘笔记(6)">生信技能树单细胞数据挖掘笔记(6)</a></li><li><a href="/2020/11/07/2020-11-07-ScRNAseq(13)/" rel="bookmark" title="生信技能树单细胞数据挖掘笔记(7)">生信技能树单细胞数据挖掘笔记(7)</a></li><li><a href="/2020/11/10/2020-11-10-RNAvelocity(1)/" rel="bookmark" title="RNA速率：软件下载与loom文件准备">RNA速率：软件下载与loom文件准备</a></li><li><a href="/2020/11/12/2020-11-12-RNAvelocity(2)/" rel="bookmark" title="RNA速率：数据读入">RNA速率：数据读入</a></li><li><a href="/2020/11/14/2020-11-14-RNAvelocity(3)/" rel="bookmark" title="RNA速率：使用Seurat的结果做RNA velocity">RNA速率：使用Seurat的结果做RNA velocity</a></li><li><a href="/2020/11/17/2020-11-17-RNAseq(1)/" rel="bookmark" title="RNAseq教程(1.1)">RNAseq教程(1.1)</a></li><li><a href="/2020/11/19/2020-11-19-annotion(1)/" rel="bookmark" title="基因结构注释（1）：从头注释">基因结构注释（1）：从头注释</a></li><li><a href="/2020/11/21/2020-11-21-annotion(2)/" rel="bookmark" title="基因结构注释（2）：同源注释">基因结构注释（2）：同源注释</a></li><li><a href="/2020/11/24/2020-11-24-annotion(3)/" rel="bookmark" title="基因结构注释（3）：转录组注释">基因结构注释（3）：转录组注释</a></li><li><a href="/2020/11/26/2020-11-26-annotion(4)/" rel="bookmark" title="基因结构注释（4）：整合预测结果">基因结构注释（4）：整合预测结果</a></li><li><a href="/2020/11/28/2020-11-28-RNAseq(2)/" rel="bookmark" title="RNAseq教程(1.2)">RNAseq教程(1.2)</a></li><li><a href="/2020/12/01/2020-12-01-RNAseq(3)/" rel="bookmark" title="RNAseq教程(1.3)">RNAseq教程(1.3)</a></li><li><a href="/2020/12/03/2020-12-03-RNAseq(4)/" rel="bookmark" title="RNAseq教程(1.4)">RNAseq教程(1.4)</a></li><li><a href="/2020/12/05/2020-12-05-RNAseq(5)/" rel="bookmark" title="RNAseq教程(1.5)">RNAseq教程(1.5)</a></li><li><a href="/2020/12/08/2020-12-08-RNAseq(6)/" rel="bookmark" title="RNAseq教程(1.6)">RNAseq教程(1.6)</a></li><li><a href="/2020/12/10/2020-12-10-RNAseq(7)/" rel="bookmark" title="RNAseq教程(2.1)">RNAseq教程(2.1)</a></li><li><a href="/2020/12/12/2020-12-12-RNAseq(8)/" rel="bookmark" title="RNAseq教程(2.2)">RNAseq教程(2.2)</a></li><li><a href="/2020/12/15/2020-12-15-RNAseq(9)/" rel="bookmark" title="RNAseq教程(2.3)">RNAseq教程(2.3)</a></li><li><a href="/2020/12/17/2020-12-17-RNAseq(10)/" rel="bookmark" title="RNAseq教程(2.4)">RNAseq教程(2.4)</a></li><li><a href="/2020/12/19/2020-12-19-RNAseq(11)/" rel="bookmark" title="RNAseq教程(2.5)">RNAseq教程(2.5)</a></li><li><a href="/2020/12/22/2020-12-22-RNAseq(12)/" rel="bookmark" title="RNAseq教程(3.1)">RNAseq教程(3.1)</a></li><li><a href="/2020/12/24/2020-12-24-RNAseq(13)/" rel="bookmark" title="RNAseq教程(3.2)">RNAseq教程(3.2)</a></li><li><a href="/2020/12/26/2020-12-26-RNAseq(14)/" rel="bookmark" title="RNAseq教程(3.3)">RNAseq教程(3.3)</a></li><li><a href="/2020/12/29/2020-12-29-RNAseq(15)/" rel="bookmark" title="RNAseq教程(3.4)">RNAseq教程(3.4)</a></li><li><a href="/2020/12/31/2020-12-31-RNAseq(16)/" rel="bookmark" title="RNAseq教程(4.1)">RNAseq教程(4.1)</a></li><li><a href="/2021/01/02/2021-01-02-RNAseq(17)/" rel="bookmark" title="RNAseq教程(4.2)">RNAseq教程(4.2)</a></li><li><a href="/2021/01/05/2021-01-05-RNAseq(18)/" rel="bookmark" title="RNAseq教程(4.3)">RNAseq教程(4.3)</a></li><li><a href="/2021/01/07/2021-01-07-RNAseq(19)/" rel="bookmark" title="RNAseq教程(4.4)">RNAseq教程(4.4)</a></li><li><a href="/2021/01/09/2021-01-09-RNAseq(20)/" rel="bookmark" title="RNAseq教程(4.5)">RNAseq教程(4.5)</a></li><li><a href="/2021/01/12/2021-01-12-RNAseq(21)/" rel="bookmark" title="RNAseq教程(5)">RNAseq教程(5)</a></li><li><a href="/2022/12/08/2022-12-08-SingleCopy/" rel="bookmark" title="比较基因组学分析1：构建单拷贝基因树">比较基因组学分析1：构建单拷贝基因树</a></li><li><a href="/2022/12/14/2022-12-14-CAFE/" rel="bookmark" title="比较基因组学分析2：基因家族收缩与扩张分析">比较基因组学分析2：基因家族收缩与扩张分析</a></li><li><a href="/2022/12/15/2022-12-15-GO/" rel="bookmark" title="比较基因组学分析3：特异节点基因家族富集分析（非模式物种GO/KEEG库构建）">比较基因组学分析3：特异节点基因家族富集分析（非模式物种GO/KEEG库构建）</a></li></ul></div><div class="overview panel" data-title="站点概览"><div class="author" itemprop="author" itemscope itemtype="http://schema.org/Person"><img class="image" itemprop="image" alt="amane" data-src="/images/avatar.png"><p class="name" itemprop="name">amane</p><div class="description" itemprop="description">生物学在读博士</div></div><nav class="state"><div class="item posts"><a href="/archives/"><span class="count">83</span> <span class="name">文章</span></a></div><div class="item categories"><a href="/categories/"><span class="count">6</span> <span class="name">分类</span></a></div><div class="item tags"><a href="/tags/"><span class="count">20</span> <span class="name">标签</span></a></div></nav><div class="social"><span class="exturl item github" data-url="aHR0cHM6Ly9naXRodWIuY29tL0Jpb2luZm9ybWF0aWNzLXJvb2tpZQ==" title="https:&#x2F;&#x2F;github.com&#x2F;Bioinformatics-rookie"><i class="ic i-github"></i></span> <a href="https://www.zhouxiaozhao.cn/about/" title="https:&#x2F;&#x2F;www.zhouxiaozhao.cn&#x2F;about&#x2F;" class="item about"><i class="ic i-address-card"></i></a> <span class="exturl item email" data-url="bWFpbHRvOnpob3VzaGl6aGFvQG1haWwua2liLmFjLmNu" title="mailto:zhoushizhao@mail.kib.ac.cn"><i class="ic i-envelope"></i></span></div><ul class="menu"><li class="item"><a href="/" rel="section"><i class="ic i-home"></i>首页</a></li><li class="item"><a href="/about/" rel="section"><i class="ic i-user"></i>关于</a></li><li class="item dropdown"><a href="javascript:void(0);"><i class="ic i-feather"></i>文章</a><ul class="submenu"><li class="item"><a href="/archives/" rel="section"><i class="ic i-list-alt"></i>归档</a></li><li class="item"><a href="/categories/" rel="section"><i class="ic i-th"></i>分类</a></li><li class="item"><a href="/tags/" rel="section"><i class="ic i-tags"></i>标签</a></li></ul></li><li class="item"><a href="/links/" rel="section"><i class="ic i-heart"></i>友達</a></li></ul></div></div></div><ul id="quick"><li class="prev pjax"><a href="/2020/08/05/2020-08-05-WGCNA2/" rel="prev" title="上一篇"><i class="ic i-chevron-left"></i></a></li><li class="up"><i class="ic i-arrow-up"></i></li><li class="down"><i class="ic i-arrow-down"></i></li><li class="next pjax"><a href="/2020/08/10/2020-08-13-R4ds3/" rel="next" title="下一篇"><i class="ic i-chevron-right"></i></a></li><li class="percent"></li></ul></div></div><div class="dimmer"></div></div></main><footer id="footer"><div class="inner"><div class="widgets"><div class="rpost pjax"><h2>随机文章</h2><ul><li class="item"><div class="breadcrumb"><a href="/categories/Bioinformation/" title="分类于 生信学习">生信学习</a></div><span><a href="/2020/08/25/2020-08-25-evolview1/" title="进化树美化之Evolview--热图（1）">进化树美化之Evolview--热图（1）</a></span></li><li class="item"><div class="breadcrumb"><a href="/categories/Python/" title="分类于 Python">Python</a></div><span><a href="/2020/10/22/2020-10-22-python9/" title="马士兵教育Python入门基础:4.流程控制（3）">马士兵教育Python入门基础:4.流程控制（3）</a></span></li><li class="item"><div class="breadcrumb"><a href="/categories/Bioinformation/" title="分类于 生信学习">生信学习</a></div><span><a href="/2020/12/19/2020-12-19-RNAseq(11)/" title="RNAseq教程(2.5)">RNAseq教程(2.5)</a></span></li><li class="item"><div class="breadcrumb"><a href="/categories/Bioinformation/" title="分类于 生信学习">生信学习</a></div><span><a href="/2020/08/20/2020-08-20-nt/" title="使用本地nt数据库对reads和Trinity结果进行blast">使用本地nt数据库对reads和Trinity结果进行blast</a></span></li><li class="item"><div class="breadcrumb"><a href="/categories/Bioinformation/" title="分类于 生信学习">生信学习</a></div><span><a href="/2021/01/09/2021-01-09-RNAseq(20)/" title="RNAseq教程(4.5)">RNAseq教程(4.5)</a></span></li><li class="item"><div class="breadcrumb"><a href="/categories/Bioinformation/" title="分类于 生信学习">生信学习</a></div><span><a href="/2020/11/14/2020-11-14-RNAvelocity(3)/" title="RNA速率：使用Seurat的结果做RNA velocity">RNA速率：使用Seurat的结果做RNA velocity</a></span></li><li class="item"><div class="breadcrumb"><a href="/categories/Bioinformation/" title="分类于 生信学习">生信学习</a></div><span><a href="/2020/09/29/2020-09-29-ScRNAseq(1)/" title="单细胞实战(1)：原始测序数据下载">单细胞实战(1)：原始测序数据下载</a></span></li><li class="item"><div class="breadcrumb"><a href="/categories/Bioinformation/" title="分类于 生信学习">生信学习</a></div><span><a href="/2020/12/22/2020-12-22-RNAseq(12)/" title="RNAseq教程(3.1)">RNAseq教程(3.1)</a></span></li><li class="item"><div class="breadcrumb"><a href="/categories/Bioinformation/" title="分类于 生信学习">生信学习</a></div><span><a href="/2020/08/18/2020-08-18-sys/" title="根据转录组数据鉴定共生物种">根据转录组数据鉴定共生物种</a></span></li><li class="item"><div class="breadcrumb"><a href="/categories/paper/" title="分类于 文献导读">文献导读</a></div><span><a href="/2020/09/19/2020-09-19-three-dimensional_shoot_development/" title="文献导读：小立碗藓三维发育的细胞特异性转录组分析">文献导读：小立碗藓三维发育的细胞特异性转录组分析</a></span></li></ul></div><div><h2>最新评论</h2><ul class="leancloud-recent-comment"></ul></div></div><div class="status"><div class="copyright">&copy; 2010 – <span itemprop="copyrightYear">2023</span> <span class="with-love"><i class="ic i-sakura rotate"></i> </span><span class="author" itemprop="copyrightHolder">amane @ Zhou Xiaozhao</span></div><div class="count"><span class="post-meta-item-icon"><i class="ic i-chart-area"></i> </span><span title="站点总字数">344k 字</span> <span class="post-meta-divider">|</span> <span class="post-meta-item-icon"><i class="ic i-coffee"></i> </span><span title="站点阅读时长">5:13</span></div><div class="powered-by">基于 <span class="exturl" data-url="aHR0cHM6Ly9oZXhvLmlv">Hexo</span> & Theme.<span class="exturl" data-url="aHR0cHM6Ly9naXRodWIuY29tL2FtZWhpbWUvaGV4by10aGVtZS1zaG9rYQ==">Shoka</span></div></div></div></footer></div><script data-config type="text/javascript">var LOCAL={path:"2020/08/08/2020-08-08-WGCNA3/",favicon:{show:"（●´3｀●）やれやれだぜ",hide:"(´Д｀)大変だ！"},search:{placeholder:"文章搜索",empty:"关于 「 ${query} 」，什么也没搜到",stats:"${time} ms 内找到 ${hits} 条结果"},valine:!0,fancybox:!0,copyright:'复制成功，转载请遵守 <i class="ic i-creative-commons"></i>BY-NC-SA 协议。',ignores:[function(e){return e.includes("#")},function(e){return new RegExp(LOCAL.path+"$").test(e)}]}</script><script src="https://cdn.polyfill.io/v2/polyfill.js"></script><script src="//cdn.jsdelivr.net/combine/npm/pace-js@1.0.2/pace.min.js,npm/pjax@0.2.8/pjax.min.js,npm/whatwg-fetch@3.4.0/dist/fetch.umd.min.js,npm/animejs@3.2.0/lib/anime.min.js,npm/algoliasearch@4/dist/algoliasearch-lite.umd.js,npm/instantsearch.js@4/dist/instantsearch.production.min.js,npm/lozad@1/dist/lozad.min.js,npm/quicklink@2/dist/quicklink.umd.js"></script><script src="/js/app.js?v=0.2.5"></script></body></html><!-- rebuild by hrmmi -->