<!-- build time:Wed Mar 08 2023 15:14:02 GMT+0800 (中国标准时间) --><!DOCTYPE html><html lang="zh-CN"><head><meta charset="UTF-8"><meta name="viewport" content="width=device-width,initial-scale=1,maximum-scale=2"><meta name="theme-color" content="#FFF"><link rel="apple-touch-icon" sizes="180x180" href="/images/apple-touch-icon.png"><link rel="icon" type="image/ico" sizes="32x32" href="/images/favicon.ico"><meta http-equiv="Cache-Control" content="no-transform"><meta http-equiv="Cache-Control" content="no-siteapp"><link rel="alternate" type="application/rss+xml" title="小钊の笔记" href="https://www.zhouxiaozhao.cn/rss.xml"><link rel="alternate" type="application/atom+xml" title="小钊の笔记" href="https://www.zhouxiaozhao.cn/atom.xml"><link rel="alternate" type="application/json" title="小钊の笔记" href="https://www.zhouxiaozhao.cn/feed.json"><link rel="stylesheet" href="//fonts.googleapis.com/css?family=Mulish:300,300italic,400,400italic,700,700italic%7CFredericka%20the%20Great:300,300italic,400,400italic,700,700italic%7CNoto%20Serif%20JP:300,300italic,400,400italic,700,700italic%7CNoto%20Serif%20SC:300,300italic,400,400italic,700,700italic%7CInconsolata:300,300italic,400,400italic,700,700italic&display=swap&subset=latin,latin-ext"><link rel="stylesheet" href="/css/app.css?v=0.2.5"><meta name="keywords" content="bioinformatics,Comparative_Genomics"><link rel="canonical" href="https://www.zhouxiaozhao.cn/2022/12/15/2022-12-15-GO/"><title>比较基因组学分析 3：特异节点基因家族富集分析（非模式物种 GO/KEEG 库构建） - 基因组 - 生信学习 | Zhou Xiaozhao = 小钊の笔记 = 前天是小兔子，昨天是小鹿，今天是你</title><meta name="generator" content="Hexo 5.4.1"></head><body itemscope itemtype="http://schema.org/WebPage"><div id="loading"><div class="cat"><div class="body"></div><div class="head"><div class="face"></div></div><div class="foot"><div class="tummy-end"></div><div class="bottom"></div><div class="legs left"></div><div class="legs right"></div></div><div class="paw"><div class="hands left"></div><div class="hands right"></div></div></div></div><div id="container"><header id="header" itemscope itemtype="http://schema.org/WPHeader"><div class="inner"><div id="brand"><div class="pjax"><h1 itemprop="name headline">比较基因组学分析 3：特异节点基因家族富集分析（非模式物种 GO/KEEG 库构建）</h1><div class="meta"><span class="item" title="创建时间：2022-12-15 00:00:00"><span class="icon"><i class="ic i-calendar"></i> </span><span class="text">发表于</span> <time itemprop="dateCreated datePublished" datetime="2022-12-15T00:00:00+08:00">2022-12-15</time> </span><span class="item" title="本文字数"><span class="icon"><i class="ic i-pen"></i> </span><span class="text">本文字数</span> <span>15k</span> <span class="text">字</span> </span><span class="item" title="阅读时长"><span class="icon"><i class="ic i-clock"></i> </span><span class="text">阅读时长</span> <span>13 分钟</span></span></div></div></div><nav id="nav"><div class="inner"><div class="toggle"><div class="lines" aria-label="切换导航栏"><span class="line"></span> <span class="line"></span> <span class="line"></span></div></div><ul class="menu"><li class="item title"><a href="/" rel="start">Zhou Xiaozhao</a></li></ul><ul class="right"><li class="item theme"><i class="ic i-sun"></i></li><li class="item search"><i class="ic i-search"></i></li></ul></div></nav></div><div id="imgs" class="pjax"><ul><li class="item" data-background-image="https://img1.imgtp.com/2023/02/01/tzn2O8cD.jpg"></li><li class="item" data-background-image="https://img1.imgtp.com/2023/02/01/XEdGqgmv.jpg"></li><li class="item" data-background-image="https://img1.imgtp.com/2023/02/01/TNhS2mtU.jpg"></li><li class="item" data-background-image="https://img1.imgtp.com/2023/02/01/6iJnFub2.jpg"></li><li class="item" data-background-image="https://img1.imgtp.com/2023/02/01/3G4nGWGt.jpg"></li><li class="item" data-background-image="https://img1.imgtp.com/2023/02/01/f1mI0PAU.jpg"></li></ul></div></header><div id="waves"><svg class="waves" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" viewBox="0 24 150 28" preserveAspectRatio="none" shape-rendering="auto"><defs><path id="gentle-wave" d="M-160 44c30 0 58-18 88-18s 58 18 88 18 58-18 88-18 58 18 88 18 v44h-352z"/></defs><g class="parallax"><use xlink:href="#gentle-wave" x="48" y="0"/><use xlink:href="#gentle-wave" x="48" y="3"/><use xlink:href="#gentle-wave" x="48" y="5"/><use xlink:href="#gentle-wave" x="48" y="7"/></g></svg></div><main><div class="inner"><div id="main" class="pjax"><div class="article wrap"><div class="breadcrumb" itemscope itemtype="https://schema.org/BreadcrumbList"><i class="ic i-home"></i> <span><a href="/">首页</a></span><i class="ic i-angle-right"></i> <span itemprop="itemListElement" itemscope itemtype="https://schema.org/ListItem"><a href="/categories/Bioinformation/" itemprop="item" rel="index" title="分类于 生信学习"><span itemprop="name">生信学习</span></a><meta itemprop="position" content="1"></span><i class="ic i-angle-right"></i> <span class="current" itemprop="itemListElement" itemscope itemtype="https://schema.org/ListItem"><a href="/categories/Bioinformation/genome/" itemprop="item" rel="index" title="分类于 基因组"><span itemprop="name">基因组</span></a><meta itemprop="position" content="2"></span></div><article itemscope itemtype="http://schema.org/Article" class="post block" lang="zh-CN"><link itemprop="mainEntityOfPage" href="https://www.zhouxiaozhao.cn/2022/12/15/2022-12-15-GO/"><span hidden itemprop="author" itemscope itemtype="http://schema.org/Person"><meta itemprop="image" content="/images/avatar.png"><meta itemprop="name" content="amane"><meta itemprop="description" content="前天是小兔子，昨天是小鹿，今天是你, 生物学在读博士"></span><span hidden itemprop="publisher" itemscope itemtype="http://schema.org/Organization"><meta itemprop="name" content="小钊の笔记"></span><div class="body md" itemprop="articleBody"><h1 id="1前言"><a class="anchor" href="#1前言">#</a> 1. 前言</h1><p>上次讲到基因家族的收缩扩张分析，这次我们以某个节点为切入点，进行特异节点的功能富集分析<br>其实这篇教程适用于所有的非模式物种 (没有现成的 GO/KEGG 库的物种) 的富集分析<br>物种特异的比较简单就不进行实践了</p><h1 id="2节点选择"><a class="anchor" href="#2节点选择">#</a> 2. 节点选择</h1><p>这次选择一个进化上比较重要的节点，水生到陆生的过程，对应进化树为 node25<br><img data-src="/img/posts/2022.12.15/1.png" alt="进化树"></p><h1 id="3获取目的基因"><a class="anchor" href="#3获取目的基因">#</a> 3. 获取目的基因</h1><p>主要是目标基因与背景基因的选择，目标基因可以是该节点显著扩张的基因家族包含的所有物种基因，背景基因一般为该节点包含的 Orthogroup</p><h2 id="31-获取node25显著扩张的基因"><a class="anchor" href="#31-获取node25显著扩张的基因">#</a> 3.1 获取 node25 显著扩张的基因</h2><figure class="highlight raw"><figcaption data-lang></figcaption><table><tr><td data-num="1"></td><td><pre>cat Gamma_p0.05change.tab | cut -f1,27 | grep &quot;+[1-9]&quot; | cut -f1 &gt; node25significant.expand</pre></td></tr><tr><td data-num="2"></td><td><pre>#node25 显著扩张的 orthogroupsID</pre></td></tr><tr><td data-num="3"></td><td><pre>grep -f node0significant.expand Orthogroups.txt |sed &quot;s&#x2F;&#x2F;\n&#x2F;g&quot; | grep -E &quot;Ath|Csa|Vvi|Aof|Osa|Zma|Nco|Atr|Tpl|Cri|Smo|Mpo|Ppa&quot; | sort | uniq &gt; node25significant.expand.genes</pre></td></tr></table></figure><p>目前我们得到了需要进行富集分析的目标基因，接下来是背景基因的选择</p><h2 id="32-node25背景基因选择"><a class="anchor" href="#32-node25背景基因选择">#</a> 3.2 node25 背景基因选择</h2><figure class="highlight raw"><figcaption data-lang></figcaption><table><tr><td data-num="1"></td><td><pre>awk &#39;NR!&#x3D;1 &amp;&amp; $27&gt;0 &#123;print $0&#125;&#39; Gamma_count.tab | cut -f1 &gt; node25.orthogroups</pre></td></tr><tr><td data-num="2"></td><td><pre>#node25 中存在基因的 orthogroupsID</pre></td></tr><tr><td data-num="3"></td><td><pre>grep -f node25.orthogroups Orthogroups.txt |sed &quot;s&#x2F;&#x2F;\n&#x2F;g&quot; | grep -E &quot;Ath|Csa|Vvi|Aof|Osa|Zma|Nco|Atr|Tpl|Cri|Smo|Mpo|Ppa&quot; | sort | uniq &gt; node25.genes</pre></td></tr></table></figure><p><span class="label primary">node25.genes</span> 和<span class="label primary"> node25significant.expand.genes</span> 就是我们进行富集分析所需要的基因号，接下来就是构建背景基因的 GO 和 KEGG 注释，由于是无参物种，所以我们需要自己构建</p><h1 id="4-背景基因的go和kegg注释"><a class="anchor" href="#4-背景基因的go和kegg注释">#</a> 4 背景基因的 GO 和 KEGG 注释</h1><h2 id="41-gokegg注释"><a class="anchor" href="#41-gokegg注释">#</a> 4.1 GO/KEGG 注释</h2><p>使用 eggnog 对背景基因进行注释</p><figure class="highlight raw"><figcaption data-lang></figcaption><table><tr><td data-num="1"></td><td><pre>seqkit grep -f node25.genes ..&#x2F;..&#x2F;..&#x2F;allpep.fa &gt; node25background.fa </pre></td></tr><tr><td data-num="2"></td><td><pre># 提取背景基因，需要严格注意基因数是否一致</pre></td></tr><tr><td data-num="3"></td><td><pre>emapper.py -m diamond -i node25background.fa -o node25 --cpu 16</pre></td></tr></table></figure><p>最后我们拿到<span class="label primary"> node25.emapper.annotations</span> 文件，将前三行和 #删除<br><img data-src="/img/posts/2022.12.15/2.png" alt="node25.emapper.annotations"></p><h2 id="42-go注释库构建"><a class="anchor" href="#42-go注释库构建">#</a> 4.2 GO 注释库构建</h2><p>获取 GO 的注释信息：下载地址：<span class="exturl" data-url="aHR0cDovL3B1cmwub2JvbGlicmFyeS5vcmcvb2JvL2dvL2dvLWJhc2ljLm9ibw==">http://purl.obolibrary.org/obo/go/go-basic.obo</span> 之后进行修饰</p><figure class="highlight raw"><figcaption data-lang></figcaption><table><tr><td data-num="1"></td><td><pre>grep &quot;^id:&quot; go-basic.obo |awk &#39;&#123;print $2&#125;&#39; &gt; GO.id</pre></td></tr><tr><td data-num="2"></td><td><pre>grep &quot;^name:&quot; go-basic.obo |awk &#39;&#123;print $2&#125;&#39; &gt; GO.name </pre></td></tr><tr><td data-num="3"></td><td><pre>grep &quot;^namespace:&quot; go-basic.obo |awk &#39;&#123;print $2&#125;&#39; &gt; GO.class</pre></td></tr><tr><td data-num="4"></td><td><pre>paste GO.id GO.name GO.class -d &quot;\t&quot; &gt; GO.library</pre></td></tr></table></figure><p>文件一共有三列</p><figure class="highlight raw"><figcaption data-lang></figcaption><table><tr><td data-num="1"></td><td><pre>GO:1903040	exon-exon junction complex assembly	biological process</pre></td></tr><tr><td data-num="2"></td><td><pre>GO:1903045	neural crest cell migration involved in sympathetic nervous system development	biological process</pre></td></tr><tr><td data-num="3"></td><td><pre>GO:1903046	meiotic cell cycle process	biological process</pre></td></tr><tr><td data-num="4"></td><td><pre>GO:1903043	positive regulation of chondrocyte hypertrophy	biological process</pre></td></tr><tr><td data-num="5"></td><td><pre>GO:1903044	protein localization to membrane raft	biological process</pre></td></tr><tr><td data-num="6"></td><td><pre>GO:1903049	negative regulation of acetylcholine-gated cation channel activity	biological process</pre></td></tr><tr><td data-num="7"></td><td><pre>GO:1903047	mitotic cell cycle process	biological process</pre></td></tr><tr><td data-num="8"></td><td><pre>GO:1903048	regulation of acetylcholine-gated cation channel activity	biological process</pre></td></tr><tr><td data-num="9"></td><td><pre>GO:1903012	positive regulation of bone development	biological process</pre></td></tr><tr><td data-num="10"></td><td><pre>GO:1903013	response to differentiation-inducing factor 1	biological process</pre></td></tr><tr><td data-num="11"></td><td><pre>GO:1903010	regulation of bone development	biological process</pre></td></tr><tr><td data-num="12"></td><td><pre>GO:1903011	negative regulation of bone development	biological process</pre></td></tr><tr><td data-num="13"></td><td><pre>GO:1903016	negative regulation of exo-alpha-sialidase activity	biological process</pre></td></tr><tr><td data-num="14"></td><td><pre>GO:1903017	positive regulation of exo-alpha-sialidase activity	biological process</pre></td></tr><tr><td data-num="15"></td><td><pre>GO:1903014	cellular response to differentiation-inducing factor 1	biological process</pre></td></tr><tr><td data-num="16"></td><td><pre>GO:1903015	regulation of exo-alpha-sialidase activity	biological process</pre></td></tr><tr><td data-num="17"></td><td><pre>GO:1903018	regulation of glycoprotein metabolic process	biological process</pre></td></tr><tr><td data-num="18"></td><td><pre>GO:1903019	negative regulation of glycoprotein metabolic process	biological process</pre></td></tr><tr><td data-num="19"></td><td><pre>GO:1903020	positive regulation of glycoprotein metabolic process	biological process</pre></td></tr><tr><td data-num="20"></td><td><pre>GO:1903023	regulation of ascospore-type prospore membrane assembly	biological process</pre></td></tr></table></figure><h2 id="43-kegg注释库构建"><a class="anchor" href="#43-kegg注释库构建">#</a> 4.3 KEGG 注释库构建</h2><p>需要下载<span class="label primary"> ko00001.json</span>： <span class="exturl" data-url="aHR0cHM6Ly93d3cuZ2Vub21lLmpwL2tlZ2ctYmluL2dldF9odGV4dD9rbzAwMDAx">https://www.genome.jp/kegg-bin/get_htext?ko00001</span></p><figure class="highlight raw"><figcaption data-lang></figcaption><table><tr><td data-num="1"></td><td><pre># 需要下载 json 文件 (这是是经常更新的)</pre></td></tr><tr><td data-num="2"></td><td><pre># https:&#x2F;&#x2F;www.genome.jp&#x2F;kegg-bin&#x2F;get_htext?ko00001</pre></td></tr><tr><td data-num="3"></td><td><pre># 代码来自：http:&#x2F;&#x2F;www.genek.tv&#x2F;course&#x2F;225&#x2F;task&#x2F;4861&#x2F;show</pre></td></tr><tr><td data-num="4"></td><td><pre>library (jsonlite)</pre></td></tr><tr><td data-num="5"></td><td><pre>library (purrr)</pre></td></tr><tr><td data-num="6"></td><td><pre>library (RCurl)</pre></td></tr><tr><td data-num="7"></td><td><pre>library (dplyr)</pre></td></tr><tr><td data-num="8"></td><td><pre>library (stringr)</pre></td></tr><tr><td data-num="9"></td><td><pre> kegg &lt;- function (json &#x3D; &quot;ko00001.json&quot;) &#123;</pre></td></tr><tr><td data-num="10"></td><td><pre>    pathway2name &lt;- tibble (Pathway &#x3D; character (), Name &#x3D; character ())</pre></td></tr><tr><td data-num="11"></td><td><pre>    ko2pathway &lt;- tibble (Ko &#x3D; character (), Pathway &#x3D; character ())</pre></td></tr><tr><td data-num="12"></td><td><pre>    </pre></td></tr><tr><td data-num="13"></td><td><pre>    kegg &lt;- fromJSON (json)</pre></td></tr><tr><td data-num="14"></td><td><pre>    </pre></td></tr><tr><td data-num="15"></td><td><pre>    for (a in seq_along (kegg [[&quot;children&quot;]][[&quot;children&quot;]])) &#123;</pre></td></tr><tr><td data-num="16"></td><td><pre>      A &lt;- kegg [[&quot;children&quot;]][[&quot;name&quot;]][[a]]</pre></td></tr><tr><td data-num="17"></td><td><pre>      </pre></td></tr><tr><td data-num="18"></td><td><pre>      for (b in seq_along (kegg [[&quot;children&quot;]][[&quot;children&quot;]][[a]][[&quot;children&quot;]])) &#123;</pre></td></tr><tr><td data-num="19"></td><td><pre>        B &lt;- kegg [[&quot;children&quot;]][[&quot;children&quot;]][[a]][[&quot;name&quot;]][[b]] </pre></td></tr><tr><td data-num="20"></td><td><pre>        </pre></td></tr><tr><td data-num="21"></td><td><pre>        for (c in seq_along (kegg [[&quot;children&quot;]][[&quot;children&quot;]][[a]][[&quot;children&quot;]][[b]][[&quot;children&quot;]])) &#123;</pre></td></tr><tr><td data-num="22"></td><td><pre>          pathway_info &lt;- kegg [[&quot;children&quot;]][[&quot;children&quot;]][[a]][[&quot;children&quot;]][[b]][[&quot;name&quot;]][[c]]</pre></td></tr><tr><td data-num="23"></td><td><pre>          </pre></td></tr><tr><td data-num="24"></td><td><pre>          pathway_id &lt;- str_match (pathway_info, &quot;ko [0-9]&#123;5&#125;&quot;)[1]</pre></td></tr><tr><td data-num="25"></td><td><pre>          pathway_name &lt;- str_replace (pathway_info, &quot;\\[PATH:ko [0-9]&#123;5&#125;\\]&quot;, &quot;&quot;) %&gt;% str_replace (&quot;[0-9]&#123;5&#125; &quot;,&quot;&quot;)</pre></td></tr><tr><td data-num="26"></td><td><pre>          pathway2name &lt;- rbind (pathway2name, tibble (Pathway &#x3D; pathway_id, Name &#x3D; pathway_name))</pre></td></tr><tr><td data-num="27"></td><td><pre>          </pre></td></tr><tr><td data-num="28"></td><td><pre>          kos_info &lt;- kegg [[&quot;children&quot;]][[&quot;children&quot;]][[a]][[&quot;children&quot;]][[b]][[&quot;children&quot;]][[c]][[&quot;name&quot;]]</pre></td></tr><tr><td data-num="29"></td><td><pre>          </pre></td></tr><tr><td data-num="30"></td><td><pre>          kos &lt;- str_match (kos_info, &quot;K [0-9]*&quot;)[,1]</pre></td></tr><tr><td data-num="31"></td><td><pre>          </pre></td></tr><tr><td data-num="32"></td><td><pre>          ko2pathway &lt;- rbind (ko2pathway, tibble (Ko &#x3D; kos, Pathway &#x3D; rep (pathway_id, length (kos))))</pre></td></tr><tr><td data-num="33"></td><td><pre>        &#125;</pre></td></tr><tr><td data-num="34"></td><td><pre>      &#125;</pre></td></tr><tr><td data-num="35"></td><td><pre>    &#125;</pre></td></tr><tr><td data-num="36"></td><td><pre>    colnames (ko2pathway) &lt;- c (&quot;KO&quot;,&quot;Pathway&quot;)</pre></td></tr><tr><td data-num="37"></td><td><pre>    save (pathway2name, ko2pathway, file &#x3D; &quot;kegg_info.RData&quot;)</pre></td></tr><tr><td data-num="38"></td><td><pre>    write.table (pathway2name,&quot;KEGG.library&quot;,sep&#x3D;&quot;\t&quot;,row.names &#x3D; F)</pre></td></tr><tr><td data-num="39"></td><td><pre>  &#125;</pre></td></tr><tr><td data-num="40"></td><td><pre>  </pre></td></tr><tr><td data-num="41"></td><td><pre>  </pre></td></tr><tr><td data-num="42"></td><td><pre>kegg (json &#x3D; &quot;ko00001.json&quot;)</pre></td></tr></table></figure><h2 id="44-背景注释构建"><a class="anchor" href="#44-背景注释构建">#</a> 4.4 背景注释构建</h2><p>主要是将自己的基因集与注释库结合</p><figure class="highlight raw"><figcaption data-lang></figcaption><table><tr><td data-num="1"></td><td><pre>library (clusterProfiler)</pre></td></tr><tr><td data-num="2"></td><td><pre>library (dplyr)</pre></td></tr><tr><td data-num="3"></td><td><pre>library (stringr)</pre></td></tr><tr><td data-num="4"></td><td><pre>options (stringsAsFactors &#x3D; F)</pre></td></tr><tr><td data-num="5"></td><td><pre>##&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;STEP1:GO 注释生成 &#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;</pre></td></tr><tr><td data-num="6"></td><td><pre># 自己构建的话，首先需要读入文件</pre></td></tr><tr><td data-num="7"></td><td><pre>egg &lt;- read.delim (&quot;node25.emapper.annotations&quot;,header &#x3D; T,sep&#x3D;&quot;\t&quot;)</pre></td></tr><tr><td data-num="8"></td><td><pre>egg [egg&#x3D;&#x3D;&quot;&quot;]&lt;-NA  #将空行变成 NA，方便下面的去除</pre></td></tr><tr><td data-num="9"></td><td><pre># 从文件中挑出基因 query 与 eggnog 注释信息</pre></td></tr><tr><td data-num="10"></td><td><pre>##gene_info &lt;- egg %&gt;% </pre></td></tr><tr><td data-num="11"></td><td><pre>##  dplyr::select (GID &#x3D; query, GENENAME &#x3D; eggNOG_OGs) %&gt;% na.omit ()</pre></td></tr><tr><td data-num="12"></td><td><pre># 挑出 query_name 与 GO 注释信息</pre></td></tr><tr><td data-num="13"></td><td><pre>gterms &lt;- egg %&gt;%</pre></td></tr><tr><td data-num="14"></td><td><pre>  dplyr::select (query, GOs) %&gt;% na.omit ()</pre></td></tr><tr><td data-num="15"></td><td><pre>gene_ids &lt;- egg$query</pre></td></tr><tr><td data-num="16"></td><td><pre>eggnog_lines_with_go &lt;- egg$GOs!&#x3D; &quot;&quot;</pre></td></tr><tr><td data-num="17"></td><td><pre>eggnog_lines_with_go</pre></td></tr><tr><td data-num="18"></td><td><pre>eggnog_annoations_go &lt;- str_split (egg [eggnog_lines_with_go,]$GOs, &quot;,&quot;)</pre></td></tr><tr><td data-num="19"></td><td><pre>gene2go &lt;- data.frame (gene &#x3D; rep (gene_ids [eggnog_lines_with_go],</pre></td></tr><tr><td data-num="20"></td><td><pre>                                    times &#x3D; sapply (eggnog_annoations_go, length)),</pre></td></tr><tr><td data-num="21"></td><td><pre>                         term &#x3D; unlist (eggnog_annoations_go))</pre></td></tr><tr><td data-num="22"></td><td><pre>names (gene2go) &lt;- c (&#39;gene_id&#39;, &#39;ID&#39;)</pre></td></tr><tr><td data-num="23"></td><td><pre>go2name &lt;- read.delim (&#39;GO.library&#39;, header &#x3D; FALSE, stringsAsFactors &#x3D; FALSE)</pre></td></tr><tr><td data-num="24"></td><td><pre>names (go2name) &lt;- c (&#39;ID&#39;, &#39;Description&#39;, &#39;Ontology&#39;)</pre></td></tr><tr><td data-num="25"></td><td><pre></pre></td></tr><tr><td data-num="26"></td><td><pre>go_anno &lt;- merge (gene2go, go2name, by &#x3D; &#39;ID&#39;, all.x &#x3D; TRUE)</pre></td></tr><tr><td data-num="27"></td><td><pre></pre></td></tr><tr><td data-num="28"></td><td><pre>## 将 GO 注释信息保存</pre></td></tr><tr><td data-num="29"></td><td><pre>save (go_anno,file &#x3D; &quot;node25_GO.rda&quot;)</pre></td></tr><tr><td data-num="30"></td><td><pre></pre></td></tr><tr><td data-num="31"></td><td><pre>##&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;STEP2:KEGG 注释生成 &#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;</pre></td></tr><tr><td data-num="32"></td><td><pre>gene2ko &lt;- egg %&gt;%</pre></td></tr><tr><td data-num="33"></td><td><pre>  dplyr::select (GID &#x3D; query, KO &#x3D; KEGG_ko) %&gt;%</pre></td></tr><tr><td data-num="34"></td><td><pre>  na.omit (）</pre></td></tr><tr><td data-num="35"></td><td><pre>pathway2name &lt;- read.delim (&quot;KEGG.library&quot;)</pre></td></tr><tr><td data-num="36"></td><td><pre>colnames (pathway2name)&lt;-c (&quot;Pathway&quot;,&quot;Name&quot;)</pre></td></tr><tr><td data-num="37"></td><td><pre>gene2ko$KO &lt;- str_replace (gene2ko$KO, &quot;ko:&quot;,&quot;&quot;)</pre></td></tr><tr><td data-num="38"></td><td><pre>gene2pathway &lt;- gene2ko %&gt;% left_join (ko2pathway, by &#x3D; &quot;KO&quot;) %&gt;% </pre></td></tr><tr><td data-num="39"></td><td><pre>  dplyr::select (GID, Pathway) %&gt;%</pre></td></tr><tr><td data-num="40"></td><td><pre>  na.omit ()</pre></td></tr><tr><td data-num="41"></td><td><pre>kegg_anno&lt;- merge (gene2pathway,pathway2name,by &#x3D; &#39;Pathway&#39;, all.x &#x3D; TRUE)[,c (2,1,3)]</pre></td></tr><tr><td data-num="42"></td><td><pre>colnames (kegg_anno) &lt;- c (&#39;gene_id&#39;,&#39;pathway_id&#39;,&#39;pathway_description&#39;)</pre></td></tr><tr><td data-num="43"></td><td><pre>save (kegg_anno,file &#x3D; &quot;node25_KEGG.rda&quot;)</pre></td></tr></table></figure><h2 id="45-gokeeg富集分析"><a class="anchor" href="#45-gokeeg富集分析">#</a> 4.5 GO/KEEG 富集分析</h2><p>利用生成的<span class="label primary"> node25_GO.rda</span>、<span class="label primary">node25_KEGG.rda</span>、以及之前的<span class="label primary"> node25significant.expand.genes</span> 文件进行富集分析<br>我在分析的时候没有设置 p 的阈值，输出了所有的富集结果，毕竟作图也只需要 TOP10 或者 TOP20，大家按照需求设定</p><figure class="highlight raw"><figcaption data-lang></figcaption><table><tr><td data-num="1"></td><td><pre>##&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;STEP3:GO 富集分析 &#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;</pre></td></tr><tr><td data-num="2"></td><td><pre># 目标基因列表 (全部基因)</pre></td></tr><tr><td data-num="3"></td><td><pre>gene_select &lt;- read.delim (file &#x3D; &#39;node25significant.expand.genes&#39;, stringsAsFactors &#x3D; FALSE,header &#x3D; F)$V1</pre></td></tr><tr><td data-num="4"></td><td><pre>#GO 富集分析</pre></td></tr><tr><td data-num="5"></td><td><pre># 默认以所有注释到 GO 的基因为背景集，也可通过 universe 参数输入背景集</pre></td></tr><tr><td data-num="6"></td><td><pre># 默认以 p&lt;0.05 为标准，Benjamini 方法校正 p 值，q 值阈值 0.2</pre></td></tr><tr><td data-num="7"></td><td><pre># 默认输出 top500 富集结果</pre></td></tr><tr><td data-num="8"></td><td><pre># 如果想输出所有富集结果（不考虑 p 值阈值等），将 p、q 等值设置为 1 即可</pre></td></tr><tr><td data-num="9"></td><td><pre># 或者直接在 enrichResult 类对象中直接提取需要的结果</pre></td></tr><tr><td data-num="10"></td><td><pre>go_rich &lt;- enricher (gene &#x3D; gene_select,</pre></td></tr><tr><td data-num="11"></td><td><pre>                    TERM2GENE &#x3D; go_anno [c (&#39;ID&#39;, &#39;gene_id&#39;)], </pre></td></tr><tr><td data-num="12"></td><td><pre>                    TERM2NAME &#x3D; go_anno [c (&#39;ID&#39;, &#39;Description&#39;)], </pre></td></tr><tr><td data-num="13"></td><td><pre>                    pvalueCutoff &#x3D; 1, </pre></td></tr><tr><td data-num="14"></td><td><pre>                    pAdjustMethod &#x3D; &#39;BH&#39;, </pre></td></tr><tr><td data-num="15"></td><td><pre>                    qvalueCutoff &#x3D; 1</pre></td></tr><tr><td data-num="16"></td><td><pre>                   )</pre></td></tr><tr><td data-num="17"></td><td><pre># 输出默认结果，即根据上述 p 值等阈值筛选后的</pre></td></tr><tr><td data-num="18"></td><td><pre></pre></td></tr><tr><td data-num="19"></td><td><pre>tmp &lt;- merge (go_rich, go2name [c (&#39;ID&#39;, &#39;Ontology&#39;)], by &#x3D; &#39;ID&#39;)</pre></td></tr><tr><td data-num="20"></td><td><pre>tmp &lt;- tmp [c (10, 1:9)]</pre></td></tr><tr><td data-num="21"></td><td><pre>tmp &lt;- tmp [order (tmp$pvalue), ]</pre></td></tr><tr><td data-num="22"></td><td><pre>write.table (tmp, &#39;node6expand.xls&#39;, sep &#x3D; &#39;\t&#39;, row.names &#x3D; FALSE, quote &#x3D; FALSE)</pre></td></tr><tr><td data-num="23"></td><td><pre></pre></td></tr><tr><td data-num="24"></td><td><pre>##&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;STEP4:KEGG 注释 &#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;&#x3D;</pre></td></tr><tr><td data-num="25"></td><td><pre>gene_select &lt;- read.delim (&#39;node25significant.expand.genes&#39;, stringsAsFactors &#x3D; FALSE,header &#x3D; F)$V1</pre></td></tr><tr><td data-num="26"></td><td><pre>#KEGG 富集分析</pre></td></tr><tr><td data-num="27"></td><td><pre># 默认以所有注释到 KEGG 的基因为背景集，也可通过 universe 参数指定其中的一个子集作为背景集</pre></td></tr><tr><td data-num="28"></td><td><pre># 默认以 p&lt;0.05 为标准，Benjamini 方法校正 p 值，q 值阈值 0.2</pre></td></tr><tr><td data-num="29"></td><td><pre># 默认输出 top500 富集结果</pre></td></tr><tr><td data-num="30"></td><td><pre>kegg_rich &lt;- enricher (gene &#x3D; gene_select,</pre></td></tr><tr><td data-num="31"></td><td><pre>                      TERM2GENE &#x3D; kegg_anno [c (&#39;pathway_id&#39;, &#39;gene_id&#39;)], </pre></td></tr><tr><td data-num="32"></td><td><pre>                      TERM2NAME &#x3D; kegg_anno [c (&#39;pathway_id&#39;, &#39;pathway_description&#39;)], </pre></td></tr><tr><td data-num="33"></td><td><pre>                      pvalueCutoff &#x3D; 1, </pre></td></tr><tr><td data-num="34"></td><td><pre>                      pAdjustMethod &#x3D; &#39;BH&#39;, </pre></td></tr><tr><td data-num="35"></td><td><pre>                      qvalueCutoff &#x3D; 1, </pre></td></tr><tr><td data-num="36"></td><td><pre>                      maxGSSize &#x3D; 500)</pre></td></tr><tr><td data-num="37"></td><td><pre></pre></td></tr><tr><td data-num="38"></td><td><pre># 输出默认结果，即根据上述 p 值等阈值筛选后的</pre></td></tr><tr><td data-num="39"></td><td><pre>write.table (kegg_rich, &#39;node25significant.expand.KEGG.xls&#39;, sep &#x3D; &#39;\t&#39;, row.names &#x3D; FALSE, quote &#x3D; FALSE)</pre></td></tr></table></figure><p><strong>需要注意生成的文件中有一些 term 可能是重复的，比如 GO 富集结果的 16,17 行，仔细看 geneID 那一列就会发现两行的基因是完全一样的，需要删掉一个。还有一点就是在 GO/KEGG 富集的结果中，有一些是动物相关的 (KEGG 结果的第一行)，这些也请酌情考虑</strong><br><img data-src="/img/posts/2022.12.15/3.png" alt="GO结果"><br><img data-src="/img/posts/2022.12.15/4.png" alt="KEGG结果"></p><h1 id="5-gokegg绘图"><a class="anchor" href="#5-gokegg绘图">#</a> 5 GO/KEGG 绘图</h1><p>此次使用 TOP10 的 term 进行绘图，看一下在陆生植物出现的关键进化节点主要是哪些基因功能发生了大规模扩张</p><figure class="highlight raw"><figcaption data-lang></figcaption><table><tr><td data-num="1"></td><td><pre>library (ggplot2)</pre></td></tr><tr><td data-num="2"></td><td><pre>pathway &#x3D; read.delim (&quot;node25significant.expand.GO.xls&quot;,header &#x3D; T,sep&#x3D;&quot;\t&quot;)</pre></td></tr><tr><td data-num="3"></td><td><pre>pathway$GeneRatio&lt;- pathway [,10]&#x2F;54700 ## 这个 54700 是 GeneRatio 的那个数值，自己修改</pre></td></tr><tr><td data-num="4"></td><td><pre></pre></td></tr><tr><td data-num="5"></td><td><pre>pathway$log&lt;- -log10 (pathway [,6])</pre></td></tr><tr><td data-num="6"></td><td><pre></pre></td></tr><tr><td data-num="7"></td><td><pre>library (dplyr)</pre></td></tr><tr><td data-num="8"></td><td><pre>library (ggrepel)</pre></td></tr><tr><td data-num="9"></td><td><pre>GO &lt;- arrange (pathway,pathway [,6])</pre></td></tr><tr><td data-num="10"></td><td><pre>GO_dataset &lt;- GO [1:10,]</pre></td></tr><tr><td data-num="11"></td><td><pre># 按照 PValue 从低到高排序 [升序]</pre></td></tr><tr><td data-num="12"></td><td><pre>GO_dataset$Description&lt;- factor (GO_dataset$Description,levels &#x3D; rev (GO_dataset$Description))</pre></td></tr><tr><td data-num="13"></td><td><pre>GO_dataset$GeneRatio &lt;- as.numeric (GO_dataset$GeneRatio)</pre></td></tr><tr><td data-num="14"></td><td><pre>GO_dataset$GeneRatio</pre></td></tr><tr><td data-num="15"></td><td><pre>GO_dataset$log</pre></td></tr><tr><td data-num="16"></td><td><pre># 图片背景设定</pre></td></tr><tr><td data-num="17"></td><td><pre>mytheme &lt;- theme (axis.title&#x3D;element_text (face&#x3D;&quot;bold&quot;, size&#x3D;8,colour &#x3D; &#39;black&#39;), #坐标轴标题</pre></td></tr><tr><td data-num="18"></td><td><pre>                 axis.text.y &#x3D; element_text (face&#x3D;&quot;bold&quot;, size&#x3D;6,colour &#x3D; &#39;black&#39;), #坐标轴标签</pre></td></tr><tr><td data-num="19"></td><td><pre>                 axis.text.x &#x3D; element_text (face &#x3D;&quot;bold&quot;,color&#x3D;&quot;black&quot;,angle&#x3D;0,vjust&#x3D;1,size&#x3D;8),</pre></td></tr><tr><td data-num="20"></td><td><pre>                 axis.line &#x3D; element_line (size&#x3D;0.5, colour &#x3D; &#39;black&#39;), #轴线</pre></td></tr><tr><td data-num="21"></td><td><pre>                 panel.background &#x3D; element_rect (color&#x3D;&#39;black&#39;), #绘图区边框</pre></td></tr><tr><td data-num="22"></td><td><pre>                 plot.title &#x3D; element_text (face&#x3D;&quot;bold&quot;, size&#x3D;8,colour &#x3D; &#39;black&#39;,hjust &#x3D; 0.8),</pre></td></tr><tr><td data-num="23"></td><td><pre>                 legend.key &#x3D; element_blank () #关闭图例边框</pre></td></tr><tr><td data-num="24"></td><td><pre>)</pre></td></tr><tr><td data-num="25"></td><td><pre></pre></td></tr><tr><td data-num="26"></td><td><pre># 绘制 GO 气泡图</pre></td></tr><tr><td data-num="27"></td><td><pre>p &lt;- ggplot (GO_dataset,aes (x&#x3D;GeneRatio,y&#x3D;Description,colour&#x3D;log,size&#x3D;Count,shape&#x3D;Ontology))+</pre></td></tr><tr><td data-num="28"></td><td><pre>  geom_point ()+</pre></td></tr><tr><td data-num="29"></td><td><pre>  scale_size (range&#x3D;c (2, 8))+</pre></td></tr><tr><td data-num="30"></td><td><pre>  scale_colour_gradient (low &#x3D; &quot;#52c2eb&quot;,high &#x3D; &quot;#EA4F30&quot;)+</pre></td></tr><tr><td data-num="31"></td><td><pre>  theme_bw ()+</pre></td></tr><tr><td data-num="32"></td><td><pre>  labs (x&#x3D;&#39;Fold Enrichment&#39;,y&#x3D;&#39;GO Terms&#39;, #自定义 x、y 轴、标题内容</pre></td></tr><tr><td data-num="33"></td><td><pre>       title &#x3D; &#39;Enriched GO Terms&#39;)+</pre></td></tr><tr><td data-num="34"></td><td><pre>  labs (color&#x3D;expression (-log [10](pvalue)))+</pre></td></tr><tr><td data-num="35"></td><td><pre>  theme (legend.title&#x3D;element_text (size&#x3D;8), legend.text &#x3D; element_text (size&#x3D;14))+</pre></td></tr><tr><td data-num="36"></td><td><pre>  theme (axis.title.y &#x3D; element_text (margin &#x3D; margin (r &#x3D; 50)),axis.title.x &#x3D; element_text (margin &#x3D; margin (t &#x3D; 20)))+</pre></td></tr><tr><td data-num="37"></td><td><pre>  theme (axis.text.x &#x3D; element_text (face &#x3D;&quot;bold&quot;,color&#x3D;&quot;black&quot;,angle&#x3D;0,vjust&#x3D;1))</pre></td></tr><tr><td data-num="38"></td><td><pre>plot &lt;- p+mytheme</pre></td></tr><tr><td data-num="39"></td><td><pre>plot</pre></td></tr><tr><td data-num="40"></td><td><pre># 保存图片</pre></td></tr><tr><td data-num="41"></td><td><pre>ggsave (plot,filename &#x3D; &quot;node25GO.pdf&quot;,width &#x3D; 210,height &#x3D; 210,units &#x3D; &quot;mm&quot;,dpi&#x3D;300)</pre></td></tr><tr><td data-num="42"></td><td><pre>ggsave (plot,filename &#x3D; &quot;node25GO.png&quot;,width &#x3D; 210,height &#x3D; 210,units &#x3D; &quot;mm&quot;,dpi&#x3D;300)</pre></td></tr></table></figure><p>KEGG 同理，大家修改运行参数和输入文件即可</p><figure class="highlight raw"><figcaption data-lang></figcaption><table><tr><td data-num="1"></td><td><pre>pathway &#x3D; read.delim (&quot;node25significant.expand.KEGG.xls&quot;,header &#x3D; T,sep&#x3D;&quot;\t&quot;)</pre></td></tr><tr><td data-num="2"></td><td><pre>pathway$GeneRatio&lt;- pathway [,9]&#x2F;31537</pre></td></tr><tr><td data-num="3"></td><td><pre></pre></td></tr><tr><td data-num="4"></td><td><pre>pathway$log&lt;- -log10 (pathway [,5])</pre></td></tr><tr><td data-num="5"></td><td><pre></pre></td></tr><tr><td data-num="6"></td><td><pre>library (dplyr)</pre></td></tr><tr><td data-num="7"></td><td><pre>library (ggrepel)</pre></td></tr><tr><td data-num="8"></td><td><pre>GO &lt;- arrange (pathway,pathway [,5])</pre></td></tr><tr><td data-num="9"></td><td><pre>GO_dataset &lt;- GO [1:10,]</pre></td></tr><tr><td data-num="10"></td><td><pre></pre></td></tr><tr><td data-num="11"></td><td><pre>#Pathway 列最好转化成因子型，否则作图时 ggplot2 会将所有 Pathway 按字母顺序重排序</pre></td></tr><tr><td data-num="12"></td><td><pre># 将 Pathway 列转化为因子型</pre></td></tr><tr><td data-num="13"></td><td><pre>GO_dataset$Description&lt;- factor (GO_dataset$Description,levels &#x3D; rev (GO_dataset$Description))</pre></td></tr><tr><td data-num="14"></td><td><pre>GO_dataset$GeneRatio &lt;- as.numeric (GO_dataset$GeneRatio)</pre></td></tr><tr><td data-num="15"></td><td><pre>GO_dataset&lt;- arrange (GO_dataset,GO_dataset [,5])</pre></td></tr><tr><td data-num="16"></td><td><pre>GO_dataset$GeneRatio</pre></td></tr><tr><td data-num="17"></td><td><pre>GO_dataset$log</pre></td></tr><tr><td data-num="18"></td><td><pre># 图片背景设定</pre></td></tr><tr><td data-num="19"></td><td><pre>mytheme &lt;- theme (axis.title&#x3D;element_text (face&#x3D;&quot;bold&quot;, size&#x3D;8,colour &#x3D; &#39;black&#39;), #坐标轴标题</pre></td></tr><tr><td data-num="20"></td><td><pre>                 axis.text.y &#x3D; element_text (face&#x3D;&quot;bold&quot;, size&#x3D;6,colour &#x3D; &#39;black&#39;), #坐标轴标签</pre></td></tr><tr><td data-num="21"></td><td><pre>                 axis.text.x &#x3D; element_text (face &#x3D;&quot;bold&quot;,color&#x3D;&quot;black&quot;,angle&#x3D;0,vjust&#x3D;1,size&#x3D;8),</pre></td></tr><tr><td data-num="22"></td><td><pre>                 axis.line &#x3D; element_line (size&#x3D;0.5, colour &#x3D; &#39;black&#39;), #轴线</pre></td></tr><tr><td data-num="23"></td><td><pre>                 panel.background &#x3D; element_rect (color&#x3D;&#39;black&#39;), #绘图区边框</pre></td></tr><tr><td data-num="24"></td><td><pre>                 plot.title &#x3D; element_text (face&#x3D;&quot;bold&quot;, size&#x3D;8,colour &#x3D; &#39;black&#39;,hjust &#x3D; 0.8),</pre></td></tr><tr><td data-num="25"></td><td><pre>                 legend.key &#x3D; element_blank () #关闭图例边框</pre></td></tr><tr><td data-num="26"></td><td><pre>)</pre></td></tr><tr><td data-num="27"></td><td><pre></pre></td></tr><tr><td data-num="28"></td><td><pre># 绘制 KEGG 气泡图</pre></td></tr><tr><td data-num="29"></td><td><pre>p &lt;- ggplot (GO_dataset,aes (x&#x3D;GeneRatio,y&#x3D;Description,colour&#x3D;log,size&#x3D;Count))+</pre></td></tr><tr><td data-num="30"></td><td><pre>  geom_point ()+</pre></td></tr><tr><td data-num="31"></td><td><pre>  scale_size (range&#x3D;c (2, 8))+</pre></td></tr><tr><td data-num="32"></td><td><pre>  scale_colour_gradient (low &#x3D; &quot;#52c2eb&quot;,high &#x3D; &quot;#EA4F30&quot;)+</pre></td></tr><tr><td data-num="33"></td><td><pre>  theme_bw ()+</pre></td></tr><tr><td data-num="34"></td><td><pre>  labs (x&#x3D;&#39;Fold Enrichment&#39;,y&#x3D;&#39;KEEG Terms&#39;, #自定义 x、y 轴、标题内容</pre></td></tr><tr><td data-num="35"></td><td><pre>       title &#x3D; &#39;Enriched KEEG Terms&#39;)+</pre></td></tr><tr><td data-num="36"></td><td><pre>  labs (color&#x3D;expression (-log [10](pvalue)))+</pre></td></tr><tr><td data-num="37"></td><td><pre>  theme (legend.title&#x3D;element_text (size&#x3D;8), legend.text &#x3D; element_text (size&#x3D;14))+</pre></td></tr><tr><td data-num="38"></td><td><pre>  theme (axis.title.y &#x3D; element_text (margin &#x3D; margin (r &#x3D; 50)),axis.title.x &#x3D; element_text (margin &#x3D; margin (t &#x3D; 20)))+</pre></td></tr><tr><td data-num="39"></td><td><pre>  theme (axis.text.x &#x3D; element_text (face &#x3D;&quot;bold&quot;,color&#x3D;&quot;black&quot;,angle&#x3D;0,vjust&#x3D;1))</pre></td></tr><tr><td data-num="40"></td><td><pre>plot &lt;- p+mytheme</pre></td></tr><tr><td data-num="41"></td><td><pre>plot</pre></td></tr><tr><td data-num="42"></td><td><pre># 保存图片</pre></td></tr><tr><td data-num="43"></td><td><pre>ggsave (plot,filename &#x3D; &quot;node25KEGG.pdf&quot;,width &#x3D; 210,height &#x3D; 210,units &#x3D; &quot;mm&quot;,dpi&#x3D;300)</pre></td></tr><tr><td data-num="44"></td><td><pre>ggsave (plot,filename &#x3D; &quot;node25KEGG.png&quot;,width &#x3D; 210,height &#x3D; 210,units &#x3D; &quot;mm&quot;,dpi&#x3D;300)</pre></td></tr></table></figure><p><strong>最后我们可以看到，在 GO 富集中，最显著的是水运输相关功能，KEGG 富集中，最显著的是昼夜节律，大家觉得陆生植物的进化和这两个功能有没有关系？欢迎讨论</strong><br><img data-src="/img/posts/2022.12.15/5.png" alt="GO注释"><br><img data-src="/img/posts/2022.12.15/6.png" alt="KEGG注释"></p><h1 id="6-结果整理"><a class="anchor" href="#6-结果整理">#</a> 6 结果整理</h1><p>结合这三篇推文的结果，我们可以整理一张主图<br><img data-src="/img/posts/2022.12.15/7.png" alt="结果"></p><h1 id="总结"><a class="anchor" href="#总结">#</a> 总结</h1><p>== 比较基因组学分析的基本内容已经介绍完了，如果说有遗漏的内容那应该就是加倍化事件，由于本次使用的数据集跨度过大所以没有增加 WGD 的分析，大家有什么问题欢迎交流 ==</p><div class="tags"><a href="/tags/bioinformatics/" rel="tag"><i class="ic i-tag"></i> bioinformatics</a> <a href="/tags/Comparative-Genomics/" rel="tag"><i class="ic i-tag"></i> Comparative_Genomics</a></div></div><footer><div class="meta"><span class="item"><span class="icon"><i class="ic i-calendar-check"></i> </span><span class="text">更新于</span> <time title="修改时间：2023-03-08 15:12:47" itemprop="dateModified" datetime="2023-03-08T15:12:47+08:00">2023-03-08</time> </span><span id="2022/12/15/2022-12-15-GO/" class="item leancloud_visitors" data-flag-title="比较基因组学分析 3：特异节点基因家族富集分析（非模式物种 GO/KEEG 库构建）" title="阅读次数"><span class="icon"><i class="ic i-eye"></i> </span><span class="text">阅读次数</span> <span class="leancloud-visitors-count"></span> <span class="text">次</span></span></div><div class="reward"><button><i class="ic i-heartbeat"></i> 赞赏</button><p>请我喝[茶]~(￣▽￣)~*</p><div id="qr"><div><img data-src="/images/wechatpay.png" alt="amane 微信支付"><p>微信支付</p></div><div><img data-src="/images/alipay.png" alt="amane 支付宝"><p>支付宝</p></div></div></div><div id="copyright"><ul><li class="author"><strong>本文作者： </strong>amane <i class="ic i-at"><em>@</em></i>小钊の笔记</li><li class="link"><strong>本文链接：</strong> <a href="https://www.zhouxiaozhao.cn/2022/12/15/2022-12-15-GO/" title="比较基因组学分析 3：特异节点基因家族富集分析（非模式物种 GO&#x2F;KEEG 库构建）">https://www.zhouxiaozhao.cn/2022/12/15/2022-12-15-GO/</a></li><li class="license"><strong>版权声明： </strong>本站所有文章除特别声明外，均采用 <span class="exturl" data-url="aHR0cHM6Ly9jcmVhdGl2ZWNvbW1vbnMub3JnL2xpY2Vuc2VzL2J5LW5jLXNhLzQuMC9kZWVkLnpo"><i class="ic i-creative-commons"><em>(CC)</em></i>BY-NC-SA</span> 许可协议。转载请注明出处！</li></ul></div></footer></article></div><div class="post-nav"><div class="item left"><a href="/2022/12/14/2022-12-14-CAFE/" itemprop="url" rel="prev" data-background-image="https:&#x2F;&#x2F;img1.imgtp.com&#x2F;2023&#x2F;02&#x2F;01&#x2F;GgJdQCED.jpg" title="比较基因组学分析2：基因家族收缩与扩张分析"><span class="type">上一篇</span> <span class="category"><i class="ic i-flag"></i> 基因组</span><h3>比较基因组学分析2：基因家族收缩与扩张分析</h3></a></div><div class="item right"><a href="/2023/02/22/2023-02-22-ST(1)/" itemprop="url" rel="next" data-background-image="https:&#x2F;&#x2F;img1.imgtp.com&#x2F;2023&#x2F;02&#x2F;01&#x2F;82Elo9NQ.jpg" title="植物空间转录组分析1：Seurat基本流程"><span class="type">下一篇</span> <span class="category"><i class="ic i-flag"></i> 空间转录组</span><h3>植物空间转录组分析1：Seurat基本流程</h3></a></div></div><div class="wrap" id="comments"></div></div><div id="sidebar"><div class="inner"><div class="panels"><div class="inner"><div class="contents panel pjax" data-title="文章目录"><ol class="toc"><li class="toc-item toc-level-1"><a class="toc-link" href="#1%E5%89%8D%E8%A8%80"><span class="toc-number">1.</span> <span class="toc-text">1. 前言</span></a></li><li class="toc-item toc-level-1"><a class="toc-link" href="#2%E8%8A%82%E7%82%B9%E9%80%89%E6%8B%A9"><span class="toc-number">2.</span> <span class="toc-text">2. 节点选择</span></a></li><li class="toc-item toc-level-1"><a class="toc-link" href="#3%E8%8E%B7%E5%8F%96%E7%9B%AE%E7%9A%84%E5%9F%BA%E5%9B%A0"><span class="toc-number">3.</span> <span class="toc-text">3. 获取目的基因</span></a><ol class="toc-child"><li class="toc-item toc-level-2"><a class="toc-link" href="#31-%E8%8E%B7%E5%8F%96node25%E6%98%BE%E8%91%97%E6%89%A9%E5%BC%A0%E7%9A%84%E5%9F%BA%E5%9B%A0"><span class="toc-number">3.1.</span> <span class="toc-text">3.1 获取 node25 显著扩张的基因</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#32-node25%E8%83%8C%E6%99%AF%E5%9F%BA%E5%9B%A0%E9%80%89%E6%8B%A9"><span class="toc-number">3.2.</span> <span class="toc-text">3.2 node25 背景基因选择</span></a></li></ol></li><li class="toc-item toc-level-1"><a class="toc-link" href="#4-%E8%83%8C%E6%99%AF%E5%9F%BA%E5%9B%A0%E7%9A%84go%E5%92%8Ckegg%E6%B3%A8%E9%87%8A"><span class="toc-number">4.</span> <span class="toc-text">4 背景基因的 GO 和 KEGG 注释</span></a><ol class="toc-child"><li class="toc-item toc-level-2"><a class="toc-link" href="#41-gokegg%E6%B3%A8%E9%87%8A"><span class="toc-number">4.1.</span> <span class="toc-text">4.1 GO&#x2F;KEGG 注释</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#42-go%E6%B3%A8%E9%87%8A%E5%BA%93%E6%9E%84%E5%BB%BA"><span class="toc-number">4.2.</span> <span class="toc-text">4.2 GO 注释库构建</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#43-kegg%E6%B3%A8%E9%87%8A%E5%BA%93%E6%9E%84%E5%BB%BA"><span class="toc-number">4.3.</span> <span class="toc-text">4.3 KEGG 注释库构建</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#44-%E8%83%8C%E6%99%AF%E6%B3%A8%E9%87%8A%E6%9E%84%E5%BB%BA"><span class="toc-number">4.4.</span> <span class="toc-text">4.4 背景注释构建</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#45-gokeeg%E5%AF%8C%E9%9B%86%E5%88%86%E6%9E%90"><span class="toc-number">4.5.</span> <span class="toc-text">4.5 GO&#x2F;KEEG 富集分析</span></a></li></ol></li><li class="toc-item toc-level-1"><a class="toc-link" href="#5-gokegg%E7%BB%98%E5%9B%BE"><span class="toc-number">5.</span> <span class="toc-text">5 GO&#x2F;KEGG 绘图</span></a></li><li class="toc-item toc-level-1"><a class="toc-link" href="#6-%E7%BB%93%E6%9E%9C%E6%95%B4%E7%90%86"><span class="toc-number">6.</span> <span class="toc-text">6 结果整理</span></a></li><li class="toc-item toc-level-1"><a class="toc-link" href="#%E6%80%BB%E7%BB%93"><span class="toc-number">7.</span> <span class="toc-text">总结</span></a></li></ol></div><div class="related panel pjax" data-title="系列文章"><ul><li><a href="/2020/07/08/2020-07-08-repeat/" rel="bookmark" title="RepeatModeler+RepeatMasker的安装与使用">RepeatModeler+RepeatMasker的安装与使用</a></li><li><a href="/2020/07/18/2020-07-18-orthofinder/" rel="bookmark" title="利用orthofinder寻找单拷贝基因构建系统发育树">利用orthofinder寻找单拷贝基因构建系统发育树</a></li><li><a href="/2022/12/08/2022-12-08-SingleCopy/" rel="bookmark" title="比较基因组学分析1：构建单拷贝基因树">比较基因组学分析1：构建单拷贝基因树</a></li><li><a href="/2022/12/14/2022-12-14-CAFE/" rel="bookmark" title="比较基因组学分析2：基因家族收缩与扩张分析">比较基因组学分析2：基因家族收缩与扩张分析</a></li><li class="active"><a href="/2022/12/15/2022-12-15-GO/" rel="bookmark" title="比较基因组学分析3：特异节点基因家族富集分析（非模式物种GO/KEEG库构建）">比较基因组学分析3：特异节点基因家族富集分析（非模式物种GO/KEEG库构建）</a></li></ul></div><div class="overview panel" data-title="站点概览"><div class="author" itemprop="author" itemscope itemtype="http://schema.org/Person"><img class="image" itemprop="image" alt="amane" data-src="/images/avatar.png"><p class="name" itemprop="name">amane</p><div class="description" itemprop="description">生物学在读博士</div></div><nav class="state"><div class="item posts"><a href="/archives/"><span class="count">84</span> <span class="name">文章</span></a></div><div class="item categories"><a href="/categories/"><span class="count">8</span> <span class="name">分类</span></a></div><div class="item tags"><a href="/tags/"><span class="count">21</span> <span class="name">标签</span></a></div></nav><div class="social"><span class="exturl item github" data-url="aHR0cHM6Ly9naXRodWIuY29tL0Jpb2luZm9ybWF0aWNzLXJvb2tpZQ==" title="https:&#x2F;&#x2F;github.com&#x2F;Bioinformatics-rookie"><i class="ic i-github"></i></span> <span class="exturl item bilibili" data-url="aHR0cHM6Ly9zcGFjZS5iaWxpYmlsaS5jb20vMTU3Nzg1NTM/c3BtX2lkX2Zyb209MzMzLjEwMDcuMC4w" title="https:&#x2F;&#x2F;space.bilibili.com&#x2F;15778553?spm_id_from&#x3D;333.1007.0.0"><i class="ic i-bilibili"></i></span> <a href="https://www.zhouxiaozhao.cn/about/" title="https:&#x2F;&#x2F;www.zhouxiaozhao.cn&#x2F;about&#x2F;" class="item about"><i class="ic i-address-card"></i></a> <span class="exturl item email" data-url="bWFpbHRvOnpob3VzaGl6aGFvQG1haWwua2liLmFjLmNu" title="mailto:zhoushizhao@mail.kib.ac.cn"><i class="ic i-envelope"></i></span></div><ul class="menu"><li class="item"><a href="/" rel="section"><i class="ic i-home"></i>首页</a></li><li class="item"><a href="/about/" rel="section"><i class="ic i-user"></i>关于</a></li><li class="item dropdown"><a href="javascript:void(0);"><i class="ic i-feather"></i>文章</a><ul class="submenu"><li class="item"><a href="/archives/" rel="section"><i class="ic i-list-alt"></i>归档</a></li><li class="item"><a href="/categories/" rel="section"><i class="ic i-th"></i>分类</a></li><li class="item"><a href="/tags/" rel="section"><i class="ic i-tags"></i>标签</a></li></ul></li><li class="item dropdown"><a href="/webstack/" rel="section"><i class="ic i-magic"></i>AI学习</a><ul class="submenu"><li class="item"><a href="/AIpainting/" rel="section"><i class="ic i-pen"></i>AI绘画</a></li><li class="item"><a href="/AIbookmark/" rel="section"><i class="ic i-comments"></i>AI书签</a></li></ul></li><li class="item"><a href="/links/" rel="section"><i class="ic i-heart"></i>友達</a></li></ul></div></div></div><ul id="quick"><li class="prev pjax"><a href="/2022/12/14/2022-12-14-CAFE/" rel="prev" title="上一篇"><i class="ic i-chevron-left"></i></a></li><li class="up"><i class="ic i-arrow-up"></i></li><li class="down"><i class="ic i-arrow-down"></i></li><li class="next pjax"><a href="/2023/02/22/2023-02-22-ST(1)/" rel="next" title="下一篇"><i class="ic i-chevron-right"></i></a></li><li class="percent"></li></ul></div></div><div class="dimmer"></div></div></main><footer id="footer"><div class="inner"><div class="widgets"><div class="rpost pjax"><h2>随机文章</h2><ul><li class="item"><div class="breadcrumb"><a href="/categories/Bioinformation/" title="分类于 生信学习">生信学习</a></div><span><a href="/2020/12/26/2020-12-26-RNAseq(14)/" title="RNAseq教程(3.3)">RNAseq教程(3.3)</a></span></li><li class="item"><div class="breadcrumb"><a href="/categories/Bioinformation/" title="分类于 生信学习">生信学习</a></div><span><a href="/2020/08/03/2020-08-03-WGCNA1/" title="WGCNA学习：WGCNA分析原理">WGCNA学习：WGCNA分析原理</a></span></li><li class="item"><div class="breadcrumb"><a href="/categories/Bioinformation/" title="分类于 生信学习">生信学习</a> <i class="ic i-angle-right"></i> <a href="/categories/Bioinformation/genome/" title="分类于 基因组">基因组</a></div><span><a href="/2022/12/14/2022-12-14-CAFE/" title="比较基因组学分析2：基因家族收缩与扩张分析">比较基因组学分析2：基因家族收缩与扩张分析</a></span></li><li class="item"><div class="breadcrumb"><a href="/categories/Bioinformation/" title="分类于 生信学习">生信学习</a></div><span><a href="/2020/08/05/2020-08-05-WGCNA2/" title="WGCNA学习：WGCNA应用场景与实例解读">WGCNA学习：WGCNA应用场景与实例解读</a></span></li><li class="item"><div class="breadcrumb"><a href="/categories/Python/" title="分类于 Python">Python</a></div><span><a href="/2020/09/12/2020-09-12-python2/" title="马士兵教育Python入门基础:1.python入门（二）">马士兵教育Python入门基础:1.python入门（二）</a></span></li><li class="item"><div class="breadcrumb"><a href="/categories/Bioinformation/" title="分类于 生信学习">生信学习</a></div><span><a href="/2020/08/30/2020-08-30-evolview3/" title="进化树+蛋白结构域">进化树+蛋白结构域</a></span></li><li class="item"><div class="breadcrumb"><a href="/categories/Bioinformation/" title="分类于 生信学习">生信学习</a></div><span><a href="/2020/08/18/2020-08-18-sys/" title="根据转录组数据鉴定共生物种">根据转录组数据鉴定共生物种</a></span></li><li class="item"><div class="breadcrumb"><a href="/categories/Python/" title="分类于 Python">Python</a></div><span><a href="/2020/10/15/2020-10-15-python6/" title="马士兵教育Python入门基础:3.运算（2）">马士兵教育Python入门基础:3.运算（2）</a></span></li><li class="item"><div class="breadcrumb"><a href="/categories/Bioinformation/" title="分类于 生信学习">生信学习</a> <i class="ic i-angle-right"></i> <a href="/categories/Bioinformation/genome/" title="分类于 基因组">基因组</a></div><span><a href="/2022/12/08/2022-12-08-SingleCopy/" title="比较基因组学分析1：构建单拷贝基因树">比较基因组学分析1：构建单拷贝基因树</a></span></li><li class="item"><div class="breadcrumb"><a href="/categories/Bioinformation/" title="分类于 生信学习">生信学习</a></div><span><a href="/2020/11/26/2020-11-26-annotion(4)/" title="基因结构注释（4）：整合预测结果">基因结构注释（4）：整合预测结果</a></span></li></ul></div><div><h2>最新评论</h2><ul class="leancloud-recent-comment"></ul></div></div><div class="status"><div class="copyright">&copy; 2010 – <span itemprop="copyrightYear">2023</span> <span class="with-love"><i class="ic i-sakura rotate"></i> </span><span class="author" itemprop="copyrightHolder">amane @ Zhou Xiaozhao</span></div><div class="count"><span class="post-meta-item-icon"><i class="ic i-chart-area"></i> </span><span title="站点总字数">381k 字</span> <span class="post-meta-divider">|</span> <span class="post-meta-item-icon"><i class="ic i-coffee"></i> </span><span title="站点阅读时长">5:46</span></div><div class="powered-by">基于 <span class="exturl" data-url="aHR0cHM6Ly9oZXhvLmlv">Hexo</span> & Theme.<span class="exturl" data-url="aHR0cHM6Ly9naXRodWIuY29tL2FtZWhpbWUvaGV4by10aGVtZS1zaG9rYQ==">Shoka</span></div></div></div></footer></div><script data-config type="text/javascript">var LOCAL={path:"2022/12/15/2022-12-15-GO/",favicon:{show:"（●´3｀●）やれやれだぜ",hide:"(´Д｀)大変だ！"},search:{placeholder:"文章搜索",empty:"关于 「 ${query} 」，什么也没搜到",stats:"${time} ms 内找到 ${hits} 条结果"},valine:!0,fancybox:!0,copyright:'复制成功，转载请遵守 <i class="ic i-creative-commons"></i>BY-NC-SA 协议。',ignores:[function(e){return e.includes("#")},function(e){return new RegExp(LOCAL.path+"$").test(e)}]}</script><script src="https://cdn.polyfill.io/v2/polyfill.js"></script><script src="//cdn.jsdelivr.net/combine/npm/pace-js@1.0.2/pace.min.js,npm/pjax@0.2.8/pjax.min.js,npm/whatwg-fetch@3.4.0/dist/fetch.umd.min.js,npm/animejs@3.2.0/lib/anime.min.js,npm/algoliasearch@4/dist/algoliasearch-lite.umd.js,npm/instantsearch.js@4/dist/instantsearch.production.min.js,npm/lozad@1/dist/lozad.min.js,npm/quicklink@2/dist/quicklink.umd.js"></script><script src="/js/app.js?v=0.2.5"></script></body></html><!-- rebuild by hrmmi -->