<!-- build time:Wed Mar 08 2023 17:10:28 GMT+0800 (中国标准时间) --><!DOCTYPE html><html lang="zh-CN"><head><meta charset="UTF-8"><meta name="viewport" content="width=device-width,initial-scale=1,maximum-scale=2"><meta name="theme-color" content="#FFF"><link rel="apple-touch-icon" sizes="180x180" href="/images/apple-touch-icon.png"><link rel="icon" type="image/ico" sizes="32x32" href="/images/favicon.ico"><meta http-equiv="Cache-Control" content="no-transform"><meta http-equiv="Cache-Control" content="no-siteapp"><link rel="alternate" type="application/rss+xml" title="小钊の笔记" href="https://www.zhouxiaozhao.cn/rss.xml"><link rel="alternate" type="application/atom+xml" title="小钊の笔记" href="https://www.zhouxiaozhao.cn/atom.xml"><link rel="alternate" type="application/json" title="小钊の笔记" href="https://www.zhouxiaozhao.cn/feed.json"><link rel="stylesheet" href="//fonts.googleapis.com/css?family=Mulish:300,300italic,400,400italic,700,700italic%7CFredericka%20the%20Great:300,300italic,400,400italic,700,700italic%7CNoto%20Serif%20JP:300,300italic,400,400italic,700,700italic%7CNoto%20Serif%20SC:300,300italic,400,400italic,700,700italic%7CInconsolata:300,300italic,400,400italic,700,700italic&display=swap&subset=latin,latin-ext"><link rel="stylesheet" href="/css/app.css?v=0.2.5"><meta name="keywords" content="bioinformatics,transcriptome"><link rel="canonical" href="https://www.zhouxiaozhao.cn/2020/08/18/2020-08-18-sys/"><title>根据转录组数据鉴定共生物种 - 生信学习 | Zhou Xiaozhao = 小钊の笔记 = 前天是小兔子，昨天是小鹿，今天是你</title><meta name="generator" content="Hexo 5.4.1"></head><body itemscope itemtype="http://schema.org/WebPage"><div id="loading"><div class="cat"><div class="body"></div><div class="head"><div class="face"></div></div><div class="foot"><div class="tummy-end"></div><div class="bottom"></div><div class="legs left"></div><div class="legs right"></div></div><div class="paw"><div class="hands left"></div><div class="hands right"></div></div></div></div><div id="container"><header id="header" itemscope itemtype="http://schema.org/WPHeader"><div class="inner"><div id="brand"><div class="pjax"><h1 itemprop="name headline">根据转录组数据鉴定共生物种</h1><div class="meta"><span class="item" title="创建时间：2020-08-18 18:38:45"><span class="icon"><i class="ic i-calendar"></i> </span><span class="text">发表于</span> <time itemprop="dateCreated datePublished" datetime="2020-08-18T18:38:45+08:00">2020-08-18</time> </span><span class="item" title="本文字数"><span class="icon"><i class="ic i-pen"></i> </span><span class="text">本文字数</span> <span>5.9k</span> <span class="text">字</span> </span><span class="item" title="阅读时长"><span class="icon"><i class="ic i-clock"></i> </span><span class="text">阅读时长</span> <span>5 分钟</span></span></div></div></div><nav id="nav"><div class="inner"><div class="toggle"><div class="lines" aria-label="切换导航栏"><span class="line"></span> <span class="line"></span> <span class="line"></span></div></div><ul class="menu"><li class="item title"><a href="/" rel="start">Zhou Xiaozhao</a></li></ul><ul class="right"><li class="item theme"><i class="ic i-sun"></i></li><li class="item search"><i class="ic i-search"></i></li></ul></div></nav></div><div id="imgs" class="pjax"><ul><li class="item" data-background-image="https://img1.imgtp.com/2023/02/01/LoNchvnE.jpg"></li><li class="item" data-background-image="https://img1.imgtp.com/2023/02/01/mlFpeQkC.jpg"></li><li class="item" data-background-image="https://img1.imgtp.com/2023/02/01/iFdcQwdu.jpg"></li><li class="item" data-background-image="https://img1.imgtp.com/2023/02/01/a4l1BNWD.jpg"></li><li class="item" data-background-image="https://img1.imgtp.com/2023/02/01/3G4nGWGt.jpg"></li><li class="item" data-background-image="https://img1.imgtp.com/2023/02/01/MRMqrWXd.jpg"></li></ul></div></header><div id="waves"><svg class="waves" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" viewBox="0 24 150 28" preserveAspectRatio="none" shape-rendering="auto"><defs><path id="gentle-wave" d="M-160 44c30 0 58-18 88-18s 58 18 88 18 58-18 88-18 58 18 88 18 v44h-352z"/></defs><g class="parallax"><use xlink:href="#gentle-wave" x="48" y="0"/><use xlink:href="#gentle-wave" x="48" y="3"/><use xlink:href="#gentle-wave" x="48" y="5"/><use xlink:href="#gentle-wave" x="48" y="7"/></g></svg></div><main><div class="inner"><div id="main" class="pjax"><div class="article wrap"><div class="breadcrumb" itemscope itemtype="https://schema.org/BreadcrumbList"><i class="ic i-home"></i> <span><a href="/">首页</a></span><i class="ic i-angle-right"></i> <span class="current" itemprop="itemListElement" itemscope itemtype="https://schema.org/ListItem"><a href="/categories/Bioinformation/" itemprop="item" rel="index" title="分类于 生信学习"><span itemprop="name">生信学习</span></a><meta itemprop="position" content="1"></span></div><article itemscope itemtype="http://schema.org/Article" class="post block" lang="zh-CN"><link itemprop="mainEntityOfPage" href="https://www.zhouxiaozhao.cn/2020/08/18/2020-08-18-sys/"><span hidden itemprop="author" itemscope itemtype="http://schema.org/Person"><meta itemprop="image" content="/images/avatar.jpg"><meta itemprop="name" content="amane"><meta itemprop="description" content="前天是小兔子，昨天是小鹿，今天是你, 生物学在读博士"></span><span hidden itemprop="publisher" itemscope itemtype="http://schema.org/Organization"><meta itemprop="name" content="小钊の笔记"></span><div class="body md" itemprop="articleBody"><p>最近做的项目是关于地衣的转录组和基因组，拿到的结果是真菌部位的基因组以及不同时期和部位的转录组，地下部位的藻类部由于太复杂而且杂质过多不好分离就没有测基因组，目前想知道地下部位的藻类到底是什么？根据之前的观察，推测是<em> Coccomyxa subellipsoidea</em>，想要具体确定的话还需要进一步证明。</p><p><img data-src="/img/posts/2020.8.18/image-20200826100329224.png" alt></p><p>可以看到该地衣的地上部位是一种担子菌，地下部位不仅有藻类还有其他的物种。</p><h2 id="转录组比对到真菌参考基因组"><a class="anchor" href="#转录组比对到真菌参考基因组">#</a> 转录组比对到真菌参考基因组</h2><p>目前有真菌的测序基因组，第一步就是想把转录组数据比对到真菌的参考基因组上</p><p>比对工具我选择 STAR，使用地上部位（真菌部_Lh3-1）和地下部位（藻类部 _Lh1-1）的转录组比对到基因组上</p><p><strong>建立索引</strong></p><figure class="highlight raw"><figcaption data-lang></figcaption><table><tr><td data-num="1"></td><td><pre>S$ATR --runMode genomeGenerate --genomeDir &#x2F;datadisk02&#x2F;mpoly&#x2F;hun_db&#x2F; --genomeFastaFiles &#x2F;datadisk02&#x2F;mpoly&#x2F;hudsoiana.fa</pre></td></tr></table></figure><p><strong>比对</strong></p><figure class="highlight raw"><figcaption data-lang></figcaption><table><tr><td data-num="1"></td><td><pre>$STAR --runThreadN 20 --genomeDir &#x2F;datadisk02&#x2F;mpoly&#x2F;hun_db&#x2F; --readFilesCommand zcat --readFilesIn _Lh2-1_S110_L007_R1_001.fastq.gz _Lh2-1_S110_L007_R2_001.fastq.gz --outSAMtype BAM SortedByCoordinate --outBAMsortingThreadN 5 --outFileNamePrefix S2-1</pre></td></tr><tr><td data-num="2"></td><td><pre></pre></td></tr><tr><td data-num="3"></td><td><pre>$STAR --runThreadN 20 --genomeDir &#x2F;datadisk02&#x2F;mpoly&#x2F;hun_db&#x2F; --readFilesCommand zcat --readFilesIn _Lh3-1_S108_L007_R1_001.fastq.gz _Lh3-1_S108_L007_R2_001.fastq.gz --outSAMtype BAM SortedByCoordinate --outBAMsortingThreadN 5 --outFileNamePrefix S1-1</pre></td></tr></table></figure><p>比对结束后我最关心的是比对率，按照预期，地下部位的比对率比地上部位低。</p><pre><code>$less S1-1Log.final.out
                                 Started job on |       Aug 20 05:48:00
                             Started mapping on |       Aug 20 05:48:01
                                    Finished on |       Aug 20 06:42:35
       Mapping speed, Million of reads per hour |       26.61

                          Number of input reads |       24198466
                      Average input read length |       300
                                    UNIQUE READS:
                   Uniquely mapped reads number |       15397710
                        Uniquely mapped reads % |       60.89%

$less S3-1Log.final.out
                                 Started job on |       Aug 20 08:38:29
                             Started mapping on |       Aug 20 08:38:30
                                    Finished on |       Aug 20 08:49:15
       Mapping speed, Million of reads per hour |       134.06

                          Number of input reads |       24018884
                      Average input read length |       300
                                    UNIQUE READS:
                   Uniquely mapped reads number |       19151198
                        Uniquely mapped reads % |       79.73%

</code></pre><p>可以看到结果符合预期，地下部位的 reads 比对率为 64%，地上部位为 80%。</p><h2 id="地下部位转录组分析"><a class="anchor" href="#地下部位转录组分析">#</a> 地下部位转录组分析</h2><p>接下来用地下部位的转录组分析确定共生藻，首先的想法是利用转录组数据比对到初步确定的共生藻<em> Coccomyxa subellipsoidea</em> 上，使用<em> Coccomyxa subellipsoidea</em> 的基因组建库进行比对</p><figure class="highlight raw"><figcaption data-lang></figcaption><table><tr><td data-num="1"></td><td><pre>S$ATR --runMode genomeGenerate --genomeDir &#x2F;datadisk02&#x2F;mpoly&#x2F;cocco&#x2F; --genomeFastaFiles &#x2F;datadisk02&#x2F;mpoly&#x2F;Coccomyxa.fa</pre></td></tr><tr><td data-num="2"></td><td><pre></pre></td></tr><tr><td data-num="3"></td><td><pre>$STAR --runThreadN 20 --genomeDir &#x2F;datadisk02&#x2F;mpoly&#x2F;cocco&#x2F; --readFilesCommand zcat --readFilesIn _Lh1-1_S110_L007_R1_001.fastq.gz _Lh1-1_S110_L007_R2_001.fastq.gz --outSAMtype BAM SortedByCoordinate --outBAMsortingThreadN 5 --outFileNamePrefix S1-1C</pre></td></tr></table></figure><p>结果有点出乎意料，只有零点几的比对率，</p><figure class="highlight raw"><figcaption data-lang></figcaption><table><tr><td data-num="1"></td><td><pre>$less S1-1CLog.final.out</pre></td></tr><tr><td data-num="2"></td><td><pre>                                 Started job on |       Aug 19 14:25:11</pre></td></tr><tr><td data-num="3"></td><td><pre>                             Started mapping on |       Aug 19 14:25:16</pre></td></tr><tr><td data-num="4"></td><td><pre>                                    Finished on |       Aug 19 20:51:41</pre></td></tr><tr><td data-num="5"></td><td><pre>       Mapping speed, Million of reads per hour |       3.96</pre></td></tr><tr><td data-num="6"></td><td><pre></pre></td></tr><tr><td data-num="7"></td><td><pre>                          Number of input reads |       25517519</pre></td></tr><tr><td data-num="8"></td><td><pre>                      Average input read length |       300</pre></td></tr><tr><td data-num="9"></td><td><pre>                                    UNIQUE READS:</pre></td></tr><tr><td data-num="10"></td><td><pre>                   Uniquely mapped reads number |       92457</pre></td></tr><tr><td data-num="11"></td><td><pre>                        Uniquely mapped reads % |       0.36%</pre></td></tr></table></figure><p>由于地下部位比对到真菌部也有 64% 的比对率，所以我重新跑了一遍 STAR，利用 unmapped reads 进行比对。</p><figure class="highlight raw"><figcaption data-lang></figcaption><table><tr><td data-num="1"></td><td><pre>$STAR --runThreadN 32 --genomeDir hun_db&#x2F; --readFilesIn _Lh1-1_S110_L007_R1_001.fastq.gz _Lh1-1_S110_L007_R2_001.fastq.gz --readFilesCommand zcat --outSAMtype SAM --outFileNamePrefix S1-1 --outReadsUnmapped Fastx</pre></td></tr><tr><td data-num="2"></td><td><pre></pre></td></tr><tr><td data-num="3"></td><td><pre>$awk -F &quot; &quot; &#39;&#123;print $1&#125;&#39; S1-1Unmapped.out.mate1| awk &#39;&#123;if (NR%4&#x3D;&#x3D;1||NR%4&#x3D;3) print&#125;&#39; &gt; S1-1.left.fa</pre></td></tr><tr><td data-num="4"></td><td><pre>$sed -i &quot;s&#x2F;@&#x2F;&gt;&#x2F;g&quot; S1-1.left.fa</pre></td></tr><tr><td data-num="5"></td><td><pre>$awk -F &quot; &quot; &#39;&#123;print $1&#125;&#39; S1-1Unmapped.out.mate2| awk &#39;&#123;if (NR%4&#x3D;&#x3D;1||NR%4&#x3D;3) print&#125;&#39; &gt; S1-1.right.fa</pre></td></tr><tr><td data-num="6"></td><td><pre>$sed -i &quot;s&#x2F;@&#x2F;&gt;&#x2F;g&quot; S1-1.right.fa</pre></td></tr><tr><td data-num="7"></td><td><pre>$STAR --runThreadN 20 --genomeDir cocco&#x2F; --readFilesIn S1-1.left.fa S1-1.right.fa --outSAMtype SAM --outFileNamePrefix S1-1UC --outReadsUnmapped Fastx</pre></td></tr><tr><td data-num="8"></td><td><pre></pre></td></tr><tr><td data-num="9"></td><td><pre>$less S1-1UCLog.final.out</pre></td></tr><tr><td data-num="10"></td><td><pre></pre></td></tr><tr><td data-num="11"></td><td><pre>                                 Started job on |       Aug 26 10:59:26</pre></td></tr><tr><td data-num="12"></td><td><pre>                             Started mapping on |       Aug 26 10:59:39</pre></td></tr><tr><td data-num="13"></td><td><pre>                                    Finished on |       Aug 26 15:32:14</pre></td></tr><tr><td data-num="14"></td><td><pre>       Mapping speed, Million of reads per hour |       1.58</pre></td></tr><tr><td data-num="15"></td><td><pre></pre></td></tr><tr><td data-num="16"></td><td><pre>                          Number of input reads |       7180320</pre></td></tr><tr><td data-num="17"></td><td><pre>                      Average input read length |       300</pre></td></tr><tr><td data-num="18"></td><td><pre>                                    UNIQUE READS:</pre></td></tr><tr><td data-num="19"></td><td><pre>                   Uniquely mapped reads number |       96710</pre></td></tr><tr><td data-num="20"></td><td><pre>                        Uniquely mapped reads % |       1.35%</pre></td></tr></table></figure><p>比对率也只有可怜的 1.35%，一开始我怀疑是不是藻类物种鉴定错了，后来我用了 ncbi 中所有藻类的基因组，比对率都不到 1%。</p><p>后来去网上搜索比对率低的原因，在<span class="exturl" data-url="aHR0cHM6Ly9naXRodWIuY29tL2FsZXhkb2Jpbi9TVEFSL2lzc3Vlcy8xNjk="> github</span> 上发现相应解答，加入 --outFilterScoreMinOverLread 0.3 和 --outFilterMatchNminOverLread 0.3 参数</p><figure class="highlight raw"><figcaption data-lang></figcaption><table><tr><td data-num="1"></td><td><pre>$STAR --runThreadN 20 --genomeDir hun_db&#x2F; --readFilesCommand zcat --readFilesIn _Lh1-1_S110_L007_R1_001.fastq.gz _Lh1-1_S110_L007_R2_001.fastq.gz --outFileNamePrefix S1-1fli --outReadsUnmapped Fastx --outSAMtype SAM --outFilterScoreMinOverLread 0.3 --outFilterMatchNminOverLread 0.3</pre></td></tr><tr><td data-num="2"></td><td><pre></pre></td></tr><tr><td data-num="3"></td><td><pre>$less S1-1fliLog.finall.out</pre></td></tr><tr><td data-num="4"></td><td><pre>                                 Started job on |       Aug 27 09:52:31</pre></td></tr><tr><td data-num="5"></td><td><pre>                             Started mapping on |       Aug 27 09:52:37</pre></td></tr><tr><td data-num="6"></td><td><pre>                                    Finished on |       Aug 27 13:02:11</pre></td></tr><tr><td data-num="7"></td><td><pre>       Mapping speed, Million of reads per hour |       8.08</pre></td></tr><tr><td data-num="8"></td><td><pre></pre></td></tr><tr><td data-num="9"></td><td><pre>                          Number of input reads |       25517519</pre></td></tr><tr><td data-num="10"></td><td><pre>                      Average input read length |       300</pre></td></tr><tr><td data-num="11"></td><td><pre>                                    UNIQUE READS:</pre></td></tr><tr><td data-num="12"></td><td><pre>                   Uniquely mapped reads number |       16371292</pre></td></tr><tr><td data-num="13"></td><td><pre>                        Uniquely mapped reads % |       64.16%</pre></td></tr><tr><td data-num="14"></td><td><pre></pre></td></tr><tr><td data-num="15"></td><td><pre>$STAR --runThreadN 20 --genomeDir cocco&#x2F; --readFilesCommand zcat --readFilesIn _Lh1-1_S110_L007_R1_001.fastq.gz _Lh1-1_S110_L007_R2_001.fastq.gz --outFileNamePrefix S1-1Cfli --outReadsUnmapped Fastx --outSAMtype SAM --outFilterScoreMinOverLread 0.3 --outFilterMatchNminOverLread 0.3</pre></td></tr><tr><td data-num="16"></td><td><pre></pre></td></tr><tr><td data-num="17"></td><td><pre>$less S1-1CfliLog.final.out</pre></td></tr><tr><td data-num="18"></td><td><pre>                                 Started job on |       Aug 27 09:52:51</pre></td></tr><tr><td data-num="19"></td><td><pre>                             Started mapping on |       Aug 27 09:52:53</pre></td></tr><tr><td data-num="20"></td><td><pre>                                    Finished on |       Aug 28 04:02:41</pre></td></tr><tr><td data-num="21"></td><td><pre>       Mapping speed, Million of reads per hour |       1.40</pre></td></tr><tr><td data-num="22"></td><td><pre></pre></td></tr><tr><td data-num="23"></td><td><pre>                          Number of input reads |       25517519</pre></td></tr><tr><td data-num="24"></td><td><pre>                      Average input read length |       300</pre></td></tr><tr><td data-num="25"></td><td><pre>                                    UNIQUE READS:</pre></td></tr><tr><td data-num="26"></td><td><pre>                   Uniquely mapped reads number |       1475579</pre></td></tr><tr><td data-num="27"></td><td><pre>                        Uniquely mapped reads % |       5.78%</pre></td></tr></table></figure><p>对率有 5.78%，提高了一点。</p><p>接下来，我使用真菌的 unmapped reads 进行比对，看看没有比到真菌上的 reads 有多少是藻类的。</p><figure class="highlight raw"><figcaption data-lang></figcaption><table><tr><td data-num="1"></td><td><pre>$awk -F &quot; &quot; &#39;&#123;print $1&#125;&#39; S1-1fliUnmapped.out.mate1| awk &#39;&#123;if (NR%4&#x3D;&#x3D;1||NR%4&#x3D;3) print&#125;&#39; &gt; S1-1fli.left.fa</pre></td></tr><tr><td data-num="2"></td><td><pre>$sed -i &quot;s&#x2F;@&#x2F;&gt;&#x2F;g&quot; S1-1.left.fa</pre></td></tr><tr><td data-num="3"></td><td><pre>$awk -F &quot; &quot; &#39;&#123;print $1&#125;&#39; S1-1fliUnmapped.out.mate2| awk &#39;&#123;if (NR%4&#x3D;&#x3D;1||NR%4&#x3D;3) print&#125;&#39; &gt; S1-1fli.right.fa</pre></td></tr><tr><td data-num="4"></td><td><pre>$sed -i &quot;s&#x2F;@&#x2F;&gt;&#x2F;g&quot; S1-1.right.fa</pre></td></tr><tr><td data-num="5"></td><td><pre>$STAR --runThreadN 20 --genomeDir cocco&#x2F; --readFilesIn S1-1fli.left.fa S1-1fli.right.fa --outSAMtype SAM --outFileNamePrefix S1-1Ufli --outReadsUnmapped Fastx --outFilterScoreMinOverLread 0.3 --outFilterMatchNminOverLread 0.3</pre></td></tr><tr><td data-num="6"></td><td><pre></pre></td></tr><tr><td data-num="7"></td><td><pre>$less S1-1UfliLog.final.out</pre></td></tr><tr><td data-num="8"></td><td><pre>                                 Started job on |       Aug 27 14:28:39</pre></td></tr><tr><td data-num="9"></td><td><pre>                             Started mapping on |       Aug 27 14:28:44</pre></td></tr><tr><td data-num="10"></td><td><pre>                                    Finished on |       Aug 27 21:38:00</pre></td></tr><tr><td data-num="11"></td><td><pre>       Mapping speed, Million of reads per hour |       0.86</pre></td></tr><tr><td data-num="12"></td><td><pre></pre></td></tr><tr><td data-num="13"></td><td><pre>                          Number of input reads |       6123885</pre></td></tr><tr><td data-num="14"></td><td><pre>                      Average input read length |       300</pre></td></tr><tr><td data-num="15"></td><td><pre>                                    UNIQUE READS:</pre></td></tr><tr><td data-num="16"></td><td><pre>                   Uniquely mapped reads number |       1441904</pre></td></tr><tr><td data-num="17"></td><td><pre>                        Uniquely mapped reads % |       23.55%</pre></td></tr></table></figure><p>也就是说，在剩余的 36% 未比对的 reads 中有 23% 属于 Coccomyxa，总比对率有 70% 左右，地上部位按照相同的步骤进行比对有 83% 的比对率，这个现象也正常，因为地下部位的组织特别复杂，在取样时肯定包含了很多杂质，比对率肯定比地上部位低。</p><h2 id="总结"><a class="anchor" href="#总结">#</a> 总结</h2><p>目前已经可以确定地下部位的藻类就是 Coccomyxa 属的，是不是<em> Coccomyxa subellipsoidea</em> 目前还不确定，接下来的工作是确定地下部位未比对上 reads 的所属。</p><div class="tags"><a href="/tags/bioinformatics/" rel="tag"><i class="ic i-tag"></i> bioinformatics</a> <a href="/tags/transcriptome/" rel="tag"><i class="ic i-tag"></i> transcriptome</a></div></div><footer><div class="meta"><span class="item"><span class="icon"><i class="ic i-calendar-check"></i> </span><span class="text">更新于</span> <time title="修改时间：2023-03-08 14:52:53" itemprop="dateModified" datetime="2023-03-08T14:52:53+08:00">2023-03-08</time> </span><span id="2020/08/18/2020-08-18-sys/" class="item leancloud_visitors" data-flag-title="根据转录组数据鉴定共生物种" title="阅读次数"><span class="icon"><i class="ic i-eye"></i> </span><span class="text">阅读次数</span> <span class="leancloud-visitors-count"></span> <span class="text">次</span></span></div><div class="reward"><button><i class="ic i-heartbeat"></i> 赞赏</button><p>请我喝[茶]~(￣▽￣)~*</p><div id="qr"><div><img data-src="/images/wechatpay.png" alt="amane 微信支付"><p>微信支付</p></div><div><img data-src="/images/alipay.png" alt="amane 支付宝"><p>支付宝</p></div></div></div><div id="copyright"><ul><li class="author"><strong>本文作者： </strong>amane <i class="ic i-at"><em>@</em></i>小钊の笔记</li><li class="link"><strong>本文链接：</strong> <a href="https://www.zhouxiaozhao.cn/2020/08/18/2020-08-18-sys/" title="根据转录组数据鉴定共生物种">https://www.zhouxiaozhao.cn/2020/08/18/2020-08-18-sys/</a></li><li class="license"><strong>版权声明： </strong>本站所有文章除特别声明外，均采用 <span class="exturl" data-url="aHR0cHM6Ly9jcmVhdGl2ZWNvbW1vbnMub3JnL2xpY2Vuc2VzL2J5LW5jLXNhLzQuMC9kZWVkLnpo"><i class="ic i-creative-commons"><em>(CC)</em></i>BY-NC-SA</span> 许可协议。转载请注明出处！</li></ul></div></footer></article></div><div class="post-nav"><div class="item left"><a href="/2020/08/16/2020-08-16-R4ds4/" itemprop="url" rel="prev" data-background-image="https:&#x2F;&#x2F;img1.imgtp.com&#x2F;2023&#x2F;02&#x2F;01&#x2F;LoNchvnE.jpg" title="R数据科学笔记:4"><span class="type">上一篇</span> <span class="category"><i class="ic i-flag"></i> R</span><h3>R数据科学笔记:4</h3></a></div><div class="item right"><a href="/2020/08/20/2020-08-20-nt/" itemprop="url" rel="next" data-background-image="https:&#x2F;&#x2F;img1.imgtp.com&#x2F;2023&#x2F;02&#x2F;01&#x2F;iP4gAcmN.jpg" title="使用本地nt数据库对reads和Trinity结果进行blast"><span class="type">下一篇</span> <span class="category"><i class="ic i-flag"></i> 生信学习</span><h3>使用本地nt数据库对reads和Trinity结果进行blast</h3></a></div></div><div class="wrap" id="comments"></div></div><div id="sidebar"><div class="inner"><div class="panels"><div class="inner"><div class="contents panel pjax" data-title="文章目录"><ol class="toc"><li class="toc-item toc-level-2"><a class="toc-link" href="#%E8%BD%AC%E5%BD%95%E7%BB%84%E6%AF%94%E5%AF%B9%E5%88%B0%E7%9C%9F%E8%8F%8C%E5%8F%82%E8%80%83%E5%9F%BA%E5%9B%A0%E7%BB%84"><span class="toc-number">1.</span> <span class="toc-text">转录组比对到真菌参考基因组</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#%E5%9C%B0%E4%B8%8B%E9%83%A8%E4%BD%8D%E8%BD%AC%E5%BD%95%E7%BB%84%E5%88%86%E6%9E%90"><span class="toc-number">2.</span> <span class="toc-text">地下部位转录组分析</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#%E6%80%BB%E7%BB%93"><span class="toc-number">3.</span> <span class="toc-text">总结</span></a></li></ol></div><div class="related panel pjax" data-title="系列文章"><ul><li><a href="/2020/07/08/2020-07-08-repeat/" rel="bookmark" title="RepeatModeler+RepeatMasker的安装与使用">RepeatModeler+RepeatMasker的安装与使用</a></li><li><a href="/2020/07/09/2020-07-09-github+jekyll/" rel="bookmark" title="利用Github+Jekyll构建自己的博客">利用Github+Jekyll构建自己的博客</a></li><li><a href="/2020/07/18/2020-07-18-orthofinder/" rel="bookmark" title="利用orthofinder寻找单拷贝基因构建系统发育树">利用orthofinder寻找单拷贝基因构建系统发育树</a></li><li><a href="/2020/07/20/2020-07-20-fungi/" rel="bookmark" title="文献导读：真菌进化">文献导读：真菌进化</a></li><li><a href="/2020/08/03/2020-08-03-WGCNA1/" rel="bookmark" title="WGCNA学习：WGCNA分析原理">WGCNA学习：WGCNA分析原理</a></li><li><a href="/2020/08/05/2020-08-05-WGCNA2/" rel="bookmark" title="WGCNA学习：WGCNA应用场景与实例解读">WGCNA学习：WGCNA应用场景与实例解读</a></li><li><a href="/2020/08/08/2020-08-08-WGCNA3/" rel="bookmark" title="WGCNA学习：WGCNA分析实战">WGCNA学习：WGCNA分析实战</a></li><li><a href="/2020/08/10/2020-08-10-cytoscape/" rel="bookmark" title="基因共表达网络图的绘制">基因共表达网络图的绘制</a></li><li class="active"><a href="/2020/08/18/2020-08-18-sys/" rel="bookmark" title="根据转录组数据鉴定共生物种">根据转录组数据鉴定共生物种</a></li><li><a href="/2020/08/20/2020-08-20-nt/" rel="bookmark" title="使用本地nt数据库对reads和Trinity结果进行blast">使用本地nt数据库对reads和Trinity结果进行blast</a></li><li><a href="/2020/08/23/2020-08-23-IGV/" rel="bookmark" title="使用IGV简单绘制甲基化分布图">使用IGV简单绘制甲基化分布图</a></li><li><a href="/2020/08/25/2020-08-25-evolview1/" rel="bookmark" title="进化树美化之Evolview--热图（1）">进化树美化之Evolview--热图（1）</a></li><li><a href="/2020/08/28/2020-08-28-evolview/" rel="bookmark" title="进化树美化之Evolview--热图（2）">进化树美化之Evolview--热图（2）</a></li><li><a href="/2020/08/30/2020-08-30-evolview3/" rel="bookmark" title="进化树+蛋白结构域">进化树+蛋白结构域</a></li><li><a href="/2020/09/01/2020-09-01-iTOL1/" rel="bookmark" title="进化树在线制作工具iTOL教程（1）：基本操作">进化树在线制作工具iTOL教程（1）：基本操作</a></li><li><a href="/2020/09/03/2020-09-03-iTOL2/" rel="bookmark" title="进化树在线制作工具iTOL教程（2）：添加文本">进化树在线制作工具iTOL教程（2）：添加文本</a></li><li><a href="/2020/09/06/2020-09-06-iTOL3/" rel="bookmark" title="进化树在线制作工具iTOL教程（3）：添加图片">进化树在线制作工具iTOL教程（3）：添加图片</a></li><li><a href="/2020/09/08/2020-09-08-iTOL4/" rel="bookmark" title="进化树在线制作工具iTOL教程（4）：添加点图">进化树在线制作工具iTOL教程（4）：添加点图</a></li><li><a href="/2020/09/29/2020-09-29-ScRNAseq(1)/" rel="bookmark" title="单细胞实战(1)：原始测序数据下载">单细胞实战(1)：原始测序数据下载</a></li><li><a href="/2020/10/01/2020-10-01-ScRNAseq(2)/" rel="bookmark" title="单细胞实战(2)：cellranger使用">单细胞实战(2)：cellranger使用</a></li><li><a href="/2020/10/03/2020-10-03-ScRNAseq(3)/" rel="bookmark" title="单细胞实战(3)：STAR分析单细胞数据">单细胞实战(3)：STAR分析单细胞数据</a></li><li><a href="/2020/10/06/2020-10-06-ScRNAseq(4)/" rel="bookmark" title="单细胞实战(4)：STAR与cellranger结果比较">单细胞实战(4)：STAR与cellranger结果比较</a></li><li><a href="/2020/10/10/2020-10-10-ScRNAseq(5)/" rel="bookmark" title="单细胞实战(5)：复现文章中的聚类图（1）">单细胞实战(5)：复现文章中的聚类图（1）</a></li><li><a href="/2020/10/13/2020-10-13-ScRNAseq(6)/" rel="bookmark" title="单细胞实战(5)：复现文章中的聚类图（2）">单细胞实战(5)：复现文章中的聚类图（2）</a></li><li><a href="/2020/10/24/2020-10-24-ScRNAseq(7)/" rel="bookmark" title="生信技能树单细胞数据挖掘笔记(1)">生信技能树单细胞数据挖掘笔记(1)</a></li><li><a href="/2020/10/27/2020-10-27-ScRNAseq(8)/" rel="bookmark" title="生信技能树单细胞数据挖掘笔记(2)">生信技能树单细胞数据挖掘笔记(2)</a></li><li><a href="/2020/10/29/2020-10-29-ScRNAseq(9)/" rel="bookmark" title="生信技能树单细胞数据挖掘笔记(3)">生信技能树单细胞数据挖掘笔记(3)</a></li><li><a href="/2020/10/31/2020-10-31-ScRNAseq(10)/" rel="bookmark" title="生信技能树单细胞数据挖掘笔记(4)">生信技能树单细胞数据挖掘笔记(4)</a></li><li><a href="/2020/11/03/2020-11-03-ScRNAseq(11)/" rel="bookmark" title="生信技能树单细胞数据挖掘笔记(5)">生信技能树单细胞数据挖掘笔记(5)</a></li><li><a href="/2020/11/05/2020-11-05-ScRNAseq(12)/" rel="bookmark" title="生信技能树单细胞数据挖掘笔记(6)">生信技能树单细胞数据挖掘笔记(6)</a></li><li><a href="/2020/11/07/2020-11-07-ScRNAseq(13)/" rel="bookmark" title="生信技能树单细胞数据挖掘笔记(7)">生信技能树单细胞数据挖掘笔记(7)</a></li><li><a href="/2020/11/10/2020-11-10-RNAvelocity(1)/" rel="bookmark" title="RNA速率：软件下载与loom文件准备">RNA速率：软件下载与loom文件准备</a></li><li><a href="/2020/11/12/2020-11-12-RNAvelocity(2)/" rel="bookmark" title="RNA速率：数据读入">RNA速率：数据读入</a></li><li><a href="/2020/11/14/2020-11-14-RNAvelocity(3)/" rel="bookmark" title="RNA速率：使用Seurat的结果做RNA velocity">RNA速率：使用Seurat的结果做RNA velocity</a></li><li><a href="/2020/11/17/2020-11-17-RNAseq(1)/" rel="bookmark" title="RNAseq教程(1.1)">RNAseq教程(1.1)</a></li><li><a href="/2020/11/19/2020-11-19-annotion(1)/" rel="bookmark" title="基因结构注释（1）：从头注释">基因结构注释（1）：从头注释</a></li><li><a href="/2020/11/21/2020-11-21-annotion(2)/" rel="bookmark" title="基因结构注释（2）：同源注释">基因结构注释（2）：同源注释</a></li><li><a href="/2020/11/24/2020-11-24-annotion(3)/" rel="bookmark" title="基因结构注释（3）：转录组注释">基因结构注释（3）：转录组注释</a></li><li><a href="/2020/11/26/2020-11-26-annotion(4)/" rel="bookmark" title="基因结构注释（4）：整合预测结果">基因结构注释（4）：整合预测结果</a></li><li><a href="/2020/11/28/2020-11-28-RNAseq(2)/" rel="bookmark" title="RNAseq教程(1.2)">RNAseq教程(1.2)</a></li><li><a href="/2020/12/01/2020-12-01-RNAseq(3)/" rel="bookmark" title="RNAseq教程(1.3)">RNAseq教程(1.3)</a></li><li><a href="/2020/12/03/2020-12-03-RNAseq(4)/" rel="bookmark" title="RNAseq教程(1.4)">RNAseq教程(1.4)</a></li><li><a href="/2020/12/05/2020-12-05-RNAseq(5)/" rel="bookmark" title="RNAseq教程(1.5)">RNAseq教程(1.5)</a></li><li><a href="/2020/12/08/2020-12-08-RNAseq(6)/" rel="bookmark" title="RNAseq教程(1.6)">RNAseq教程(1.6)</a></li><li><a href="/2020/12/10/2020-12-10-RNAseq(7)/" rel="bookmark" title="RNAseq教程(2.1)">RNAseq教程(2.1)</a></li><li><a href="/2020/12/12/2020-12-12-RNAseq(8)/" rel="bookmark" title="RNAseq教程(2.2)">RNAseq教程(2.2)</a></li><li><a href="/2020/12/15/2020-12-15-RNAseq(9)/" rel="bookmark" title="RNAseq教程(2.3)">RNAseq教程(2.3)</a></li><li><a href="/2020/12/17/2020-12-17-RNAseq(10)/" rel="bookmark" title="RNAseq教程(2.4)">RNAseq教程(2.4)</a></li><li><a href="/2020/12/19/2020-12-19-RNAseq(11)/" rel="bookmark" title="RNAseq教程(2.5)">RNAseq教程(2.5)</a></li><li><a href="/2020/12/22/2020-12-22-RNAseq(12)/" rel="bookmark" title="RNAseq教程(3.1)">RNAseq教程(3.1)</a></li><li><a href="/2020/12/24/2020-12-24-RNAseq(13)/" rel="bookmark" title="RNAseq教程(3.2)">RNAseq教程(3.2)</a></li><li><a href="/2020/12/26/2020-12-26-RNAseq(14)/" rel="bookmark" title="RNAseq教程(3.3)">RNAseq教程(3.3)</a></li><li><a href="/2020/12/29/2020-12-29-RNAseq(15)/" rel="bookmark" title="RNAseq教程(3.4)">RNAseq教程(3.4)</a></li><li><a href="/2020/12/31/2020-12-31-RNAseq(16)/" rel="bookmark" title="RNAseq教程(4.1)">RNAseq教程(4.1)</a></li><li><a href="/2021/01/02/2021-01-02-RNAseq(17)/" rel="bookmark" title="RNAseq教程(4.2)">RNAseq教程(4.2)</a></li><li><a href="/2021/01/05/2021-01-05-RNAseq(18)/" rel="bookmark" title="RNAseq教程(4.3)">RNAseq教程(4.3)</a></li><li><a href="/2021/01/07/2021-01-07-RNAseq(19)/" rel="bookmark" title="RNAseq教程(4.4)">RNAseq教程(4.4)</a></li><li><a href="/2021/01/09/2021-01-09-RNAseq(20)/" rel="bookmark" title="RNAseq教程(4.5)">RNAseq教程(4.5)</a></li><li><a href="/2021/01/12/2021-01-12-RNAseq(21)/" rel="bookmark" title="RNAseq教程(5)">RNAseq教程(5)</a></li><li><a href="/2022/12/08/2022-12-08-SingleCopy/" rel="bookmark" title="比较基因组学分析1：构建单拷贝基因树">比较基因组学分析1：构建单拷贝基因树</a></li><li><a href="/2022/12/14/2022-12-14-CAFE/" rel="bookmark" title="比较基因组学分析2：基因家族收缩与扩张分析">比较基因组学分析2：基因家族收缩与扩张分析</a></li><li><a href="/2022/12/15/2022-12-15-GO/" rel="bookmark" title="比较基因组学分析3：特异节点基因家族富集分析（非模式物种GO/KEEG库构建）">比较基因组学分析3：特异节点基因家族富集分析（非模式物种GO/KEEG库构建）</a></li><li><a href="/2023/02/22/2023-02-22-ST(1)/" rel="bookmark" title="植物空间转录组分析1：Seurat基本流程">植物空间转录组分析1：Seurat基本流程</a></li><li><a href="/2023/02/24/2023-02-24-ST(2)/" rel="bookmark" title="植物空间转录组分析2：STEEL+Seurat">植物空间转录组分析2：STEEL+Seurat</a></li></ul></div><div class="overview panel" data-title="站点概览"><div class="author" itemprop="author" itemscope itemtype="http://schema.org/Person"><img class="image" itemprop="image" alt="amane" data-src="/images/avatar.jpg"><p class="name" itemprop="name">amane</p><div class="description" itemprop="description">生物学在读博士</div></div><nav class="state"><div class="item posts"><a href="/archives/"><span class="count">84</span> <span class="name">文章</span></a></div><div class="item categories"><a href="/categories/"><span class="count">8</span> <span class="name">分类</span></a></div><div class="item tags"><a href="/tags/"><span class="count">21</span> <span class="name">标签</span></a></div></nav><div class="social"><span class="exturl item github" data-url="aHR0cHM6Ly9naXRodWIuY29tL0Jpb2luZm9ybWF0aWNzLXJvb2tpZQ==" title="https:&#x2F;&#x2F;github.com&#x2F;Bioinformatics-rookie"><i class="ic i-github"></i></span> <span class="exturl item bilibili" data-url="aHR0cHM6Ly9zcGFjZS5iaWxpYmlsaS5jb20vMTU3Nzg1NTM/c3BtX2lkX2Zyb209MzMzLjEwMDcuMC4w" title="https:&#x2F;&#x2F;space.bilibili.com&#x2F;15778553?spm_id_from&#x3D;333.1007.0.0"><i class="ic i-bilibili"></i></span> <a href="https://www.zhouxiaozhao.cn/about/" title="https:&#x2F;&#x2F;www.zhouxiaozhao.cn&#x2F;about&#x2F;" class="item about"><i class="ic i-address-card"></i></a> <span class="exturl item email" data-url="bWFpbHRvOnpob3VzaGl6aGFvQG1haWwua2liLmFjLmNu" title="mailto:zhoushizhao@mail.kib.ac.cn"><i class="ic i-envelope"></i></span></div><ul class="menu"><li class="item"><a href="/" rel="section"><i class="ic i-home"></i>首页</a></li><li class="item"><a href="/about/" rel="section"><i class="ic i-user"></i>关于</a></li><li class="item dropdown"><a href="javascript:void(0);"><i class="ic i-feather"></i>文章</a><ul class="submenu"><li class="item"><a href="/archives/" rel="section"><i class="ic i-list-alt"></i>归档</a></li><li class="item"><a href="/categories/" rel="section"><i class="ic i-th"></i>分类</a></li><li class="item"><a href="/tags/" rel="section"><i class="ic i-tags"></i>标签</a></li></ul></li><li class="item dropdown"><a href="/webstack/" rel="section"><i class="ic i-magic"></i>AI学习</a><ul class="submenu"><li class="item"><a href="/AIpainting/" rel="section"><i class="ic i-pen"></i>AI绘画</a></li><li class="item"><a href="/AIbookmark/" rel="section"><i class="ic i-comments"></i>AI书签</a></li></ul></li><li class="item"><a href="/links/" rel="section"><i class="ic i-heart"></i>友達</a></li></ul></div></div></div><ul id="quick"><li class="prev pjax"><a href="/2020/08/16/2020-08-16-R4ds4/" rel="prev" title="上一篇"><i class="ic i-chevron-left"></i></a></li><li class="up"><i class="ic i-arrow-up"></i></li><li class="down"><i class="ic i-arrow-down"></i></li><li class="next pjax"><a href="/2020/08/20/2020-08-20-nt/" rel="next" title="下一篇"><i class="ic i-chevron-right"></i></a></li><li class="percent"></li></ul></div></div><div class="dimmer"></div></div></main><footer id="footer"><div class="inner"><div class="widgets"><div class="rpost pjax"><h2>随机文章</h2><ul><li class="item"><div class="breadcrumb"><a href="/categories/Bioinformation/" title="分类于 生信学习">生信学习</a> <i class="ic i-angle-right"></i> <a href="/categories/Bioinformation/genome/" title="分类于 基因组">基因组</a></div><span><a href="/2020/07/08/2020-07-08-repeat/" title="RepeatModeler+RepeatMasker的安装与使用">RepeatModeler+RepeatMasker的安装与使用</a></span></li><li class="item"><div class="breadcrumb"><a href="/categories/Bioinformation/" title="分类于 生信学习">生信学习</a></div><span><a href="/2020/07/20/2020-07-20-fungi/" title="文献导读：真菌进化">文献导读：真菌进化</a></span></li><li class="item"><div class="breadcrumb"><a href="/categories/Bioinformation/" title="分类于 生信学习">生信学习</a></div><span><a href="/2020/12/12/2020-12-12-RNAseq(8)/" title="RNAseq教程(2.2)">RNAseq教程(2.2)</a></span></li><li class="item"><div class="breadcrumb"><a href="/categories/Bioinformation/" title="分类于 生信学习">生信学习</a> <i class="ic i-angle-right"></i> <a href="/categories/Bioinformation/SingleCell/" title="分类于 单细胞">单细胞</a></div><span><a href="/2020/10/24/2020-10-24-ScRNAseq(7)/" title="生信技能树单细胞数据挖掘笔记(1)">生信技能树单细胞数据挖掘笔记(1)</a></span></li><li class="item"><div class="breadcrumb"><a href="/categories/Bioinformation/" title="分类于 生信学习">生信学习</a></div><span><a href="/2020/12/15/2020-12-15-RNAseq(9)/" title="RNAseq教程(2.3)">RNAseq教程(2.3)</a></span></li><li class="item"><div class="breadcrumb"><a href="/categories/Python/" title="分类于 Python">Python</a></div><span><a href="/2020/10/22/2020-10-22-python9/" title="马士兵教育Python入门基础:4.流程控制（3）">马士兵教育Python入门基础:4.流程控制（3）</a></span></li><li class="item"><div class="breadcrumb"><a href="/categories/Python/" title="分类于 Python">Python</a></div><span><a href="/2020/10/20/2020-10-20-python8/" title="马士兵教育Python入门基础:4.流程控制（2）">马士兵教育Python入门基础:4.流程控制（2）</a></span></li><li class="item"><div class="breadcrumb"><a href="/categories/R/" title="分类于 R">R</a></div><span><a href="/2020/08/10/2020-08-13-R4ds3/" title="R数据科学笔记:3">R数据科学笔记:3</a></span></li><li class="item"><div class="breadcrumb"><a href="/categories/Bioinformation/" title="分类于 生信学习">生信学习</a></div><span><a href="/2020/12/24/2020-12-24-RNAseq(13)/" title="RNAseq教程(3.2)">RNAseq教程(3.2)</a></span></li><li class="item"><div class="breadcrumb"><a href="/categories/paper/" title="分类于 文献导读">文献导读</a></div><span><a href="/2020/09/22/2020-09-22-Cool_your_github/" title="炫酷你的Github主页">炫酷你的Github主页</a></span></li></ul></div><div><h2>最新评论</h2><ul class="leancloud-recent-comment"></ul></div></div><div class="status"><div class="copyright">&copy; 2010 – <span itemprop="copyrightYear">2023</span> <span class="with-love"><i class="ic i-sakura rotate"></i> </span><span class="author" itemprop="copyrightHolder">amane @ Zhou Xiaozhao</span></div><div class="count"><span class="post-meta-item-icon"><i class="ic i-chart-area"></i> </span><span title="站点总字数">381k 字</span> <span class="post-meta-divider">|</span> <span class="post-meta-item-icon"><i class="ic i-coffee"></i> </span><span title="站点阅读时长">5:46</span></div><div class="powered-by">基于 <span class="exturl" data-url="aHR0cHM6Ly9oZXhvLmlv">Hexo</span> & Theme.<span class="exturl" data-url="aHR0cHM6Ly9naXRodWIuY29tL2FtZWhpbWUvaGV4by10aGVtZS1zaG9rYQ==">Shoka</span></div></div></div></footer></div><script data-config type="text/javascript">var LOCAL={path:"2020/08/18/2020-08-18-sys/",favicon:{show:"（●´3｀●）やれやれだぜ",hide:"(´Д｀)大変だ！"},search:{placeholder:"文章搜索",empty:"关于 「 ${query} 」，什么也没搜到",stats:"${time} ms 内找到 ${hits} 条结果"},valine:!0,fancybox:!0,copyright:'复制成功，转载请遵守 <i class="ic i-creative-commons"></i>BY-NC-SA 协议。',ignores:[function(e){return e.includes("#")},function(e){return new RegExp(LOCAL.path+"$").test(e)}]}</script><script src="https://cdn.polyfill.io/v2/polyfill.js"></script><script src="//cdn.jsdelivr.net/combine/npm/pace-js@1.0.2/pace.min.js,npm/pjax@0.2.8/pjax.min.js,npm/whatwg-fetch@3.4.0/dist/fetch.umd.min.js,npm/animejs@3.2.0/lib/anime.min.js,npm/algoliasearch@4/dist/algoliasearch-lite.umd.js,npm/instantsearch.js@4/dist/instantsearch.production.min.js,npm/lozad@1/dist/lozad.min.js,npm/quicklink@2/dist/quicklink.umd.js"></script><script src="/js/app.js?v=0.2.5"></script></body></html><!-- rebuild by hrmmi -->