<!-- build time:Fri Feb 24 2023 14:40:38 GMT+0800 (中国标准时间) --><!DOCTYPE html><html lang="zh-CN"><head><meta charset="UTF-8"><meta name="viewport" content="width=device-width,initial-scale=1,maximum-scale=2"><meta name="theme-color" content="#FFF"><link rel="apple-touch-icon" sizes="180x180" href="/images/apple-touch-icon.png"><link rel="icon" type="image/ico" sizes="32x32" href="/images/favicon.ico"><meta http-equiv="Cache-Control" content="no-transform"><meta http-equiv="Cache-Control" content="no-siteapp"><link rel="alternate" type="application/rss+xml" title="小钊の笔记" href="https://www.zhouxiaozhao.cn/rss.xml"><link rel="alternate" type="application/atom+xml" title="小钊の笔记" href="https://www.zhouxiaozhao.cn/atom.xml"><link rel="alternate" type="application/json" title="小钊の笔记" href="https://www.zhouxiaozhao.cn/feed.json"><link rel="stylesheet" href="//fonts.googleapis.com/css?family=Mulish:300,300italic,400,400italic,700,700italic%7CFredericka%20the%20Great:300,300italic,400,400italic,700,700italic%7CNoto%20Serif%20JP:300,300italic,400,400italic,700,700italic%7CNoto%20Serif%20SC:300,300italic,400,400italic,700,700italic%7CInconsolata:300,300italic,400,400italic,700,700italic&display=swap&subset=latin,latin-ext"><link rel="stylesheet" href="/css/app.css?v=0.2.5"><meta name="keywords" content="bioinformatics,Comparative_Genomics"><link rel="canonical" href="https://www.zhouxiaozhao.cn/2022/12/14/2022-12-14-CAFE/"><title>比较基因组学分析 2：基因家族收缩与扩张分析 - 基因组 - 生信学习 | Zhou Xiaozhao = 小钊の笔记 = 前天是小兔子，昨天是小鹿，今天是你</title><meta name="generator" content="Hexo 5.4.1"></head><body itemscope itemtype="http://schema.org/WebPage"><div id="loading"><div class="cat"><div class="body"></div><div class="head"><div class="face"></div></div><div class="foot"><div class="tummy-end"></div><div class="bottom"></div><div class="legs left"></div><div class="legs right"></div></div><div class="paw"><div class="hands left"></div><div class="hands right"></div></div></div></div><div id="container"><header id="header" itemscope itemtype="http://schema.org/WPHeader"><div class="inner"><div id="brand"><div class="pjax"><h1 itemprop="name headline">比较基因组学分析 2：基因家族收缩与扩张分析</h1><div class="meta"><span class="item" title="创建时间：2022-12-14 00:00:00"><span class="icon"><i class="ic i-calendar"></i> </span><span class="text">发表于</span> <time itemprop="dateCreated datePublished" datetime="2022-12-14T00:00:00+08:00">2022-12-14</time> </span><span class="item" title="本文字数"><span class="icon"><i class="ic i-pen"></i> </span><span class="text">本文字数</span> <span>8.9k</span> <span class="text">字</span> </span><span class="item" title="阅读时长"><span class="icon"><i class="ic i-clock"></i> </span><span class="text">阅读时长</span> <span>8 分钟</span></span></div></div></div><nav id="nav"><div class="inner"><div class="toggle"><div class="lines" aria-label="切换导航栏"><span class="line"></span> <span class="line"></span> <span class="line"></span></div></div><ul class="menu"><li class="item title"><a href="/" rel="start">Zhou Xiaozhao</a></li></ul><ul class="right"><li class="item theme"><i class="ic i-sun"></i></li><li class="item search"><i class="ic i-search"></i></li></ul></div></nav></div><div id="imgs" class="pjax"><ul><li class="item" data-background-image="https://img1.imgtp.com/2023/02/01/DnAfhAaN.jpg"></li><li class="item" data-background-image="https://img1.imgtp.com/2023/02/01/j5n64NGl.jpg"></li><li class="item" data-background-image="https://img1.imgtp.com/2023/02/01/UJhRWazi.jpg"></li><li class="item" data-background-image="https://img1.imgtp.com/2023/02/01/4GFP7sNy.jpg"></li><li class="item" data-background-image="https://img1.imgtp.com/2023/02/01/szJGZgtF.jpg"></li><li class="item" data-background-image="https://img1.imgtp.com/2023/02/01/I8YcsVBp.jpg"></li></ul></div></header><div id="waves"><svg class="waves" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" viewBox="0 24 150 28" preserveAspectRatio="none" shape-rendering="auto"><defs><path id="gentle-wave" d="M-160 44c30 0 58-18 88-18s 58 18 88 18 58-18 88-18 58 18 88 18 v44h-352z"/></defs><g class="parallax"><use xlink:href="#gentle-wave" x="48" y="0"/><use xlink:href="#gentle-wave" x="48" y="3"/><use xlink:href="#gentle-wave" x="48" y="5"/><use xlink:href="#gentle-wave" x="48" y="7"/></g></svg></div><main><div class="inner"><div id="main" class="pjax"><div class="article wrap"><div class="breadcrumb" itemscope itemtype="https://schema.org/BreadcrumbList"><i class="ic i-home"></i> <span><a href="/">首页</a></span><i class="ic i-angle-right"></i> <span itemprop="itemListElement" itemscope itemtype="https://schema.org/ListItem"><a href="/categories/Bioinformation/" itemprop="item" rel="index" title="分类于 生信学习"><span itemprop="name">生信学习</span></a><meta itemprop="position" content="1"></span><i class="ic i-angle-right"></i> <span class="current" itemprop="itemListElement" itemscope itemtype="https://schema.org/ListItem"><a href="/categories/Bioinformation/genome/" itemprop="item" rel="index" title="分类于 基因组"><span itemprop="name">基因组</span></a><meta itemprop="position" content="2"></span></div><article itemscope itemtype="http://schema.org/Article" class="post block" lang="zh-CN"><link itemprop="mainEntityOfPage" href="https://www.zhouxiaozhao.cn/2022/12/14/2022-12-14-CAFE/"><span hidden itemprop="author" itemscope itemtype="http://schema.org/Person"><meta itemprop="image" content="/images/avatar.png"><meta itemprop="name" content="amane"><meta itemprop="description" content="前天是小兔子，昨天是小鹿，今天是你, 生物学在读博士"></span><span hidden itemprop="publisher" itemscope itemtype="http://schema.org/Organization"><meta itemprop="name" content="小钊の笔记"></span><div class="body md" itemprop="articleBody"><h1 id="前言"><a class="anchor" href="#前言">#</a> 前言</h1><h2 id="上篇推文中介绍到比较基因组学分析常用套路的第一步利用单拷贝基因构建具有分化时间的物种树补充一点对于跨度较大的物种可以选择单拷贝基因的方法比如此次分析使用的物种-对于目级或者科级水平来讲推荐使用共线性基因建树-以十字花科为例如果用单拷贝基因可能只有1000多组但是使用共线性可能有接近5000组-共线性基因建树可以使用wgdi的方法这部分内容我以后会探索一下-sun-et-al-2022-wgdi本篇推文主要讲基因家族的收缩与扩张分析使用的软件是cafe52020年发表相较于cafe4来讲操作更加方便并且新增了模型gamma"><a class="anchor" href="#上篇推文中介绍到比较基因组学分析常用套路的第一步利用单拷贝基因构建具有分化时间的物种树补充一点对于跨度较大的物种可以选择单拷贝基因的方法比如此次分析使用的物种-对于目级或者科级水平来讲推荐使用共线性基因建树-以十字花科为例如果用单拷贝基因可能只有1000多组但是使用共线性可能有接近5000组-共线性基因建树可以使用wgdi的方法这部分内容我以后会探索一下-sun-et-al-2022-wgdi本篇推文主要讲基因家族的收缩与扩张分析使用的软件是cafe52020年发表相较于cafe4来讲操作更加方便并且新增了模型gamma">#</a> 上篇推文中介绍到比较基因组学分析常用套路的第一步，利用单拷贝基因构建具有分化时间的物种树，补充一点，<span class="pink">对于跨度较大的物种，可以选择单拷贝基因的方法，比如此次分析使用的物种。对于目级或者科级水平来讲，推荐使用共线性基因建树。以十字花科为例，如果用单拷贝基因，可能只有 1000 多组，但是使用共线性可能有接近 5000 组。</span>共线性基因建树可以使用 WGDI 的方法，这部分内容我以后会探索一下。<br><img data-src="/img/posts/2022.12.14/1.png" alt="Sun et al 2022 WGDI"><br>本篇推文主要讲基因家族的收缩与扩张分析，使用的软件是<span class="exturl" data-url="aHR0cHM6Ly9naXRodWIuY29tL2hhaG5sYWIvQ0FGRTU="> cafe5</span>，2020 年发表，相较于 cafe4 来讲操作更加方便并且新增了模型 (Gamma)</h2><h1 id="安装"><a class="anchor" href="#安装">#</a> 安装</h1><pre><code class="language-bash">git clone https://github.com/hahnlab/CAFE5.git
cd CAFE5
./configure
make
</code></pre><p>安装过程其实有点复杂，可能不同的服务器会出现不同的错误，这个请自行解决</p><h1 id="cafe5使用"><a class="anchor" href="#cafe5使用">#</a> CAFE5 使用</h1><p>输入文件至少要两个，一个是基因家族数目统计文件 Genefamilies_Count.tsv，一个是树文件 tree.txt（带有分化时间），还可以增加一个 lambda 文件</p><h2 id="主要参数"><a class="anchor" href="#主要参数">#</a> 主要参数</h2><pre><code class="language-bash">--fixed_alpha, -a

Alpha value of the discrete gamma distribution to use in category calculations. If not specified, the alpha parameter will be estimated by maximum likelihood.

--lambda_per_family, -b

Estimate lambda by family (for testing purposes only).

--cores, -c

Number of processing cores to use, requires an integer argument. Default=All available cores.

--error_model, -e

Run with no file name to estimate the global error model file. This file can be provided in subsequent runs by providing the path to the Error model file with no spaces (e.g. -eBase_error_model.txt).

--Expansion, -E

Expansion parameter for Nelder-Mead optimizer, Default=2.

--rootdist, -f

Path to root distribution file for simulating datasets.

--help, -h

Help menu with a list of all commands.

--infile, -i

Path to tab delimited gene families file to be analyzed - Required for estimation.

--Iterations, -I

Maximum number of iterations that will be performed in lambda search. Default=300 (increase this number if likelihood is still improving when limit is hit).

--n_gamma_cats, -k

Number of gamma categories to use. If specified, the Gamma model will be used to run calculations; otherwise the Base model will be used.

--fixed_lambda, -l

Value (between 0 and 1) for a single user provided lambda value, otherwise lambda is estimated.

--log_config, -L

Turn on logging, provide name of the configuration file for logging (see example log.config file).

--fixed_multiple_lambdas, -m

Multiple lambda values, comma separated, must be used in conjunction with lambda tree (-y).

--output_prefix, -o

Output directory - Name of directory automatically created for output. Default=results.

--poisson, -p

Use a Poisson distribution for the root frequency distribution. If no -p flag is given, a uniform distribution will be used. A value can be specified (-p10, or --poisson=10); otherwise the distribution will be estimated from the gene families.

--pvalue, -P

P-value to use for determining significance of family size change, Default=0.05.

--chisquare_compare, -r

Chi square compare (not tested).

--Reflection, -R

Reflection parameter for Nelder-Mead optimizer, Default=1.

--simulate, -s

Simulate families. Either provide an argument of the number of families to simulate (-s100, or --simulate=100) or provide a rootdist file giving a set of root family sizes to match. Without such a file, the families will be generated with root sizes selected randomly between 0 and 100.

--tree, -t

Path to file containing newick formatted tree - Required for estimation.

--lambda_tree, -y

Path to lambda tree, for use with multiple lambdas.

--zero_root, -z

Include gene families that don't exist at the root, not recommended.
</code></pre><p>其实主要用的就是 - i -p -k -y -t 这些参数</p><h2 id="输入文件准备"><a class="anchor" href="#输入文件准备">#</a> 输入文件准备</h2><h3 id="genefamilies_counttsv"><a class="anchor" href="#genefamilies_counttsv">#</a> Genefamilies_Count.tsv</h3><p>制表符分隔的基因家族计数文件，通常用 OrthoMCL,<br>OrthoFinder 等软件获取计数信息。<br>示例格式</p><pre><code class="language-bash">Desc	Family ID	human	chimp	orang	baboon	gibbon	macaque	marmoset rat	mouse	cat	horse	cow
ATPase	ORTHOMCL1	 52	 55	 54	 57	 54	  56	  56	 53	 52	57	55	 54
(null)	ORTHOMCL2	 76	 51	 41	 39	 45	  36	  37	 67	 79	37	41	 49
HMG box	ORTHOMCL3	 50	 49	 48	 48	 46	  49	  48	 55	 52	51	47	 55
(null)	ORTHOMCL4	 43	 43	 47	 53	 44	  47	  46	 59	 58	51	50	 55
Dynamin	ORTHOMCL5	 43	 40	 43	 44	 31	  46	  33	 79	 70	43	49	 50
......
....
..
DnaJ	ORTHOMCL10016	 45	 46	 50	 46	 46 	  47	  46	 48	 49	45	44	 48
</code></pre><p>我们首先利用 OrthoFinder 的 Orthogroups.GeneCount.tsv 文件生成符合要求的输入文件</p><pre><code class="language-bash">cp Results_May02/Orthogroups/Orthogroups.GeneCount.tsv CAFE/
awk 'OFS=&quot;\t&quot; &#123;$NF=&quot;&quot;; print&#125;' Orthogroups.GeneCount.tsv &gt; tmp &amp;&amp; awk '&#123;print &quot;(null)&quot;&quot;\t&quot;$0&#125;' tmp &gt; cafe.input.tsv &amp;&amp; sed -i '1s/(null)/Desc/g' cafe.input.tsv &amp;&amp; rm tmp
</code></pre><p>查看文件格式</p><pre><code class="language-bash">Desc    Orthogroup      Aof.pro Ath.pro Atr.pro Cba.pro Cri.pro Csa.pro Csu.pro Kle.pro Mpo.pro Nco.pro Osa.pro Ppa.pro Smo.pro Tpl.pro Vca.pro Vvi.pro Zma.pro
(null)  OG0000000       145     112     95      5       372     129     3       1       2       217     126     16      206     419     4       177     117
(null)  OG0000001       9       4       3       1691    9       96      2       56      2       4       21      0       2       5       3       2       0
(null)  OG0000002       32      117     62      1       92      117     2       0       20      81      119     77      40      193     5       107     161
(null)  OG0000003       37      104     54      3       89      76      4       5       10      74      144     22      47      134     8       79      154
(null)  OG0000004       73      104     51      4       40      80      2       10      12      76      87      33      22      136     5       97      135
(null)  OG0000005       28      46      36      11      37      47      0       3       50      81      81      32      48      120     0       54      73
(null)  OG0000006       41      43      74      6       38      57      0       4       25      57      52      19      33      155     0       87      40
(null)  OG0000007       58      52      60      0       18      42      0       0       12      50      56      17      57      99      1       82      52
(null)  OG0000008       38      57      26      7       52      47      4       6       19      40      59      43      20      29      1       41      80
(null)  OG0000009       46      57      26      1       25      46      1       2       11      52      65      29      13      50      1       48      87
</code></pre><p>生成之后还需要剔除不同物种间拷贝数差异过大的基因家族，否则会报错，有内置脚本可以使用，我在运行的时候需要去掉第一行才能使用</p><pre><code class="language-bash">python ~/soft/CAFE5/tutorial/clade_and_size_filter.py -i cafe.input.tsv -o gene_family_filter.txt -s 

</code></pre><p>笨方法</p><pre><code class="language-bash">awk 'NR==1 || $3&lt;100 &amp;&amp; $4&lt;100 &amp;&amp; $5&lt;100 &amp;&amp; $6&lt;100 &amp;&amp; $7&lt;100 &amp;&amp; $8&lt;100 &amp;&amp; $9&lt;100 &amp;&amp; $10&lt;100 &amp;&amp; $11&lt;100 &amp;&amp; $12&lt;100 &amp;&amp; $13&lt;100 &amp;&amp; $14&lt;100 &amp;&amp; $15&lt;100 &amp;&amp; $16&lt;100 &amp;&amp; $17&lt;100 &amp;&amp; $18&lt;100 &amp;&amp; $19&lt;100 &#123;print $0&#125;' cafe.input.tsv &gt;gene_family_filter.txt
</code></pre><p>最后的文件格式，保证第一行的物种名字与进化树的一致即可</p><pre><code class="language-bash">Desc    Orthogroup      Aof     Ath     Atr     Cba     Cri     Csa     Csu     Kle     Mpo     Nco     Osa     Ppa     Smo     Tpl     Vca     Vvi     Zma
(null)  OG0000020       26      37      23      4       35      28      0       1       24      28      43      24      27      47      0       35      42
(null)  OG0000021       49      41      31      7       30      31      8       2       7       26      49      15      11      31      0       36      45
(null)  OG0000022       27      25      31      0       27      34      2       1       23      25      46      18      27      44      1       39      45
(null)  OG0000024       37      40      27      0       22      30      1       11      9       33      38      18      25      43      0       37      39
(null)  OG0000029       28      26      23      2       24      25      1       2       5       32      34      31      17      35      1       30      40
(null)  OG0000030       23      30      16      1       23      27      1       1       27      26      26      16      15      49      1       28      35
(null)  OG0000031       28      36      26      3       27      23      8       1       3       17      37      10      18      38      3       34      28
(null)  OG0000032       18      16      25      0       24      19      0       5       4       25      36      6       38      49      1       38      35
(null)  OG0000033       19      17      20      0       12      16      4       6       18      35      42      4       23      39      3       45      28
(null)  OG0000035       17      37      17      8       28      24      2       2       5       30      41      8       6       26      2       32      37
(null)  OG0000036       22      15      17      3       22      19      7       12      13      20      24      20      30      38      3       35      22
(null)  OG0000039       14      27      36      0       34      24      0       2       2       12      41      4       47      13      0       37      26
(null)  OG0000040       15      30      9       1       19      35      0       2       11      25      26      19      12      48      0       39      27


</code></pre><h3 id="treetxt"><a class="anchor" href="#treetxt">#</a> tree.txt</h3><p>本步骤直接使用 mcmctree 生成的 FigTree.tre 文件修改一下即可使用</p><pre><code class="language-bash">grep  &quot;UTREE 1 =&quot; FigTree.tre | sed -E -e &quot;s/\[[^]]*\]//g&quot; -e &quot;s/[ \t]//g&quot; -e &quot;/^$/d&quot; -e &quot;s/UTREE1=//&quot; &gt; tree.txt
</code></pre><h2 id="运行cafe5"><a class="anchor" href="#运行cafe5">#</a> 运行 CAFE5</h2><pre><code class="language-bash"> cafe5 -i gene_family_filter.txt -t tree.txt -o out -c 1
## 如果使用Gamma模型与泊松分布
 cafe5 -i gene_family_filter.txt -t tree.txt -o out -c 1 -k 2 -p ##注意-k可以调，一般为2-5
</code></pre><p>结果还是报错<br><img data-src="/img/posts/2022.12.14/2.png" alt="报错"><br>查了一下解决方法，可以将 lambda 值调低 (0.0001) 再进行计算<br><img data-src="/img/posts/2022.12.14/3.png" alt="解决方案"></p><p>重新运行</p><pre><code class="language-bash"> cafe5 -i gene_family_filter.txt -t tree.txt -o out -c 16 -l 0.0001 -k 2 -p
</code></pre><p>当大家出现这种错误时，可以尝试这种解决方案，单单删除这些家族是没用的。</p><h2 id="cafe5输出结果"><a class="anchor" href="#cafe5输出结果">#</a> CAFE5 输出结果</h2><h3 id="结果文件"><a class="anchor" href="#结果文件">#</a> 结果文件</h3><pre><code class="language-bash">Gamma_asr.tre ## 每个基因家族的树文件
Gamma_branch_probabilities.tab  ## 每个分支计算的概率
Gamma_category_likelihoods.txt 
Gamma_change.tab ## 每一个基因家族在每个节点的收缩与扩增数目
Gamma_clade_results.txt ##每个节点基因家族的扩增/收缩数目
Gamma_count.txt ## 每一个基因家族在每个节点的数目
Gamma_family_likelihoods.txt
Gamma_family_results.txt ## 基因家族变化的p值和是否显著的结果
Gamma_results.txt ## 模型，最终似然值，最终Lambda值等参数信息。
</code></pre><p>我们主要用的文件有<span class="label primary"> Gamma_asr.tre</span> (主要对应后面表格中的节点)、<span class="label primary">Gamma_change.tab</span> (看哪些基因家族在哪个节点发生变化)、<span class="label primary">Gamma_clade_results.txt</span> (体现在树上，每个节点基因家族的收缩 / 扩增数目)、<span class="label primary">Gamma_family_results.txt</span> (显著扩增 / 收缩的基因家族)</p><h3 id="每个节点基因家族收缩扩增数目的体现"><a class="anchor" href="#每个节点基因家族收缩扩增数目的体现">#</a> 每个节点基因家族收缩 / 扩增数目的体现</h3><p>其实有绘图脚本，但是很久没有更新，可能不适用于 CAFE5，我们可以自己画<br>将基因家族的扩增 / 收缩数目体现在树上，需要两个文件，<span class="label primary">Gamma_asr.tre</span>，<span class="label primary">Gamma_clade_results.txt</span></p><pre><code class="language-bash">cat Gamma_clade_results.txt
#Taxon_ID       Increase        Decrease
Mpo&lt;21&gt; 232     1298
Ppa&lt;20&gt; 2231    371
&lt;31&gt;    134     65
&lt;25&gt;    949     220
&lt;23&gt;    134     209
Atr&lt;13&gt; 516     922
Kle&lt;29&gt; 493     741
&lt;12&gt;    245     56
&lt;28&gt;    314     340
&lt;4&gt;     118     176
Cri&lt;17&gt; 1669    287
&lt;22&gt;    214     184
&lt;19&gt;    445     93
&lt;16&gt;    291     352
Osa&lt;3&gt;  579     572
Aof&lt;6&gt;  935     840
&lt;8&gt;     142     112
Csa&lt;1&gt;  326     834
&lt;7&gt;     640     138
Tpl&lt;15&gt; 1147    395
&lt;14&gt;    204     273
Nco&lt;11&gt; 631     559
Zma&lt;2&gt;  1776    232
Vvi&lt;5&gt;  958     433
&lt;10&gt;    413     112
&lt;9&gt;     345     66
Smo&lt;18&gt; 842     1315
Cba&lt;24&gt; 744     1664
&lt;30&gt;    23      17
Ath&lt;0&gt;  1090    291
Csu&lt;27&gt; 305     1560
Vca&lt;26&gt; 438     1040
</code></pre><pre><code class="language-bash">less Gamma_asr.tre
BEGIN TREES;
  TREE OG0000021 = ((Kle&lt;29&gt;*_2:820.007,(((Mpo&lt;21&gt;*_7:428.285,Ppa&lt;20&gt;*_15:428.285)&lt;23&gt;_12:70.3982,((Cri&lt;17&gt;_30:404.796,(Tpl&lt;15&gt;_31:308.175,(Atr&lt;13&gt;_31:208.47,(Nco&lt;11&gt;*_26:176.909,(((Osa&lt;3&gt;*_49:45.2652,Zma&lt;2&gt;_45:45.2652)&lt;7&gt;_45:60.4652,Aof&lt;6&gt;*_49:105.73)&lt;9&gt;*_44:25.8965,(Vvi&lt;5&gt;_36:101.378,(Csa&lt;1&gt;*_31:83.0358,Ath&lt;0&gt;*_41:83.0358)&lt;4&gt;_36:18.3421)&lt;8&gt;_36:30.2489)&lt;10&gt;*_36:45.2824)&lt;12&gt;_31:31.5605)&lt;14&gt;_31:99.7053)&lt;16&gt;*_30:96.6205)&lt;19&gt;*_27:48.1939,Smo&lt;18&gt;_11:452.99)&lt;22&gt;*_13:45.694)&lt;25&gt;*_12:190.787,Cba&lt;24&gt;_7:689.47)&lt;28&gt;*_7:130.537)&lt;31&gt;_5:177.864,(Csu&lt;27&gt;*_8:874.03,Vca&lt;26&gt;*_0:874.03)&lt;30&gt;_5:123.841)&lt;32&gt;_5
</code></pre><p>可以看到少了个 32<br>我们可以利用<span class="label primary"> Gamma_change.tab</span> 文件去找一下<br>经查看确实没有，忽略这一部分<br>将两个文件的 nodeid 对应即可绘图<br><img data-src="/img/posts/2022.12.14/4.png" alt="基因家族收缩和扩张"></p><h3 id="其他整理"><a class="anchor" href="#其他整理">#</a> 其他整理</h3><p>相较于 CAFE4，这些结果并没有直接体现显著扩张 / 收缩的基因家族，或者我们想找一下某个节点具体扩张的基因，可以结合目前拿到的输出文件进行进一步整理</p><pre><code class="language-bash">cat Gamma_family_results.txt |grep &quot;y&quot;|cut -f1 &gt;p0.05.significant 
#提取显著扩张或收缩的orthogroupsID
grep -f p0.05.significant Gamma_change.tab &gt; Gamma_p0.05change.tab
#显著扩张/收缩的基因家族在每个节点的收缩与扩增数目
cat Gamma_p0.05change.tab | cut -f1,2 | grep &quot;+[1-9]&quot; | cut -f1 &gt; node0significant.expand
#Ath显著扩张的orthogroupsID
wc -l node0significant.expand
#Ath显著扩张的基因家族数目
cp ../../Results_May02/Orthogroups/Orthogroups.tsv ./
grep -f node0significant.expand Orthogroups.tsv | cut -f3 | sed &quot;s/ /\n/g&quot; | sed &quot;s/\t/\n/g&quot; | sed &quot;s/,//g&quot; | sort | uniq &gt; node0significant.expand.genes
#提取Ath显著扩张的基因，方法一
cp ../../Results_May02/Orthogroups/Orthogroups.txt ./
grep -f node0significant.expand Orthogroups.txt |sed &quot;s/ /\n/g&quot; | grep &quot;Ath&quot; | sort | uniq  &gt; node0significant.expand.genes
#提取Ath显著扩张的基因，方法二
</code></pre><h1 id="结语"><a class="anchor" href="#结语">#</a> 结语</h1><p>总体来说，个人感觉 CAFE5 要比 CAFE4 方便很多，但是输出结果的可视化方面还需要加强，下一篇推文将简单实践某个节点的基因功能富集分析 (GO/KEGG), 类似于分析下图的红色节点。<br><img data-src="/img/posts/2022.12.14/5.png" alt="image.png"></p><div class="tags"><a href="/tags/bioinformatics/" rel="tag"><i class="ic i-tag"></i> bioinformatics</a> <a href="/tags/Comparative-Genomics/" rel="tag"><i class="ic i-tag"></i> Comparative_Genomics</a></div></div><footer><div class="meta"><span class="item"><span class="icon"><i class="ic i-calendar-check"></i> </span><span class="text">更新于</span> <time title="修改时间：2022-12-20 10:33:53" itemprop="dateModified" datetime="2022-12-20T10:33:53+08:00">2022-12-20</time> </span><span id="2022/12/14/2022-12-14-CAFE/" class="item leancloud_visitors" data-flag-title="比较基因组学分析 2：基因家族收缩与扩张分析" title="阅读次数"><span class="icon"><i class="ic i-eye"></i> </span><span class="text">阅读次数</span> <span class="leancloud-visitors-count"></span> <span class="text">次</span></span></div><div class="reward"><button><i class="ic i-heartbeat"></i> 赞赏</button><p>请我喝[茶]~(￣▽￣)~*</p><div id="qr"><div><img data-src="/images/wechatpay.png" alt="amane 微信支付"><p>微信支付</p></div><div><img data-src="/images/alipay.png" alt="amane 支付宝"><p>支付宝</p></div></div></div><div id="copyright"><ul><li class="author"><strong>本文作者： </strong>amane <i class="ic i-at"><em>@</em></i>小钊の笔记</li><li class="link"><strong>本文链接：</strong> <a href="https://www.zhouxiaozhao.cn/2022/12/14/2022-12-14-CAFE/" title="比较基因组学分析 2：基因家族收缩与扩张分析">https://www.zhouxiaozhao.cn/2022/12/14/2022-12-14-CAFE/</a></li><li class="license"><strong>版权声明： </strong>本站所有文章除特别声明外，均采用 <span class="exturl" data-url="aHR0cHM6Ly9jcmVhdGl2ZWNvbW1vbnMub3JnL2xpY2Vuc2VzL2J5LW5jLXNhLzQuMC9kZWVkLnpo"><i class="ic i-creative-commons"><em>(CC)</em></i>BY-NC-SA</span> 许可协议。转载请注明出处！</li></ul></div></footer></article></div><div class="post-nav"><div class="item left"><a href="/2022/12/08/2022-12-08-SingleCopy/" itemprop="url" rel="prev" data-background-image="https:&#x2F;&#x2F;img1.imgtp.com&#x2F;2023&#x2F;02&#x2F;01&#x2F;GOGkua2f.jpg" title="比较基因组学分析1：构建单拷贝基因树"><span class="type">上一篇</span> <span class="category"><i class="ic i-flag"></i> 基因组</span><h3>比较基因组学分析1：构建单拷贝基因树</h3></a></div><div class="item right"><a href="/2022/12/15/2022-12-15-GO/" itemprop="url" rel="next" data-background-image="https:&#x2F;&#x2F;img1.imgtp.com&#x2F;2023&#x2F;02&#x2F;01&#x2F;mlFpeQkC.jpg" title="比较基因组学分析3：特异节点基因家族富集分析（非模式物种GO&#x2F;KEEG库构建）"><span class="type">下一篇</span> <span class="category"><i class="ic i-flag"></i> 基因组</span><h3>比较基因组学分析3：特异节点基因家族富集分析（非模式物种GO/KEEG库构建）</h3></a></div></div><div class="wrap" id="comments"></div></div><div id="sidebar"><div class="inner"><div class="panels"><div class="inner"><div class="contents panel pjax" data-title="文章目录"><ol class="toc"><li class="toc-item toc-level-1"><a class="toc-link" href="#%E5%89%8D%E8%A8%80"><span class="toc-number">1.</span> <span class="toc-text">前言</span></a><ol class="toc-child"><li class="toc-item toc-level-2"><a class="toc-link" href="#%E4%B8%8A%E7%AF%87%E6%8E%A8%E6%96%87%E4%B8%AD%E4%BB%8B%E7%BB%8D%E5%88%B0%E6%AF%94%E8%BE%83%E5%9F%BA%E5%9B%A0%E7%BB%84%E5%AD%A6%E5%88%86%E6%9E%90%E5%B8%B8%E7%94%A8%E5%A5%97%E8%B7%AF%E7%9A%84%E7%AC%AC%E4%B8%80%E6%AD%A5%E5%88%A9%E7%94%A8%E5%8D%95%E6%8B%B7%E8%B4%9D%E5%9F%BA%E5%9B%A0%E6%9E%84%E5%BB%BA%E5%85%B7%E6%9C%89%E5%88%86%E5%8C%96%E6%97%B6%E9%97%B4%E7%9A%84%E7%89%A9%E7%A7%8D%E6%A0%91%E8%A1%A5%E5%85%85%E4%B8%80%E7%82%B9%E5%AF%B9%E4%BA%8E%E8%B7%A8%E5%BA%A6%E8%BE%83%E5%A4%A7%E7%9A%84%E7%89%A9%E7%A7%8D%E5%8F%AF%E4%BB%A5%E9%80%89%E6%8B%A9%E5%8D%95%E6%8B%B7%E8%B4%9D%E5%9F%BA%E5%9B%A0%E7%9A%84%E6%96%B9%E6%B3%95%E6%AF%94%E5%A6%82%E6%AD%A4%E6%AC%A1%E5%88%86%E6%9E%90%E4%BD%BF%E7%94%A8%E7%9A%84%E7%89%A9%E7%A7%8D-%E5%AF%B9%E4%BA%8E%E7%9B%AE%E7%BA%A7%E6%88%96%E8%80%85%E7%A7%91%E7%BA%A7%E6%B0%B4%E5%B9%B3%E6%9D%A5%E8%AE%B2%E6%8E%A8%E8%8D%90%E4%BD%BF%E7%94%A8%E5%85%B1%E7%BA%BF%E6%80%A7%E5%9F%BA%E5%9B%A0%E5%BB%BA%E6%A0%91-%E4%BB%A5%E5%8D%81%E5%AD%97%E8%8A%B1%E7%A7%91%E4%B8%BA%E4%BE%8B%E5%A6%82%E6%9E%9C%E7%94%A8%E5%8D%95%E6%8B%B7%E8%B4%9D%E5%9F%BA%E5%9B%A0%E5%8F%AF%E8%83%BD%E5%8F%AA%E6%9C%891000%E5%A4%9A%E7%BB%84%E4%BD%86%E6%98%AF%E4%BD%BF%E7%94%A8%E5%85%B1%E7%BA%BF%E6%80%A7%E5%8F%AF%E8%83%BD%E6%9C%89%E6%8E%A5%E8%BF%915000%E7%BB%84-%E5%85%B1%E7%BA%BF%E6%80%A7%E5%9F%BA%E5%9B%A0%E5%BB%BA%E6%A0%91%E5%8F%AF%E4%BB%A5%E4%BD%BF%E7%94%A8wgdi%E7%9A%84%E6%96%B9%E6%B3%95%E8%BF%99%E9%83%A8%E5%88%86%E5%86%85%E5%AE%B9%E6%88%91%E4%BB%A5%E5%90%8E%E4%BC%9A%E6%8E%A2%E7%B4%A2%E4%B8%80%E4%B8%8B-sun-et-al-2022-wgdi%E6%9C%AC%E7%AF%87%E6%8E%A8%E6%96%87%E4%B8%BB%E8%A6%81%E8%AE%B2%E5%9F%BA%E5%9B%A0%E5%AE%B6%E6%97%8F%E7%9A%84%E6%94%B6%E7%BC%A9%E4%B8%8E%E6%89%A9%E5%BC%A0%E5%88%86%E6%9E%90%E4%BD%BF%E7%94%A8%E7%9A%84%E8%BD%AF%E4%BB%B6%E6%98%AFcafe52020%E5%B9%B4%E5%8F%91%E8%A1%A8%E7%9B%B8%E8%BE%83%E4%BA%8Ecafe4%E6%9D%A5%E8%AE%B2%E6%93%8D%E4%BD%9C%E6%9B%B4%E5%8A%A0%E6%96%B9%E4%BE%BF%E5%B9%B6%E4%B8%94%E6%96%B0%E5%A2%9E%E4%BA%86%E6%A8%A1%E5%9E%8Bgamma"><span class="toc-number">1.1.</span> <span class="toc-text">上篇推文中介绍到比较基因组学分析常用套路的第一步，利用单拷贝基因构建具有分化时间的物种树，补充一点，对于跨度较大的物种，可以选择单拷贝基因的方法，比如此次分析使用的物种。对于目级或者科级水平来讲，推荐使用共线性基因建树。以十字花科为例，如果用单拷贝基因，可能只有 1000 多组，但是使用共线性可能有接近 5000 组。共线性基因建树可以使用 WGDI 的方法，这部分内容我以后会探索一下。 本篇推文主要讲基因家族的收缩与扩张分析，使用的软件是 cafe5，2020 年发表，相较于 cafe4 来讲操作更加方便并且新增了模型 (Gamma)</span></a></li></ol></li><li class="toc-item toc-level-1"><a class="toc-link" href="#%E5%AE%89%E8%A3%85"><span class="toc-number">2.</span> <span class="toc-text">安装</span></a></li><li class="toc-item toc-level-1"><a class="toc-link" href="#cafe5%E4%BD%BF%E7%94%A8"><span class="toc-number">3.</span> <span class="toc-text">CAFE5 使用</span></a><ol class="toc-child"><li class="toc-item toc-level-2"><a class="toc-link" href="#%E4%B8%BB%E8%A6%81%E5%8F%82%E6%95%B0"><span class="toc-number">3.1.</span> <span class="toc-text">主要参数</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#%E8%BE%93%E5%85%A5%E6%96%87%E4%BB%B6%E5%87%86%E5%A4%87"><span class="toc-number">3.2.</span> <span class="toc-text">输入文件准备</span></a><ol class="toc-child"><li class="toc-item toc-level-3"><a class="toc-link" href="#genefamilies_counttsv"><span class="toc-number">3.2.1.</span> <span class="toc-text">Genefamilies_Count.tsv</span></a></li><li class="toc-item toc-level-3"><a class="toc-link" href="#treetxt"><span class="toc-number">3.2.2.</span> <span class="toc-text">tree.txt</span></a></li></ol></li><li class="toc-item toc-level-2"><a class="toc-link" href="#%E8%BF%90%E8%A1%8Ccafe5"><span class="toc-number">3.3.</span> <span class="toc-text">运行 CAFE5</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#cafe5%E8%BE%93%E5%87%BA%E7%BB%93%E6%9E%9C"><span class="toc-number">3.4.</span> <span class="toc-text">CAFE5 输出结果</span></a><ol class="toc-child"><li class="toc-item toc-level-3"><a class="toc-link" href="#%E7%BB%93%E6%9E%9C%E6%96%87%E4%BB%B6"><span class="toc-number">3.4.1.</span> <span class="toc-text">结果文件</span></a></li><li class="toc-item toc-level-3"><a class="toc-link" href="#%E6%AF%8F%E4%B8%AA%E8%8A%82%E7%82%B9%E5%9F%BA%E5%9B%A0%E5%AE%B6%E6%97%8F%E6%94%B6%E7%BC%A9%E6%89%A9%E5%A2%9E%E6%95%B0%E7%9B%AE%E7%9A%84%E4%BD%93%E7%8E%B0"><span class="toc-number">3.4.2.</span> <span class="toc-text">每个节点基因家族收缩 &#x2F; 扩增数目的体现</span></a></li><li class="toc-item toc-level-3"><a class="toc-link" href="#%E5%85%B6%E4%BB%96%E6%95%B4%E7%90%86"><span class="toc-number">3.4.3.</span> <span class="toc-text">其他整理</span></a></li></ol></li></ol></li><li class="toc-item toc-level-1"><a class="toc-link" href="#%E7%BB%93%E8%AF%AD"><span class="toc-number">4.</span> <span class="toc-text">结语</span></a></li></ol></div><div class="related panel pjax" data-title="系列文章"><ul><li><a href="/2020/07/08/2020-07-08-repeat/" rel="bookmark" title="RepeatModeler+RepeatMasker的安装与使用">RepeatModeler+RepeatMasker的安装与使用</a></li><li><a href="/2020/07/18/2020-07-18-orthofinder/" rel="bookmark" title="利用orthofinder寻找单拷贝基因构建系统发育树">利用orthofinder寻找单拷贝基因构建系统发育树</a></li><li><a href="/2022/12/08/2022-12-08-SingleCopy/" rel="bookmark" title="比较基因组学分析1：构建单拷贝基因树">比较基因组学分析1：构建单拷贝基因树</a></li><li class="active"><a href="/2022/12/14/2022-12-14-CAFE/" rel="bookmark" title="比较基因组学分析2：基因家族收缩与扩张分析">比较基因组学分析2：基因家族收缩与扩张分析</a></li><li><a href="/2022/12/15/2022-12-15-GO/" rel="bookmark" title="比较基因组学分析3：特异节点基因家族富集分析（非模式物种GO/KEEG库构建）">比较基因组学分析3：特异节点基因家族富集分析（非模式物种GO/KEEG库构建）</a></li></ul></div><div class="overview panel" data-title="站点概览"><div class="author" itemprop="author" itemscope itemtype="http://schema.org/Person"><img class="image" itemprop="image" alt="amane" data-src="/images/avatar.png"><p class="name" itemprop="name">amane</p><div class="description" itemprop="description">生物学在读博士</div></div><nav class="state"><div class="item posts"><a href="/archives/"><span class="count">84</span> <span class="name">文章</span></a></div><div class="item categories"><a href="/categories/"><span class="count">7</span> <span class="name">分类</span></a></div><div class="item tags"><a href="/tags/"><span class="count">21</span> <span class="name">标签</span></a></div></nav><div class="social"><span class="exturl item github" data-url="aHR0cHM6Ly9naXRodWIuY29tL0Jpb2luZm9ybWF0aWNzLXJvb2tpZQ==" title="https:&#x2F;&#x2F;github.com&#x2F;Bioinformatics-rookie"><i class="ic i-github"></i></span> <a href="https://www.zhouxiaozhao.cn/about/" title="https:&#x2F;&#x2F;www.zhouxiaozhao.cn&#x2F;about&#x2F;" class="item about"><i class="ic i-address-card"></i></a> <span class="exturl item email" data-url="bWFpbHRvOnpob3VzaGl6aGFvQG1haWwua2liLmFjLmNu" title="mailto:zhoushizhao@mail.kib.ac.cn"><i class="ic i-envelope"></i></span></div><ul class="menu"><li class="item"><a href="/" rel="section"><i class="ic i-home"></i>首页</a></li><li class="item"><a href="/about/" rel="section"><i class="ic i-user"></i>关于</a></li><li class="item dropdown"><a href="javascript:void(0);"><i class="ic i-feather"></i>文章</a><ul class="submenu"><li class="item"><a href="/archives/" rel="section"><i class="ic i-list-alt"></i>归档</a></li><li class="item"><a href="/categories/" rel="section"><i class="ic i-th"></i>分类</a></li><li class="item"><a href="/tags/" rel="section"><i class="ic i-tags"></i>标签</a></li></ul></li><li class="item"><a href="/links/" rel="section"><i class="ic i-heart"></i>友達</a></li></ul></div></div></div><ul id="quick"><li class="prev pjax"><a href="/2022/12/08/2022-12-08-SingleCopy/" rel="prev" title="上一篇"><i class="ic i-chevron-left"></i></a></li><li class="up"><i class="ic i-arrow-up"></i></li><li class="down"><i class="ic i-arrow-down"></i></li><li class="next pjax"><a href="/2022/12/15/2022-12-15-GO/" rel="next" title="下一篇"><i class="ic i-chevron-right"></i></a></li><li class="percent"></li></ul></div></div><div class="dimmer"></div></div></main><footer id="footer"><div class="inner"><div class="widgets"><div class="rpost pjax"><h2>随机文章</h2><ul><li class="item"><div class="breadcrumb"><a href="/categories/Bioinformation/" title="分类于 生信学习">生信学习</a></div><span><a href="/2020/12/19/2020-12-19-RNAseq(11)/" title="RNAseq教程(2.5)">RNAseq教程(2.5)</a></span></li><li class="item"><div class="breadcrumb"><a href="/categories/Python/" title="分类于 Python">Python</a></div><span><a href="/2020/09/10/2020-09-10-python1/" title="马士兵教育Python入门基础:1.python入门（一）">马士兵教育Python入门基础:1.python入门（一）</a></span></li><li class="item"><div class="breadcrumb"><a href="/categories/Bioinformation/" title="分类于 生信学习">生信学习</a> <i class="ic i-angle-right"></i> <a href="/categories/Bioinformation/SingleCell/" title="分类于 单细胞">单细胞</a></div><span><a href="/2023/02/22/2023-02-22-ST(1)/" title="植物空间转录组分析1：Seurat基本流程">植物空间转录组分析1：Seurat基本流程</a></span></li><li class="item"><div class="breadcrumb"><a href="/categories/paper/" title="分类于 文献导读">文献导读</a></div><span><a href="/2020/09/22/2020-09-22-Cool_your_github/" title="炫酷你的Github主页">炫酷你的Github主页</a></span></li><li class="item"><div class="breadcrumb"><a href="/categories/Bioinformation/" title="分类于 生信学习">生信学习</a></div><span><a href="/2020/12/22/2020-12-22-RNAseq(12)/" title="RNAseq教程(3.1)">RNAseq教程(3.1)</a></span></li><li class="item"><div class="breadcrumb"><a href="/categories/Python/" title="分类于 Python">Python</a></div><span><a href="/2020/10/20/2020-10-20-python8/" title="马士兵教育Python入门基础:4.流程控制（2）">马士兵教育Python入门基础:4.流程控制（2）</a></span></li><li class="item"><div class="breadcrumb"><a href="/categories/Bioinformation/" title="分类于 生信学习">生信学习</a></div><span><a href="/2020/12/05/2020-12-05-RNAseq(5)/" title="RNAseq教程(1.5)">RNAseq教程(1.5)</a></span></li><li class="item"><div class="breadcrumb"><a href="/categories/Bioinformation/" title="分类于 生信学习">生信学习</a> <i class="ic i-angle-right"></i> <a href="/categories/Bioinformation/SingleCell/" title="分类于 单细胞">单细胞</a></div><span><a href="/2020/11/10/2020-11-10-RNAvelocity(1)/" title="RNA速率：软件下载与loom文件准备">RNA速率：软件下载与loom文件准备</a></span></li><li class="item"><div class="breadcrumb"><a href="/categories/paper/" title="分类于 文献导读">文献导读</a></div><span><a href="/2020/07/27/2020-07-27-Symbiotic/" title="文献导读:菌虫共生真菌测序">文献导读:菌虫共生真菌测序</a></span></li><li class="item"><div class="breadcrumb"><a href="/categories/paper/" title="分类于 文献导读">文献导读</a></div><span><a href="/2020/07/30/2020-07-30-HGT/" title="文献导读:真菌和陆栖藻类水平基因转移">文献导读:真菌和陆栖藻类水平基因转移</a></span></li></ul></div><div><h2>最新评论</h2><ul class="leancloud-recent-comment"></ul></div></div><div class="status"><div class="copyright">&copy; 2010 – <span itemprop="copyrightYear">2023</span> <span class="with-love"><i class="ic i-sakura rotate"></i> </span><span class="author" itemprop="copyrightHolder">amane @ Zhou Xiaozhao</span></div><div class="count"><span class="post-meta-item-icon"><i class="ic i-chart-area"></i> </span><span title="站点总字数">351k 字</span> <span class="post-meta-divider">|</span> <span class="post-meta-item-icon"><i class="ic i-coffee"></i> </span><span title="站点阅读时长">5:19</span></div><div class="powered-by">基于 <span class="exturl" data-url="aHR0cHM6Ly9oZXhvLmlv">Hexo</span> & Theme.<span class="exturl" data-url="aHR0cHM6Ly9naXRodWIuY29tL2FtZWhpbWUvaGV4by10aGVtZS1zaG9rYQ==">Shoka</span></div></div></div></footer></div><script data-config type="text/javascript">var LOCAL={path:"2022/12/14/2022-12-14-CAFE/",favicon:{show:"（●´3｀●）やれやれだぜ",hide:"(´Д｀)大変だ！"},search:{placeholder:"文章搜索",empty:"关于 「 ${query} 」，什么也没搜到",stats:"${time} ms 内找到 ${hits} 条结果"},valine:!0,fancybox:!0,copyright:'复制成功，转载请遵守 <i class="ic i-creative-commons"></i>BY-NC-SA 协议。',ignores:[function(e){return e.includes("#")},function(e){return new RegExp(LOCAL.path+"$").test(e)}]}</script><script src="https://cdn.polyfill.io/v2/polyfill.js"></script><script src="//cdn.jsdelivr.net/combine/npm/pace-js@1.0.2/pace.min.js,npm/pjax@0.2.8/pjax.min.js,npm/whatwg-fetch@3.4.0/dist/fetch.umd.min.js,npm/animejs@3.2.0/lib/anime.min.js,npm/algoliasearch@4/dist/algoliasearch-lite.umd.js,npm/instantsearch.js@4/dist/instantsearch.production.min.js,npm/lozad@1/dist/lozad.min.js,npm/quicklink@2/dist/quicklink.umd.js"></script><script src="/js/app.js?v=0.2.5"></script></body></html><!-- rebuild by hrmmi -->