<!-- build time:Wed Feb 01 2023 15:15:28 GMT+0800 (中国标准时间) --><!DOCTYPE html><html lang="zh-CN"><head><meta charset="UTF-8"><meta name="viewport" content="width=device-width,initial-scale=1,maximum-scale=2"><meta name="theme-color" content="#FFF"><link rel="apple-touch-icon" sizes="180x180" href="/images/apple-touch-icon.png"><link rel="icon" type="image/ico" sizes="32x32" href="/images/favicon.ico"><meta http-equiv="Cache-Control" content="no-transform"><meta http-equiv="Cache-Control" content="no-siteapp"><link rel="alternate" type="application/rss+xml" title="小钊の笔记" href="http://example.com/rss.xml"><link rel="alternate" type="application/atom+xml" title="小钊の笔记" href="http://example.com/atom.xml"><link rel="alternate" type="application/json" title="小钊の笔记" href="http://example.com/feed.json"><link rel="stylesheet" href="//fonts.googleapis.com/css?family=Mulish:300,300italic,400,400italic,700,700italic%7CFredericka%20the%20Great:300,300italic,400,400italic,700,700italic%7CNoto%20Serif%20JP:300,300italic,400,400italic,700,700italic%7CNoto%20Serif%20SC:300,300italic,400,400italic,700,700italic%7CInconsolata:300,300italic,400,400italic,700,700italic&display=swap&subset=latin,latin-ext"><link rel="stylesheet" href="/css/app.css?v=0.2.5"><meta name="keywords" content="bioinformatics,ScRNAseq"><link rel="canonical" href="http://example.com/2020/10/03/2020-10-03-ScRNAseq(3)/"><title>单细胞实战 (3)：STAR 分析单细胞数据 - 生信学习 | Zhou Xiaozhao = 小钊の笔记 = 前天是小兔子，昨天是小鹿，今天是你</title><meta name="generator" content="Hexo 5.4.1"></head><body itemscope itemtype="http://schema.org/WebPage"><div id="loading"><div class="cat"><div class="body"></div><div class="head"><div class="face"></div></div><div class="foot"><div class="tummy-end"></div><div class="bottom"></div><div class="legs left"></div><div class="legs right"></div></div><div class="paw"><div class="hands left"></div><div class="hands right"></div></div></div></div><div id="container"><header id="header" itemscope itemtype="http://schema.org/WPHeader"><div class="inner"><div id="brand"><div class="pjax"><h1 itemprop="name headline">单细胞实战 (3)：STAR 分析单细胞数据</h1><div class="meta"><span class="item" title="创建时间：2020-10-03 18:38:45"><span class="icon"><i class="ic i-calendar"></i> </span><span class="text">发表于</span> <time itemprop="dateCreated datePublished" datetime="2020-10-03T18:38:45+08:00">2020-10-03</time> </span><span class="item" title="本文字数"><span class="icon"><i class="ic i-pen"></i> </span><span class="text">本文字数</span> <span>6.7k</span> <span class="text">字</span> </span><span class="item" title="阅读时长"><span class="icon"><i class="ic i-clock"></i> </span><span class="text">阅读时长</span> <span>6 分钟</span></span></div></div></div><nav id="nav"><div class="inner"><div class="toggle"><div class="lines" aria-label="切换导航栏"><span class="line"></span> <span class="line"></span> <span class="line"></span></div></div><ul class="menu"><li class="item title"><a href="/" rel="start">Zhou Xiaozhao</a></li></ul><ul class="right"><li class="item theme"><i class="ic i-sun"></i></li><li class="item search"><i class="ic i-search"></i></li></ul></div></nav></div><div id="imgs" class="pjax"><ul><li class="item" data-background-image="https://img1.imgtp.com/2023/02/01/ID2PmCx1.jpg"></li><li class="item" data-background-image="https://img1.imgtp.com/2023/02/01/UJhRWazi.jpg"></li><li class="item" data-background-image="https://img1.imgtp.com/2023/02/01/mlFpeQkC.jpg"></li><li class="item" data-background-image="https://img1.imgtp.com/2023/02/01/i9tPHlaZ.jpg"></li><li class="item" data-background-image="https://img1.imgtp.com/2023/02/01/xioKAX7K.jpg"></li><li class="item" data-background-image="https://img1.imgtp.com/2023/02/01/uNH0Qiaw.jpg"></li></ul></div></header><div id="waves"><svg class="waves" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" viewBox="0 24 150 28" preserveAspectRatio="none" shape-rendering="auto"><defs><path id="gentle-wave" d="M-160 44c30 0 58-18 88-18s 58 18 88 18 58-18 88-18 58 18 88 18 v44h-352z"/></defs><g class="parallax"><use xlink:href="#gentle-wave" x="48" y="0"/><use xlink:href="#gentle-wave" x="48" y="3"/><use xlink:href="#gentle-wave" x="48" y="5"/><use xlink:href="#gentle-wave" x="48" y="7"/></g></svg></div><main><div class="inner"><div id="main" class="pjax"><div class="article wrap"><div class="breadcrumb" itemscope itemtype="https://schema.org/BreadcrumbList"><i class="ic i-home"></i> <span><a href="/">首页</a></span><i class="ic i-angle-right"></i> <span class="current" itemprop="itemListElement" itemscope itemtype="https://schema.org/ListItem"><a href="/categories/Bioinformation/" itemprop="item" rel="index" title="分类于 生信学习"><span itemprop="name">生信学习</span></a><meta itemprop="position" content="1"></span></div><article itemscope itemtype="http://schema.org/Article" class="post block" lang="zh-CN"><link itemprop="mainEntityOfPage" href="http://example.com/2020/10/03/2020-10-03-ScRNAseq(3)/"><span hidden itemprop="author" itemscope itemtype="http://schema.org/Person"><meta itemprop="image" content="/images/avatar.png"><meta itemprop="name" content="kuchi Shū"><meta itemprop="description" content="前天是小兔子，昨天是小鹿，今天是你, 生物学在读博士"></span><span hidden itemprop="publisher" itemscope itemtype="http://schema.org/Organization"><meta itemprop="name" content="小钊の笔记"></span><div class="body md" itemprop="articleBody"><h2 id="前言"><a class="anchor" href="#前言">#</a> 前言</h2><p>在利用 cellranger 比对单细胞 reads 时，可以发现有 STAR 的进程夹杂在里面，那么 STAR 可以用来比对单细胞数据吗？在 STAR 的 2.7 版本中 (2.7.6a) 出现了 STARsolo，可以进行单细胞数据的比对，由此可见 STAR 的强大</p><h2 id="cellranger输出结果"><a class="anchor" href="#cellranger输出结果">#</a> Cellranger 输出结果</h2><p>在使用 STAR 之前，先看一下 cellranger 的输出结果</p><pre><code>.
├── analysis
│   ├── clustering
│   ├── diffexp
│   ├── pca
│   ├── tsne
│   └── umap
├── cloupe.cloupe
├── filtered_feature_bc_matrix
│   ├── barcodes.tsv.gz
│   ├── features.tsv.gz
│   └── matrix.mtx.gz
├── filtered_feature_bc_matrix.h5
├── metrics_summary.csv
├── molecule_info.h5
├── possorted_genome_bam.bam
├── possorted_genome_bam.bam.bai
├── raw_feature_bc_matrix
│   ├── barcodes.tsv.gz
│   ├── features.tsv.gz
│   └── matrix.mtx.gz
├── raw_feature_bc_matrix.h5
└── web_summary.html

</code></pre><p>为方便查看，cellranger 提供了一个网页端的结果，我们主要观察细胞和基因数目的评估即可，后续的聚类工作由 seurat 完成</p><p><img data-src="/img/posts/2020.10.3/image-20201006123448015.png" alt="image-20201006123448015"></p><p>在结果目录，可以看到如下两个目录</p><p>raw_feature_bc_matrix</p><p>filtered_gene_bc_matrices</p><p>raw 目录下是所有的 barcode 信息，包含了细胞相关的 barcoed 和背景 barcode, 而 filter 目录下只包含细胞相关的 barcode 信息，内容如下</p><pre><code>│   ├── barcodes.tsv.gz
│   ├── features.tsv.gz
│   └── matrix.mtx.gz
</code></pre><p>后缀为 mtx 的文件记录的就是基因的表达量信息，可以导入 R 或者 python 中查看，barcodes 相当于一个细胞，features 代表不同的基因，barcodes 文件在 STARsolo 中会用到，这就是为什么我要先说明一下 cellranger 的输出结果</p><h2 id="利用star分析10x数据"><a class="anchor" href="#利用star分析10x数据">#</a> 利用 STAR 分析 10X 数据</h2><p>STARsolo 被设计为替代 10X CellRanger 基因定量比对软件。而且 STARsolo 的速度是 cellranger 的十倍（具体怎么样我也不知道）</p><p>建立索引：</p><pre><code>STAR --runMode genomeGenerate --genomeDir ghg38/ --genomeFastaFiles Homo_sapiens.GRCh38.dna.primary_assembly.fa --sjdbGTFfile Homo_sapiens.GRCh38.93.filtered.gtf
</code></pre><p>其实利用 cellranger 构建的索引原则上也能用，在 GRCh38/star/ 下，但是由于 STAR 版本问题可能会导致无法识别，因为 cellranger 用的 STAR 构建的索引和我们自己用的 STAR 版本是不一致的。</p><p>STARsolo 与普通的转录组比对区别在于你需要在比对时加上 whitelist，<a href="(https://kb.10xgenomics.com/hc/en-us/articles/115004506263-What-is-a-barcode-whitelist-)">whitelist</a> 文件格式在 10X 官网有写，我们可以利用 cellranger 的 barcodes.tsv.gz 文件获得</p><pre><code>zcat barcodes.tsv.gz&gt;whitelist
sed -i &quot;s\-1\\g&quot; whitelist
</code></pre><p>需要注意 ReadFilesIn 先读入测序数据，再读入 barcode+UMI 文件，即先读入 R2 再读入 R1</p><pre><code>STAR  --genomeDir ghg38/ --readFilesCommand zcat --readFilesIn SRR7722939/SRR7722939_S1_L001_R2_001.fastq.gz SRR7722939/SRR7722939_S1_L001_R1_001.fastq.gz --soloType CB_UMI_Simple --soloCBwhitelist whitelist --runThreadN 8
</code></pre><p>结果默认保存在 Solo.out 文件中，8 线程只用了 10min 左右，确实要快一点</p><pre><code>├── Barcodes.stats
└── Gene
    ├── Features.stats
    ├── filtered
    │   ├── barcodes.tsv
    │   ├── features.tsv
    │   └── matrix.mtx
    ├── raw
    │   ├── barcodes.tsv
    │   ├── features.tsv
    │   └── matrix.mtx
    ├── Summary.csv
    └── UMIperCellSorted.txt
</code></pre><p>看一下 Summary 里是啥</p><pre><code>Number of Reads,23095815
Reads With Valid Barcodes,0.979732
Sequencing Saturation,0.529367
Q30 Bases in CB+UMI,0.991351
Q30 Bases in RNA read,0.842989
Reads Mapped to Genome: Unique+Multiple,0.959769
Reads Mapped to Genome: Unique,0.874368
Reads Mapped to Transcriptome: Unique+Multipe Genes,0.63797
Reads Mapped to Transcriptome: Unique Genes,0.613655
Estimated Number of Cells,2048
Reads in Cells Mapped to Unique Genes,12815372
Fraction of Reads in Cells,0.904219
Mean Reads per Cell,6257
Median Reads per Cell,5544
UMIs in Cells,5978718
Mean UMI per Cell,2919
Median UMI per Cell,2577
Mean Genes per Cell,915
Median Genes per Cell,873
Total Genes Detected,16265
</code></pre><p>可以看到比起 cellranger，STAR 捕获到的细胞数少一点，而且每个细胞的 reads 要低一点，其他差不多，后续将使用 Seurat 包对两组数据进行比较</p><h2 id="starsolo其他命令"><a class="anchor" href="#starsolo其他命令">#</a> STARsolo 其他命令</h2><p><strong>--soloType</strong><br>default: None<br>string(s): type of single-cell RNA-seq<br><code>CB_UMI_Simple</code><br>(a.k.a. Droplet) one UMI and one Cell Barcode ofxed length in<br>read2, e.g. Drop-seq and 10X Chromium.<br><code>CB_UMI_Complex</code><br>one UMI ofxed length, but multiple Cell Barcodes of varying length,<br>as well as adapters sequences are allowed in read2 only, e.g. inDrop.<br><code>CB_samTagOut</code><br>output Cell Barcode as CR and/or CB SAm tag. No UMI counting.<br>{readFilesIn cDNA read1 [cDNA read2 if paired-end]<br>CellBarcode read . Requires {outSAMtype BAM Unsorted [and/or<br>SortedByCoordinate]<br><code>SmartSeq</code><br>Smart-seq: each cell in a separate FASTQ (paired- or single-end),<br>barcodes are corresponding read-groups, no UMI sequences,<br>alignments deduplicated according to alignment start and end (after<br>extending soft-clipped bases)</p><p><strong>--soloCBwhitelist</strong><br>default: -<br>string(s):le(s) with whitelist(s) of cell barcodes. Only {soloType<br>CB UMI Complex allows more than one whitelistle.<br>None<br>no whitelist: all cell barcodes are allowed</p><p><strong>--soloCBstart</strong><br>default: 1<br>int&gt;0: cell barcode start base</p><p><strong>--soloCBlen</strong><br>default: 16<br>int&gt;0: cell barcode length<br><strong>--soloUMIstart</strong><br>default: 17<br>int&gt;0: UMI start base</p><p><strong>--soloUMIlen</strong><br>default: 10<br>int&gt;0: UMI length</p><p><strong>--soloBarcodeReadLength</strong><br>default: 1<br>int: length of the barcode read<br>1<br>equal to sum of soloCBlen+soloUMIlen<br>0<br>not dened, do not check<br><strong>--soloCBposition</strong><br>default: -<br>strings(s) position of Cell Barcode(s) on the barcode read.<br>Presently only works with {soloType CB UMI Complex, and barcodes are<br>assumed to be on Read2.<br>Format for each barcode: startAnchor startPosition endAnchor endPosition<br>start(end)Anchor denes the Anchor Base for the CB: 0: read start; 1: read<br>end; 2: adapter start; 3: adapter end<br>start(end)Position is the 0-based position with of the CB start(end) with<br>respect to the Anchor Base<br>String for dierent barcodes are separated by space.<br>Example: inDrop (Zilionis et al, Nat. Protocols, 2017):<br>{soloCBposition 0 0 2 -1 3 1 3 8<br><strong>--soloUMIposition</strong><br>default: -<br>string position of the UMI on the barcode read, same as soloCBposition<br><strong>--soloAdapterSequence</strong><br>default: -<br>string: adapter sequence to anchor barcodes.<br>--<strong>soloAdapterMismatchesNmax</strong></p><p>default: 1<br>int&gt;0: maximum number of mismatches allowed in adapter sequence.<br><strong>--soloCBmatchWLtype</strong><br>default: 1MM multi<br>string: matching the Cell Barcodes to the WhiteList<br>Exact<br>only exact matches allowed<br><code>1MM</code><br>only one match in whitelist with 1 mismatched base allowed. Allowed<br>CBs have to have at least one read with exact match.<br><code>1MM_multi</code><br>multiple matches in whitelist with 1 mismatched base allowed,<br>posterior probability calculation is used choose one of the matches.<br>Allowed CBs have to have at least one read with exact match. Similar to<br>CellRanger 2.2.0<br><code>1MM_multi_pseudocounts</code><br>same as 1MM Multi, but pseudocounts of 1 are added to all whitelist<br>barcodes.<br>Similar to CellRanger 3.x.x<br><strong>--soloStrand</strong><br>default: Forward<br>string: strandedness of the solo libraries:<br>Unstranded<br>no strand information<br>Forward<br>read strand same as the original RNA molecule<br>Reverse<br>read strand opposite to the original RNA molecule<br><strong>--soloFeatures</strong><br>default: Gene<br>string(s): genomic features for which the UMI counts per Cell Barcode are<br>collected</p><p><code>Gene</code><br>genes: reads match the gene transcript<br><code>SJ</code><br>splice junctions: reported in SJ.out.tab<br><code>GeneFull</code><br>full genes: count all reads overlapping genes' exons and introns<br><strong>--soloUMIdedup</strong><br>default: 1MM_All<br>string(s): type of UMI deduplication (collapsing) algorithm<br><code>1MM_All</code><br>all UMIs with 1 mismatch distance to each other are collapsed (i.e.<br>counted once)<br><code>1MM_Directional</code><br>follows the &quot;directional&quot; method from the UMI-tools by Smith, Heger<br>and Sudbery (Genome Research 2017).<br><code>Exact</code><br>only exactly matching UMIs are collapsed<br><code>NoDedup</code><br>no deduplication of UMIs, count all reads. Allowed for --soloType<br>SmartSeq</p><p><strong>--soloUMIfiltering</strong><br>default: -<br>string(s) type of UMIltering<br><code>-</code></p><p>basicltering: remove UMIs with N and homopolymers (similar to<br>CellRanger 2.2.0)<br><code>MultiGeneUMI</code><br>remove lower-count UMIs that map to more than one gene<br>(introduced in CellRanger 3.x.x)</p><p><strong>--soloOutFileNames</strong><br>default: Solo.out/ features.tsv barcodes.tsv matrix.mtx<br>string(s)le names for STARsolo output:<br>le name prex gene names barcode sequences cell feature count matrix</p><p><strong>--soloCellFilter</strong><br>default: CellRanger2.2 3000 0.99 10</p><p>string(s): cellltering type and parameters<br><code>CellRanger2.2</code><br>simpleltering of CellRanger 2.2, followed by three numbers: number<br>of expected cells, robust maximum percentile for UMI count,<br>maximum to minimum ratio for UMI count<br><code>TopCells</code><br>only report top cells by UMI count, followed by the exact number of<br>cells<br><code>None</code><br>do not outputltered cells</p><p><strong>--soloOutFormatFeaturesGeneField3</strong><br>default: &quot;Gene Expression&quot;<br>string(s):eld 3 in the Gene features.tsv le. If &quot;-&quot;, then no 3rdld is output.</p><div class="tags"><a href="/tags/bioinformatics/" rel="tag"><i class="ic i-tag"></i> bioinformatics</a> <a href="/tags/ScRNAseq/" rel="tag"><i class="ic i-tag"></i> ScRNAseq</a></div></div><footer><div class="meta"><span class="item"><span class="icon"><i class="ic i-calendar-check"></i> </span><span class="text">更新于</span> <time title="修改时间：2022-12-14 10:55:44" itemprop="dateModified" datetime="2022-12-14T10:55:44+08:00">2022-12-14</time> </span><span id="2020/10/03/2020-10-03-ScRNAseq(3)/" class="item leancloud_visitors" data-flag-title="单细胞实战 (3)：STAR 分析单细胞数据" title="阅读次数"><span class="icon"><i class="ic i-eye"></i> </span><span class="text">阅读次数</span> <span class="leancloud-visitors-count"></span> <span class="text">次</span></span></div><div class="reward"><button><i class="ic i-heartbeat"></i> 赞赏</button><p>请我喝[茶]~(￣▽￣)~*</p><div id="qr"><div><img data-src="/images/wechatpay.png" alt="kuchi Shū 微信支付"><p>微信支付</p></div><div><img data-src="/images/alipay.png" alt="kuchi Shū 支付宝"><p>支付宝</p></div></div></div><div id="copyright"><ul><li class="author"><strong>本文作者： </strong>kuchi Shū <i class="ic i-at"><em>@</em></i>小钊の笔记</li><li class="link"><strong>本文链接：</strong> <a href="http://example.com/2020/10/03/2020-10-03-ScRNAseq(3)/" title="单细胞实战 (3)：STAR 分析单细胞数据">http://example.com/2020/10/03/2020-10-03-ScRNAseq(3)/</a></li><li class="license"><strong>版权声明： </strong>本站所有文章除特别声明外，均采用 <span class="exturl" data-url="aHR0cHM6Ly9jcmVhdGl2ZWNvbW1vbnMub3JnL2xpY2Vuc2VzL2J5LW5jLXNhLzQuMC9kZWVkLnpo"><i class="ic i-creative-commons"><em>(CC)</em></i>BY-NC-SA</span> 许可协议。转载请注明出处！</li></ul></div></footer></article></div><div class="post-nav"><div class="item left"><a href="/2020/10/01/2020-10-01-ScRNAseq(2)/" itemprop="url" rel="prev" data-background-image="https:&#x2F;&#x2F;img1.imgtp.com&#x2F;2023&#x2F;02&#x2F;01&#x2F;C6np4Aob.jpg" title="单细胞实战(2)：cellranger使用"><span class="type">上一篇</span> <span class="category"><i class="ic i-flag"></i> 生信学习</span><h3>单细胞实战(2)：cellranger使用</h3></a></div><div class="item right"><a href="/2020/10/06/2020-10-06-ScRNAseq(4)/" itemprop="url" rel="next" data-background-image="https:&#x2F;&#x2F;img1.imgtp.com&#x2F;2023&#x2F;02&#x2F;01&#x2F;QRWxkJxN.jpg" title="单细胞实战(4)：STAR与cellranger结果比较"><span class="type">下一篇</span> <span class="category"><i class="ic i-flag"></i> 生信学习</span><h3>单细胞实战(4)：STAR与cellranger结果比较</h3></a></div></div><div class="wrap" id="comments"></div></div><div id="sidebar"><div class="inner"><div class="panels"><div class="inner"><div class="contents panel pjax" data-title="文章目录"><ol class="toc"><li class="toc-item toc-level-2"><a class="toc-link" href="#%E5%89%8D%E8%A8%80"><span class="toc-number">1.</span> <span class="toc-text">前言</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#cellranger%E8%BE%93%E5%87%BA%E7%BB%93%E6%9E%9C"><span class="toc-number">2.</span> <span class="toc-text">Cellranger 输出结果</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#%E5%88%A9%E7%94%A8star%E5%88%86%E6%9E%9010x%E6%95%B0%E6%8D%AE"><span class="toc-number">3.</span> <span class="toc-text">利用 STAR 分析 10X 数据</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#starsolo%E5%85%B6%E4%BB%96%E5%91%BD%E4%BB%A4"><span class="toc-number">4.</span> <span class="toc-text">STARsolo 其他命令</span></a></li></ol></div><div class="related panel pjax" data-title="系列文章"><ul><li><a href="/2020/07/08/2020-07-08-repeat/" rel="bookmark" title="RepeatModeler+RepeatMasker的安装与使用">RepeatModeler+RepeatMasker的安装与使用</a></li><li><a href="/2020/07/09/2020-07-09-github+jekyll/" rel="bookmark" title="利用Github+Jekyll构建自己的博客">利用Github+Jekyll构建自己的博客</a></li><li><a href="/2020/07/18/2020-07-18-orthofinder/" rel="bookmark" title="利用orthofinder寻找单拷贝基因构建系统发育树">利用orthofinder寻找单拷贝基因构建系统发育树</a></li><li><a href="/2020/07/20/2020-07-20-fungi/" rel="bookmark" title="文献导读：真菌进化">文献导读：真菌进化</a></li><li><a href="/2020/08/03/2020-08-03-WGCNA1/" rel="bookmark" title="WGCNA学习：WGCNA分析原理">WGCNA学习：WGCNA分析原理</a></li><li><a href="/2020/08/05/2020-08-05-WGCNA2/" rel="bookmark" title="WGCNA学习：WGCNA应用场景与实例解读">WGCNA学习：WGCNA应用场景与实例解读</a></li><li><a href="/2020/08/08/2020-08-08-WGCNA3/" rel="bookmark" title="WGCNA学习：WGCNA分析实战">WGCNA学习：WGCNA分析实战</a></li><li><a href="/2020/08/10/2020-08-10-cytoscape/" rel="bookmark" title="基因共表达网络图的绘制">基因共表达网络图的绘制</a></li><li><a href="/2020/08/18/2020-08-18-sys/" rel="bookmark" title="根据转录组数据鉴定共生物种">根据转录组数据鉴定共生物种</a></li><li><a href="/2020/08/20/2020-08-20-nt/" rel="bookmark" title="使用本地nt数据库对reads和Trinity结果进行blast">使用本地nt数据库对reads和Trinity结果进行blast</a></li><li><a href="/2020/08/23/2020-08-23-IGV/" rel="bookmark" title="使用IGV简单绘制甲基化分布图">使用IGV简单绘制甲基化分布图</a></li><li><a href="/2020/08/25/2020-08-25-evolview1/" rel="bookmark" title="进化树美化之Evolview--热图（1）">进化树美化之Evolview--热图（1）</a></li><li><a href="/2020/08/28/2020-08-28-evolview/" rel="bookmark" title="进化树美化之Evolview--热图（2）">进化树美化之Evolview--热图（2）</a></li><li><a href="/2020/08/30/2020-08-30-evolview3/" rel="bookmark" title="进化树+蛋白结构域">进化树+蛋白结构域</a></li><li><a href="/2020/09/01/2020-09-01-iTOL1/" rel="bookmark" title="进化树在线制作工具iTOL教程（1）：基本操作">进化树在线制作工具iTOL教程（1）：基本操作</a></li><li><a href="/2020/09/03/2020-09-03-iTOL2/" rel="bookmark" title="进化树在线制作工具iTOL教程（2）：添加文本">进化树在线制作工具iTOL教程（2）：添加文本</a></li><li><a href="/2020/09/06/2020-09-06-iTOL3/" rel="bookmark" title="进化树在线制作工具iTOL教程（3）：添加图片">进化树在线制作工具iTOL教程（3）：添加图片</a></li><li><a href="/2020/09/08/2020-09-08-iTOL4/" rel="bookmark" title="进化树在线制作工具iTOL教程（4）：添加点图">进化树在线制作工具iTOL教程（4）：添加点图</a></li><li><a href="/2020/09/29/2020-09-29-ScRNAseq(1)/" rel="bookmark" title="单细胞实战(1)：原始测序数据下载">单细胞实战(1)：原始测序数据下载</a></li><li><a href="/2020/10/01/2020-10-01-ScRNAseq(2)/" rel="bookmark" title="单细胞实战(2)：cellranger使用">单细胞实战(2)：cellranger使用</a></li><li class="active"><a href="/2020/10/03/2020-10-03-ScRNAseq(3)/" rel="bookmark" title="单细胞实战(3)：STAR分析单细胞数据">单细胞实战(3)：STAR分析单细胞数据</a></li><li><a href="/2020/10/06/2020-10-06-ScRNAseq(4)/" rel="bookmark" title="单细胞实战(4)：STAR与cellranger结果比较">单细胞实战(4)：STAR与cellranger结果比较</a></li><li><a href="/2020/10/10/2020-10-10-ScRNAseq(5)/" rel="bookmark" title="单细胞实战(5)：复现文章中的聚类图（1）">单细胞实战(5)：复现文章中的聚类图（1）</a></li><li><a href="/2020/10/13/2020-10-13-ScRNAseq(6)/" rel="bookmark" title="单细胞实战(5)：复现文章中的聚类图（2）">单细胞实战(5)：复现文章中的聚类图（2）</a></li><li><a href="/2020/10/24/2020-10-24-ScRNAseq(7)/" rel="bookmark" title="生信技能树单细胞数据挖掘笔记(1)">生信技能树单细胞数据挖掘笔记(1)</a></li><li><a href="/2020/10/27/2020-10-27-ScRNAseq(8)/" rel="bookmark" title="生信技能树单细胞数据挖掘笔记(2)">生信技能树单细胞数据挖掘笔记(2)</a></li><li><a href="/2020/10/29/2020-10-29-ScRNAseq(9)/" rel="bookmark" title="生信技能树单细胞数据挖掘笔记(3)">生信技能树单细胞数据挖掘笔记(3)</a></li><li><a href="/2020/10/31/2020-10-31-ScRNAseq(10)/" rel="bookmark" title="生信技能树单细胞数据挖掘笔记(4)">生信技能树单细胞数据挖掘笔记(4)</a></li><li><a href="/2020/11/03/2020-11-03-ScRNAseq(11)/" rel="bookmark" title="生信技能树单细胞数据挖掘笔记(5)">生信技能树单细胞数据挖掘笔记(5)</a></li><li><a href="/2020/11/05/2020-11-05-ScRNAseq(12)/" rel="bookmark" title="生信技能树单细胞数据挖掘笔记(6)">生信技能树单细胞数据挖掘笔记(6)</a></li><li><a href="/2020/11/07/2020-11-07-ScRNAseq(13)/" rel="bookmark" title="生信技能树单细胞数据挖掘笔记(7)">生信技能树单细胞数据挖掘笔记(7)</a></li><li><a href="/2020/11/10/2020-11-10-RNAvelocity(1)/" rel="bookmark" title="RNA速率：软件下载与loom文件准备">RNA速率：软件下载与loom文件准备</a></li><li><a href="/2020/11/12/2020-11-12-RNAvelocity(2)/" rel="bookmark" title="RNA速率：数据读入">RNA速率：数据读入</a></li><li><a href="/2020/11/14/2020-11-14-RNAvelocity(3)/" rel="bookmark" title="RNA速率：使用Seurat的结果做RNA velocity">RNA速率：使用Seurat的结果做RNA velocity</a></li><li><a href="/2020/11/17/2020-11-17-RNAseq(1)/" rel="bookmark" title="RNAseq教程(1.1)">RNAseq教程(1.1)</a></li><li><a href="/2020/11/19/2020-11-19-annotion(1)/" rel="bookmark" title="基因结构注释（1）：从头注释">基因结构注释（1）：从头注释</a></li><li><a href="/2020/11/21/2020-11-21-annotion(2)/" rel="bookmark" title="基因结构注释（2）：同源注释">基因结构注释（2）：同源注释</a></li><li><a href="/2020/11/24/2020-11-24-annotion(3)/" rel="bookmark" title="基因结构注释（3）：转录组注释">基因结构注释（3）：转录组注释</a></li><li><a href="/2020/11/26/2020-11-26-annotion(4)/" rel="bookmark" title="基因结构注释（4）：整合预测结果">基因结构注释（4）：整合预测结果</a></li><li><a href="/2020/11/28/2020-11-28-RNAseq(2)/" rel="bookmark" title="RNAseq教程(1.2)">RNAseq教程(1.2)</a></li><li><a href="/2020/12/01/2020-12-01-RNAseq(3)/" rel="bookmark" title="RNAseq教程(1.3)">RNAseq教程(1.3)</a></li><li><a href="/2020/12/03/2020-12-03-RNAseq(4)/" rel="bookmark" title="RNAseq教程(1.4)">RNAseq教程(1.4)</a></li><li><a href="/2020/12/05/2020-12-05-RNAseq(5)/" rel="bookmark" title="RNAseq教程(1.5)">RNAseq教程(1.5)</a></li><li><a href="/2020/12/08/2020-12-08-RNAseq(6)/" rel="bookmark" title="RNAseq教程(1.6)">RNAseq教程(1.6)</a></li><li><a href="/2020/12/10/2020-12-10-RNAseq(7)/" rel="bookmark" title="RNAseq教程(2.1)">RNAseq教程(2.1)</a></li><li><a href="/2020/12/12/2020-12-12-RNAseq(8)/" rel="bookmark" title="RNAseq教程(2.2)">RNAseq教程(2.2)</a></li><li><a href="/2020/12/15/2020-12-15-RNAseq(9)/" rel="bookmark" title="RNAseq教程(2.3)">RNAseq教程(2.3)</a></li><li><a href="/2020/12/17/2020-12-17-RNAseq(10)/" rel="bookmark" title="RNAseq教程(2.4)">RNAseq教程(2.4)</a></li><li><a href="/2020/12/19/2020-12-19-RNAseq(11)/" rel="bookmark" title="RNAseq教程(2.5)">RNAseq教程(2.5)</a></li><li><a href="/2020/12/22/2020-12-22-RNAseq(12)/" rel="bookmark" title="RNAseq教程(3.1)">RNAseq教程(3.1)</a></li><li><a href="/2020/12/24/2020-12-24-RNAseq(13)/" rel="bookmark" title="RNAseq教程(3.2)">RNAseq教程(3.2)</a></li><li><a href="/2020/12/26/2020-12-26-RNAseq(14)/" rel="bookmark" title="RNAseq教程(3.3)">RNAseq教程(3.3)</a></li><li><a href="/2020/12/29/2020-12-29-RNAseq(15)/" rel="bookmark" title="RNAseq教程(3.4)">RNAseq教程(3.4)</a></li><li><a href="/2020/12/31/2020-12-31-RNAseq(16)/" rel="bookmark" title="RNAseq教程(4.1)">RNAseq教程(4.1)</a></li><li><a href="/2021/01/02/2021-01-02-RNAseq(17)/" rel="bookmark" title="RNAseq教程(4.2)">RNAseq教程(4.2)</a></li><li><a href="/2021/01/05/2021-01-05-RNAseq(18)/" rel="bookmark" title="RNAseq教程(4.3)">RNAseq教程(4.3)</a></li><li><a href="/2021/01/07/2021-01-07-RNAseq(19)/" rel="bookmark" title="RNAseq教程(4.4)">RNAseq教程(4.4)</a></li><li><a href="/2021/01/09/2021-01-09-RNAseq(20)/" rel="bookmark" title="RNAseq教程(4.5)">RNAseq教程(4.5)</a></li><li><a href="/2021/01/12/2021-01-12-RNAseq(21)/" rel="bookmark" title="RNAseq教程(5)">RNAseq教程(5)</a></li><li><a href="/2022/12/08/2022-12-08-SingleCopy/" rel="bookmark" title="比较基因组学分析1：构建单拷贝基因树">比较基因组学分析1：构建单拷贝基因树</a></li><li><a href="/2022/12/14/2022-12-14-CAFE/" rel="bookmark" title="比较基因组学分析2：基因家族收缩与扩张分析">比较基因组学分析2：基因家族收缩与扩张分析</a></li><li><a href="/2022/12/15/2022-12-15-GO/" rel="bookmark" title="比较基因组学分析3：特异节点基因家族富集分析（非模式物种GO/KEEG库构建）">比较基因组学分析3：特异节点基因家族富集分析（非模式物种GO/KEEG库构建）</a></li></ul></div><div class="overview panel" data-title="站点概览"><div class="author" itemprop="author" itemscope itemtype="http://schema.org/Person"><img class="image" itemprop="image" alt="kuchi Shū" data-src="/images/avatar.png"><p class="name" itemprop="name">kuchi Shū</p><div class="description" itemprop="description">生物学在读博士</div></div><nav class="state"><div class="item posts"><a href="/archives/"><span class="count">83</span> <span class="name">文章</span></a></div><div class="item categories"><a href="/categories/"><span class="count">6</span> <span class="name">分类</span></a></div><div class="item tags"><a href="/tags/"><span class="count">20</span> <span class="name">标签</span></a></div></nav><div class="social"><span class="exturl item github" data-url="aHR0cHM6Ly9naXRodWIuY29tL0Jpb2luZm9ybWF0aWNzLXJvb2tpZQ==" title="https:&#x2F;&#x2F;github.com&#x2F;Bioinformatics-rookie"><i class="ic i-github"></i></span> <span class="exturl item about" data-url="aHR0cHM6Ly93d3cuemhvdXhpYW96aGFvLmNuL2Fib3V0Lw==" title="https:&#x2F;&#x2F;www.zhouxiaozhao.cn&#x2F;about&#x2F;"><i class="ic i-address-card"></i></span> <span class="exturl item email" data-url="bWFpbHRvOnpob3VzaGl6aGFvQG1haWwua2liLmFjLmNu" title="mailto:zhoushizhao@mail.kib.ac.cn"><i class="ic i-envelope"></i></span></div><ul class="menu"><li class="item"><a href="/" rel="section"><i class="ic i-home"></i>首页</a></li><li class="item"><a href="/about/" rel="section"><i class="ic i-user"></i>关于</a></li><li class="item dropdown"><a href="javascript:void(0);"><i class="ic i-feather"></i>文章</a><ul class="submenu"><li class="item"><a href="/archives/" rel="section"><i class="ic i-list-alt"></i>归档</a></li><li class="item"><a href="/categories/" rel="section"><i class="ic i-th"></i>分类</a></li><li class="item"><a href="/tags/" rel="section"><i class="ic i-tags"></i>标签</a></li></ul></li><li class="item"><a href="/links/" rel="section"><i class="ic i-heart"></i>友達</a></li></ul></div></div></div><ul id="quick"><li class="prev pjax"><a href="/2020/10/01/2020-10-01-ScRNAseq(2)/" rel="prev" title="上一篇"><i class="ic i-chevron-left"></i></a></li><li class="up"><i class="ic i-arrow-up"></i></li><li class="down"><i class="ic i-arrow-down"></i></li><li class="next pjax"><a href="/2020/10/06/2020-10-06-ScRNAseq(4)/" rel="next" title="下一篇"><i class="ic i-chevron-right"></i></a></li><li class="percent"></li></ul></div></div><div class="dimmer"></div></div></main><footer id="footer"><div class="inner"><div class="widgets"><div class="rpost pjax"><h2>随机文章</h2><ul><li class="item"><div class="breadcrumb"><a href="/categories/Bioinformation/" title="分类于 生信学习">生信学习</a></div><span><a href="/2020/11/21/2020-11-21-annotion(2)/" title="基因结构注释（2）：同源注释">基因结构注释（2）：同源注释</a></span></li><li class="item"><div class="breadcrumb"><a href="/categories/Bioinformation/" title="分类于 生信学习">生信学习</a></div><span><a href="/2021/01/12/2021-01-12-RNAseq(21)/" title="RNAseq教程(5)">RNAseq教程(5)</a></span></li><li class="item"><div class="breadcrumb"><a href="/categories/Bioinformation/" title="分类于 生信学习">生信学习</a></div><span><a href="/2020/08/03/2020-08-03-WGCNA1/" title="WGCNA学习：WGCNA分析原理">WGCNA学习：WGCNA分析原理</a></span></li><li class="item"><div class="breadcrumb"><a href="/categories/Bioinformation/" title="分类于 生信学习">生信学习</a></div><span><a href="/2020/10/01/2020-10-01-ScRNAseq(2)/" title="单细胞实战(2)：cellranger使用">单细胞实战(2)：cellranger使用</a></span></li><li class="item"><div class="breadcrumb"><a href="/categories/paper/" title="分类于 文献导读">文献导读</a></div><span><a href="/2020/10/08/2020-10-08-2Dto3D/" title="文献导读：CLAVATA在小立碗藓3D生长中的作用">文献导读：CLAVATA在小立碗藓3D生长中的作用</a></span></li><li class="item"><div class="breadcrumb"><a href="/categories/Bioinformation/" title="分类于 生信学习">生信学习</a></div><span><a href="/2020/10/27/2020-10-27-ScRNAseq(8)/" title="生信技能树单细胞数据挖掘笔记(2)">生信技能树单细胞数据挖掘笔记(2)</a></span></li><li class="item"><div class="breadcrumb"><a href="/categories/Bioinformation/" title="分类于 生信学习">生信学习</a></div><span><a href="/2020/12/17/2020-12-17-RNAseq(10)/" title="RNAseq教程(2.4)">RNAseq教程(2.4)</a></span></li><li class="item"><div class="breadcrumb"><a href="/categories/Bioinformation/" title="分类于 生信学习">生信学习</a></div><span><a href="/2020/09/06/2020-09-06-iTOL3/" title="进化树在线制作工具iTOL教程（3）：添加图片">进化树在线制作工具iTOL教程（3）：添加图片</a></span></li><li class="item"><div class="breadcrumb"><a href="/categories/Python/" title="分类于 Python">Python</a></div><span><a href="/2020/10/17/2020-10-17-python7/" title="马士兵教育Python入门基础:4.流程控制（1）">马士兵教育Python入门基础:4.流程控制（1）</a></span></li><li class="item"><div class="breadcrumb"><a href="/categories/Bioinformation/" title="分类于 生信学习">生信学习</a></div><span><a href="/2020/08/30/2020-08-30-evolview3/" title="进化树+蛋白结构域">进化树+蛋白结构域</a></span></li></ul></div><div><h2>最新评论</h2><ul class="leancloud-recent-comment"></ul></div></div><div class="status"><div class="copyright">&copy; 2010 – <span itemprop="copyrightYear">2023</span> <span class="with-love"><i class="ic i-sakura rotate"></i> </span><span class="author" itemprop="copyrightHolder">kuchi Shū @ Zhou Xiaozhao</span></div><div class="count"><span class="post-meta-item-icon"><i class="ic i-chart-area"></i> </span><span title="站点总字数">344k 字</span> <span class="post-meta-divider">|</span> <span class="post-meta-item-icon"><i class="ic i-coffee"></i> </span><span title="站点阅读时长">5:13</span></div><div class="powered-by">基于 <span class="exturl" data-url="aHR0cHM6Ly9oZXhvLmlv">Hexo</span> & Theme.<span class="exturl" data-url="aHR0cHM6Ly9naXRodWIuY29tL2FtZWhpbWUvaGV4by10aGVtZS1zaG9rYQ==">Shoka</span></div></div></div></footer></div><script data-config type="text/javascript">var LOCAL={path:"2020/10/03/2020-10-03-ScRNAseq(3)/",favicon:{show:"（●´3｀●）やれやれだぜ",hide:"(´Д｀)大変だ！"},search:{placeholder:"文章搜索",empty:"关于 「 ${query} 」，什么也没搜到",stats:"${time} ms 内找到 ${hits} 条结果"},valine:!0,fancybox:!0,copyright:'复制成功，转载请遵守 <i class="ic i-creative-commons"></i>BY-NC-SA 协议。',ignores:[function(e){return e.includes("#")},function(e){return new RegExp(LOCAL.path+"$").test(e)}]}</script><script src="https://cdn.polyfill.io/v2/polyfill.js"></script><script src="//cdn.jsdelivr.net/combine/npm/pace-js@1.0.2/pace.min.js,npm/pjax@0.2.8/pjax.min.js,npm/whatwg-fetch@3.4.0/dist/fetch.umd.min.js,npm/animejs@3.2.0/lib/anime.min.js,npm/algoliasearch@4/dist/algoliasearch-lite.umd.js,npm/instantsearch.js@4/dist/instantsearch.production.min.js,npm/lozad@1/dist/lozad.min.js,npm/quicklink@2/dist/quicklink.umd.js"></script><script src="/js/app.js?v=0.2.5"></script></body></html><!-- rebuild by hrmmi -->